{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyNwpTVu3VoVDK77c2/xYDMk",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/Lynxlazy/genomics-data-analysis/blob/main/Project2.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "QT601tcUc912",
        "outputId": "7c5a2f46-c3fe-416e-e8bd-acc7c37a1150"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-03-03 07:59:06--  https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-ubuntu64.tar.gz\n",
            "Resolving ftp-trace.ncbi.nlm.nih.gov (ftp-trace.ncbi.nlm.nih.gov)... 130.14.250.7, 130.14.250.10, 2607:f220:41f:250::230, ...\n",
            "Connecting to ftp-trace.ncbi.nlm.nih.gov (ftp-trace.ncbi.nlm.nih.gov)|130.14.250.7|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 86400775 (82M) [application/x-gzip]\n",
            "Saving to: ‘sratoolkit.tar.gz’\n",
            "\n",
            "sratoolkit.tar.gz   100%[===================>]  82.40M   145MB/s    in 0.6s    \n",
            "\n",
            "2023-03-03 07:59:07 (145 MB/s) - ‘sratoolkit.tar.gz’ saved [86400775/86400775]\n",
            "\n",
            "sratoolkit.3.0.1-ubuntu64/\n",
            "sratoolkit.3.0.1-ubuntu64/README.md\n",
            "sratoolkit.3.0.1-ubuntu64/schema/\n",
            "sratoolkit.3.0.1-ubuntu64/schema/insdc/\n",
            "sratoolkit.3.0.1-ubuntu64/schema/insdc/sra.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/insdc/insdc.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/insdc/seq.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/align/\n",
            "sratoolkit.3.0.1-ubuntu64/schema/align/refseq.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/align/qstat.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/align/pileup-stats.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/align/mate-cache.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/align/align.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/align/seq.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/helicos.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/454.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/pacbio.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/generic-fastq.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/nanopore.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/pevents.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/ion-torrent.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/illumina.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/abi.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/vdb/\n",
            "sratoolkit.3.0.1-ubuntu64/schema/vdb/built-in.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/vdb/vdb.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/sra.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/seq-graph.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/spotname.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/wgs-contig.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/pnbrdb.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/varloc.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/clip.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/seq.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/ncbi.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/stats.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/trace.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/csra2/\n",
            "sratoolkit.3.0.1-ubuntu64/schema/csra2/csra2.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/csra2/read.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/csra2/reference.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/csra2/stats.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/CHANGES\n",
            "sratoolkit.3.0.1-ubuntu64/bin/\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-decrypt.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-encrypt\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fastq-dump.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-sort-cg.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fasterq-dump-orig.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-validate.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/srapath.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/cg-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/illumina-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/abi-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/kdbmeta\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-unlock.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/srf-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/abi-dump.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/srf-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/dump-ref-fasta.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-config\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-copy.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-search.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/test-sra\n",
            "sratoolkit.3.0.1-ubuntu64/bin/abi-dump.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-lock.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/srapath.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-dump\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-dump.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/align-info.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/cg-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/kar.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/srapath-orig.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-pileup\n",
            "sratoolkit.3.0.1-ubuntu64/bin/dump-ref-fasta\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-pileup.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fasterq-dump.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/helicos-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fastq-dump\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sam-dump.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/kar.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/align-info.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sam-dump.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sff-dump.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-config.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/bam-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-copy\n",
            "sratoolkit.3.0.1-ubuntu64/bin/bam-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/helicos-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-pileup.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-stat\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-lock\n",
            "sratoolkit.3.0.1-ubuntu64/bin/illumina-dump\n",
            "sratoolkit.3.0.1-ubuntu64/bin/pacbio-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fastq-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-copy.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/dump-ref-fasta.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/cache-mgr.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/pacbio-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/abi-dump\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-encrypt.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/cg-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sff-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/abi-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-stat.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-unlock.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/srapath\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sff-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/kar\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-sort.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-stat.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/ncbi/\n",
            "sratoolkit.3.0.1-ubuntu64/bin/ncbi/default.kfg\n",
            "sratoolkit.3.0.1-ubuntu64/bin/ncbi/vdb-copy.kfg\n",
            "sratoolkit.3.0.1-ubuntu64/bin/ncbi/certs.kfg\n",
            "sratoolkit.3.0.1-ubuntu64/bin/kdbmeta.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fastq-dump-orig.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-config.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-search\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sam-dump-orig.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/illumina-dump.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/illumina-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/abi-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/test-sra.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/test-sra.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fastq-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/prefetch-orig.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sratools.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/cache-mgr.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fasterq-dump\n",
            "sratoolkit.3.0.1-ubuntu64/bin/illumina-dump.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fastq-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/helicos-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/latf-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-validate\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-lock.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/srf-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/kdbmeta.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/cache-mgr\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sff-dump.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/rcexplain.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-sort.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-search.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fastq-dump.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/bam-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-pileup-orig.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-decrypt\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-sort-cg\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-decrypt.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-unlock\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-encrypt.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-validate.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/align-info\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sff-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-dump-orig.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/prefetch\n",
            "sratoolkit.3.0.1-ubuntu64/bin/rcexplain.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sam-dump\n",
            "sratoolkit.3.0.1-ubuntu64/bin/prefetch.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sff-dump\n",
            "sratoolkit.3.0.1-ubuntu64/bin/rcexplain\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fasterq-dump.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/illumina-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-dump.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/latf-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-sort\n",
            "sratoolkit.3.0.1-ubuntu64/bin/prefetch.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-sort-cg.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/pacbio-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/latf-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/example/\n",
            "sratoolkit.3.0.1-ubuntu64/example/perl/\n",
            "sratoolkit.3.0.1-ubuntu64/example/perl/base-stats.pl\n",
            "sratoolkit.3.0.1-ubuntu64/example/perl/mismatch-stats.pl\n",
            "sratoolkit.3.0.1-ubuntu64/example/perl/gene-lookup.pl\n",
            "sratoolkit.3.0.1-ubuntu64/example/perl/splitfastq.pl\n",
            "sratoolkit.3.0.1-ubuntu64/example/perl/quality-stats.pl\n",
            "sratoolkit.3.0.1-ubuntu64/example/perl/simplefastq.pl\n",
            "sratoolkit.3.0.1-ubuntu64/example/perl/dump-reference.pl\n",
            "sratoolkit.3.0.1-ubuntu64/README-blastn\n",
            "sratoolkit.3.0.1-ubuntu64/README-vdb-config\n"
          ]
        }
      ],
      "source": [
        "!wget --output-document sratoolkit.tar.gz https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-ubuntu64.tar.gz\n",
        "!tar -vxzf sratoolkit.tar.gz\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!export PATH=$PATH:$PWD/sratoolkit.3.0.1-ubuntu64/bin\n",
        "!which fastq-dump"
      ],
      "metadata": {
        "id": "vkkUS4ARe5o2"
      },
      "execution_count": 8,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "# 1. Exploring the dataset"
      ],
      "metadata": {
        "id": "QqN3noYueLkj"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!./sratoolkit.3.0.1-ubuntu64/bin/fasterq-dump SRR292678\n",
        "!./sratoolkit.3.0.1-ubuntu64/bin/fasterq-dump SRR292862\n",
        "!./sratoolkit.3.0.1-ubuntu64/bin/fasterq-dump SRR292770"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "n7NJm1MLfNEv",
        "outputId": "306809b2-a034-4006-8c01-3adedf224086"
      },
      "execution_count": 16,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "spots read      : 6,111,112\n",
            "reads read      : 12,222,224\n",
            "reads written   : 12,222,224\n",
            "spots read      : 5,102,041\n",
            "reads read      : 10,204,082\n",
            "reads written   : 10,204,082\n",
            "spots read      : 5,102,041\n",
            "reads read      : 10,204,082\n",
            "reads written   : 10,204,082\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# 2. K-mer profile and genome size estimation\n"
      ],
      "metadata": {
        "id": "pB5h8KoIglO_"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!sudo apt install jellyfish"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "_lksBntbgjpZ",
        "outputId": "324407bb-34ed-46de-8b6a-221072f3ec91"
      },
      "execution_count": 17,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  libhts3 libjellyfish-2.0-2\n",
            "The following NEW packages will be installed:\n",
            "  jellyfish libhts3 libjellyfish-2.0-2\n",
            "0 upgraded, 3 newly installed, 0 to remove and 22 not upgraded.\n",
            "Need to get 771 kB of archives.\n",
            "After this operation, 1,727 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 libhts3 amd64 1.10.2-3ubuntu0.1 [350 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu focal/universe amd64 libjellyfish-2.0-2 amd64 2.3.0-4build1 [60.5 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu focal/universe amd64 jellyfish amd64 2.3.0-4build1 [361 kB]\n",
            "Fetched 771 kB in 0s (2,706 kB/s)\n",
            "debconf: unable to initialize frontend: Dialog\n",
            "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 3.)\n",
            "debconf: falling back to frontend: Readline\n",
            "debconf: unable to initialize frontend: Readline\n",
            "debconf: (This frontend requires a controlling tty.)\n",
            "debconf: falling back to frontend: Teletype\n",
            "dpkg-preconfigure: unable to re-open stdin: \n",
            "Selecting previously unselected package libhts3:amd64.\n",
            "(Reading database ... 128213 files and directories currently installed.)\n",
            "Preparing to unpack .../libhts3_1.10.2-3ubuntu0.1_amd64.deb ...\n",
            "Unpacking libhts3:amd64 (1.10.2-3ubuntu0.1) ...\n",
            "Selecting previously unselected package libjellyfish-2.0-2:amd64.\n",
            "Preparing to unpack .../libjellyfish-2.0-2_2.3.0-4build1_amd64.deb ...\n",
            "Unpacking libjellyfish-2.0-2:amd64 (2.3.0-4build1) ...\n",
            "Selecting previously unselected package jellyfish.\n",
            "Preparing to unpack .../jellyfish_2.3.0-4build1_amd64.deb ...\n",
            "Unpacking jellyfish (2.3.0-4build1) ...\n",
            "Setting up libhts3:amd64 (1.10.2-3ubuntu0.1) ...\n",
            "Setting up libjellyfish-2.0-2:amd64 (2.3.0-4build1) ...\n",
            "Setting up jellyfish (2.3.0-4build1) ...\n",
            "Processing triggers for man-db (2.9.1-1) ...\n",
            "Processing triggers for libc-bin (2.31-0ubuntu9.9) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://trace.ncbi.nlm.nih.gov/Traces/sra-reads-be/fasta?acc=SRR292678"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "p5v_ePwFkylY",
        "outputId": "805b0c4d-f401-4383-ea99-c72411da188d"
      },
      "execution_count": 19,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-03-03 08:25:14--  https://trace.ncbi.nlm.nih.gov/Traces/sra-reads-be/fasta?acc=SRR292678\n",
            "Resolving trace.ncbi.nlm.nih.gov (trace.ncbi.nlm.nih.gov)... 130.14.29.113, 2607:f220:41e:4290::113\n",
            "Connecting to trace.ncbi.nlm.nih.gov (trace.ncbi.nlm.nih.gov)|130.14.29.113|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: unspecified [application/x-gzip]\n",
            "Saving to: ‘fasta?acc=SRR292678’\n",
            "\n",
            "fasta?acc=SRR292678     [           <=>      ] 438.04M  1.09MB/s    in 6m 44s  \n",
            "\n",
            "2023-03-03 08:31:59 (1.08 MB/s) - ‘fasta?acc=SRR292678’ saved [459321528]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!mv fasta?acc=SRR292678 SRR292678.fasta"
      ],
      "metadata": {
        "id": "WsK_POegmcRd"
      },
      "execution_count": 21,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!jellyfish count -m 21 -s 5G -t 10 -C SRR292678_1.fastq"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "-CEjfL77goB0",
        "outputId": "fe7ab0f3-29fd-4fdc-d5bd-b5f7a11ecd0c"
      },
      "execution_count": 22,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "terminate called after throwing an instance of 'std::runtime_error'\n",
            "  what():  Unsupported format\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# 3. Assembling E. coli X genome from paired reads"
      ],
      "metadata": {
        "id": "V6e_ohwElCfr"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!apt -y install spades"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bpe7qcdzlBgF",
        "outputId": "beb460a9-d092-4902-dbd4-061dbd31ba43"
      },
      "execution_count": 20,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  bamtools bwa cwltool fonts-glyphicons-halflings libbamtools2.5.1 libc-ares2\n",
            "  libhat-trie0 libjs-bootstrap libnlopt0 libnode64 libssw0 nodejs nodejs-doc\n",
            "  python3-atomicwrites python3-attr python3-bs4 python3-cachecontrol\n",
            "  python3-coloredlogs python3-dateutil python3-decorator python3-html5lib\n",
            "  python3-humanfriendly python3-importlib-metadata python3-isodate\n",
            "  python3-joblib python3-lockfile python3-lxml python3-mistune\n",
            "  python3-more-itertools python3-msgpack python3-mypy-extensions\n",
            "  python3-networkx python3-packaging python3-pluggy python3-prov\n",
            "  python3-psutil python3-py python3-pyparsing python3-pytest python3-rdflib\n",
            "  python3-rdflib-jsonld python3-ruamel.yaml python3-schema-salad\n",
            "  python3-shellescape python3-simplejson python3-soupsieve\n",
            "  python3-sparqlwrapper python3-typing-extensions python3-wcwidth\n",
            "  python3-webencodings python3-yaml python3-zipp samtools\n",
            "Suggested packages:\n",
            "  docker.io | singularity-container npm python-attr-doc python3-genshi\n",
            "  python-lockfile-doc python3-lxml-dbg python-lxml-doc python-networkx-doc\n",
            "  python3-matplotlib python3-pygraphviz | python3-pydot python3-scipy\n",
            "  python-prov-doc python3-pydot python-psutil-doc subversion\n",
            "  python-pyparsing-doc python-rdflib-doc\n",
            "The following NEW packages will be installed:\n",
            "  bamtools bwa cwltool fonts-glyphicons-halflings libbamtools2.5.1 libc-ares2\n",
            "  libhat-trie0 libjs-bootstrap libnlopt0 libnode64 libssw0 nodejs nodejs-doc\n",
            "  python3-atomicwrites python3-attr python3-bs4 python3-cachecontrol\n",
            "  python3-coloredlogs python3-dateutil python3-decorator python3-html5lib\n",
            "  python3-humanfriendly python3-importlib-metadata python3-isodate\n",
            "  python3-joblib python3-lockfile python3-lxml python3-mistune\n",
            "  python3-more-itertools python3-msgpack python3-mypy-extensions\n",
            "  python3-networkx python3-packaging python3-pluggy python3-prov\n",
            "  python3-psutil python3-py python3-pyparsing python3-pytest python3-rdflib\n",
            "  python3-rdflib-jsonld python3-ruamel.yaml python3-schema-salad\n",
            "  python3-shellescape python3-simplejson python3-soupsieve\n",
            "  python3-sparqlwrapper python3-typing-extensions python3-wcwidth\n",
            "  python3-webencodings python3-yaml python3-zipp samtools spades\n",
            "0 upgraded, 54 newly installed, 0 to remove and 22 not upgraded.\n",
            "Need to get 18.1 MB of archives.\n",
            "After this operation, 86.7 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 python3-yaml amd64 5.3.1-1ubuntu0.1 [136 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu focal/universe amd64 libbamtools2.5.1 amd64 2.5.1+dfsg-5build1 [158 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu focal/universe amd64 bamtools amd64 2.5.1+dfsg-5build1 [177 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu focal/universe amd64 bwa amd64 0.7.17-4 [204 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-humanfriendly all 4.18-2 [60.8 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-coloredlogs all 7.3-2 [28.0 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-mypy-extensions all 0.4.3-1 [5,436 B]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-dateutil all 2.7.3-3ubuntu1 [63.3 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 python3-lxml amd64 4.5.0-1ubuntu0.5 [1,384 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-decorator all 4.4.2-0ubuntu1 [10.3 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-networkx all 2.4-3ubuntu1 [855 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-isodate all 0.6.0-2 [24.0 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-pyparsing all 2.4.6-1 [61.3 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-rdflib all 4.2.2-5 [241 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-prov all 1.5.2-2 [92.8 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-psutil amd64 5.5.1-1ubuntu4 [141 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-ruamel.yaml amd64 0.15.89-3build1 [228 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-msgpack amd64 0.6.2-1 [73.5 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-lockfile all 1:0.12.2-2ubuntu2 [14.6 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-cachecontrol all 0.12.6-1ubuntu2 [16.7 kB]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-mistune all 0.8.4-2 [14.3 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-rdflib-jsonld all 0.4.0-5 [12.4 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-typing-extensions all 3.7.4.1-1 [27.5 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu focal/universe amd64 fonts-glyphicons-halflings all 1.009~3.4.1+dfsg-1 [117 kB]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu focal/universe amd64 libjs-bootstrap all 3.4.1+dfsg-1 [124 kB]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-schema-salad all 5.0.20200220195218-1 [295 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-shellescape all 3.4.1-4 [4,928 B]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu focal/universe amd64 cwltool all 2.0.20200224214940+dfsg-1 [203 kB]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu focal/universe amd64 libhat-trie0 amd64 0.1.2-2 [11.7 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libc-ares2 amd64 1.15.0-1ubuntu0.1 [38.2 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu focal/universe amd64 libnode64 amd64 10.19.0~dfsg-3ubuntu1 [5,765 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu focal/universe amd64 libssw0 amd64 1.1-9 [18.9 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu focal/universe amd64 nodejs-doc all 10.19.0~dfsg-3ubuntu1 [942 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-atomicwrites all 1.1.5-2build1 [7,016 B]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-attr all 19.3.0-2 [33.9 kB]\n",
            "Get:36 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-soupsieve all 1.9.5+dfsg-1 [29.1 kB]\n",
            "Get:37 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-bs4 all 4.8.2-1 [83.0 kB]\n",
            "Get:38 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-webencodings all 0.5.1-1ubuntu1 [11.0 kB]\n",
            "Get:39 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-html5lib all 1.0.1-2 [84.3 kB]\n",
            "Get:40 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-more-itertools all 4.2.0-1build1 [39.4 kB]\n",
            "Get:41 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-zipp all 1.0.0-1 [5,312 B]\n",
            "Get:42 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-importlib-metadata all 1.5.0-1 [9,992 B]\n",
            "Get:43 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-joblib all 0.14.0-3 [192 kB]\n",
            "Get:44 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-packaging all 20.3-1 [26.8 kB]\n",
            "Get:45 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-pluggy all 0.13.0-2 [18.4 kB]\n",
            "Get:46 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 python3-py all 1.8.1-1ubuntu0.1 [65.6 kB]\n",
            "Get:47 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-wcwidth all 0.1.8+dfsg1-3 [17.4 kB]\n",
            "Get:48 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-pytest all 4.6.9-1 [178 kB]\n",
            "Get:49 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-simplejson amd64 3.16.0-2ubuntu2 [50.8 kB]\n",
            "Get:50 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-sparqlwrapper all 1.8.5-1 [32.3 kB]\n",
            "Get:51 http://archive.ubuntu.com/ubuntu focal/universe amd64 samtools amd64 1.10-3 [496 kB]\n",
            "Get:52 http://archive.ubuntu.com/ubuntu focal/universe amd64 libnlopt0 amd64 2.6.1-8ubuntu2 [167 kB]\n",
            "Get:53 http://archive.ubuntu.com/ubuntu focal/universe amd64 spades amd64 3.13.1+dfsg-2build2 [4,928 kB]\n",
            "Get:54 http://archive.ubuntu.com/ubuntu focal/universe amd64 nodejs amd64 10.19.0~dfsg-3ubuntu1 [61.1 kB]\n",
            "Fetched 18.1 MB in 5s (3,533 kB/s)\n",
            "Extracting templates from packages: 100%\n",
            "Selecting previously unselected package python3-yaml.\n",
            "(Reading database ... 128233 files and directories currently installed.)\n",
            "Preparing to unpack .../00-python3-yaml_5.3.1-1ubuntu0.1_amd64.deb ...\n",
            "Unpacking python3-yaml (5.3.1-1ubuntu0.1) ...\n",
            "Selecting previously unselected package libbamtools2.5.1:amd64.\n",
            "Preparing to unpack .../01-libbamtools2.5.1_2.5.1+dfsg-5build1_amd64.deb ...\n",
            "Unpacking libbamtools2.5.1:amd64 (2.5.1+dfsg-5build1) ...\n",
            "Selecting previously unselected package bamtools.\n",
            "Preparing to unpack .../02-bamtools_2.5.1+dfsg-5build1_amd64.deb ...\n",
            "Unpacking bamtools (2.5.1+dfsg-5build1) ...\n",
            "Selecting previously unselected package bwa.\n",
            "Preparing to unpack .../03-bwa_0.7.17-4_amd64.deb ...\n",
            "Unpacking bwa (0.7.17-4) ...\n",
            "Selecting previously unselected package python3-humanfriendly.\n",
            "Preparing to unpack .../04-python3-humanfriendly_4.18-2_all.deb ...\n",
            "Unpacking python3-humanfriendly (4.18-2) ...\n",
            "Selecting previously unselected package python3-coloredlogs.\n",
            "Preparing to unpack .../05-python3-coloredlogs_7.3-2_all.deb ...\n",
            "Unpacking python3-coloredlogs (7.3-2) ...\n",
            "Selecting previously unselected package python3-mypy-extensions.\n",
            "Preparing to unpack .../06-python3-mypy-extensions_0.4.3-1_all.deb ...\n",
            "Unpacking python3-mypy-extensions (0.4.3-1) ...\n",
            "Selecting previously unselected package python3-dateutil.\n",
            "Preparing to unpack .../07-python3-dateutil_2.7.3-3ubuntu1_all.deb ...\n",
            "Unpacking python3-dateutil (2.7.3-3ubuntu1) ...\n",
            "Selecting previously unselected package python3-lxml:amd64.\n",
            "Preparing to unpack .../08-python3-lxml_4.5.0-1ubuntu0.5_amd64.deb ...\n",
            "Unpacking python3-lxml:amd64 (4.5.0-1ubuntu0.5) ...\n",
            "Selecting previously unselected package python3-decorator.\n",
            "Preparing to unpack .../09-python3-decorator_4.4.2-0ubuntu1_all.deb ...\n",
            "Unpacking python3-decorator (4.4.2-0ubuntu1) ...\n",
            "Selecting previously unselected package python3-networkx.\n",
            "Preparing to unpack .../10-python3-networkx_2.4-3ubuntu1_all.deb ...\n",
            "Unpacking python3-networkx (2.4-3ubuntu1) ...\n",
            "Selecting previously unselected package python3-isodate.\n",
            "Preparing to unpack .../11-python3-isodate_0.6.0-2_all.deb ...\n",
            "Unpacking python3-isodate (0.6.0-2) ...\n",
            "Selecting previously unselected package python3-pyparsing.\n",
            "Preparing to unpack .../12-python3-pyparsing_2.4.6-1_all.deb ...\n",
            "Unpacking python3-pyparsing (2.4.6-1) ...\n",
            "Selecting previously unselected package python3-rdflib.\n",
            "Preparing to unpack .../13-python3-rdflib_4.2.2-5_all.deb ...\n",
            "Unpacking python3-rdflib (4.2.2-5) ...\n",
            "Selecting previously unselected package python3-prov.\n",
            "Preparing to unpack .../14-python3-prov_1.5.2-2_all.deb ...\n",
            "Unpacking python3-prov (1.5.2-2) ...\n",
            "Selecting previously unselected package python3-psutil.\n",
            "Preparing to unpack .../15-python3-psutil_5.5.1-1ubuntu4_amd64.deb ...\n",
            "Unpacking python3-psutil (5.5.1-1ubuntu4) ...\n",
            "Selecting previously unselected package python3-ruamel.yaml.\n",
            "Preparing to unpack .../16-python3-ruamel.yaml_0.15.89-3build1_amd64.deb ...\n",
            "Unpacking python3-ruamel.yaml (0.15.89-3build1) ...\n",
            "Selecting previously unselected package python3-msgpack.\n",
            "Preparing to unpack .../17-python3-msgpack_0.6.2-1_amd64.deb ...\n",
            "Unpacking python3-msgpack (0.6.2-1) ...\n",
            "Selecting previously unselected package python3-lockfile.\n",
            "Preparing to unpack .../18-python3-lockfile_1%3a0.12.2-2ubuntu2_all.deb ...\n",
            "Unpacking python3-lockfile (1:0.12.2-2ubuntu2) ...\n",
            "Selecting previously unselected package python3-cachecontrol.\n",
            "Preparing to unpack .../19-python3-cachecontrol_0.12.6-1ubuntu2_all.deb ...\n",
            "Unpacking python3-cachecontrol (0.12.6-1ubuntu2) ...\n",
            "Selecting previously unselected package python3-mistune.\n",
            "Preparing to unpack .../20-python3-mistune_0.8.4-2_all.deb ...\n",
            "Unpacking python3-mistune (0.8.4-2) ...\n",
            "Selecting previously unselected package python3-rdflib-jsonld.\n",
            "Preparing to unpack .../21-python3-rdflib-jsonld_0.4.0-5_all.deb ...\n",
            "Unpacking python3-rdflib-jsonld (0.4.0-5) ...\n",
            "Selecting previously unselected package python3-typing-extensions.\n",
            "Preparing to unpack .../22-python3-typing-extensions_3.7.4.1-1_all.deb ...\n",
            "Unpacking python3-typing-extensions (3.7.4.1-1) ...\n",
            "Selecting previously unselected package fonts-glyphicons-halflings.\n",
            "Preparing to unpack .../23-fonts-glyphicons-halflings_1.009~3.4.1+dfsg-1_all.deb ...\n",
            "Unpacking fonts-glyphicons-halflings (1.009~3.4.1+dfsg-1) ...\n",
            "Selecting previously unselected package libjs-bootstrap.\n",
            "Preparing to unpack .../24-libjs-bootstrap_3.4.1+dfsg-1_all.deb ...\n",
            "Unpacking libjs-bootstrap (3.4.1+dfsg-1) ...\n",
            "Selecting previously unselected package python3-schema-salad.\n",
            "Preparing to unpack .../25-python3-schema-salad_5.0.20200220195218-1_all.deb ...\n",
            "Unpacking python3-schema-salad (5.0.20200220195218-1) ...\n",
            "Selecting previously unselected package python3-shellescape.\n",
            "Preparing to unpack .../26-python3-shellescape_3.4.1-4_all.deb ...\n",
            "Unpacking python3-shellescape (3.4.1-4) ...\n",
            "Selecting previously unselected package cwltool.\n",
            "Preparing to unpack .../27-cwltool_2.0.20200224214940+dfsg-1_all.deb ...\n",
            "Unpacking cwltool (2.0.20200224214940+dfsg-1) ...\n",
            "Selecting previously unselected package libhat-trie0:amd64.\n",
            "Preparing to unpack .../28-libhat-trie0_0.1.2-2_amd64.deb ...\n",
            "Unpacking libhat-trie0:amd64 (0.1.2-2) ...\n",
            "Selecting previously unselected package libc-ares2:amd64.\n",
            "Preparing to unpack .../29-libc-ares2_1.15.0-1ubuntu0.1_amd64.deb ...\n",
            "Unpacking libc-ares2:amd64 (1.15.0-1ubuntu0.1) ...\n",
            "Selecting previously unselected package libnode64:amd64.\n",
            "Preparing to unpack .../30-libnode64_10.19.0~dfsg-3ubuntu1_amd64.deb ...\n",
            "Unpacking libnode64:amd64 (10.19.0~dfsg-3ubuntu1) ...\n",
            "Selecting previously unselected package libssw0:amd64.\n",
            "Preparing to unpack .../31-libssw0_1.1-9_amd64.deb ...\n",
            "Unpacking libssw0:amd64 (1.1-9) ...\n",
            "Selecting previously unselected package nodejs-doc.\n",
            "Preparing to unpack .../32-nodejs-doc_10.19.0~dfsg-3ubuntu1_all.deb ...\n",
            "Unpacking nodejs-doc (10.19.0~dfsg-3ubuntu1) ...\n",
            "Selecting previously unselected package python3-atomicwrites.\n",
            "Preparing to unpack .../33-python3-atomicwrites_1.1.5-2build1_all.deb ...\n",
            "Unpacking python3-atomicwrites (1.1.5-2build1) ...\n",
            "Selecting previously unselected package python3-attr.\n",
            "Preparing to unpack .../34-python3-attr_19.3.0-2_all.deb ...\n",
            "Unpacking python3-attr (19.3.0-2) ...\n",
            "Selecting previously unselected package python3-soupsieve.\n",
            "Preparing to unpack .../35-python3-soupsieve_1.9.5+dfsg-1_all.deb ...\n",
            "Unpacking python3-soupsieve (1.9.5+dfsg-1) ...\n",
            "Selecting previously unselected package python3-bs4.\n",
            "Preparing to unpack .../36-python3-bs4_4.8.2-1_all.deb ...\n",
            "Unpacking python3-bs4 (4.8.2-1) ...\n",
            "Selecting previously unselected package python3-webencodings.\n",
            "Preparing to unpack .../37-python3-webencodings_0.5.1-1ubuntu1_all.deb ...\n",
            "Unpacking python3-webencodings (0.5.1-1ubuntu1) ...\n",
            "Selecting previously unselected package python3-html5lib.\n",
            "Preparing to unpack .../38-python3-html5lib_1.0.1-2_all.deb ...\n",
            "Unpacking python3-html5lib (1.0.1-2) ...\n",
            "Selecting previously unselected package python3-more-itertools.\n",
            "Preparing to unpack .../39-python3-more-itertools_4.2.0-1build1_all.deb ...\n",
            "Unpacking python3-more-itertools (4.2.0-1build1) ...\n",
            "Selecting previously unselected package python3-zipp.\n",
            "Preparing to unpack .../40-python3-zipp_1.0.0-1_all.deb ...\n",
            "Unpacking python3-zipp (1.0.0-1) ...\n",
            "Selecting previously unselected package python3-importlib-metadata.\n",
            "Preparing to unpack .../41-python3-importlib-metadata_1.5.0-1_all.deb ...\n",
            "Unpacking python3-importlib-metadata (1.5.0-1) ...\n",
            "Selecting previously unselected package python3-joblib.\n",
            "Preparing to unpack .../42-python3-joblib_0.14.0-3_all.deb ...\n",
            "Unpacking python3-joblib (0.14.0-3) ...\n",
            "Selecting previously unselected package python3-packaging.\n",
            "Preparing to unpack .../43-python3-packaging_20.3-1_all.deb ...\n",
            "Unpacking python3-packaging (20.3-1) ...\n",
            "Selecting previously unselected package python3-pluggy.\n",
            "Preparing to unpack .../44-python3-pluggy_0.13.0-2_all.deb ...\n",
            "Unpacking python3-pluggy (0.13.0-2) ...\n",
            "Selecting previously unselected package python3-py.\n",
            "Preparing to unpack .../45-python3-py_1.8.1-1ubuntu0.1_all.deb ...\n",
            "Unpacking python3-py (1.8.1-1ubuntu0.1) ...\n",
            "Selecting previously unselected package python3-wcwidth.\n",
            "Preparing to unpack .../46-python3-wcwidth_0.1.8+dfsg1-3_all.deb ...\n",
            "Unpacking python3-wcwidth (0.1.8+dfsg1-3) ...\n",
            "Selecting previously unselected package python3-pytest.\n",
            "Preparing to unpack .../47-python3-pytest_4.6.9-1_all.deb ...\n",
            "Unpacking python3-pytest (4.6.9-1) ...\n",
            "Selecting previously unselected package python3-simplejson.\n",
            "Preparing to unpack .../48-python3-simplejson_3.16.0-2ubuntu2_amd64.deb ...\n",
            "Unpacking python3-simplejson (3.16.0-2ubuntu2) ...\n",
            "Selecting previously unselected package python3-sparqlwrapper.\n",
            "Preparing to unpack .../49-python3-sparqlwrapper_1.8.5-1_all.deb ...\n",
            "Unpacking python3-sparqlwrapper (1.8.5-1) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../50-samtools_1.10-3_amd64.deb ...\n",
            "Unpacking samtools (1.10-3) ...\n",
            "Selecting previously unselected package libnlopt0:amd64.\n",
            "Preparing to unpack .../51-libnlopt0_2.6.1-8ubuntu2_amd64.deb ...\n",
            "Unpacking libnlopt0:amd64 (2.6.1-8ubuntu2) ...\n",
            "Selecting previously unselected package spades.\n",
            "Preparing to unpack .../52-spades_3.13.1+dfsg-2build2_amd64.deb ...\n",
            "Unpacking spades (3.13.1+dfsg-2build2) ...\n",
            "Selecting previously unselected package nodejs.\n",
            "Preparing to unpack .../53-nodejs_10.19.0~dfsg-3ubuntu1_amd64.deb ...\n",
            "Unpacking nodejs (10.19.0~dfsg-3ubuntu1) ...\n",
            "Setting up python3-more-itertools (4.2.0-1build1) ...\n",
            "Setting up python3-attr (19.3.0-2) ...\n",
            "Setting up python3-lockfile (1:0.12.2-2ubuntu2) ...\n",
            "Setting up python3-py (1.8.1-1ubuntu0.1) ...\n",
            "Setting up python3-joblib (0.14.0-3) ...\n",
            "Setting up bwa (0.7.17-4) ...\n",
            "Setting up fonts-glyphicons-halflings (1.009~3.4.1+dfsg-1) ...\n",
            "Setting up python3-mypy-extensions (0.4.3-1) ...\n",
            "Setting up python3-yaml (5.3.1-1ubuntu0.1) ...\n",
            "Setting up libnlopt0:amd64 (2.6.1-8ubuntu2) ...\n",
            "Setting up libc-ares2:amd64 (1.15.0-1ubuntu0.1) ...\n",
            "Setting up python3-zipp (1.0.0-1) ...\n",
            "Setting up python3-webencodings (0.5.1-1ubuntu1) ...\n",
            "Setting up python3-psutil (5.5.1-1ubuntu4) ...\n",
            "Setting up samtools (1.10-3) ...\n",
            "Setting up python3-atomicwrites (1.1.5-2build1) ...\n",
            "Setting up python3-simplejson (3.16.0-2ubuntu2) ...\n",
            "Setting up python3-isodate (0.6.0-2) ...\n",
            "Setting up python3-decorator (4.4.2-0ubuntu1) ...\n",
            "Setting up libhat-trie0:amd64 (0.1.2-2) ...\n",
            "Setting up python3-wcwidth (0.1.8+dfsg1-3) ...\n",
            "Setting up python3-pyparsing (2.4.6-1) ...\n",
            "Setting up python3-humanfriendly (4.18-2) ...\n",
            "Setting up libbamtools2.5.1:amd64 (2.5.1+dfsg-5build1) ...\n",
            "Setting up libnode64:amd64 (10.19.0~dfsg-3ubuntu1) ...\n",
            "Setting up python3-typing-extensions (3.7.4.1-1) ...\n",
            "Setting up python3-html5lib (1.0.1-2) ...\n",
            "Setting up nodejs-doc (10.19.0~dfsg-3ubuntu1) ...\n",
            "Setting up python3-ruamel.yaml (0.15.89-3build1) ...\n",
            "Setting up python3-lxml:amd64 (4.5.0-1ubuntu0.5) ...\n",
            "Setting up python3-rdflib (4.2.2-5) ...\n",
            "/usr/lib/python3/dist-packages/rdflib/plugins/memory.py:354: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
            "  if triple is None or triple is (None,None,None):\n",
            "Setting up bamtools (2.5.1+dfsg-5build1) ...\n",
            "Setting up python3-dateutil (2.7.3-3ubuntu1) ...\n",
            "Setting up python3-msgpack (0.6.2-1) ...\n",
            "Setting up python3-cachecontrol (0.12.6-1ubuntu2) ...\n",
            "Setting up python3-shellescape (3.4.1-4) ...\n",
            "Setting up python3-coloredlogs (7.3-2) ...\n",
            "Setting up python3-mistune (0.8.4-2) ...\n",
            "Setting up libssw0:amd64 (1.1-9) ...\n",
            "Setting up python3-soupsieve (1.9.5+dfsg-1) ...\n",
            "Setting up libjs-bootstrap (3.4.1+dfsg-1) ...\n",
            "Setting up python3-rdflib-jsonld (0.4.0-5) ...\n",
            "Setting up python3-importlib-metadata (1.5.0-1) ...\n",
            "Setting up python3-sparqlwrapper (1.8.5-1) ...\n",
            "Setting up python3-schema-salad (5.0.20200220195218-1) ...\n",
            "Setting up nodejs (10.19.0~dfsg-3ubuntu1) ...\n",
            "update-alternatives: using /usr/bin/nodejs to provide /usr/bin/js (js) in auto mode\n",
            "Setting up python3-networkx (2.4-3ubuntu1) ...\n",
            "/usr/lib/python3/dist-packages/networkx/readwrite/tests/test_graphml.py:933: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
            "  assert H.nodes['n1']['special'] is 0\n",
            "/usr/lib/python3/dist-packages/networkx/readwrite/tests/test_graphml.py:935: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
            "  assert H.edges['n0','n1',1]['special'] is 0\n",
            "Setting up python3-bs4 (4.8.2-1) ...\n",
            "Setting up python3-prov (1.5.2-2) ...\n",
            "Setting up python3-packaging (20.3-1) ...\n",
            "Setting up python3-pluggy (0.13.0-2) ...\n",
            "Setting up spades (3.13.1+dfsg-2build2) ...\n",
            "Setting up python3-pytest (4.6.9-1) ...\n",
            "Setting up cwltool (2.0.20200224214940+dfsg-1) ...\n",
            "/usr/lib/python3/dist-packages/cwltool/docker.py:423: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
            "  if res_req is not None and (\"ramMin\" in res_req or \"ramMax\" is res_req):\n",
            "Processing triggers for fontconfig (2.13.1-2ubuntu3) ...\n",
            "Processing triggers for libc-bin (2.31-0ubuntu9.9) ...\n",
            "Processing triggers for man-db (2.9.1-1) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir result"
      ],
      "metadata": {
        "id": "GDF0TUfglzll"
      },
      "execution_count": 23,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!spades --test"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "QHrhhLZSmu_v",
        "outputId": "ee1f25e4-33ac-4466-9ea8-3619c178a268"
      },
      "execution_count": 25,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Command line: /usr/lib/spades/bin/spades.py\t--test\t\n",
            "\n",
            "System information:\n",
            "  SPAdes version: 3.13.1\n",
            "  Python version: 3.8.10\n",
            "  OS: Linux-5.10.147+-x86_64-with-glibc2.29\n",
            "\n",
            "Output dir: /content/spades_test\n",
            "Mode: read error correction and assembling\n",
            "Debug mode is turned OFF\n",
            "\n",
            "Dataset parameters:\n",
            "  Multi-cell mode (you should set '--sc' flag if input data was obtained with MDA (single-cell) technology or --meta flag if processing metagenomic dataset)\n",
            "  Reads:\n",
            "    Library number: 1, library type: paired-end\n",
            "      orientation: fr\n",
            "      left reads: ['/usr/share/spades/test_dataset/ecoli_1K_1.fq.gz']\n",
            "      right reads: ['/usr/share/spades/test_dataset/ecoli_1K_2.fq.gz']\n",
            "      interlaced reads: not specified\n",
            "      single reads: not specified\n",
            "      merged reads: not specified\n",
            "Read error correction parameters:\n",
            "  Iterations: 1\n",
            "  PHRED offset will be auto-detected\n",
            "  Corrected reads will be compressed\n",
            "Assembly parameters:\n",
            "  k: automatic selection based on read length\n",
            "  Repeat resolution is enabled\n",
            "  Mismatch careful mode is turned OFF\n",
            "  MismatchCorrector will be SKIPPED\n",
            "  Coverage cutoff is turned OFF\n",
            "Other parameters:\n",
            "  Dir for temp files: /content/spades_test/tmp\n",
            "  Threads: 16\n",
            "  Memory limit (in Gb): 12\n",
            "\n",
            "\n",
            "======= SPAdes pipeline started. Log can be found here: /content/spades_test/spades.log\n",
            "\n",
            "\n",
            "===== Read error correction started. \n",
            "\n",
            "\n",
            "== Running read error correction tool: /usr/lib/spades/bin/spades-hammer /content/spades_test/corrected/configs/config.info\n",
            "\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from N/A, git revision N/A\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /content/spades_test/corrected/configs/config.info\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 12 Gb\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
            "  0:00:00.001     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
            "     === ITERATION 0 begins ===\n",
            "  0:00:00.001     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:00:00.001     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 32 files using 2 threads. This might take a while.\n",
            "  0:00:00.001     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.99935 Gb\n",
            "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2097152\n",
            "  0:00:00.001     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /usr/share/spades/test_dataset/ecoli_1K_1.fq.gz\n",
            "  0:00:00.048     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /usr/share/spades/test_dataset/ecoli_1K_2.fq.gz\n",
            "  0:00:00.096     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 4108 reads processed\n",
            "  0:00:00.098     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:00:00.102     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 1974 kmers in total.\n",
            "  0:00:00.103     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:00:00.105     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:00:00.133     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
            "  0:00:00.134     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 10136 bytes occupied (41.078 bits per kmer).\n",
            "  0:00:00.134     8M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
            "  0:00:00.135     8M / 1G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
            "  0:00:00.141     8M / 1G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
            "  0:00:00.141     8M / 1G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 1960\n",
            "  0:00:00.142     8M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
            "  0:00:00.142     8M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /usr/share/spades/test_dataset/ecoli_1K_1.fq.gz\n",
            "  0:00:00.274     8M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /usr/share/spades/test_dataset/ecoli_1K_2.fq.gz\n",
            "  0:00:00.395     8M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
            "  0:00:00.396     8M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 402)   There are 1974 kmers in total. Among them 0 (0%) are singletons.\n",
            "  0:00:00.396     8M / 1G    INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
            "  0:00:00.396     8M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 0 non-read kmers were generated.\n",
            "  0:00:00.397     8M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
            "  0:00:00.397     8M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 1946. Among them 1946 (100%) are good\n",
            "  0:00:00.397     8M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 8. Among them 8 (100%) are good\n",
            "  0:00:00.397     8M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 10. Among them 10 (100%) are good\n",
            "  0:00:00.397     8M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2.8 kmers\n",
            "  0:00:00.397     8M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.28571\n",
            "  0:00:00.397     8M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 1964\n",
            "  0:00:00.397     8M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.875,0.125,0,0),(0,1,0,0),(0,0,1,0),(0,0,0.125,0.875))\n",
            "  0:00:00.397     8M / 1G    INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
            "  0:00:00.397     8M / 1G    INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 2 threads.\n",
            "  0:00:00.455     8M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 0 new k-mers.\n",
            "  0:00:00.455     8M / 1G    INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
            "  0:00:00.455     8M / 1G    INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 2 threads.\n",
            "  0:00:00.455     8M / 1G    INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /usr/share/spades/test_dataset/ecoli_1K_1.fq.gz and /usr/share/spades/test_dataset/ecoli_1K_2.fq.gz\n",
            "  0:00:00.496    56M / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 2054 reads.\n",
            "  0:00:00.517    56M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
            "  0:00:00.521    56M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
            "  0:00:00.533     8M / 1G    INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 4 bases in 4 reads.\n",
            "  0:00:00.533     8M / 1G    INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 353915.\n",
            "  0:00:00.533     8M / 1G    INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /content/spades_test/corrected/corrected.yaml\n",
            "  0:00:00.534     8M / 1G    INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
            "\n",
            "== Compressing corrected reads (with gzip)\n",
            "\n",
            "== Dataset description file was created: /content/spades_test/corrected/corrected.yaml\n",
            "\n",
            "\n",
            "===== Read error correction finished. \n",
            "\n",
            "\n",
            "===== Assembling started.\n",
            "\n",
            "\n",
            "== Running assembler: K21\n",
            "\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /content/spades_test/K21/configs/config.info\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 12 Gb\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  84)   Starting SPAdes, built from N/A, git revision N/A\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/content/spades_test/dataset.info) with K=21\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  52)   SPAdes started\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  59)   Starting from stage: construction\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  66)   Two-step RR enabled: 0\n",
            "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
            "  0:00:00.124    68M / 100M  INFO    General                 (binary_converter.hpp      : 117)   2054 reads written\n",
            "  0:00:00.129     4M / 100M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
            "  0:00:00.321   132M / 196M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
            "  0:00:00.329     4M / 196M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
            "  0:00:00.516   132M / 196M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
            "  0:00:00.523     4M / 196M  INFO    General                 (construction.cpp          : 111)   Max read length 100\n",
            "  0:00:00.524     4M / 196M  INFO    General                 (construction.cpp          : 117)   Average read length 86.1526\n",
            "  0:00:00.524     4M / 196M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
            "  0:00:00.524     4M / 196M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 4 files using 2 threads. This might take a while.\n",
            "  0:00:00.524     4M / 196M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:00.524     4M / 196M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.99935 Gb\n",
            "  0:00:00.524     4M / 196M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 16777216\n",
            "  0:00:00.556     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
            "  0:00:00.563     8M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 8216 reads\n",
            "  0:00:00.563     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:00:00.564     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 984 kmers in total.\n",
            "  0:00:00.570     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:00:00.571     8M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
            "  0:00:00.571     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:00:00.571     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 32 files using 2 threads. This might take a while.\n",
            "  0:00:00.571     8M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:00.571     8M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.9987 Gb\n",
            "  0:00:00.572     8M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2097152\n",
            "  0:00:00.581     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 984 kmers.\n",
            "  0:00:00.582     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:00:00.587     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 985 kmers in total.\n",
            "  0:00:00.587     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:00:00.590     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:00:00.620     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
            "  0:00:00.622     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 9776 bytes occupied (79.399 bits per kmer).\n",
            "  0:00:00.622     8M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
            "  0:00:00.622     8M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
            "  0:00:00.623     8M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Early tip clipping\n",
            "  0:00:00.623     8M / 1G    INFO    General                 (construction.cpp          : 253)   Early tip clipper length bound set as (RL - K)\n",
            "  0:00:00.623     8M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
            "  0:00:00.624     8M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 184)   5 22-mers were removed by early tip clipper\n",
            "  0:00:00.624     8M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
            "  0:00:00.624     8M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
            "  0:00:00.625     8M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 1 sequences extracted\n",
            "  0:00:00.625     8M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
            "  0:00:00.625     8M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 0 loops collected\n",
            "  0:00:00.626     8M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
            "  0:00:00.626     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:00:00.626     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:00:00.629     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1584 bytes occupied (12.878 bits per kmer).\n",
            "  0:00:00.629     8M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
            "  0:00:00.657     8M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
            "  0:00:00.658     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
            "  0:00:00.658     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 0\n",
            "  0:00:00.658     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 348\n",
            "  0:00:00.658     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 348. Coverage mad: 62.2692\n",
            "  0:00:00.658     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
            "  0:00:00.722     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
            "  0:00:00.860     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
            "  0:00:01.065     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
            "  0:00:01.335     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
            "  0:00:01.803     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 32\n",
            "  0:00:02.987     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 342.845. Fitted coverage std. dev: 68.3997\n",
            "  0:00:02.988     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.999994\n",
            "  0:00:02.988     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 355\n",
            "  0:00:02.988     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 171\n",
            "  0:00:02.988     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 749\n",
            "  0:00:02.988     8M / 1G    INFO    General                 (genomic_info_filler.cpp   : 112)   Mean coverage was calculated as 342.845\n",
            "  0:00:02.988     8M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 171\n",
            "  0:00:02.988     8M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
            "  0:00:02.988     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
            "  0:00:02.988     8M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
            "  0:00:02.988     8M / 1G    INFO    General                 (graph_simplification.hpp  : 669)   Flanking coverage based disconnection disabled\n",
            "  0:00:02.988     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial tip clipper\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial tip clipper triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial ec remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial ec remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial isolated edge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial isolated edge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (graph_simplification.hpp  : 641)   Creating parallel br instance\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (graph_simplification.hpp  : 458)   Disconnection of relatively low covered edges disabled\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (graph_simplification.hpp  : 494)   Complex tip clipping disabled\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (graph_simplification.hpp  : 641)   Creating parallel br instance\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
            "  0:00:02.989     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 0 times\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 273.35\n",
            "  0:00:02.989     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
            "  0:00:02.989     8M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/spades_test//K21/simplified_contigs.fasta\n",
            "  0:00:02.990     8M / 1G    INFO    General                 (launch.hpp                : 151)   SPAdes finished\n",
            "  0:00:02.990     8M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 2 seconds\n",
            "Max read length detected as 100\n",
            "\n",
            "== Running assembler: K33\n",
            "\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /content/spades_test/K33/configs/config.info\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 12 Gb\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  84)   Starting SPAdes, built from N/A, git revision N/A\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/content/spades_test/dataset.info) with K=33\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  52)   SPAdes started\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  59)   Starting from stage: construction\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  66)   Two-step RR enabled: 0\n",
            "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  59)   Binary reads detected\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (construction.cpp          : 111)   Max read length 100\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (construction.cpp          : 117)   Average read length 86.1526\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 4 files using 2 threads. This might take a while.\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.99935 Gb\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 8388608\n",
            "  0:00:00.031     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
            "  0:00:00.032     8M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 8216 reads\n",
            "  0:00:00.032     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:00:00.033     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 972 kmers in total.\n",
            "  0:00:00.033     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:00:00.033     8M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
            "  0:00:00.033     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:00:00.033     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 32 files using 2 threads. This might take a while.\n",
            "  0:00:00.034     8M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:00.034     8M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.9987 Gb\n",
            "  0:00:00.034     8M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 1048576\n",
            "  0:00:00.037     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 972 kmers.\n",
            "  0:00:00.038     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:00:00.042     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 973 kmers in total.\n",
            "  0:00:00.042     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:00:00.045     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:00:00.076     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
            "  0:00:00.077     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 9776 bytes occupied (80.3782 bits per kmer).\n",
            "  0:00:00.077     8M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
            "  0:00:00.078     8M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
            "  0:00:00.078     8M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Early tip clipping\n",
            "  0:00:00.078     8M / 1G    INFO    General                 (construction.cpp          : 253)   Early tip clipper length bound set as (RL - K)\n",
            "  0:00:00.078     8M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
            "  0:00:00.078     8M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 184)   5 34-mers were removed by early tip clipper\n",
            "  0:00:00.078     8M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
            "  0:00:00.078     8M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
            "  0:00:00.079     8M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 1 sequences extracted\n",
            "  0:00:00.079     8M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
            "  0:00:00.079     8M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 0 loops collected\n",
            "  0:00:00.079     8M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
            "  0:00:00.079     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:00:00.079     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:00:00.083     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1592 bytes occupied (13.1029 bits per kmer).\n",
            "  0:00:00.083     8M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
            "  0:00:00.106     8M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
            "  0:00:00.107     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
            "  0:00:00.107     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 0\n",
            "  0:00:00.107     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 17\n",
            "  0:00:00.107     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 17. Coverage mad: 11.8608\n",
            "  0:00:00.107     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
            "  0:00:00.139     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
            "  0:00:00.187     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
            "  0:00:00.252     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
            "  0:00:00.335     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
            "  0:00:00.485     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 32\n",
            "  0:00:00.784     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 64\n",
            "  0:00:01.407     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 128\n",
            "  0:00:02.162     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 108.181. Fitted coverage std. dev: 65.0594\n",
            "  0:00:02.163     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.0555142\n",
            "  0:00:02.163     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 3\n",
            "  0:00:02.163     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 3\n",
            "  0:00:02.163     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 400\n",
            "  0:00:02.163     8M / 1G    INFO    General                 (genomic_info_filler.cpp   : 112)   Mean coverage was calculated as 108.181\n",
            "  0:00:02.163     8M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 3\n",
            "  0:00:02.163     8M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
            "  0:00:02.163     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
            "  0:00:02.163     8M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
            "  0:00:02.163     8M / 1G    INFO    General                 (graph_simplification.hpp  : 669)   Flanking coverage based disconnection disabled\n",
            "  0:00:02.163     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
            "  0:00:02.163     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
            "  0:00:02.163     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial tip clipper\n",
            "  0:00:02.163     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial tip clipper triggered 0 times\n",
            "  0:00:02.164     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial ec remover\n",
            "  0:00:02.164     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial ec remover triggered 0 times\n",
            "  0:00:02.164     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial isolated edge remover\n",
            "  0:00:02.164     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial isolated edge remover triggered 0 times\n",
            "  0:00:02.164     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
            "  0:00:02.164     8M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
            "  0:00:02.164     8M / 1G    INFO    General                 (graph_simplification.hpp  : 641)   Creating parallel br instance\n",
            "  0:00:02.164     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
            "  0:00:02.164     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.164     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.164     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.164     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.164     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.164     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.164     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
            "  0:00:02.165     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.165     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.165     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.165     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.165     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.165     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.165     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
            "  0:00:02.165     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.165     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.165     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.165     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.165     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.165     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.165     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
            "  0:00:02.165     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.165     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.165     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.166     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.166     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.166     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.166     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
            "  0:00:02.166     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.166     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.166     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.166     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.166     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.166     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.166     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
            "  0:00:02.166     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.166     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.166     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.166     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.166     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.166     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.166     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
            "  0:00:02.167     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.167     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.167     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.167     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.167     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.167     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.167     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
            "  0:00:02.167     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.167     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.167     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.167     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.167     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.167     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.167     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
            "  0:00:02.167     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.167     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.167     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.168     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.168     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.168     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.168     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
            "  0:00:02.168     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.168     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.168     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.168     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.168     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.168     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.168     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
            "  0:00:02.168     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.174     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.174     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.174     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.174     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:02.174     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:02.175     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
            "  0:00:02.175     8M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
            "  0:00:02.175     8M / 1G    INFO    General                 (graph_simplification.hpp  : 458)   Disconnection of relatively low covered edges disabled\n",
            "  0:00:02.175     8M / 1G    INFO    General                 (graph_simplification.hpp  : 494)   Complex tip clipping disabled\n",
            "  0:00:02.175     8M / 1G    INFO    General                 (graph_simplification.hpp  : 641)   Creating parallel br instance\n",
            "  0:00:02.175     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.175     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.175     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.175     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.175     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:02.175     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:02.175     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:02.175     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:02.175     8M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
            "  0:00:02.175     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
            "  0:00:02.175     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 0 times\n",
            "  0:00:02.176     8M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
            "  0:00:02.176     8M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 225.839\n",
            "  0:00:02.176     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
            "  0:00:02.176     8M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/spades_test//K33/simplified_contigs.fasta\n",
            "  0:00:02.176     8M / 1G    INFO    General                 (launch.hpp                : 151)   SPAdes finished\n",
            "  0:00:02.176     8M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 2 seconds\n",
            "\n",
            "== Running assembler: K55\n",
            "\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /content/spades_test/K55/configs/config.info\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 12 Gb\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  84)   Starting SPAdes, built from N/A, git revision N/A\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/content/spades_test/dataset.info) with K=55\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  52)   SPAdes started\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  59)   Starting from stage: construction\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  66)   Two-step RR enabled: 0\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  77)   Will need read mapping, kmer mapper will be attached\n",
            "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  59)   Binary reads detected\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (construction.cpp          : 111)   Max read length 100\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (construction.cpp          : 117)   Average read length 86.1526\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 4 files using 2 threads. This might take a while.\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.99935 Gb\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 8388608\n",
            "  0:00:00.021     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
            "  0:00:00.022     8M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 8216 reads\n",
            "  0:00:00.022     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:00:00.023     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 949 kmers in total.\n",
            "  0:00:00.023     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:00:00.023     8M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
            "  0:00:00.023     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:00:00.024     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 32 files using 2 threads. This might take a while.\n",
            "  0:00:00.024     8M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:00.024     8M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.9987 Gb\n",
            "  0:00:00.024     8M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 1048576\n",
            "  0:00:00.027     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 949 kmers.\n",
            "  0:00:00.028     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:00:00.032     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 951 kmers in total.\n",
            "  0:00:00.032     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:00:00.035     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:00:00.061     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
            "  0:00:00.062     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 9768 bytes occupied (82.1703 bits per kmer).\n",
            "  0:00:00.062     8M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
            "  0:00:00.063     8M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
            "  0:00:00.063     8M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
            "  0:00:00.063     8M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
            "  0:00:00.063     8M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 4 sequences extracted\n",
            "  0:00:00.063     8M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
            "  0:00:00.063     8M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 0 loops collected\n",
            "  0:00:00.063     8M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
            "  0:00:00.063     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:00:00.063     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:00:00.067     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1568 bytes occupied (13.2181 bits per kmer).\n",
            "  0:00:00.068     8M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
            "  0:00:00.084     8M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
            "  0:00:00.085     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
            "  0:00:00.085     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 0\n",
            "  0:00:00.085     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 6\n",
            "  0:00:00.085     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 6. Coverage mad: 4.4478\n",
            "  0:00:00.085     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
            "  0:00:00.092     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
            "  0:00:00.110     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
            "  0:00:00.166     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
            "  0:00:00.251     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
            "  0:00:00.320     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 13.7375. Fitted coverage std. dev: 9.60863\n",
            "  0:00:00.321     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 2.02732e-07\n",
            "  0:00:00.321     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 1\n",
            "  0:00:00.321     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 1\n",
            "  0:00:00.321     8M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 238\n",
            "  0:00:00.321     8M / 1G    INFO    General                 (genomic_info_filler.cpp   : 112)   Mean coverage was calculated as 13.7375\n",
            "  0:00:00.321     8M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 1\n",
            "  0:00:00.321     8M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
            "  0:00:00.321     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
            "  0:00:00.321     8M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
            "  0:00:00.321     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:00:00.321     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 32 files using 2 threads. This might take a while.\n",
            "  0:00:00.321     8M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:00.321     8M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.9987 Gb\n",
            "  0:00:00.321     8M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 1048576\n",
            "  0:00:00.325     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 8 sequences.\n",
            "  0:00:00.326     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:00:00.331     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 949 kmers in total.\n",
            "  0:00:00.331     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:00:00.333     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:00:00.360     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
            "  0:00:00.361     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 9752 bytes occupied (82.2086 bits per kmer).\n",
            "  0:00:00.361     8M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
            "  0:00:00.361     8M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
            "  0:00:00.361     8M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
            "  0:00:00.361     8M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
            "  0:00:00.368     8M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 2054 paired reads\n",
            "  0:00:00.368     8M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
            "  0:00:00.368     8M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
            "  0:00:00.368     8M / 1G    INFO   GapCloser                (gap_closer.cpp            : 378)   Closing short gaps complete: filled 0 gaps after checking 4 candidates\n",
            "  0:00:00.368     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
            "  0:00:00.369     8M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
            "  0:00:00.369     8M / 1G    INFO    General                 (graph_simplification.hpp  : 669)   Flanking coverage based disconnection disabled\n",
            "  0:00:00.369     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
            "  0:00:00.369     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
            "  0:00:00.369     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial tip clipper\n",
            "  0:00:00.369     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial tip clipper triggered 0 times\n",
            "  0:00:00.369     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial ec remover\n",
            "  0:00:00.369     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial ec remover triggered 0 times\n",
            "  0:00:00.369     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial isolated edge remover\n",
            "  0:00:00.369     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial isolated edge remover triggered 1 times\n",
            "  0:00:00.369     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
            "  0:00:00.369     8M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
            "  0:00:00.369     8M / 1G    INFO    General                 (graph_simplification.hpp  : 641)   Creating parallel br instance\n",
            "  0:00:00.369     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
            "  0:00:00.369     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:00.369     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.370     8M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
            "  0:00:00.370     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:00.373     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:00.373     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:00.373     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:00.373     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:00:00.373     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:00:00.373     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
            "  0:00:00.373     8M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
            "  0:00:00.373     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:00:00.373     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 32 files using 2 threads. This might take a while.\n",
            "  0:00:00.373     8M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:00.373     8M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.9987 Gb\n",
            "  0:00:00.373     8M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 1048576\n",
            "  0:00:00.378     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 2 sequences.\n",
            "  0:00:00.379     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:00:00.392     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 945 kmers in total.\n",
            "  0:00:00.392     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:00:00.396     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:00:00.425     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
            "  0:00:00.426     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 9752 bytes occupied (82.5566 bits per kmer).\n",
            "  0:00:00.426     8M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
            "  0:00:00.426     8M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
            "  0:00:00.426     8M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
            "  0:00:00.426     8M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
            "  0:00:00.439     8M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 2054 paired reads\n",
            "  0:00:00.439     8M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
            "  0:00:00.439     8M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
            "  0:00:00.439     8M / 1G    INFO   GapCloser                (gap_closer.cpp            : 378)   Closing short gaps complete: filled 0 gaps after checking 0 candidates\n",
            "  0:00:00.440     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
            "  0:00:00.440     8M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
            "  0:00:00.440     8M / 1G    INFO    General                 (graph_simplification.hpp  : 458)   Disconnection of relatively low covered edges disabled\n",
            "  0:00:00.440     8M / 1G    INFO    General                 (graph_simplification.hpp  : 494)   Complex tip clipping disabled\n",
            "  0:00:00.440     8M / 1G    INFO    General                 (graph_simplification.hpp  : 641)   Creating parallel br instance\n",
            "  0:00:00.440     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:00.440     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:00.440     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:00.440     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:00.440     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:00:00.443     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:00:00.443     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:00:00.444     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:00:00.444     8M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
            "  0:00:00.444     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
            "  0:00:00.444     8M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 0 times\n",
            "  0:00:00.444     8M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
            "  0:00:00.444     8M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 140.62\n",
            "  0:00:00.444     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
            "  0:00:00.444     8M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 0 kmers to process\n",
            "  0:00:00.444     8M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
            "  0:00:00.471     8M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
            "  0:00:00.471     8M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
            "  0:00:00.471     8M / 1G    INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 0 nucleotides\n",
            "  0:00:00.471     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
            "  0:00:00.471     8M / 1G    INFO    General                 (contig_output_stage.cpp   :  45)   Writing GFA to /content/spades_test//K55/assembly_graph_with_scaffolds.gfa\n",
            "  0:00:00.471     8M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/spades_test//K55/before_rr.fasta\n",
            "  0:00:00.471     8M / 1G    INFO    General                 (contig_output_stage.cpp   :  56)   Outputting FastG graph to /content/spades_test//K55/assembly_graph.fastg\n",
            "  0:00:00.472     8M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/spades_test//K55/final_contigs.fasta\n",
            "  0:00:00.472     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
            "  0:00:00.472     8M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 0 kmers to process\n",
            "  0:00:00.472     8M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
            "  0:00:00.472     8M / 1G    INFO    General                 (pair_info_count.cpp       : 323)   Min edge length for estimation: 945\n",
            "  0:00:00.472     8M / 1G    INFO    General                 (pair_info_count.cpp       : 334)   Estimating insert size for library #0\n",
            "  0:00:00.472     8M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
            "  0:00:00.508    56M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
            "  0:00:00.578    56M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 2054 reads processed\n",
            "  0:00:00.644    56M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
            "  0:00:00.645    56M / 1G    INFO    General                 (pair_info_count.cpp       : 211)   1636 paired reads (79.6495% of all) aligned to long edges\n",
            "  0:00:00.648     8M / 1G    INFO    General                 (pair_info_count.cpp       : 353)     Insert size = 214.696, deviation = 10.4821, left quantile = 201, right quantile = 228, read length = 100\n",
            "  0:00:00.787   200M / 1G    INFO    General                 (pair_info_count.cpp       : 374)   Filtering data for library #0\n",
            "  0:00:00.787   200M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
            "  0:00:00.794   200M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 2054 reads processed\n",
            "  0:00:00.795   200M / 1G    INFO    General                 (pair_info_count.cpp       : 386)   Mapping library #0\n",
            "  0:00:00.795   200M / 1G    INFO    General                 (pair_info_count.cpp       : 388)   Mapping paired reads (takes a while)\n",
            "  0:00:00.795   200M / 1G    INFO    General                 (pair_info_count.cpp       : 286)   Left insert size quantile 201, right insert size quantile 228, filtering threshold 2, rounding threshold 0\n",
            "  0:00:00.806   212M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
            "  0:00:00.817   212M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 2054 reads processed\n",
            "  0:00:00.837     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
            "  0:00:00.837     8M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
            "  0:00:00.837     8M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
            "  0:00:00.837     8M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
            "  0:00:00.838     8M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
            "  0:00:00.838     8M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 2\n",
            "  0:00:00.838     8M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
            "  0:00:00.838     8M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
            "  0:00:00.838     8M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
            "  0:00:00.838     8M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
            "  0:00:00.838     8M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 102)   Paired info stats: missing = 0; contradictional = 0\n",
            "  0:00:00.838     8M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 102)   Paired info stats: missing = 0; contradictional = 0\n",
            "  0:00:00.839     8M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
            "  0:00:00.839     8M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
            "  0:00:00.839     8M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
            "  0:00:00.839     8M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 2\n",
            "  0:00:00.839     8M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
            "  0:00:00.839     8M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
            "  0:00:00.839     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\n",
            "  0:00:00.839     8M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
            "  0:00:00.839     8M / 1G    INFO    General                 (launcher.cpp              : 481)   ExSPAnder repeat resolving tool started\n",
            "  0:00:00.839     8M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
            "  0:00:00.840     8M / 1G    INFO    General                 (extenders_logic.cpp       : 278)   Estimated coverage of library #0 is 140.62\n",
            "  0:00:00.840     8M / 1G    INFO    General                 (extenders_logic.cpp       : 278)   Estimated coverage of library #0 is 140.62\n",
            "  0:00:00.840     8M / 1G    INFO    General                 (extenders_logic.cpp       : 475)   Using 1 paired-end library\n",
            "  0:00:00.840     8M / 1G    INFO    General                 (extenders_logic.cpp       : 476)   Using 1 paired-end scaffolding library\n",
            "  0:00:00.840     8M / 1G    INFO    General                 (extenders_logic.cpp       : 477)   Using 0 single read libraries\n",
            "  0:00:00.840     8M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 3\n",
            "  0:00:00.840     8M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
            "  0:00:00.840     8M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
            "  0:00:00.840     8M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
            "  0:00:00.840     8M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
            "  0:00:00.840     8M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
            "  0:00:00.840     8M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
            "  0:00:00.840     8M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
            "  0:00:00.841     8M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
            "  0:00:00.841     8M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
            "  0:00:00.841     8M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
            "  0:00:00.841     8M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
            "  0:00:00.841     8M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
            "  0:00:00.841     8M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
            "  0:00:00.841     8M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
            "  0:00:00.841     8M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
            "  0:00:00.841     8M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 0 loops\n",
            "  0:00:00.841     8M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
            "  0:00:00.841     8M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
            "  0:00:00.841     8M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
            "  0:00:00.841     8M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
            "  0:00:00.841     8M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
            "  0:00:00.841     8M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
            "  0:00:00.841     8M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
            "  0:00:00.841     8M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
            "  0:00:00.842     8M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
            "  0:00:00.842     8M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
            "  0:00:00.842     8M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
            "  0:00:00.842     8M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
            "  0:00:00.842     8M / 1G    INFO    General                 (launcher.cpp              : 534)   ExSPAnder repeat resolving tool finished\n",
            "  0:00:00.842     8M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
            "  0:00:00.842     8M / 1G    INFO    General                 (contig_output_stage.cpp   :  45)   Writing GFA to /content/spades_test//K55/assembly_graph_with_scaffolds.gfa\n",
            "  0:00:00.842     8M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/spades_test//K55/before_rr.fasta\n",
            "  0:00:00.842     8M / 1G    INFO    General                 (contig_output_stage.cpp   :  56)   Outputting FastG graph to /content/spades_test//K55/assembly_graph.fastg\n",
            "  0:00:00.843     8M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /content/spades_test//K55/final_contigs.paths\n",
            "  0:00:00.843     8M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /content/spades_test//K55/scaffolds.paths\n",
            "  0:00:00.843     8M / 1G    INFO    General                 (launch.hpp                : 151)   SPAdes finished\n",
            "  0:00:00.844     8M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 0 seconds\n",
            "\n",
            "===== Assembling finished. Used k-mer sizes: 21, 33, 55 \n",
            "\n",
            " * Corrected reads are in /content/spades_test/corrected/\n",
            " * Assembled contigs are in /content/spades_test/contigs.fasta\n",
            " * Assembled scaffolds are in /content/spades_test/scaffolds.fasta\n",
            " * Paths in the assembly graph corresponding to the contigs are in /content/spades_test/contigs.paths\n",
            " * Paths in the assembly graph corresponding to the scaffolds are in /content/spades_test/scaffolds.paths\n",
            " * Assembly graph is in /content/spades_test/assembly_graph.fastg\n",
            " * Assembly graph in GFA format is in /content/spades_test/assembly_graph_with_scaffolds.gfa\n",
            "\n",
            "======= SPAdes pipeline finished.\n",
            "\n",
            "========= TEST PASSED CORRECTLY.\n",
            "\n",
            "SPAdes log can be found here: /content/spades_test/spades.log\n",
            "\n",
            "Thank you for using SPAdes!\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!spades -1 SRR292678_1.fastq -2 SRR292678_2.fastq -o output"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "-js_Ijqyl19V",
        "outputId": "c797bc08-4bab-41d5-9e1a-da52f0338f76"
      },
      "execution_count": 26,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Command line: /usr/lib/spades/bin/spades.py\t-1\t/content/SRR292678_1.fastq\t-2\t/content/SRR292678_2.fastq\t-o\t/content/output\t\n",
            "\n",
            "System information:\n",
            "  SPAdes version: 3.13.1\n",
            "  Python version: 3.8.10\n",
            "  OS: Linux-5.10.147+-x86_64-with-glibc2.29\n",
            "\n",
            "Output dir: /content/output\n",
            "Mode: read error correction and assembling\n",
            "Debug mode is turned OFF\n",
            "\n",
            "Dataset parameters:\n",
            "  Multi-cell mode (you should set '--sc' flag if input data was obtained with MDA (single-cell) technology or --meta flag if processing metagenomic dataset)\n",
            "  Reads:\n",
            "    Library number: 1, library type: paired-end\n",
            "      orientation: fr\n",
            "      left reads: ['/content/SRR292678_1.fastq']\n",
            "      right reads: ['/content/SRR292678_2.fastq']\n",
            "      interlaced reads: not specified\n",
            "      single reads: not specified\n",
            "      merged reads: not specified\n",
            "Read error correction parameters:\n",
            "  Iterations: 1\n",
            "  PHRED offset will be auto-detected\n",
            "  Corrected reads will be compressed\n",
            "Assembly parameters:\n",
            "  k: automatic selection based on read length\n",
            "  Repeat resolution is enabled\n",
            "  Mismatch careful mode is turned OFF\n",
            "  MismatchCorrector will be SKIPPED\n",
            "  Coverage cutoff is turned OFF\n",
            "Other parameters:\n",
            "  Dir for temp files: /content/output/tmp\n",
            "  Threads: 16\n",
            "  Memory limit (in Gb): 12\n",
            "\n",
            "\n",
            "======= SPAdes pipeline started. Log can be found here: /content/output/spades.log\n",
            "\n",
            "\n",
            "===== Read error correction started. \n",
            "\n",
            "\n",
            "== Running read error correction tool: /usr/lib/spades/bin/spades-hammer /content/output/corrected/configs/config.info\n",
            "\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from N/A, git revision N/A\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /content/output/corrected/configs/config.info\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 12 Gb\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
            "     === ITERATION 0 begins ===\n",
            "  0:00:00.001     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:00:00.001     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 32 files using 2 threads. This might take a while.\n",
            "  0:00:00.001     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.99935 Gb\n",
            "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2097152\n",
            "  0:00:00.001     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /content/SRR292678_1.fastq\n",
            "  0:00:18.385     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 476436 reads\n",
            "  0:00:38.170     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 952951 reads\n",
            "  0:00:56.577     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1429685 reads\n",
            "  0:01:14.532     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1906716 reads\n",
            "  0:01:34.004     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2383922 reads\n",
            "  0:01:57.257     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2860589 reads\n",
            "  0:02:16.787     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 3336852 reads\n",
            "  0:02:55.110     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 4289500 reads\n",
            "  0:04:08.362     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /content/SRR292678_2.fastq\n",
            "  0:05:44.870     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 8495954 reads\n",
            "  0:08:13.973     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 12222224 reads processed\n",
            "  0:08:14.100     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:08:39.590     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 38444962 kmers in total.\n",
            "  0:08:39.590     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:08:40.281     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:08:49.203    24M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
            "  0:08:49.895    24M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 17834992 bytes occupied (3.71128 bits per kmer).\n",
            "  0:08:49.896    24M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
            "  0:08:54.905   616M / 1G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
            "  0:18:32.157   616M / 1G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
            "  0:19:26.805   616M / 1G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 13311677\n",
            "  0:19:26.826   320M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
            "  0:19:27.488     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /content/SRR292678_1.fastq\n",
            "  0:43:59.937     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /content/SRR292678_2.fastq\n",
            "  1:06:23.650     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
            "  1:06:23.863     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 402)   There are 38444962 kmers in total. Among them 25446372 (66.1891%) are singletons.\n",
            "  1:06:23.863     1G / 1G    INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
            "  1:07:17.250     1G / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 8 non-read kmers were generated.\n",
            "  1:07:17.250     1G / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
            "  1:07:17.250     1G / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 4372702. Among them 2137886 (48.8916%) are good\n",
            "  1:07:17.250     1G / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 38483. Among them 38044 (98.8592%) are good\n",
            "  1:07:17.250     1G / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 9075523. Among them 7721233 (85.0776%) are good\n",
            "  1:07:17.250     1G / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 3.7543 kmers\n",
            "  1:07:17.250     1G / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.01958\n",
            "  1:07:17.250     1G / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 9897163\n",
            "  1:07:17.250     1G / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.929505,0.0466435,0.0183443,0.00550736),(0.0177785,0.970831,0.00429558,0.00709526),(0.00707197,0.00428477,0.970846,0.0177969),(0.00550123,0.0183173,0.0466262,0.929555))\n",
            "  1:07:17.272     1G / 1G    INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
            "  1:07:17.272     1G / 1G    INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 2 threads.\n",
            "  1:24:32.242     1G / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 59820 new k-mers.\n",
            "  1:40:58.814     1G / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 1338 new k-mers.\n",
            "  1:54:13.519     1G / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 30 new k-mers.\n",
            "  2:08:41.832     1G / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 3 produced 0 new k-mers.\n",
            "  2:08:41.833     1G / 1G    INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
            "  2:08:41.833     1G / 1G    INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 2 threads.\n",
            "  2:08:41.833     1G / 1G    INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /content/SRR292678_1.fastq and /content/SRR292678_2.fastq\n",
            "  2:08:42.839     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 200000 reads.\n",
            "  2:08:56.338     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
            "  2:08:56.883     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
            "  2:08:57.763     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 1 of 200000 reads.\n",
            "  2:09:12.847     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 1\n",
            "  2:09:13.480     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 1\n",
            "  2:09:14.856     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 2 of 200000 reads.\n",
            "  2:09:30.591     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 2\n",
            "  2:09:31.084     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 2\n",
            "  2:09:31.975     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 3 of 200000 reads.\n",
            "  2:09:46.642     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 3\n",
            "  2:09:47.174     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 3\n",
            "  2:09:47.890     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 4 of 200000 reads.\n",
            "  2:10:02.357     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 4\n",
            "  2:10:02.896     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 4\n",
            "  2:10:03.564     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 5 of 200000 reads.\n",
            "  2:10:17.703     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 5\n",
            "  2:10:18.286     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 5\n",
            "  2:10:18.853     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 6 of 200000 reads.\n",
            "  2:10:35.196     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 6\n",
            "  2:10:36.161     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 6\n",
            "  2:10:36.964     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 7 of 200000 reads.\n",
            "  2:10:53.691     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 7\n",
            "  2:10:54.272     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 7\n",
            "  2:10:54.778     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 8 of 200000 reads.\n",
            "  2:11:10.553     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 8\n",
            "  2:11:10.994     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 8\n",
            "  2:11:11.399     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 9 of 200000 reads.\n",
            "  2:11:24.508     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 9\n",
            "  2:11:24.985     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 9\n",
            "  2:11:25.389     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 10 of 200000 reads.\n",
            "  2:11:38.456     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 10\n",
            "  2:11:38.989     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 10\n",
            "  2:11:39.465     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 11 of 200000 reads.\n",
            "  2:11:54.933     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 11\n",
            "  2:11:55.545     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 11\n",
            "  2:11:55.962     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 12 of 200000 reads.\n",
            "  2:12:12.086     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 12\n",
            "  2:12:12.991     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 12\n",
            "  2:12:13.658     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 13 of 200000 reads.\n",
            "  2:12:28.825     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 13\n",
            "  2:12:29.330     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 13\n",
            "  2:12:29.740     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 14 of 200000 reads.\n",
            "  2:12:45.228     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 14\n",
            "  2:12:45.705     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 14\n",
            "  2:12:46.100     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 15 of 200000 reads.\n",
            "  2:13:01.060     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 15\n",
            "  2:13:01.635     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 15\n",
            "  2:13:02.065     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 16 of 200000 reads.\n",
            "  2:13:16.959     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 16\n",
            "  2:13:17.462     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 16\n",
            "  2:13:17.993     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 17 of 200000 reads.\n",
            "  2:13:33.971     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 17\n",
            "  2:13:34.735     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 17\n",
            "  2:13:35.216     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 18 of 200000 reads.\n",
            "  2:13:49.876     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 18\n",
            "  2:13:51.345     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 18\n",
            "  2:13:52.673     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 19 of 200000 reads.\n",
            "  2:14:09.309     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 19\n",
            "  2:14:09.831     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 19\n",
            "  2:14:10.248     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 20 of 200000 reads.\n",
            "  2:14:28.703     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 20\n",
            "  2:14:29.276     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 20\n",
            "  2:14:29.687     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 21 of 200000 reads.\n",
            "  2:14:45.352     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 21\n",
            "  2:14:45.909     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 21\n",
            "  2:14:46.325     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 22 of 200000 reads.\n",
            "  2:14:59.777     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 22\n",
            "  2:15:00.258     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 22\n",
            "  2:15:00.679     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 23 of 200000 reads.\n",
            "  2:15:16.178     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 23\n",
            "  2:15:16.714     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 23\n",
            "  2:15:17.113     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 24 of 200000 reads.\n",
            "  2:15:32.152     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 24\n",
            "  2:15:33.140     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 24\n",
            "  2:15:33.817     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 25 of 200000 reads.\n",
            "  2:15:48.974     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 25\n",
            "  2:15:49.510     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 25\n",
            "  2:15:49.923     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 26 of 200000 reads.\n",
            "  2:16:04.173     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 26\n",
            "  2:16:04.722     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 26\n",
            "  2:16:05.203     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 27 of 200000 reads.\n",
            "  2:16:19.427     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 27\n",
            "  2:16:19.913     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 27\n",
            "  2:16:20.331     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 28 of 200000 reads.\n",
            "  2:16:34.930     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 28\n",
            "  2:16:35.417     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 28\n",
            "  2:16:35.831     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 29 of 200000 reads.\n",
            "  2:16:50.413     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 29\n",
            "  2:16:50.925     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 29\n",
            "  2:16:51.254     1G / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 30 of 111112 reads.\n",
            "  2:16:59.884     1G / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 30\n",
            "  2:17:00.166     1G / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 30\n",
            "  2:17:00.279     1G / 1G    INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 1037990 bases in 874856 reads.\n",
            "  2:17:00.280     1G / 1G    INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 1099999338.\n",
            "  2:17:00.385     8M / 1G    INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /content/output/corrected/corrected.yaml\n",
            "  2:17:00.388     8M / 1G    INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
            "\n",
            "== Compressing corrected reads (with gzip)\n",
            "\n",
            "== Dataset description file was created: /content/output/corrected/corrected.yaml\n",
            "\n",
            "\n",
            "===== Read error correction finished. \n",
            "\n",
            "\n",
            "===== Assembling started.\n",
            "\n",
            "\n",
            "== Running assembler: K21\n",
            "\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /content/output/K21/configs/config.info\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 12 Gb\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  84)   Starting SPAdes, built from N/A, git revision N/A\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/content/output/dataset.info) with K=21\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.001     4M / 4M    INFO    General                 (launch.hpp                :  52)   SPAdes started\n",
            "  0:00:00.001     4M / 4M    INFO    General                 (launch.hpp                :  59)   Starting from stage: construction\n",
            "  0:00:00.001     4M / 4M    INFO    General                 (launch.hpp                :  66)   Two-step RR enabled: 0\n",
            "  0:00:00.001     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
            "  0:00:00.008     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
            "  0:00:00.008     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
            "  0:00:00.228    76M / 100M  INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
            "  0:00:00.363    84M / 100M  INFO    General                 (binary_converter.hpp      :  93)   32768 reads processed\n",
            "  0:00:00.640   100M / 100M  INFO    General                 (binary_converter.hpp      :  93)   65536 reads processed\n",
            "  0:00:01.421   132M / 132M  INFO    General                 (binary_converter.hpp      :  93)   131072 reads processed\n",
            "  0:00:03.133   200M / 200M  INFO    General                 (binary_converter.hpp      :  93)   262144 reads processed\n",
            "  0:00:06.460   240M / 240M  INFO    General                 (binary_converter.hpp      :  93)   524288 reads processed\n",
            "  0:00:12.299   252M / 252M  INFO    General                 (binary_converter.hpp      :  93)   1048576 reads processed\n",
            "  0:00:24.450   268M / 268M  INFO    General                 (binary_converter.hpp      :  93)   2097152 reads processed\n",
            "  0:00:49.979   276M / 276M  INFO    General                 (binary_converter.hpp      :  93)   4194304 reads processed\n",
            "  0:01:12.105   204M / 276M  INFO    General                 (binary_converter.hpp      : 117)   6110636 reads written\n",
            "  0:01:12.290     4M / 276M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
            "  0:01:12.524   132M / 276M  INFO    General                 (binary_converter.hpp      : 117)   447 reads written\n",
            "  0:01:12.540     4M / 276M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
            "  0:01:12.765   132M / 276M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
            "  0:01:12.782     4M / 276M  INFO    General                 (construction.cpp          : 111)   Max read length 90\n",
            "  0:01:12.782     4M / 276M  INFO    General                 (construction.cpp          : 117)   Average read length 89.9999\n",
            "  0:01:12.782     4M / 276M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
            "  0:01:12.782     4M / 276M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 4 files using 2 threads. This might take a while.\n",
            "  0:01:12.783     4M / 276M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:01:12.783     4M / 276M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.99935 Gb\n",
            "  0:01:12.783     4M / 276M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 16777216\n",
            "  0:01:48.202     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 3887294 reads\n",
            "  0:02:15.840     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 7774482 reads\n",
            "  0:02:43.027     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 11661426 reads\n",
            "  0:03:12.525     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 15549022 reads\n",
            "  0:03:39.308     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 19435531 reads\n",
            "  0:04:06.212     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 23322730 reads\n",
            "  0:04:15.143     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 24443438 reads\n",
            "  0:04:15.144     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
            "  0:04:15.282     8M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 24443438 reads\n",
            "  0:04:15.283     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:04:17.306     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 6387455 kmers in total.\n",
            "  0:04:17.307     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:04:17.372     8M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
            "  0:04:17.372     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:04:17.372     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 32 files using 2 threads. This might take a while.\n",
            "  0:04:17.372     8M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:04:17.372     8M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.9987 Gb\n",
            "  0:04:17.372     8M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2097152\n",
            "  0:04:19.851     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 6387455 kmers\n",
            "  0:04:19.851     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 6387455 kmers.\n",
            "  0:04:19.865     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:04:20.340     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 6387532 kmers in total.\n",
            "  0:04:20.340     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:04:20.436     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:04:22.105     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
            "  0:04:22.277     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2970240 bytes occupied (3.72005 bits per kmer).\n",
            "  0:04:22.287    16M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
            "  0:04:26.055    16M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
            "  0:04:26.055    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Early tip clipping\n",
            "  0:04:26.055    16M / 1G    INFO    General                 (construction.cpp          : 253)   Early tip clipper length bound set as (RL - K)\n",
            "  0:04:26.055    16M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
            "  0:04:30.944    16M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 184)   344601 22-mers were removed by early tip clipper\n",
            "  0:04:30.944    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
            "  0:04:30.945    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
            "  0:04:36.572    20M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 36144 sequences extracted\n",
            "  0:04:39.936    20M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
            "  0:04:40.520    20M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 5 loops collected\n",
            "  0:04:40.602    28M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
            "  0:04:40.603    28M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:04:40.603    28M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:04:41.806    32M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2962880 bytes occupied (3.71087 bits per kmer).\n",
            "  0:04:41.829    60M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
            "  0:08:10.488    60M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
            "  0:08:12.580    60M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 72258 edges\n",
            "  0:08:12.608    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
            "  0:08:12.608    20M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 5\n",
            "  0:08:12.608    20M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 161\n",
            "  0:08:12.609    20M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 161. Coverage mad: 57.8214\n",
            "  0:08:12.609    20M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
            "  0:08:12.814    20M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
            "  0:08:13.328    20M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
            "  0:08:14.953    20M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
            "  0:08:21.005    20M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
            "  0:08:23.642    20M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 153.603. Fitted coverage std. dev: 54.1727\n",
            "  0:08:23.649    20M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.786382\n",
            "  0:08:23.649    20M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 40\n",
            "  0:08:23.649    20M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 40\n",
            "  0:08:23.649    20M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 5162671\n",
            "  0:08:23.649    20M / 1G    INFO    General                 (genomic_info_filler.cpp   : 112)   Mean coverage was calculated as 153.603\n",
            "  0:08:23.649    20M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 40\n",
            "  0:08:23.649    20M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
            "  0:08:23.649    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
            "  0:08:23.649    20M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
            "  0:08:23.650    20M / 1G    INFO    General                 (graph_simplification.hpp  : 669)   Flanking coverage based disconnection disabled\n",
            "  0:08:23.650    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
            "  0:08:23.660    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
            "  0:08:23.660    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial tip clipper\n",
            "  0:08:23.684    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial tip clipper triggered 250 times\n",
            "  0:08:23.684    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial ec remover\n",
            "  0:08:23.910    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial ec remover triggered 4629 times\n",
            "  0:08:23.910    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial isolated edge remover\n",
            "  0:08:23.954    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial isolated edge remover triggered 8666 times\n",
            "  0:08:23.956    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
            "  0:08:23.956    20M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
            "  0:08:23.956    20M / 1G    INFO    General                 (graph_simplification.hpp  : 641)   Creating parallel br instance\n",
            "  0:08:23.956    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
            "  0:08:23.956    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:08:23.964    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 24 times\n",
            "  0:08:23.964    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:08:24.140    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 2873 times\n",
            "  0:08:24.140    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:08:24.150    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 128 times\n",
            "  0:08:24.150    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
            "  0:08:24.150    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:08:24.151    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
            "  0:08:24.151    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:08:24.157    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 55 times\n",
            "  0:08:24.158    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:08:24.160    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 73 times\n",
            "  0:08:24.160    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
            "  0:08:24.160    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:08:24.160    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:08:24.160    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:08:24.164    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 62 times\n",
            "  0:08:24.164    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:08:24.167    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 31 times\n",
            "  0:08:24.167    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
            "  0:08:24.167    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:08:24.167    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:08:24.167    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:08:24.169    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 27 times\n",
            "  0:08:24.169    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:08:24.170    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 14 times\n",
            "  0:08:24.170    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
            "  0:08:24.170    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:08:24.170    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:08:24.170    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:08:24.171    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 8 times\n",
            "  0:08:24.171    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:08:24.171    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 9 times\n",
            "  0:08:24.171    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
            "  0:08:24.172    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:08:24.172    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:08:24.172    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:08:24.172    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\n",
            "  0:08:24.172    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:08:24.173    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 7 times\n",
            "  0:08:24.173    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
            "  0:08:24.173    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:08:24.173    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:08:24.173    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:08:24.173    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:08:24.173    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:08:24.174    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 5 times\n",
            "  0:08:24.174    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
            "  0:08:24.174    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:08:24.175    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:08:24.175    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:08:24.175    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\n",
            "  0:08:24.175    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:08:24.175    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2 times\n",
            "  0:08:24.175    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
            "  0:08:24.175    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:08:24.175    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:08:24.175    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:08:24.175    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:08:24.175    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:08:24.176    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
            "  0:08:24.176    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
            "  0:08:24.176    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:08:24.176    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:08:24.176    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:08:24.176    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:08:24.176    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:08:24.177    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 6 times\n",
            "  0:08:24.177    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
            "  0:08:24.177    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:08:24.179    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:08:24.179    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:08:24.188    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 9 times\n",
            "  0:08:24.188    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:08:24.193    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:08:24.193    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
            "  0:08:24.193    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:08:24.193    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:08:24.193    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:08:24.193    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:08:24.194    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:08:24.194    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:08:24.194    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
            "  0:08:24.194    20M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
            "  0:08:24.194    20M / 1G    INFO    General                 (graph_simplification.hpp  : 458)   Disconnection of relatively low covered edges disabled\n",
            "  0:08:24.194    20M / 1G    INFO    General                 (graph_simplification.hpp  : 494)   Complex tip clipping disabled\n",
            "  0:08:24.194    20M / 1G    INFO    General                 (graph_simplification.hpp  : 641)   Creating parallel br instance\n",
            "  0:08:24.194    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:08:24.199    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:08:24.199    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:08:24.219    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:08:24.220    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:08:24.221    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:08:24.221    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:08:24.229    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:08:24.229    20M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
            "  0:08:24.232    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
            "  0:08:24.234    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 357 times\n",
            "  0:08:24.234    20M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
            "  0:08:24.235    20M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 161.611\n",
            "  0:08:24.236    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
            "  0:08:24.236    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/output//K21/simplified_contigs.fasta\n",
            "  0:08:24.318    20M / 1G    INFO    General                 (launch.hpp                : 151)   SPAdes finished\n",
            "  0:08:24.333     8M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 8 minutes 24 seconds\n",
            "Max read length detected as 90\n",
            "\n",
            "== Running assembler: K33\n",
            "\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /content/output/K33/configs/config.info\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 12 Gb\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  84)   Starting SPAdes, built from N/A, git revision N/A\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/content/output/dataset.info) with K=33\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  52)   SPAdes started\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  59)   Starting from stage: construction\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  66)   Two-step RR enabled: 0\n",
            "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  59)   Binary reads detected\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (construction.cpp          : 111)   Max read length 90\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (construction.cpp          : 117)   Average read length 89.9999\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
            "  0:00:00.001     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 4 files using 2 threads. This might take a while.\n",
            "  0:00:00.001     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.99935 Gb\n",
            "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 8388608\n",
            "  0:00:17.121     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 2353636 reads\n",
            "  0:00:32.194     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 4707185 reads\n",
            "  0:00:48.282     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 7059878 reads\n",
            "  0:01:03.889     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 9412852 reads\n",
            "  0:01:19.158     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 11766313 reads\n",
            "  0:01:34.131     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 14119732 reads\n",
            "  0:01:48.873     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 16472983 reads\n",
            "  0:02:05.483     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 18825495 reads\n",
            "  0:02:20.506     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 21178707 reads\n",
            "  0:02:35.637     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 23531763 reads\n",
            "  0:02:42.752     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
            "  0:02:44.314     8M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 24443438 reads\n",
            "  0:02:44.314     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:02:48.069     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 6353220 kmers in total.\n",
            "  0:02:48.070     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:02:48.218     8M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
            "  0:02:48.219     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:02:48.219     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 32 files using 2 threads. This might take a while.\n",
            "  0:02:48.219     8M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:02:48.219     8M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.9987 Gb\n",
            "  0:02:48.219     8M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 1048576\n",
            "  0:02:51.146     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 6353220 kmers\n",
            "  0:02:51.147     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 6353220 kmers.\n",
            "  0:02:51.171     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:02:51.571     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 6357744 kmers in total.\n",
            "  0:02:51.571     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:02:51.687     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:02:52.606     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
            "  0:02:52.845     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2956592 bytes occupied (3.7203 bits per kmer).\n",
            "  0:02:52.865    16M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
            "  0:02:56.954    16M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
            "  0:02:56.954    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Early tip clipping\n",
            "  0:02:56.954    16M / 1G    INFO    General                 (construction.cpp          : 253)   Early tip clipper length bound set as (RL - K)\n",
            "  0:02:56.954    16M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
            "  0:03:01.397    16M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 184)   407703 34-mers were removed by early tip clipper\n",
            "  0:03:01.397    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
            "  0:03:01.398    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
            "  0:03:04.763    20M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 24499 sequences extracted\n",
            "  0:03:06.996    20M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
            "  0:03:10.321    20M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 3 loops collected\n",
            "  0:03:10.679    24M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
            "  0:03:10.688    24M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:03:10.689    24M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:03:15.465    28M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2947024 bytes occupied (3.7109 bits per kmer).\n",
            "  0:03:15.547    56M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
            "  0:06:10.629    56M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
            "  0:06:13.362    56M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 48994 edges\n",
            "  0:06:13.430    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
            "  0:06:13.431    16M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 3\n",
            "  0:06:13.431    16M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 134\n",
            "  0:06:13.431    16M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 134. Coverage mad: 50.4084\n",
            "  0:06:13.431    16M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
            "  0:06:13.614    16M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
            "  0:06:13.953    16M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
            "  0:06:14.939    16M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
            "  0:06:19.012    16M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
            "  0:06:21.335    16M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 126.892. Fitted coverage std. dev: 45.9955\n",
            "  0:06:21.341    16M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.850305\n",
            "  0:06:21.341    16M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 34\n",
            "  0:06:21.341    16M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 34\n",
            "  0:06:21.341    16M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 5165120\n",
            "  0:06:21.341    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 112)   Mean coverage was calculated as 126.892\n",
            "  0:06:21.341    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 34\n",
            "  0:06:21.341    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
            "  0:06:21.341    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
            "  0:06:21.341    16M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
            "  0:06:21.342    16M / 1G    INFO    General                 (graph_simplification.hpp  : 669)   Flanking coverage based disconnection disabled\n",
            "  0:06:21.342    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
            "  0:06:21.349    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
            "  0:06:21.350    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial tip clipper\n",
            "  0:06:21.376    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial tip clipper triggered 304 times\n",
            "  0:06:21.378    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial ec remover\n",
            "  0:06:21.513    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial ec remover triggered 2131 times\n",
            "  0:06:21.513    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial isolated edge remover\n",
            "  0:06:21.555    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial isolated edge remover triggered 9109 times\n",
            "  0:06:21.556    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
            "  0:06:21.556    16M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
            "  0:06:21.556    16M / 1G    INFO    General                 (graph_simplification.hpp  : 641)   Creating parallel br instance\n",
            "  0:06:21.556    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
            "  0:06:21.556    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:06:21.561    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 15 times\n",
            "  0:06:21.562    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:06:21.731    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1826 times\n",
            "  0:06:21.731    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:06:21.741    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 69 times\n",
            "  0:06:21.741    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
            "  0:06:21.741    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:06:21.743    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 4 times\n",
            "  0:06:21.743    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:06:21.748    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 52 times\n",
            "  0:06:21.748    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:06:21.752    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 72 times\n",
            "  0:06:21.752    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
            "  0:06:21.752    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:06:21.752    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 3 times\n",
            "  0:06:21.753    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:06:21.760    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 53 times\n",
            "  0:06:21.760    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:06:21.761    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 26 times\n",
            "  0:06:21.761    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
            "  0:06:21.761    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:06:21.762    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
            "  0:06:21.762    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:06:21.763    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 17 times\n",
            "  0:06:21.763    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:06:21.763    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 5 times\n",
            "  0:06:21.764    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
            "  0:06:21.764    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:06:21.764    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:06:21.764    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:06:21.764    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 4 times\n",
            "  0:06:21.764    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:06:21.764    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2 times\n",
            "  0:06:21.764    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
            "  0:06:21.764    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:06:21.765    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:06:21.765    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:06:21.765    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:06:21.765    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:06:21.765    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 5 times\n",
            "  0:06:21.765    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
            "  0:06:21.765    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:06:21.765    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:06:21.765    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:06:21.765    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\n",
            "  0:06:21.765    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:06:21.766    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2 times\n",
            "  0:06:21.766    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
            "  0:06:21.766    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:06:21.766    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:06:21.766    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:06:21.766    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:06:21.766    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:06:21.766    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
            "  0:06:21.767    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
            "  0:06:21.767    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:06:21.767    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:06:21.767    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:06:21.767    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:06:21.767    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:06:21.767    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 3 times\n",
            "  0:06:21.767    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
            "  0:06:21.767    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:06:21.767    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:06:21.767    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:06:21.767    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\n",
            "  0:06:21.767    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:06:21.767    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 3 times\n",
            "  0:06:21.767    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
            "  0:06:21.767    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:06:21.768    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:06:21.768    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:06:21.771    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\n",
            "  0:06:21.771    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:06:21.771    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:06:21.771    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
            "  0:06:21.771    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:06:21.771    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:06:21.771    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:06:21.771    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:06:21.771    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:06:21.771    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:06:21.772    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
            "  0:06:21.772    16M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
            "  0:06:21.772    16M / 1G    INFO    General                 (graph_simplification.hpp  : 458)   Disconnection of relatively low covered edges disabled\n",
            "  0:06:21.772    16M / 1G    INFO    General                 (graph_simplification.hpp  : 494)   Complex tip clipping disabled\n",
            "  0:06:21.772    16M / 1G    INFO    General                 (graph_simplification.hpp  : 641)   Creating parallel br instance\n",
            "  0:06:21.772    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:06:21.772    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:06:21.772    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:06:21.775    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:06:21.775    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:06:21.776    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:06:21.776    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:06:21.778    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:06:21.778    16M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
            "  0:06:21.779    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
            "  0:06:21.781    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 189 times\n",
            "  0:06:21.781    16M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
            "  0:06:21.782    16M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 132.498\n",
            "  0:06:21.782    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
            "  0:06:21.782    16M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/output//K33/simplified_contigs.fasta\n",
            "  0:06:21.860    16M / 1G    INFO    General                 (launch.hpp                : 151)   SPAdes finished\n",
            "  0:06:21.869     8M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 6 minutes 21 seconds\n",
            "\n",
            "== Running assembler: K55\n",
            "\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /content/output/K55/configs/config.info\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 12 Gb\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  84)   Starting SPAdes, built from N/A, git revision N/A\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/content/output/dataset.info) with K=55\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  52)   SPAdes started\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  59)   Starting from stage: construction\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  66)   Two-step RR enabled: 0\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  77)   Will need read mapping, kmer mapper will be attached\n",
            "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
            "  0:00:00.001     4M / 4M    INFO    General                 (read_converter.hpp        :  59)   Binary reads detected\n",
            "  0:00:00.002     4M / 4M    INFO    General                 (construction.cpp          : 111)   Max read length 90\n",
            "  0:00:00.002     4M / 4M    INFO    General                 (construction.cpp          : 117)   Average read length 89.9999\n",
            "  0:00:00.002     4M / 4M    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
            "  0:00:00.002     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 4 files using 2 threads. This might take a while.\n",
            "  0:00:00.002     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:00.002     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.99935 Gb\n",
            "  0:00:00.002     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 8388608\n",
            "  0:00:17.277     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 3830187 reads\n",
            "  0:00:34.159     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 7659626 reads\n",
            "  0:00:50.710     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 11489393 reads\n",
            "  0:01:06.520     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 15319718 reads\n",
            "  0:01:22.394     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 19148951 reads\n",
            "  0:01:38.226     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 22979927 reads\n",
            "  0:01:44.691     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 24443438 reads\n",
            "  0:01:44.698     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
            "  0:01:46.890     8M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 24443438 reads\n",
            "  0:01:46.890     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:01:49.341     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 6194393 kmers in total.\n",
            "  0:01:49.342     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:01:49.462     8M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
            "  0:01:49.462     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:01:49.462     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 32 files using 2 threads. This might take a while.\n",
            "  0:01:49.462     8M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:01:49.462     8M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.9987 Gb\n",
            "  0:01:49.462     8M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 1048576\n",
            "  0:01:52.314     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 6194393 kmers\n",
            "  0:01:52.315     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 6194393 kmers.\n",
            "  0:01:52.367     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:01:52.830     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 6203799 kmers in total.\n",
            "  0:01:52.830     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:01:53.012     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:01:53.948     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
            "  0:01:54.142     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2884856 bytes occupied (3.72012 bits per kmer).\n",
            "  0:01:54.152    16M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
            "  0:01:56.801    16M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
            "  0:01:56.801    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
            "  0:01:56.801    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
            "  0:02:02.127    32M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 164849 sequences extracted\n",
            "  0:02:04.288    32M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
            "  0:02:04.859    32M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 0 loops collected\n",
            "  0:02:05.174    72M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
            "  0:02:05.175    72M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:02:05.175    72M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:02:06.294    72M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2873368 bytes occupied (3.71093 bits per kmer).\n",
            "  0:02:06.316    96M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
            "  0:04:02.278    96M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
            "  0:04:06.828    96M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 329697 edges\n",
            "  0:04:06.864    60M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
            "  0:04:06.865    60M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 8\n",
            "  0:04:06.865    60M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 80\n",
            "  0:04:06.865    60M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 80. Coverage mad: 32.6172\n",
            "  0:04:06.865    60M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
            "  0:04:06.987    60M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
            "  0:04:07.335    60M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
            "  0:04:08.586    60M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
            "  0:04:10.829    60M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
            "  0:04:11.945    60M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 77.8251. Fitted coverage std. dev: 29.6431\n",
            "  0:04:11.948    60M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.337292\n",
            "  0:04:11.948    60M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 23\n",
            "  0:04:11.948    60M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 23\n",
            "  0:04:11.948    60M / 1G    INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 5128459\n",
            "  0:04:11.948    60M / 1G    INFO    General                 (genomic_info_filler.cpp   : 112)   Mean coverage was calculated as 77.8251\n",
            "  0:04:11.948    60M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 23\n",
            "  0:04:11.948    60M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
            "  0:04:11.948    60M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
            "  0:04:11.948    60M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
            "  0:04:11.949    60M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:04:11.949    60M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 32 files using 2 threads. This might take a while.\n",
            "  0:04:11.949    60M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:04:11.949    60M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.99023 Gb\n",
            "  0:04:11.949    60M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 1048576\n",
            "  0:04:13.311     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 329697 edges\n",
            "  0:04:13.312     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 329697 sequences.\n",
            "  0:04:13.322    56M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:04:13.735    56M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 6194393 kmers in total.\n",
            "  0:04:13.736    56M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:04:13.871    56M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:04:14.774    64M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
            "  0:04:15.245    64M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2880560 bytes occupied (3.72022 bits per kmer).\n",
            "  0:04:15.381   208M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
            "  0:04:18.444   208M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
            "  0:04:18.444   208M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
            "  0:04:18.792   220M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
            "  0:04:53.049   224M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 6110636 paired reads\n",
            "  0:04:53.049   224M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
            "  0:04:53.098   208M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
            "  0:04:53.408   208M / 1G    INFO   GapCloser                (gap_closer.cpp            : 378)   Closing short gaps complete: filled 7 gaps after checking 288 candidates\n",
            "  0:04:53.425   208M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
            "  0:04:53.441    64M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
            "  0:04:53.442    64M / 1G    INFO    General                 (graph_simplification.hpp  : 669)   Flanking coverage based disconnection disabled\n",
            "  0:04:53.442    64M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
            "  0:04:53.481    64M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
            "  0:04:53.481    64M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial tip clipper\n",
            "  0:04:54.929    64M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial tip clipper triggered 74453 times\n",
            "  0:04:54.929    64M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial ec remover\n",
            "  0:04:54.950    64M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial ec remover triggered 187 times\n",
            "  0:04:54.951    64M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial isolated edge remover\n",
            "  0:04:55.016    48M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial isolated edge remover triggered 11049 times\n",
            "  0:04:55.021    44M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
            "  0:04:55.021    44M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
            "  0:04:55.021    44M / 1G    INFO    General                 (graph_simplification.hpp  : 641)   Creating parallel br instance\n",
            "  0:04:55.021    44M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
            "  0:04:55.021    44M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:04:55.023    44M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 15 times\n",
            "  0:04:55.024    44M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:04:55.492    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1069 times\n",
            "  0:04:55.492    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:04:55.494    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
            "  0:04:55.494    36M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
            "  0:04:55.494    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:04:55.495    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
            "  0:04:55.495    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:04:55.499    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 7 times\n",
            "  0:04:55.499    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:04:55.514    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 53 times\n",
            "  0:04:55.515    36M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
            "  0:04:55.515    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:04:55.515    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 7 times\n",
            "  0:04:55.515    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:04:55.528    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 42 times\n",
            "  0:04:55.528    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:04:55.530    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 7 times\n",
            "  0:04:55.530    36M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
            "  0:04:55.530    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:04:55.530    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:04:55.530    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:04:55.530    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 2 times\n",
            "  0:04:55.530    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:04:55.531    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 3 times\n",
            "  0:04:55.531    36M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
            "  0:04:55.531    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:04:55.531    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:04:55.531    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:04:55.531    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\n",
            "  0:04:55.531    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:04:55.532    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
            "  0:04:55.532    36M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
            "  0:04:55.532    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:04:55.532    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:04:55.532    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:04:55.533    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 2 times\n",
            "  0:04:55.533    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2 times\n",
            "  0:04:55.534    36M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1 times\n",
            "  0:04:55.534    36M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:04:55.534    36M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2 times\n",
            "  0:04:55.534    36M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:04:55.534    36M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
            "  0:04:55.534    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:04:55.536    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:04:55.536    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:04:55.539    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 5 times\n",
            "  0:04:55.539    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:04:55.539    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:04:55.540    36M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
            "  0:04:55.540    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:04:55.540    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:04:55.540    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:04:55.540    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:04:55.540    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
            "  0:04:55.540    36M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
            "  0:04:55.540    36M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
            "  0:04:55.540    36M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
            "  0:04:55.540    36M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:04:55.540    36M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 32 files using 2 threads. This might take a while.\n",
            "  0:04:55.541    36M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:04:55.541    36M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.99414 Gb\n",
            "  0:04:55.541    36M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 1048576\n",
            "  0:04:56.557     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 2421 edges\n",
            "  0:04:56.557     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 2421 sequences.\n",
            "  0:04:56.566    36M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:04:56.900    36M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 5286744 kmers in total.\n",
            "  0:04:56.900    36M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:04:57.000    36M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:04:57.960    36M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
            "  0:04:58.229    36M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2459808 bytes occupied (3.72223 bits per kmer).\n",
            "  0:04:58.380   160M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
            "  0:05:00.839   160M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
            "  0:05:00.839   160M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
            "  0:05:00.841   160M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
            "  0:05:35.248   160M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 6110636 paired reads\n",
            "  0:05:35.248   160M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
            "  0:05:35.249   160M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
            "  0:05:35.252   160M / 1G    INFO   GapCloser                (gap_closer.cpp            : 378)   Closing short gaps complete: filled 8 gaps after checking 112 candidates\n",
            "  0:05:35.954   160M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
            "  0:05:35.955   160M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
            "  0:05:35.955   160M / 1G    INFO    General                 (graph_simplification.hpp  : 458)   Disconnection of relatively low covered edges disabled\n",
            "  0:05:35.955   160M / 1G    INFO    General                 (graph_simplification.hpp  : 494)   Complex tip clipping disabled\n",
            "  0:05:35.955   160M / 1G    INFO    General                 (graph_simplification.hpp  : 641)   Creating parallel br instance\n",
            "  0:05:35.955   160M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:05:35.956   160M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:05:35.956   160M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:05:35.958   160M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:05:35.958   160M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
            "  0:05:35.958   160M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
            "  0:05:35.958   160M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
            "  0:05:35.960   160M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
            "  0:05:35.960   160M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
            "  0:05:35.961   160M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
            "  0:05:35.966   160M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 35 times\n",
            "  0:05:35.966   160M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
            "  0:05:35.967   160M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 80.7985\n",
            "  0:05:35.967   160M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
            "  0:05:35.967   160M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 146816 kmers to process\n",
            "  0:05:36.102   160M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
            "  0:07:19.458   160M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
            "  0:07:19.514   160M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
            "  0:07:19.515   160M / 1G    INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 4 nucleotides\n",
            "  0:07:19.515   160M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
            "  0:07:19.516   160M / 1G    INFO    General                 (contig_output_stage.cpp   :  45)   Writing GFA to /content/output//K55/assembly_graph_with_scaffolds.gfa\n",
            "  0:07:19.586   160M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/output//K55/before_rr.fasta\n",
            "  0:07:19.670   160M / 1G    INFO    General                 (contig_output_stage.cpp   :  56)   Outputting FastG graph to /content/output//K55/assembly_graph.fastg\n",
            "  0:07:19.877   160M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/output//K55/final_contigs.fasta\n",
            "  0:07:19.960   160M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
            "  0:07:19.965   160M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 146998 kmers to process\n",
            "  0:07:20.111   160M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
            "  0:07:20.112   160M / 1G    INFO    General                 (pair_info_count.cpp       : 323)   Min edge length for estimation: 58577\n",
            "  0:07:20.113   160M / 1G    INFO    General                 (pair_info_count.cpp       : 334)   Estimating insert size for library #0\n",
            "  0:07:20.113   160M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
            "  0:07:20.209   208M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
            "  0:07:23.778   208M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 200000 reads\n",
            "  0:07:23.989   208M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 400000 reads\n",
            "  0:07:26.558   208M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 600000 reads\n",
            "  0:07:26.712   208M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 800000 reads\n",
            "  0:07:29.060   208M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 1000000 reads\n",
            "  0:07:29.213   208M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 1200000 reads\n",
            "  0:07:38.110   208M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 2200000 reads\n",
            "  0:07:50.934   208M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 4200000 reads\n",
            "  0:08:01.903   208M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 6110636 reads processed\n",
            "  0:08:01.965   208M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
            "  0:08:01.965   208M / 1G    INFO    General                 (pair_info_count.cpp       : 211)   2936367 paired reads (48.0534% of all) aligned to long edges\n",
            "  0:08:01.970   160M / 1G    INFO    General                 (pair_info_count.cpp       : 353)     Insert size = 469.016, deviation = 14.6954, left quantile = 450, right quantile = 486, read length = 90\n",
            "  0:08:02.293   352M / 1G    INFO    General                 (pair_info_count.cpp       : 374)   Filtering data for library #0\n",
            "  0:08:02.293   352M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
            "  0:08:06.072   352M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 200000 reads\n",
            "  0:08:06.359   352M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 400000 reads\n",
            "  0:08:08.799   352M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 600000 reads\n",
            "  0:08:08.951   352M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 800000 reads\n",
            "  0:08:11.183   352M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 1000000 reads\n",
            "  0:08:11.290   352M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 1200000 reads\n",
            "  0:08:19.745   352M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 2200000 reads\n",
            "  0:08:32.790   352M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 4200000 reads\n",
            "  0:08:43.839   352M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 6110636 reads processed\n",
            "  0:08:43.839   352M / 1G    INFO    General                 (pair_info_count.cpp       : 386)   Mapping library #0\n",
            "  0:08:43.839   352M / 1G    INFO    General                 (pair_info_count.cpp       : 388)   Mapping paired reads (takes a while)\n",
            "  0:08:43.840   352M / 1G    INFO    General                 (pair_info_count.cpp       : 286)   Left insert size quantile 450, right insert size quantile 486, filtering threshold 2, rounding threshold 0\n",
            "  0:08:43.853   364M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
            "  0:08:48.392   364M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 200000 reads\n",
            "  0:08:48.587   364M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 400000 reads\n",
            "  0:08:51.044   364M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 600000 reads\n",
            "  0:08:51.269   364M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 800000 reads\n",
            "  0:08:53.543   364M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 1000000 reads\n",
            "  0:08:53.778   364M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 1200000 reads\n",
            "  0:09:03.196   364M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 2200000 reads\n",
            "  0:09:18.131   364M / 1G    INFO    General                 (sequence_mapper_notifier.h:  80)   Processed 4200000 reads\n",
            "  0:09:31.741   364M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 6110636 reads processed\n",
            "  0:09:31.772   160M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
            "  0:09:31.772   160M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
            "  0:09:31.772   160M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
            "  0:09:31.772   160M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
            "  0:09:31.808   160M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
            "  0:09:31.808   160M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 10901\n",
            "  0:09:31.814   164M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
            "  0:09:31.814   160M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
            "  0:09:31.818   160M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
            "  0:09:31.818   160M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
            "  0:09:31.871   160M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 102)   Paired info stats: missing = 1691; contradictional = 206\n",
            "  0:09:31.922   160M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 102)   Paired info stats: missing = 83; contradictional = 42\n",
            "  0:09:31.922   160M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
            "  0:09:31.922   160M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
            "  0:09:31.988   160M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
            "  0:09:31.988   160M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 2535\n",
            "  0:09:31.989   164M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
            "  0:09:31.989   160M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
            "  0:09:31.991   160M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\n",
            "  0:09:31.991   160M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
            "  0:09:31.991   160M / 1G    INFO    General                 (launcher.cpp              : 481)   ExSPAnder repeat resolving tool started\n",
            "  0:09:31.998   164M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
            "  0:09:31.998   164M / 1G    INFO    General                 (extenders_logic.cpp       : 278)   Estimated coverage of library #0 is 80.7985\n",
            "  0:09:31.999   164M / 1G    INFO    General                 (extenders_logic.cpp       : 278)   Estimated coverage of library #0 is 80.7985\n",
            "  0:09:32.000   164M / 1G    INFO    General                 (extenders_logic.cpp       : 475)   Using 1 paired-end library\n",
            "  0:09:32.000   164M / 1G    INFO    General                 (extenders_logic.cpp       : 476)   Using 1 paired-end scaffolding library\n",
            "  0:09:32.000   164M / 1G    INFO    General                 (extenders_logic.cpp       : 477)   Using 0 single read libraries\n",
            "  0:09:32.000   164M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 3\n",
            "  0:09:32.000   164M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 1127 (0%)\n",
            "  0:09:32.121   168M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 113 paths from 1127 (10%)\n",
            "  0:09:32.128   168M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 128 paths from 1127 (11%)\n",
            "  0:09:32.184   168M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 226 paths from 1127 (20%)\n",
            "  0:09:32.190   168M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 256 paths from 1127 (22%)\n",
            "  0:09:32.235   172M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 339 paths from 1127 (30%)\n",
            "  0:09:32.251   172M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 452 paths from 1127 (40%)\n",
            "  0:09:32.257   172M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 512 paths from 1127 (45%)\n",
            "  0:09:32.258   172M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 565 paths from 1127 (50%)\n",
            "  0:09:32.263   172M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 678 paths from 1127 (60%)\n",
            "  0:09:32.268   172M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 791 paths from 1127 (70%)\n",
            "  0:09:32.277   172M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 904 paths from 1127 (80%)\n",
            "  0:09:32.277   172M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 1017 paths from 1127 (90%)\n",
            "  0:09:32.278   172M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 1024 paths from 1127 (90%)\n",
            "  0:09:32.278   172M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
            "  0:09:32.278   172M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
            "  0:09:32.282   172M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
            "  0:09:32.282   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
            "  0:09:32.282   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
            "  0:09:32.282   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
            "  0:09:32.282   172M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
            "  0:09:32.287   172M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
            "  0:09:32.292   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
            "  0:09:32.295   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
            "  0:09:32.298   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
            "  0:09:32.299   172M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
            "  0:09:32.300   172M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
            "  0:09:32.302   172M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
            "  0:09:32.304   172M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
            "  0:09:32.309   172M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 1 loops\n",
            "  0:09:32.309   172M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
            "  0:09:32.309   172M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
            "  0:09:32.310   172M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
            "  0:09:32.310   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
            "  0:09:32.310   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
            "  0:09:32.310   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
            "  0:09:32.310   172M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
            "  0:09:32.311   172M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
            "  0:09:32.312   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
            "  0:09:32.313   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
            "  0:09:32.313   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
            "  0:09:32.314   172M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
            "  0:09:32.314   172M / 1G    INFO    General                 (launcher.cpp              : 534)   ExSPAnder repeat resolving tool finished\n",
            "  0:09:32.330   164M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
            "  0:09:32.330   164M / 1G    INFO    General                 (contig_output_stage.cpp   :  45)   Writing GFA to /content/output//K55/assembly_graph_with_scaffolds.gfa\n",
            "  0:09:32.406   164M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /content/output//K55/before_rr.fasta\n",
            "  0:09:32.495   164M / 1G    INFO    General                 (contig_output_stage.cpp   :  56)   Outputting FastG graph to /content/output//K55/assembly_graph.fastg\n",
            "  0:09:32.701   164M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /content/output//K55/final_contigs.paths\n",
            "  0:09:32.794   164M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /content/output//K55/scaffolds.paths\n",
            "  0:09:32.896   164M / 1G    INFO    General                 (launch.hpp                : 151)   SPAdes finished\n",
            "  0:09:32.989    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 9 minutes 32 seconds\n",
            "\n",
            "===== Assembling finished. Used k-mer sizes: 21, 33, 55 \n",
            "\n",
            " * Corrected reads are in /content/output/corrected/\n",
            " * Assembled contigs are in /content/output/contigs.fasta\n",
            " * Assembled scaffolds are in /content/output/scaffolds.fasta\n",
            " * Paths in the assembly graph corresponding to the contigs are in /content/output/contigs.paths\n",
            " * Paths in the assembly graph corresponding to the scaffolds are in /content/output/scaffolds.paths\n",
            " * Assembly graph is in /content/output/assembly_graph.fastg\n",
            " * Assembly graph in GFA format is in /content/output/assembly_graph_with_scaffolds.gfa\n",
            "\n",
            "======= SPAdes pipeline finished.\n",
            "\n",
            "SPAdes log can be found here: /content/output/spades.log\n",
            "\n",
            "Thank you for using SPAdes!\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install quast"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "LHxlvoZnnnw2",
        "outputId": "f23d2b20-5e9e-4406-914d-8b22c58ee4bb"
      },
      "execution_count": 27,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Collecting quast\n",
            "  Downloading quast-5.2.0.tar.gz (34.2 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m34.2/34.2 MB\u001b[0m \u001b[31m8.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: joblib in /usr/local/lib/python3.8/dist-packages (from quast) (1.2.0)\n",
            "Requirement already satisfied: simplejson in /usr/lib/python3/dist-packages (from quast) (3.16.0)\n",
            "Building wheels for collected packages: quast\n",
            "  Building wheel for quast (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for quast: filename=quast-5.2.0-py3-none-any.whl size=31206447 sha256=fe581ab34a1ec72852a46135a93952c82abbde029626a3d80adac06f022d3ade\n",
            "  Stored in directory: /root/.cache/pip/wheels/79/fe/84/429b22fc6184f0807dfdcb67bdf0d95441b3415bce83219ad2\n",
            "Successfully built quast\n",
            "Installing collected packages: quast\n",
            "Successfully installed quast-5.2.0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!quast.py output/contigs.fasta"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "FVmXaOYhq6vp",
        "outputId": "b9f867dd-bfab-4f95-9004-64335ececf94"
      },
      "execution_count": 29,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/usr/local/bin/quast.py output/contigs.fasta\n",
            "\n",
            "Version: 5.2.0\n",
            "\n",
            "System information:\n",
            "  OS: Linux-5.10.147+-x86_64-with-glibc2.29 (linux_64)\n",
            "  Python version: 3.8.10\n",
            "  CPUs number: 2\n",
            "\n",
            "Started: 2023-03-03 11:27:07\n",
            "\n",
            "Logging to /content/quast_results/results_2023_03_03_11_27_07/quast.log\n",
            "NOTICE: Maximum number of threads is set to 1 (use --threads option to set it manually)\n",
            "\n",
            "CWD: /content\n",
            "Main parameters: \n",
            "  MODE: default, threads: 1, min contig length: 500, min alignment length: 65, min alignment IDY: 95.0, \\\n",
            "  ambiguity: one, min local misassembly length: 200, min extensive misassembly length: 1000\n",
            "\n",
            "Contigs:\n",
            "  Pre-processing...\n",
            "  output/contigs.fasta ==> contigs\n",
            "\n",
            "2023-03-03 11:27:08\n",
            "Running Basic statistics processor...\n",
            "  Contig files: \n",
            "    contigs\n",
            "  Calculating N50 and L50...\n",
            "    contigs, N50 = 105346, L50 = 15, auN = 127832.1, Total length = 5245723, GC % = 50.53, # N's per 100 kbp =  0.00\n",
            "  Drawing Nx plot...\n",
            "    saved to /content/quast_results/results_2023_03_03_11_27_07/basic_stats/Nx_plot.pdf\n",
            "  Drawing cumulative plot...\n",
            "    saved to /content/quast_results/results_2023_03_03_11_27_07/basic_stats/cumulative_plot.pdf\n",
            "  Drawing GC content plot...\n",
            "    saved to /content/quast_results/results_2023_03_03_11_27_07/basic_stats/GC_content_plot.pdf\n",
            "  Drawing contigs GC content plot...\n",
            "    saved to /content/quast_results/results_2023_03_03_11_27_07/basic_stats/contigs_GC_content_plot.pdf\n",
            "  Drawing Coverage histogram (bin size: 1x)...\n",
            "    saved to /content/quast_results/results_2023_03_03_11_27_07/basic_stats/coverage_histogram.pdf\n",
            "  Drawing contigs coverage histogram (bin size: 1x)...\n",
            "    saved to /content/quast_results/results_2023_03_03_11_27_07/basic_stats/contigs_coverage_histogram.pdf\n",
            "Done.\n",
            "\n",
            "NOTICE: Genes are not predicted by default. Use --gene-finding or --glimmer option to enable it.\n",
            "\n",
            "2023-03-03 11:27:11\n",
            "Creating large visual summaries...\n",
            "This may take a while: press Ctrl-C to skip this step..\n",
            "  1 of 2: Creating PDF with all tables and plots...\n",
            "  2 of 2: Creating Icarus viewers...\n",
            "Done\n",
            "\n",
            "2023-03-03 11:27:11\n",
            "RESULTS:\n",
            "  Text versions of total report are saved to /content/quast_results/results_2023_03_03_11_27_07/report.txt, report.tsv, and report.tex\n",
            "  Text versions of transposed total report are saved to /content/quast_results/results_2023_03_03_11_27_07/transposed_report.txt, transposed_report.tsv, and transposed_report.tex\n",
            "  HTML version (interactive tables and plots) is saved to /content/quast_results/results_2023_03_03_11_27_07/report.html\n",
            "  PDF version (tables and plots) is saved to /content/quast_results/results_2023_03_03_11_27_07/report.pdf\n",
            "  Icarus (contig browser) is saved to /content/quast_results/results_2023_03_03_11_27_07/icarus.html\n",
            "  Log is saved to /content/quast_results/results_2023_03_03_11_27_07/quast.log\n",
            "\n",
            "Finished: 2023-03-03 11:27:11\n",
            "Elapsed time: 0:00:04.517275\n",
            "NOTICEs: 2; WARNINGs: 0; non-fatal ERRORs: 0\n",
            "\n",
            "Thank you for using QUAST!\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cat quast_results/latest/report.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "03mxoOrlriNm",
        "outputId": "64b3c9e1-3b6c-4af5-c49e-193e9aeba1d3"
      },
      "execution_count": 31,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "All statistics are based on contigs of size >= 500 bp, unless otherwise noted (e.g., \"# contigs (>= 0 bp)\" and \"Total length (>= 0 bp)\" include all contigs).\n",
            "\n",
            "Assembly                    contigs \n",
            "# contigs (>= 0 bp)         510     \n",
            "# contigs (>= 1000 bp)      147     \n",
            "# contigs (>= 5000 bp)      82      \n",
            "# contigs (>= 10000 bp)     69      \n",
            "# contigs (>= 25000 bp)     51      \n",
            "# contigs (>= 50000 bp)     33      \n",
            "Total length (>= 0 bp)      5316686 \n",
            "Total length (>= 1000 bp)   5206945 \n",
            "Total length (>= 5000 bp)   5043449 \n",
            "Total length (>= 10000 bp)  4953165 \n",
            "Total length (>= 25000 bp)  4664641 \n",
            "Total length (>= 50000 bp)  4002075 \n",
            "# contigs                   201     \n",
            "Largest contig              300784  \n",
            "Total length                5245723 \n",
            "GC (%)                      50.53   \n",
            "N50                         105346  \n",
            "N90                         21421   \n",
            "auN                         127832.1\n",
            "L50                         15      \n",
            "L90                         54      \n",
            "# N's per 100 kbp           0.00    \n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# 3a Effect of read correction."
      ],
      "metadata": {
        "id": "tGI7s-Ssr4Vk"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "# 4. Impact of reads with large insert size"
      ],
      "metadata": {
        "id": "maRo7YOpsCLw"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!spades.py --pe1-1 SRR292678_1.fastq --pe1-2 SRR292678_2.fastq \\\n",
        "    --mp1-1 SRR292770_1.fastq --mp1-2 SRR292770_2.fastq \\\n",
        "    --mp2-1 SRR292862_1.fastq --mp2-2 SRR292862_2.fastq \\\n",
        "    -o spades_output"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "09KFbVsVsskd",
        "outputId": "ebead611-1f69-461d-e831-c6ddb1a82956"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Command line: /usr/lib/spades/bin/spades.py\t--pe1-1\t/content/SRR292678_1.fastq\t--pe1-2\t/content/SRR292678_2.fastq\t--mp1-1\t/content/SRR292770_1.fastq\t--mp1-2\t/content/SRR292770_2.fastq\t--mp2-1\t/content/SRR292862_1.fastq\t--mp2-2\t/content/SRR292862_2.fastq\t-o\t/content/spades_output\t\n",
            "\n",
            "System information:\n",
            "  SPAdes version: 3.13.1\n",
            "  Python version: 3.8.10\n",
            "  OS: Linux-5.10.147+-x86_64-with-glibc2.29\n",
            "\n",
            "Output dir: /content/spades_output\n",
            "Mode: read error correction and assembling\n",
            "Debug mode is turned OFF\n",
            "\n",
            "Dataset parameters:\n",
            "  Multi-cell mode (you should set '--sc' flag if input data was obtained with MDA (single-cell) technology or --meta flag if processing metagenomic dataset)\n",
            "  Reads:\n",
            "    Library number: 1, library type: mate-pairs\n",
            "      orientation: rf\n",
            "      left reads: ['/content/SRR292770_1.fastq']\n",
            "      right reads: ['/content/SRR292770_2.fastq']\n",
            "      interlaced reads: not specified\n",
            "      single reads: not specified\n",
            "      merged reads: not specified\n",
            "    Library number: 2, library type: mate-pairs\n",
            "      orientation: rf\n",
            "      left reads: ['/content/SRR292862_1.fastq']\n",
            "      right reads: ['/content/SRR292862_2.fastq']\n",
            "      interlaced reads: not specified\n",
            "      single reads: not specified\n",
            "      merged reads: not specified\n",
            "    Library number: 3, library type: paired-end\n",
            "      orientation: fr\n",
            "      left reads: ['/content/SRR292678_1.fastq']\n",
            "      right reads: ['/content/SRR292678_2.fastq']\n",
            "      interlaced reads: not specified\n",
            "      single reads: not specified\n",
            "      merged reads: not specified\n",
            "Read error correction parameters:\n",
            "  Iterations: 1\n",
            "  PHRED offset will be auto-detected\n",
            "  Corrected reads will be compressed\n",
            "Assembly parameters:\n",
            "  k: automatic selection based on read length\n",
            "  Repeat resolution is enabled\n",
            "  Mismatch careful mode is turned OFF\n",
            "  MismatchCorrector will be SKIPPED\n",
            "  Coverage cutoff is turned OFF\n",
            "Other parameters:\n",
            "  Dir for temp files: /content/spades_output/tmp\n",
            "  Threads: 16\n",
            "  Memory limit (in Gb): 12\n",
            "\n",
            "\n",
            "======= SPAdes pipeline started. Log can be found here: /content/spades_output/spades.log\n",
            "\n",
            "\n",
            "===== Read error correction started. \n",
            "\n",
            "\n",
            "== Running read error correction tool: /usr/lib/spades/bin/spades-hammer /content/spades_output/corrected/configs/config.info\n",
            "\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from N/A, git revision N/A\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /content/spades_output/corrected/configs/config.info\n",
            "  0:00:00.002     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.002     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 12 Gb\n",
            "  0:00:00.002     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
            "  0:00:00.005     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
            "  0:00:00.005     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
            "  0:00:00.005     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
            "     === ITERATION 0 begins ===\n",
            "  0:00:00.005     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:00:00.005     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 32 files using 2 threads. This might take a while.\n",
            "  0:00:00.005     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:00.005     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.99935 Gb\n",
            "  0:00:00.005     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2097152\n",
            "  0:00:00.006     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /content/SRR292770_1.fastq\n",
            "  0:00:22.571     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1151751 reads\n",
            "  0:00:41.437     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2301589 reads\n",
            "  0:01:05.532     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 3452535 reads\n",
            "  0:01:24.956     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 4604364 reads\n",
            "  0:01:35.054     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 5102041 reads\n",
            "  0:01:35.054     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /content/SRR292770_2.fastq\n",
            "  0:01:55.863     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 6252803 reads\n",
            "  0:02:17.402     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 7404737 reads\n",
            "  0:02:37.461     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 8556627 reads\n",
            "  0:02:58.909     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 9708905 reads\n",
            "  0:03:06.393     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /content/SRR292862_1.fastq\n",
            "  0:04:46.591     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /content/SRR292862_2.fastq\n",
            "  0:05:26.496     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 17608756 reads\n",
            "  0:06:15.778     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /content/SRR292678_1.fastq\n",
            "  0:10:10.165     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /content/SRR292678_2.fastq\n",
            "  0:14:12.877     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 32630388 reads processed\n",
            "  0:14:12.949     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:15:34.821     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 172000368 kmers in total.\n",
            "  0:15:34.822     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:15:39.921     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:16:40.580    84M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
            "  0:16:45.554    84M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 79767094 bytes occupied (3.71009 bits per kmer).\n",
            "  0:16:45.555    84M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
            "  0:17:27.026     2G / 2G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
            "  1:07:23.394     2G / 2G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
            "  1:10:46.529     2G / 8G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 112913170\n",
            "  1:10:46.634     1G / 8G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
            "  1:10:53.021     5G / 8G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /content/SRR292770_1.fastq\n",
            "  1:19:52.417     5G / 8G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /content/SRR292770_2.fastq\n",
            "  1:28:47.778     5G / 8G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /content/SRR292862_1.fastq\n",
            "  1:37:48.343     5G / 8G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /content/SRR292862_2.fastq\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!quast test_data/contigs_1.fasta \\\n",
        "               test_data/contigs_2.fasta \\\n",
        "               -r test_data/reference.fasta.gz \\\n",
        "               -g test_data/genes.gff"
      ],
      "metadata": {
        "id": "G-AC5tVWtGvb"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!less quast_results/latest/report.txt"
      ],
      "metadata": {
        "id": "R2AaRaiQ52pj"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "# 5. Genome Annotation"
      ],
      "metadata": {
        "id": "baQ4j_uutLc4"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!sudo apt-get install libdatetime-perl libxml-simple-perl libdigest-md5-perl git default-jre bioperl\n",
        "!sudo cpan Bio::Perl\n",
        "!git clone https://github.com/tseemann/prokka.git $HOME/prokka\n",
        "!$HOME/prokka/bin/prokka --setupdb"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ot13p0ZatM93",
        "outputId": "a74b4ed8-f003-45b3-8e7e-306e0ebb992d"
      },
      "execution_count": 32,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[1;30;43mВыходные данные были обрезаны до нескольких последних строк (5000).\u001b[0m\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         90\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         90\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"R -> P (in ND; dbSNP:rs104894867)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"R -> P (in ND; dbSNP:rs104894867)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:1307245\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:1307245\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_005494\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_005494\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         92\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         92\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"S -> P (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"S -> P (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:20340138\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:20340138\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_064012\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_064012\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         94..96\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         94..96\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"HCC -> QCGL (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"HCC -> QCGL (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_005495\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_005495\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         95\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         95\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"C -> F (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"C -> F (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:15609522\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:15609522\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_064013\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_064013\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         95\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         95\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"C -> R (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"C -> R (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:7662640\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:7662640\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_064014\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_064014\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         96\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         96\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"C -> W (in ND; dbSNP:rs104894877)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"C -> W (in ND; dbSNP:rs104894877)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:10484772\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:10484772\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_009275\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_009275\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         96\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         96\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"C -> Y (in ND; dbSNP:rs104894871)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"C -> Y (in ND; dbSNP:rs104894871)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:10544980,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:10544980,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   ECO:0000269|PubMed:1303264, ECO:0000269|PubMed:1307245\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   ECO:0000269|PubMed:1303264, ECO:0000269|PubMed:1307245\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_005496\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_005496\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         97\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         97\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"R -> P (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"R -> P (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:17325173\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:17325173\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_064015\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_064015\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         98\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         98\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"P -> L (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"P -> L (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:20340138\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:20340138\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_064016\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_064016\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         101\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         101\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"S -> F (in ND; dbSNP:rs104894883)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"S -> F (in ND; dbSNP:rs104894883)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:8990009\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:8990009\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_005497\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_005497\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         103\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         103\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"L -> V (in EVR2)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"L -> V (in EVR2)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:16970763\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:16970763\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_064017\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_064017\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         104\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         104\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"K -> N (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"K -> N (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:17128466\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:17128466\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_064018\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_064018\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         104\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         104\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"K -> Q (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"K -> Q (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:7795608\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:7795608\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_005498\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_005498\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         105\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         105\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"A -> T (in ND; dbSNP:rs104894875)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"A -> T (in ND; dbSNP:rs104894875)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:9382152\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:9382152\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_016050\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_016050\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         108\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         108\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"L -> P (in retinopathy of prematurity)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"L -> P (in retinopathy of prematurity)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:9143917\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:9143917\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_064019\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_064019\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         110\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         110\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"C -> G (in ND; dbSNP:rs104894876)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"C -> G (in ND; dbSNP:rs104894876)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:9382152\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:9382152\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_016051\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_016051\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         110\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         110\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"C -> R (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"C -> R (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:8807344\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:8807344\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_064020\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_064020\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         112\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         112\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"G -> E (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"G -> E (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:15776010\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:15776010\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_064021\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_064021\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         115\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         115\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"R -> L (in EVR2)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"R -> L (in EVR2)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:17325173\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:17325173\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_064022\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_064022\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         118\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         118\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"A -> D (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"A -> D (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:10544980\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:10544980\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_064023\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_064023\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         120\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         120\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Y -> C (in EVR2)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"Y -> C (in EVR2)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:9143917\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:9143917\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_005499\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_005499\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         121..123\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         121..123\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Missing (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"Missing (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:7627181\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:7627181\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_005503\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_005503\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         121\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         121\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"R -> G (in EVR2)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"R -> G (in EVR2)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_005500\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_005500\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         121\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         121\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"R -> L (in EVR2; dbSNP:rs137852220)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"R -> L (in EVR2; dbSNP:rs137852220)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:8946107\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:8946107\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_064024\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_064024\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         121\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         121\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"R -> Q (in EVR2 and ND; reduced amount of protein in\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"R -> Q (in EVR2 and ND; reduced amount of protein in\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   the extracellular matrix)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   the extracellular matrix)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:16163268,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:16163268,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   ECO:0000269|PubMed:7795608, ECO:0000269|PubMed:8281159,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   ECO:0000269|PubMed:7795608, ECO:0000269|PubMed:8281159,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   ECO:0000269|PubMed:9407136\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   ECO:0000269|PubMed:9407136\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_005501\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_005501\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         121\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         121\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"R -> W (in ND and EVR2; dbSNP:rs104894878)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"R -> W (in ND and EVR2; dbSNP:rs104894878)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:17296899,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:17296899,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   ECO:0000269|PubMed:7795608\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   ECO:0000269|PubMed:7795608\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_005502\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_005502\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         123\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         123\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"I -> N (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"I -> N (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:7627181\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:7627181\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_005504\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_005504\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         124\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         124\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"L -> F (in EVR2; dbSNP:rs28933684)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"L -> F (in EVR2; dbSNP:rs28933684)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:8252044\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:8252044\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_005505\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_005505\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         126\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         126\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"C -> S (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"C -> S (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:8741107\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:8741107\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_064025\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_064025\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         128\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   VARIANT         128\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"C -> R (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"C -> R (in ND)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:14635119\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:14635119\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_064026\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /id=\"VAR_064026\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MUTAGEN         95\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   MUTAGEN         95\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"C->A: Impairs oligomerization.\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /note=\"C->A: Impairs oligomerization.\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:9407136\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:9407136\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          37..48\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   STRAND          37..48\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:5BQ8\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:5BQ8\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          51..53\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   STRAND          51..53\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:5BQ8\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:5BQ8\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          58..67\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   STRAND          58..67\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:5BQ8\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:5BQ8\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          73..77\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   STRAND          73..77\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:5BQ8\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:5BQ8\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          80..83\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   STRAND          80..83\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:5BQB\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:5BQB\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          89..92\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   STRAND          89..92\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:5BQ8\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:5BQ8\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          94..110\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   STRAND          94..110\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:5BQ8\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:5BQ8\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   TURN            111..113\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   TURN            111..113\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:5BQ8\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:5BQ8\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          115..133\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT   STRAND          115..133\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:5BQ8\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq NDP_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:5BQ8\"\n",
            "\n",
            "---------------------------------------------------\n",
            "t/SwissProt.t ............ ok\n",
            "All tests successful.\n",
            "Files=6, Tests=23,  8 wallclock secs ( 0.03 usr  0.02 sys +  0.60 cusr  0.13 csys =  0.78 CPU)\n",
            "Result: PASS\n",
            "  CJFIELDS/Bio-DB-SwissProt-1.7.4.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for CJFIELDS/Bio-DB-SwissProt-1.7.4.tar.gz\n",
            "Manifying 1 pod document\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/SwissProt.pm\n",
            "Installing /usr/local/man/man3/Bio::DB::SwissProt.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  CJFIELDS/Bio-DB-SwissProt-1.7.4.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "Running install for module 'Bio::Tools::Run::RemoteBlast'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/C/CJ/CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz\n",
            "Checksum for /root/.cpan/sources/authors/id/C/CJ/CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz ok\n",
            "Configuring C/CJ/CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for Bio::Tools::Run::RemoteBlast\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for C/CJ/CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz\n",
            "cp lib/Bio/Tools/Run/RemoteBlast.pm blib/lib/Bio/Tools/Run/RemoteBlast.pm\n",
            "cp bin/bp_remote_blast.pl blib/script/bp_remote_blast.pl\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_remote_blast.pl\n",
            "Manifying 1 pod document\n",
            "Manifying 1 pod document\n",
            "  CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/00-compile.t .................... ok\n",
            "t/author-mojibake.t ............... skipped: these tests are for testing by the author\n",
            "t/author-no-tabs.t ................ skipped: these tests are for testing by the author\n",
            "t/author-pod-syntax.t ............. skipped: these tests are for testing by the author\n",
            "t/release-RemoteBlast.t ........... skipped: these tests are for release candidate testing\n",
            "t/release-RemoteBlast_rpsblast.t .. skipped: these tests are for release candidate testing\n",
            "All tests successful.\n",
            "Files=6, Tests=2,  0 wallclock secs ( 0.04 usr  0.01 sys +  0.26 cusr  0.05 csys =  0.36 CPU)\n",
            "Result: PASS\n",
            "  CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz\n",
            "Manifying 1 pod document\n",
            "Manifying 1 pod document\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Run/RemoteBlast.pm\n",
            "Installing /usr/local/man/man1/bp_remote_blast.pl.1p\n",
            "Installing /usr/local/man/man3/Bio::Tools::Run::RemoteBlast.3pm\n",
            "Installing /usr/local/bin/bp_remote_blast.pl\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "  CJFIELDS/Bio-Procedural-1.7.4.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/Bio-Procedural-1.7.4-0\n",
            "  CJFIELDS/Bio-Procedural-1.7.4.tar.gz\n",
            "  Has already been prepared\n",
            "Running make for C/CJ/CJFIELDS/Bio-Procedural-1.7.4.tar.gz\n",
            "cp lib/Bio/Perl.pm blib/lib/Bio/Perl.pm\n",
            "cp lib/Bio/Procedural.pm blib/lib/Bio/Procedural.pm\n",
            "Manifying 2 pod documents\n",
            "  CJFIELDS/Bio-Procedural-1.7.4.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for CJFIELDS/Bio-Procedural-1.7.4.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/00-compile.t ........... ok\n",
            "t/author-eol.t ........... skipped: these tests are for testing by the author\n",
            "t/author-mojibake.t ...... skipped: these tests are for testing by the author\n",
            "t/author-no-tabs.t ....... skipped: these tests are for testing by the author\n",
            "t/author-pod-coverage.t .. skipped: these tests are for testing by the author\n",
            "t/author-pod-syntax.t .... skipped: these tests are for testing by the author\n",
            "t/Perl.t ................. 4/31 \n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   CHAIN           1..372\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   CHAIN           1..372\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Heterogeneous nuclear ribonucleoprotein A1\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Heterogeneous nuclear ribonucleoprotein A1\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"PRO_0000424509\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /id=\"PRO_0000424509\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   INIT_MET        1\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   INIT_MET        1\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Removed; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Removed; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|Ref.9, ECO:0007744|PubMed:20068231,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|Ref.9, ECO:0007744|PubMed:20068231,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   ECO:0007744|PubMed:22814378\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   ECO:0007744|PubMed:22814378\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   CHAIN           2..372\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   CHAIN           2..372\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Heterogeneous nuclear ribonucleoprotein A1, N-\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Heterogeneous nuclear ribonucleoprotein A1, N-\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   terminally processed\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   terminally processed\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"PRO_0000081828\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /id=\"PRO_0000081828\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   DOMAIN          14..97\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   DOMAIN          14..97\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"RRM 1\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"RRM 1\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000255|PROSITE-ProRule:PRU00176\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000255|PROSITE-ProRule:PRU00176\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   DOMAIN          105..184\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   DOMAIN          105..184\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"RRM 2\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"RRM 2\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000255|PROSITE-ProRule:PRU00176\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000255|PROSITE-ProRule:PRU00176\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   REGION          4..94\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   REGION          4..94\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Globular A domain\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Globular A domain\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   REGION          95..185\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   REGION          95..185\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Globular B domain\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Globular B domain\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   REGION          182..216\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   REGION          182..216\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Disordered\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Disordered\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000256|SAM:MobiDB-lite\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000256|SAM:MobiDB-lite\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   REGION          218..240\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   REGION          218..240\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"RNA-binding RGG-box\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"RNA-binding RGG-box\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   REGION          317..372\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   REGION          317..372\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Disordered\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Disordered\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000256|SAM:MobiDB-lite\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000256|SAM:MobiDB-lite\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   REGION          320..357\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   REGION          320..357\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Nuclear targeting sequence (M9)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Nuclear targeting sequence (M9)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   COMPBIAS        184..199\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   COMPBIAS        184..199\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Polar residues\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Polar residues\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000256|SAM:MobiDB-lite\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000256|SAM:MobiDB-lite\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   COMPBIAS        355..372\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   COMPBIAS        355..372\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Polar residues\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Polar residues\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000256|SAM:MobiDB-lite\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000256|SAM:MobiDB-lite\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         1\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         1\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"N-acetylmethionine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"N-acetylmethionine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:22814378\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:22814378\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         2\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         2\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"N-acetylserine; in Heterogeneous nuclear\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"N-acetylserine; in Heterogeneous nuclear\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   ribonucleoprotein A1, N-terminally processed\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   ribonucleoprotein A1, N-terminally processed\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|Ref.9, ECO:0007744|PubMed:20068231,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|Ref.9, ECO:0007744|PubMed:20068231,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   ECO:0007744|PubMed:22814378\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   ECO:0007744|PubMed:22814378\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         2\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         2\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Phosphoserine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Phosphoserine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:20068231,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:20068231,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   ECO:0007744|PubMed:23186163\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   ECO:0007744|PubMed:23186163\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         3\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         3\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"N6-acetyllysine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"N6-acetyllysine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:19608861\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:19608861\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         4\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         4\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Phosphoserine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Phosphoserine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:20068231,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:20068231,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   ECO:0007744|PubMed:21406692, ECO:0007744|PubMed:23186163\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   ECO:0007744|PubMed:21406692, ECO:0007744|PubMed:23186163\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         6\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         6\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Phosphoserine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Phosphoserine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:18669648,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:18669648,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   ECO:0007744|PubMed:20068231, ECO:0007744|PubMed:21406692,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   ECO:0007744|PubMed:20068231, ECO:0007744|PubMed:21406692,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   ECO:0007744|PubMed:23186163\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   ECO:0007744|PubMed:23186163\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         22\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         22\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Phosphoserine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Phosphoserine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000250|UniProtKB:P49312\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000250|UniProtKB:P49312\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         192\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         192\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Phosphoserine; by MKNK2\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Phosphoserine; by MKNK2\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:16111636\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:16111636\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         194\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         194\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Asymmetric dimethylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Asymmetric dimethylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000250|UniProtKB:P09867\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000250|UniProtKB:P09867\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         194\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         194\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Dimethylated arginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Dimethylated arginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:15782174\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:15782174\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         194\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         194\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Omega-N-methylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Omega-N-methylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000250|UniProtKB:P49312\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000250|UniProtKB:P49312\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         199\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         199\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Phosphoserine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Phosphoserine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:21406692\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:21406692\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         206\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         206\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Asymmetric dimethylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Asymmetric dimethylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|Ref.9, ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|Ref.9, ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         206\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         206\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Dimethylated arginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Dimethylated arginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:15782174\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:15782174\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         206\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         206\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Omega-N-methylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Omega-N-methylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|Ref.9, ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|Ref.9, ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         218\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         218\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Asymmetric dimethylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Asymmetric dimethylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         218\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         218\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Omega-N-methylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Omega-N-methylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         225\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         225\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Asymmetric dimethylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Asymmetric dimethylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|Ref.9, ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|Ref.9, ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         225\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         225\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Dimethylated arginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Dimethylated arginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:15782174\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:15782174\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         225\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         225\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Omega-N-methylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Omega-N-methylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|Ref.9, ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|Ref.9, ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         232\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         232\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Asymmetric dimethylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Asymmetric dimethylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000250|UniProtKB:P04256\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000250|UniProtKB:P04256\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         232\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         232\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Omega-N-methylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Omega-N-methylarginine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         336\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         336\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Omega-N-methylarginine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Omega-N-methylarginine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         337\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         337\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Phosphoserine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Phosphoserine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:23186163\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:23186163\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         350\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         350\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"N6-acetyllysine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"N6-acetyllysine; alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:19608861\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:19608861\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         352\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         352\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Omega-N-methylarginine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Omega-N-methylarginine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         361\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         361\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Phosphoserine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Phosphoserine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:23186163\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:23186163\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         362\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         362\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Phosphoserine; by MKNK2\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Phosphoserine; by MKNK2\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:16111636\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:16111636\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         363\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         363\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Phosphoserine; by MKNK2\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Phosphoserine; by MKNK2\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:16111636\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:16111636\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         364\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         364\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Phosphoserine; by MKNK2\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Phosphoserine; by MKNK2\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:16111636,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:16111636,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   ECO:0007744|PubMed:23186163\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   ECO:0007744|PubMed:23186163\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         365\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         365\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Phosphoserine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Phosphoserine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:21406692,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:21406692,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   ECO:0007744|PubMed:23186163\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   ECO:0007744|PubMed:23186163\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         368\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         368\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Phosphoserine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Phosphoserine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:18669648,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:18669648,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   ECO:0007744|PubMed:21406692, ECO:0007744|PubMed:23186163\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   ECO:0007744|PubMed:21406692, ECO:0007744|PubMed:23186163\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MOD_RES         370\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MOD_RES         370\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Omega-N-methylarginine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Omega-N-methylarginine\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:24129315\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   CROSSLNK        3\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   CROSSLNK        3\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Glycyl lysine isopeptide (Lys-Gly) (interchain with\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Glycyl lysine isopeptide (Lys-Gly) (interchain with\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   G-Cter in SUMO2); alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   G-Cter in SUMO2); alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:28112733\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:28112733\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   CROSSLNK        8\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   CROSSLNK        8\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Glycyl lysine isopeptide (Lys-Gly) (interchain with\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Glycyl lysine isopeptide (Lys-Gly) (interchain with\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   G-Cter in SUMO2)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   G-Cter in SUMO2)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:25755297,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:25755297,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   ECO:0007744|PubMed:28112733\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   ECO:0007744|PubMed:28112733\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   CROSSLNK        78\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   CROSSLNK        78\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Glycyl lysine isopeptide (Lys-Gly) (interchain with\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Glycyl lysine isopeptide (Lys-Gly) (interchain with\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   G-Cter in SUMO2)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   G-Cter in SUMO2)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:28112733\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:28112733\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   CROSSLNK        113\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   CROSSLNK        113\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Glycyl lysine isopeptide (Lys-Gly) (interchain with\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Glycyl lysine isopeptide (Lys-Gly) (interchain with\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   G-Cter in SUMO)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   G-Cter in SUMO)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:15161980\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:15161980\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   CROSSLNK        179\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   CROSSLNK        179\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Glycyl lysine isopeptide (Lys-Gly) (interchain with\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Glycyl lysine isopeptide (Lys-Gly) (interchain with\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   G-Cter in SUMO2)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   G-Cter in SUMO2)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:28112733\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:28112733\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   CROSSLNK        183\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   CROSSLNK        183\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Glycyl lysine isopeptide (Lys-Gly) (interchain with\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Glycyl lysine isopeptide (Lys-Gly) (interchain with\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   G-Cter in SUMO2)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   G-Cter in SUMO2)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:28112733\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:28112733\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   CROSSLNK        350\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   CROSSLNK        350\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Glycyl lysine isopeptide (Lys-Gly) (interchain with\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Glycyl lysine isopeptide (Lys-Gly) (interchain with\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   G-Cter in SUMO2); alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   G-Cter in SUMO2); alternate\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007744|PubMed:28112733\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007744|PubMed:28112733\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VAR_SEQ         203..307\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   VAR_SEQ         203..307\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Missing (in isoform 2)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Missing (in isoform 2)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000303|PubMed:15489334\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000303|PubMed:15489334\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VSP_034076\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /id=\"VSP_034076\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VAR_SEQ         252..303\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   VAR_SEQ         252..303\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Missing (in isoform A1-A)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Missing (in isoform A1-A)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000303|PubMed:14702039,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000303|PubMed:14702039,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   ECO:0000303|PubMed:15489334, ECO:0000303|PubMed:2836799,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   ECO:0000303|PubMed:15489334, ECO:0000303|PubMed:2836799,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   ECO:0000303|Ref.3\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   ECO:0000303|Ref.3\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VSP_005824\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /id=\"VSP_005824\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         277\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   VARIANT         277\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Q -> K (in ALS20; unknown pathological\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Q -> K (in ALS20; unknown pathological\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   significance)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   significance)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:27694260\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:27694260\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_077531\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /id=\"VAR_077531\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         283\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   VARIANT         283\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"G -> R (in dbSNP:rs375259222)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"G -> R (in dbSNP:rs375259222)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:27694260\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:27694260\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_077532\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /id=\"VAR_077532\"\n",
            "\n",
            "---------------------------------------------------\n",
            "t/Perl.t ................. 20/31 \n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         314\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   VARIANT         314\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"D -> N (in ALS20; dbSNP:rs397518453)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"D -> N (in ALS20; dbSNP:rs397518453)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:23455423\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:23455423\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_070588\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /id=\"VAR_070588\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         314\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   VARIANT         314\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"D -> V (in IBMPFD3; reduces binding to UBQLN2;\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"D -> V (in IBMPFD3; reduces binding to UBQLN2;\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   dbSNP:rs397518452)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   dbSNP:rs397518452)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:23455423,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:23455423,\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   ECO:0000269|PubMed:25616961\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   ECO:0000269|PubMed:25616961\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_070589\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /id=\"VAR_070589\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         319\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   VARIANT         319\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"N -> S (in ALS20; dbSNP:rs397518454)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"N -> S (in ALS20; dbSNP:rs397518454)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:23455423\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:23455423\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_070590\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /id=\"VAR_070590\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   VARIANT         340\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   VARIANT         340\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"P -> S (in ALS20; increases subcellular localization\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"P -> S (in ALS20; increases subcellular localization\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   of HNRNPA1 in cytoplasmic inclusions with stress granules)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   of HNRNPA1 in cytoplasmic inclusions with stress granules)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:27694260\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:27694260\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /id=\"VAR_077533\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /id=\"VAR_077533\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MUTAGEN         218\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MUTAGEN         218\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"R->A: Abolishes interaction with HOXB-AS3 peptide;\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"R->A: Abolishes interaction with HOXB-AS3 peptide;\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   when associated with A-225 and A-232.\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   when associated with A-225 and A-232.\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:28985503\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:28985503\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MUTAGEN         225\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MUTAGEN         225\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"R->A: Abolishes interaction with HOXB-AS3 peptide;\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"R->A: Abolishes interaction with HOXB-AS3 peptide;\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   when associated with A-218 and A-232.\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   when associated with A-218 and A-232.\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:28985503\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:28985503\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MUTAGEN         232\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MUTAGEN         232\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"R->A: Abolishes interaction with HOXB-AS3 peptide;\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"R->A: Abolishes interaction with HOXB-AS3 peptide;\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   when associated with A-218 and A-225.\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   when associated with A-218 and A-225.\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:28985503\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:28985503\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MUTAGEN         326\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MUTAGEN         326\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"G->A: No nuclear import nor export.\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"G->A: No nuclear import nor export.\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:7730395\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:7730395\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MUTAGEN         327\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MUTAGEN         327\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"P->A: No nuclear import nor export.\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"P->A: No nuclear import nor export.\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:7730395\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:7730395\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   MUTAGEN         334..335\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   MUTAGEN         334..335\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"GG->LL: Normal nuclear import and export.\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"GG->LL: Normal nuclear import and export.\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000269|PubMed:7730395\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000269|PubMed:7730395\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   CONFLICT        128\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   CONFLICT        128\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"Y -> F (in Ref. 2; CAA29922)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"Y -> F (in Ref. 2; CAA29922)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000305\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000305\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   CONFLICT        140\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   CONFLICT        140\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"R -> P (in Ref. 8; CAA27874)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"R -> P (in Ref. 8; CAA27874)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000305\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000305\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   CONFLICT        146\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   CONFLICT        146\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /note=\"R -> K (in Ref. 2; CAA29922)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /note=\"R -> K (in Ref. 2; CAA29922)\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0000305\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0000305\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   HELIX           11..14\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   HELIX           11..14\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          15..20\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   STRAND          15..20\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   HELIX           27..34\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   HELIX           27..34\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   HELIX           35..37\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   HELIX           35..37\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          40..47\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   STRAND          40..47\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   TURN            49..51\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   TURN            49..51\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          54..64\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   STRAND          54..64\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   HELIX           65..73\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   HELIX           65..73\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          76..79\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   STRAND          76..79\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1U1R\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1U1R\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          85..88\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   STRAND          85..88\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   HELIX           94..96\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   HELIX           94..96\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1U1Q\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1U1Q\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   TURN            98..101\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   TURN            98..101\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1U1Q\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1U1Q\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          105..110\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   STRAND          105..110\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   TURN            113..115\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   TURN            113..115\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   HELIX           118..125\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   HELIX           118..125\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   TURN            126..128\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   TURN            126..128\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          131..138\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   STRAND          131..138\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   TURN            140..142\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   TURN            140..142\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          145..155\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   STRAND          145..155\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   HELIX           156..164\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   HELIX           156..164\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          168..170\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   STRAND          168..170\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1U1Q\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1U1Q\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          176..179\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   STRAND          176..179\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1L3K\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   HELIX           183..188\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   HELIX           183..188\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:1U1Q\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:1U1Q\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          305..307\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   STRAND          305..307\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:7BX7\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:7BX7\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          312..314\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   STRAND          312..314\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:7BX7\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:7BX7\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          317..326\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   STRAND          317..326\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:7BX7\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:7BX7\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT   STRAND          330..333\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT   STRAND          330..333\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: No feature key in putative feature table line: FT                   /evidence=\"ECO:0007829|PDB:7BX7\"\n",
            "\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: failed to parse feature table line for seq ROA1_HUMAN\n",
            "FT                   /evidence=\"ECO:0007829|PDB:7BX7\"\n",
            "\n",
            "---------------------------------------------------\n",
            "t/Perl.t ................. ok\n",
            "All tests successful.\n",
            "Files=7, Tests=33, 140 wallclock secs ( 0.04 usr  0.01 sys +  1.08 cusr  0.25 csys =  1.38 CPU)\n",
            "Result: PASS\n",
            "  CJFIELDS/Bio-Procedural-1.7.4.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for CJFIELDS/Bio-Procedural-1.7.4.tar.gz\n",
            "Manifying 2 pod documents\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Perl.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Procedural.pm\n",
            "Installing /usr/local/man/man3/Bio::Perl.3pm\n",
            "Installing /usr/local/man/man3/Bio::Procedural.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  CJFIELDS/Bio-Procedural-1.7.4.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "Cloning into '/root/prokka'...\n",
            "remote: Enumerating objects: 1478, done.\u001b[K\n",
            "remote: Total 1478 (delta 0), reused 0 (delta 0), pack-reused 1478\u001b[K\n",
            "Receiving objects: 100% (1478/1478), 321.93 MiB | 16.99 MiB/s, done.\n",
            "Resolving deltas: 100% (901/901), done.\n",
            "Updating files: 100% (67/67), done.\n",
            "[11:53:15] Appending to PATH: /root/prokka/bin/../binaries/linux\n",
            "[11:53:15] Appending to PATH: /root/prokka/bin/../binaries/linux/../common\n",
            "[11:53:15] Appending to PATH: /root/prokka/bin\n",
            "[11:53:15] Cleaning databases in /root/prokka/db\n",
            "[11:53:15] Cleaning complete.\n",
            "[11:53:15] Looking for 'makeblastdb' - found /usr/bin/makeblastdb\n",
            "[11:53:16] Determined makeblastdb version is 002009 from 'makeblastdb: 2.9.0+'\n",
            "[11:53:16] Making kingdom BLASTP database: /root/prokka/db/kingdom/Archaea/sprot\n",
            "[11:53:16] Running: makeblastdb -hash_index -dbtype prot -in \\/root\\/prokka\\/db\\/kingdom\\/Archaea\\/sprot -logfile /dev/null\n",
            "[11:53:16] Making kingdom BLASTP database: /root/prokka/db/kingdom/Bacteria/sprot\n",
            "[11:53:16] Running: makeblastdb -hash_index -dbtype prot -in \\/root\\/prokka\\/db\\/kingdom\\/Bacteria\\/sprot -logfile /dev/null\n",
            "[11:53:17] Making kingdom BLASTP database: /root/prokka/db/kingdom/Mitochondria/sprot\n",
            "[11:53:17] Running: makeblastdb -hash_index -dbtype prot -in \\/root\\/prokka\\/db\\/kingdom\\/Mitochondria\\/sprot -logfile /dev/null\n",
            "[11:53:17] Making kingdom BLASTP database: /root/prokka/db/kingdom/Viruses/sprot\n",
            "[11:53:17] Running: makeblastdb -hash_index -dbtype prot -in \\/root\\/prokka\\/db\\/kingdom\\/Viruses\\/sprot -logfile /dev/null\n",
            "[11:53:17] Making kingdom BLASTP database: /root/prokka/db/kingdom/Bacteria/IS\n",
            "[11:53:17] Running: makeblastdb -hash_index -dbtype prot -in \\/root\\/prokka\\/db\\/kingdom\\/Bacteria\\/IS -logfile /dev/null\n",
            "[11:53:18] Making kingdom BLASTP database: /root/prokka/db/kingdom/Bacteria/AMR\n",
            "[11:53:18] Running: makeblastdb -hash_index -dbtype prot -in \\/root\\/prokka\\/db\\/kingdom\\/Bacteria\\/AMR -logfile /dev/null\n",
            "[11:53:18] Making genus BLASTP database: /root/prokka/db/genus/Enterococcus\n",
            "[11:53:18] Running: makeblastdb -hash_index -dbtype prot -in \\/root\\/prokka\\/db\\/genus\\/Enterococcus -logfile /dev/null\n",
            "[11:53:18] Making genus BLASTP database: /root/prokka/db/genus/Escherichia\n",
            "[11:53:18] Running: makeblastdb -hash_index -dbtype prot -in \\/root\\/prokka\\/db\\/genus\\/Escherichia -logfile /dev/null\n",
            "[11:53:18] Making genus BLASTP database: /root/prokka/db/genus/Staphylococcus\n",
            "[11:53:18] Running: makeblastdb -hash_index -dbtype prot -in \\/root\\/prokka\\/db\\/genus\\/Staphylococcus -logfile /dev/null\n",
            "[11:53:18] Looking for 'hmmpress' - found /usr/bin/hmmpress\n",
            "[11:53:18] Determined hmmpress version is 003003 from '# HMMER 3.3 (Nov 2019); http://hmmer.org/'\n",
            "[11:53:18] Pressing HMM database: /root/prokka/db/hmm/HAMAP.hmm\n",
            "[11:53:18] Running: hmmpress \\/root\\/prokka\\/db\\/hmm\\/HAMAP\\.hmm\n",
            "Working...    done.\n",
            "Pressed and indexed 1536 HMMs (1536 names).\n",
            "Models pressed into binary file:   /root/prokka/db/hmm/HAMAP.hmm.h3m\n",
            "SSI index for binary model file:   /root/prokka/db/hmm/HAMAP.hmm.h3i\n",
            "Profiles (MSV part) pressed into:  /root/prokka/db/hmm/HAMAP.hmm.h3f\n",
            "Profiles (remainder) pressed into: /root/prokka/db/hmm/HAMAP.hmm.h3p\n",
            "[11:53:20] Looking for 'cmpress' - found /usr/bin/cmpress\n",
            "[11:53:20] Determined cmpress version is 001001 from '# INFERNAL 1.1.3 (Nov 2019)'\n",
            "[11:53:20] Pressing CM database: /root/prokka/db/cm/Viruses\n",
            "[11:53:20] Running: cmpress \\/root\\/prokka\\/db\\/cm\\/Viruses\n",
            "Working...    done.\n",
            "Pressed and indexed 239 CMs and p7 HMM filters (239 names and 239 accessions).\n",
            "Covariance models and p7 filters pressed into binary file:  /root/prokka/db/cm/Viruses.i1m\n",
            "SSI index for binary covariance model file:                 /root/prokka/db/cm/Viruses.i1i\n",
            "Optimized p7 filter profiles (MSV part)  pressed into:      /root/prokka/db/cm/Viruses.i1f\n",
            "Optimized p7 filter profiles (remainder) pressed into:      /root/prokka/db/cm/Viruses.i1p\n",
            "[11:53:20] Pressing CM database: /root/prokka/db/cm/Bacteria\n",
            "[11:53:20] Running: cmpress \\/root\\/prokka\\/db\\/cm\\/Bacteria\n",
            "Working...    done.\n",
            "Pressed and indexed 1127 CMs and p7 HMM filters (1127 names and 1127 accessions).\n",
            "Covariance models and p7 filters pressed into binary file:  /root/prokka/db/cm/Bacteria.i1m\n",
            "SSI index for binary covariance model file:                 /root/prokka/db/cm/Bacteria.i1i\n",
            "Optimized p7 filter profiles (MSV part)  pressed into:      /root/prokka/db/cm/Bacteria.i1f\n",
            "Optimized p7 filter profiles (remainder) pressed into:      /root/prokka/db/cm/Bacteria.i1p\n",
            "[11:53:21] Pressing CM database: /root/prokka/db/cm/Archaea\n",
            "[11:53:21] Running: cmpress \\/root\\/prokka\\/db\\/cm\\/Archaea\n",
            "Working...    done.\n",
            "Pressed and indexed 152 CMs and p7 HMM filters (152 names and 152 accessions).\n",
            "Covariance models and p7 filters pressed into binary file:  /root/prokka/db/cm/Archaea.i1m\n",
            "SSI index for binary covariance model file:                 /root/prokka/db/cm/Archaea.i1i\n",
            "Optimized p7 filter profiles (MSV part)  pressed into:      /root/prokka/db/cm/Archaea.i1f\n",
            "Optimized p7 filter profiles (remainder) pressed into:      /root/prokka/db/cm/Archaea.i1p\n",
            "[11:53:21] Looking for databases in: /root/prokka/db\n",
            "[11:53:21] * Kingdoms: Archaea Bacteria Mitochondria Viruses\n",
            "[11:53:21] * Genera: Enterococcus Escherichia Staphylococcus\n",
            "[11:53:21] * HMMs: HAMAP\n",
            "[11:53:21] * CMs: Archaea Bacteria Viruses\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!$HOME/prokka/bin/prokka"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ixSfeDzbtXIg",
        "outputId": "88c7c85c-38b4-4fd9-e5d7-b1d09ffc638f"
      },
      "execution_count": 34,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Name:\n",
            "  Prokka 1.14.6 by Torsten Seemann <torsten.seemann@gmail.com>\n",
            "Synopsis:\n",
            "  rapid bacterial genome annotation\n",
            "Usage:\n",
            "  prokka [options] <contigs.fasta>\n",
            "General:\n",
            "  --help             This help\n",
            "  --version          Print version and exit\n",
            "  --citation         Print citation for referencing Prokka\n",
            "  --quiet            No screen output (default OFF)\n",
            "  --debug            Debug mode: keep all temporary files (default OFF)\n",
            "Setup:\n",
            "  --dbdir [X]        Prokka database root folders (default '/root/prokka/db')\n",
            "  --listdb           List all configured databases\n",
            "  --setupdb          Index all installed databases\n",
            "  --cleandb          Remove all database indices\n",
            "  --depends          List all software dependencies\n",
            "Outputs:\n",
            "  --outdir [X]       Output folder [auto] (default '')\n",
            "  --force            Force overwriting existing output folder (default OFF)\n",
            "  --prefix [X]       Filename output prefix [auto] (default '')\n",
            "  --addgenes         Add 'gene' features for each 'CDS' feature (default OFF)\n",
            "  --addmrna          Add 'mRNA' features for each 'CDS' feature (default OFF)\n",
            "  --locustag [X]     Locus tag prefix [auto] (default '')\n",
            "  --increment [N]    Locus tag counter increment (default '1')\n",
            "  --gffver [N]       GFF version (default '3')\n",
            "  --compliant        Force Genbank/ENA/DDJB compliance: --addgenes --mincontiglen 200 --centre XXX (default OFF)\n",
            "  --centre [X]       Sequencing centre ID. (default '')\n",
            "  --accver [N]       Version to put in Genbank file (default '1')\n",
            "Organism details:\n",
            "  --genus [X]        Genus name (default 'Genus')\n",
            "  --species [X]      Species name (default 'species')\n",
            "  --strain [X]       Strain name (default 'strain')\n",
            "  --plasmid [X]      Plasmid name or identifier (default '')\n",
            "Annotations:\n",
            "  --kingdom [X]      Annotation mode: Archaea|Bacteria|Mitochondria|Viruses (default 'Bacteria')\n",
            "  --gcode [N]        Genetic code / Translation table (set if --kingdom is set) (default '0')\n",
            "  --prodigaltf [X]   Prodigal training file (default '')\n",
            "  --gram [X]         Gram: -/neg +/pos (default '')\n",
            "  --usegenus         Use genus-specific BLAST databases (needs --genus) (default OFF)\n",
            "  --proteins [X]     FASTA or GBK file to use as 1st priority (default '')\n",
            "  --hmms [X]         Trusted HMM to first annotate from (default '')\n",
            "  --metagenome       Improve gene predictions for highly fragmented genomes (default OFF)\n",
            "  --rawproduct       Do not clean up /product annotation (default OFF)\n",
            "  --cdsrnaolap       Allow [tr]RNA to overlap CDS (default OFF)\n",
            "Matching:\n",
            "  --evalue [n.n]     Similarity e-value cut-off (default '1e-09')\n",
            "  --coverage [n.n]   Minimum coverage on query protein (default '80')\n",
            "Computation:\n",
            "  --cpus [N]         Number of CPUs to use [0=all] (default '8')\n",
            "  --fast             Fast mode - only use basic BLASTP databases (default OFF)\n",
            "  --noanno           For CDS just set /product=\"unannotated protein\" (default OFF)\n",
            "  --mincontiglen [N] Minimum contig size [NCBI needs 200] (default '1')\n",
            "  --rfam             Enable searching for ncRNAs with Infernal+Rfam (SLOW!) (default '0')\n",
            "  --norrna           Don't run rRNA search (default OFF)\n",
            "  --notrna           Don't run tRNA search (default OFF)\n",
            "  --rnammer          Prefer RNAmmer over Barrnap for rRNA prediction (default OFF)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# 6. Finding the closest relative of E. coli X"
      ],
      "metadata": {
        "id": "EpwLZccbtjYA"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!git clone https://github.com/tseemann/barrnap.git\n",
        "cd barrnap/bin\n",
        "./barrnap --help"
      ],
      "metadata": {
        "id": "4UFAq2D8tubO"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "# 7. What is the genetic cause of HUS?"
      ],
      "metadata": {
        "id": "F36dS8kEtvR6"
      }
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "3iUcy38ht505"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}