{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/Lynxlazy/genomics-data-analysis/blob/main/Project2.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "QT601tcUc912",
        "outputId": "f551122a-4120-476d-d783-2c9842946f7d"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-03-04 06:43:02--  https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-ubuntu64.tar.gz\n",
            "Resolving ftp-trace.ncbi.nlm.nih.gov (ftp-trace.ncbi.nlm.nih.gov)... 130.14.250.13, 130.14.250.11, 2607:f220:41e:250::13, ...\n",
            "Connecting to ftp-trace.ncbi.nlm.nih.gov (ftp-trace.ncbi.nlm.nih.gov)|130.14.250.13|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 86400775 (82M) [application/x-gzip]\n",
            "Saving to: ‘sratoolkit.tar.gz’\n",
            "\n",
            "sratoolkit.tar.gz   100%[===================>]  82.40M  78.7MB/s    in 1.0s    \n",
            "\n",
            "2023-03-04 06:43:04 (78.7 MB/s) - ‘sratoolkit.tar.gz’ saved [86400775/86400775]\n",
            "\n",
            "sratoolkit.3.0.1-ubuntu64/\n",
            "sratoolkit.3.0.1-ubuntu64/README.md\n",
            "sratoolkit.3.0.1-ubuntu64/schema/\n",
            "sratoolkit.3.0.1-ubuntu64/schema/insdc/\n",
            "sratoolkit.3.0.1-ubuntu64/schema/insdc/sra.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/insdc/insdc.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/insdc/seq.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/align/\n",
            "sratoolkit.3.0.1-ubuntu64/schema/align/refseq.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/align/qstat.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/align/pileup-stats.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/align/mate-cache.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/align/align.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/align/seq.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/helicos.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/454.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/pacbio.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/generic-fastq.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/nanopore.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/pevents.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/ion-torrent.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/illumina.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/sra/abi.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/vdb/\n",
            "sratoolkit.3.0.1-ubuntu64/schema/vdb/built-in.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/vdb/vdb.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/sra.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/seq-graph.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/spotname.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/wgs-contig.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/pnbrdb.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/varloc.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/clip.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/seq.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/ncbi.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/stats.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/ncbi/trace.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/csra2/\n",
            "sratoolkit.3.0.1-ubuntu64/schema/csra2/csra2.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/csra2/read.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/csra2/reference.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/schema/csra2/stats.vschema\n",
            "sratoolkit.3.0.1-ubuntu64/CHANGES\n",
            "sratoolkit.3.0.1-ubuntu64/bin/\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-decrypt.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-encrypt\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fastq-dump.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-sort-cg.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fasterq-dump-orig.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-validate.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/srapath.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/cg-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/illumina-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/abi-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/kdbmeta\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-unlock.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/srf-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/abi-dump.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/srf-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/dump-ref-fasta.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-config\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-copy.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-search.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/test-sra\n",
            "sratoolkit.3.0.1-ubuntu64/bin/abi-dump.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-lock.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/srapath.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-dump\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-dump.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/align-info.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/cg-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/kar.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/srapath-orig.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-pileup\n",
            "sratoolkit.3.0.1-ubuntu64/bin/dump-ref-fasta\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-pileup.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fasterq-dump.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/helicos-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fastq-dump\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sam-dump.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/kar.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/align-info.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sam-dump.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sff-dump.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-config.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/bam-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-copy\n",
            "sratoolkit.3.0.1-ubuntu64/bin/bam-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/helicos-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-pileup.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-stat\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-lock\n",
            "sratoolkit.3.0.1-ubuntu64/bin/illumina-dump\n",
            "sratoolkit.3.0.1-ubuntu64/bin/pacbio-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fastq-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-copy.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/dump-ref-fasta.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/cache-mgr.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/pacbio-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/abi-dump\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-encrypt.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/cg-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sff-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/abi-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-stat.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-unlock.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/srapath\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sff-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/kar\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-sort.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-stat.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/ncbi/\n",
            "sratoolkit.3.0.1-ubuntu64/bin/ncbi/default.kfg\n",
            "sratoolkit.3.0.1-ubuntu64/bin/ncbi/vdb-copy.kfg\n",
            "sratoolkit.3.0.1-ubuntu64/bin/ncbi/certs.kfg\n",
            "sratoolkit.3.0.1-ubuntu64/bin/kdbmeta.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fastq-dump-orig.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-config.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-search\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sam-dump-orig.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/illumina-dump.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/illumina-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/abi-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/test-sra.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/test-sra.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fastq-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/prefetch-orig.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sratools.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/cache-mgr.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fasterq-dump\n",
            "sratoolkit.3.0.1-ubuntu64/bin/illumina-dump.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fastq-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/helicos-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/latf-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-validate\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-lock.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/srf-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/kdbmeta.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/cache-mgr\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sff-dump.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/rcexplain.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-sort.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-search.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fastq-dump.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/bam-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-pileup-orig.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-decrypt\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-sort-cg\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-decrypt.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-unlock\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-encrypt.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-validate.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/align-info\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sff-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-dump-orig.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/prefetch\n",
            "sratoolkit.3.0.1-ubuntu64/bin/rcexplain.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sam-dump\n",
            "sratoolkit.3.0.1-ubuntu64/bin/prefetch.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sff-dump\n",
            "sratoolkit.3.0.1-ubuntu64/bin/rcexplain\n",
            "sratoolkit.3.0.1-ubuntu64/bin/fasterq-dump.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/illumina-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/vdb-dump.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/latf-load.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-sort\n",
            "sratoolkit.3.0.1-ubuntu64/bin/prefetch.3\n",
            "sratoolkit.3.0.1-ubuntu64/bin/sra-sort-cg.3.0.1\n",
            "sratoolkit.3.0.1-ubuntu64/bin/pacbio-load\n",
            "sratoolkit.3.0.1-ubuntu64/bin/latf-load.3\n",
            "sratoolkit.3.0.1-ubuntu64/example/\n",
            "sratoolkit.3.0.1-ubuntu64/example/perl/\n",
            "sratoolkit.3.0.1-ubuntu64/example/perl/base-stats.pl\n",
            "sratoolkit.3.0.1-ubuntu64/example/perl/mismatch-stats.pl\n",
            "sratoolkit.3.0.1-ubuntu64/example/perl/gene-lookup.pl\n",
            "sratoolkit.3.0.1-ubuntu64/example/perl/splitfastq.pl\n",
            "sratoolkit.3.0.1-ubuntu64/example/perl/quality-stats.pl\n",
            "sratoolkit.3.0.1-ubuntu64/example/perl/simplefastq.pl\n",
            "sratoolkit.3.0.1-ubuntu64/example/perl/dump-reference.pl\n",
            "sratoolkit.3.0.1-ubuntu64/README-blastn\n",
            "sratoolkit.3.0.1-ubuntu64/README-vdb-config\n"
          ]
        }
      ],
      "source": [
        "!wget --output-document sratoolkit.tar.gz https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-ubuntu64.tar.gz\n",
        "!tar -vxzf sratoolkit.tar.gz\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "vkkUS4ARe5o2"
      },
      "outputs": [],
      "source": [
        "!export PATH=$PATH:$PWD/sratoolkit.3.0.1-ubuntu64/bin\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "QqN3noYueLkj"
      },
      "source": [
        "# 1. Exploring the dataset"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "n7NJm1MLfNEv",
        "outputId": "660ef941-fd80-4059-ba05-7e1f1320559b"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "spots read      : 6,111,112\n",
            "reads read      : 12,222,224\n",
            "reads written   : 12,222,224\n",
            "spots read      : 5,102,041\n",
            "reads read      : 10,204,082\n",
            "reads written   : 10,204,082\n",
            "spots read      : 5,102,041\n",
            "reads read      : 10,204,082\n",
            "reads written   : 10,204,082\n"
          ]
        }
      ],
      "source": [
        "!./sratoolkit.3.0.1-ubuntu64/bin/fasterq-dump SRR292678\n",
        "!./sratoolkit.3.0.1-ubuntu64/bin/fasterq-dump SRR292862\n",
        "!./sratoolkit.3.0.1-ubuntu64/bin/fasterq-dump SRR292770"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "pB5h8KoIglO_"
      },
      "source": [
        "# 2. K-mer profile and genome size estimation\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "_lksBntbgjpZ",
        "outputId": "ebadb7a1-d99b-404e-8bde-2719089387e1"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  libhts3 libjellyfish-2.0-2\n",
            "The following NEW packages will be installed:\n",
            "  jellyfish libhts3 libjellyfish-2.0-2\n",
            "0 upgraded, 3 newly installed, 0 to remove and 22 not upgraded.\n",
            "Need to get 771 kB of archives.\n",
            "After this operation, 1,727 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 libhts3 amd64 1.10.2-3ubuntu0.1 [350 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu focal/universe amd64 libjellyfish-2.0-2 amd64 2.3.0-4build1 [60.5 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu focal/universe amd64 jellyfish amd64 2.3.0-4build1 [361 kB]\n",
            "Fetched 771 kB in 1s (892 kB/s)\n",
            "debconf: unable to initialize frontend: Dialog\n",
            "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 3.)\n",
            "debconf: falling back to frontend: Readline\n",
            "debconf: unable to initialize frontend: Readline\n",
            "debconf: (This frontend requires a controlling tty.)\n",
            "debconf: falling back to frontend: Teletype\n",
            "dpkg-preconfigure: unable to re-open stdin: \n",
            "Selecting previously unselected package libhts3:amd64.\n",
            "(Reading database ... 128215 files and directories currently installed.)\n",
            "Preparing to unpack .../libhts3_1.10.2-3ubuntu0.1_amd64.deb ...\n",
            "Unpacking libhts3:amd64 (1.10.2-3ubuntu0.1) ...\n",
            "Selecting previously unselected package libjellyfish-2.0-2:amd64.\n",
            "Preparing to unpack .../libjellyfish-2.0-2_2.3.0-4build1_amd64.deb ...\n",
            "Unpacking libjellyfish-2.0-2:amd64 (2.3.0-4build1) ...\n",
            "Selecting previously unselected package jellyfish.\n",
            "Preparing to unpack .../jellyfish_2.3.0-4build1_amd64.deb ...\n",
            "Unpacking jellyfish (2.3.0-4build1) ...\n",
            "Setting up libhts3:amd64 (1.10.2-3ubuntu0.1) ...\n",
            "Setting up libjellyfish-2.0-2:amd64 (2.3.0-4build1) ...\n",
            "Setting up jellyfish (2.3.0-4build1) ...\n",
            "Processing triggers for man-db (2.9.1-1) ...\n",
            "Processing triggers for libc-bin (2.31-0ubuntu9.9) ...\n"
          ]
        }
      ],
      "source": [
        "!sudo apt install jellyfish"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "-CEjfL77goB0"
      },
      "outputs": [],
      "source": [
        "!jellyfish count -m 21 -s 100M -t 10 -C SRR292678_1.fastq\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!jellyfish histo -o mer_counts.histo mer_counts.jf\n"
      ],
      "metadata": {
        "id": "T5rrNa3iWnMf"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install rpy2==3.4.2\n",
        "%reload_ext rpy2.ipython"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "I_rXvDxVXd50",
        "outputId": "32d29df3-4a06-418c-968e-f28a1aeb9cce"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Requirement already satisfied: rpy2==3.4.2 in /usr/local/lib/python3.8/dist-packages (3.4.2)\n",
            "Requirement already satisfied: cffi>=1.10.0 in /usr/local/lib/python3.8/dist-packages (from rpy2==3.4.2) (1.15.1)\n",
            "Requirement already satisfied: jinja2 in /usr/local/lib/python3.8/dist-packages (from rpy2==3.4.2) (3.1.2)\n",
            "Requirement already satisfied: tzlocal in /usr/local/lib/python3.8/dist-packages (from rpy2==3.4.2) (1.5.1)\n",
            "Requirement already satisfied: pytz in /usr/local/lib/python3.8/dist-packages (from rpy2==3.4.2) (2022.7.1)\n",
            "Requirement already satisfied: pycparser in /usr/local/lib/python3.8/dist-packages (from cffi>=1.10.0->rpy2==3.4.2) (2.21)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.8/dist-packages (from jinja2->rpy2==3.4.2) (2.1.2)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wc -l mer_counts.histo"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "mbsfQxcKdVmV",
        "outputId": "2bf005f9-cc2f-4067-b2bb-a9c2e4326503"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "1134 mer_counts.histo\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%R\n",
        "mer_counts <- read.table(\"mer_counts.histo\")\n",
        "plot(mer_counts[5:200,],type=\"l\")\n",
        "points(mer_counts[16:200,])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 497
        },
        "id": "gonHnn2RXG-Z",
        "outputId": "a184e48b-c4e8-4441-907d-ac8a5ca47b25"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAADAFBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////isF19AAAgAElEQVR4nO2dd0BUx/qwc7/7uzE2ioBIV5oiIIoIbN9l6R1FQBRFERR7rKiIsZfYewt2Y4s1xh6NElsswY6NqFhQ7Epx2Z1P0CTKLrK758yZOeN5/iCGs+edN/e5wnnPzLzzFeAgmq9QJ8ABF04w4XCCCYcTTDicYMLhBBMOJ5hwOMGEwwkmHE4w4XCCCYcTTDicYMLhBBMOJ5hwOMGEwwkmHE4w4XCCCYcTTDicYMLhBBMOJ5hwOMGEwwkmHE4w4XCCCYcTTDicYMLhBBMOJ5hwOMGEwwkmHE4w4XCCCYcTTDicYMLhBBMOJ5hwOMGEwwkmHE4w4XCCCYcTTDicYMLhBBMOJ5hwOMGEwwkmHE4w4XCCCYcTTDicYMLhBBMOJ5hwtBFcmnfuugJ6JhxQqFnw/Xa1zJuY1Ul+ykA2HLRTs2D/jOfvvhamR1W98HgjBwZsfktRsHF55T/KTKpe2JC4mAM9ghsUBTc7UvmP/e5qgufXeC8HfFKoCv7FRN53aG8/s/1VL3CCsYCyYPB8w/gRE356qfZ9TjAWUBdcXZnECcYCyoKrLZM4wVhAWXC1ZRInGAsoC66+TOIE4wBlwVyZhDdfbJn0sgB1Bszw5ZVJimXp438KtLCIFpxAnQoTfGFlUtmZ86HTr3xXv9sPFwW3BSWo02GAL6tMyhUkmdXmt7GNqJ2yKqm7Q/OAxSrUKcHmiyqTTll3bD57nFd969w6Tgm1Hd1Eq4d+jzon2EAokza0rsQiglpm9LM9WLq1iZfnNytNDVwTGvvWkxT4KCWok4INvDJpQLyeKUFD9iwgw6FFkOHwpuYh7v/PybiFoOU9KeqkYAOvTMJO8HHzsCYiUcS3ccYxpac8fV6LHxV/E+5wCXVakIFXJuEmeKtAcNLJxef/nPh9YmWxuTtDsz2MGzjs8CO8WIJXJmEl+NX3LvVTPOtlysQb//7WuaFdWx3igQcxKPOCD7wyCSfBb8Q9B7ry0xwmhq396LtHM0Gfo4Dw38LwyiScBK9Y0OV6pH2HlT9In3z03Sd+qvPJZ7oiy4oR4M0mYST4bUS4l/Pg5r6xLZd/8v25bafXMxDtRJMUQ3wBZVLZ9EYOscK6udNTm12pcumkQ/zcl9E5SNJiiC+gTOowpPdNU5s0I5P461UvbZv5SgYu90SRFVOQXybdDp05b3brFhO8lerXVi8DSVcKOjKfFHNQFlxU3QVMBOe1cg+vNblUujNTw8X8IMWOceNWMJ4Ug1AW/JXfBc0XMBEccEEw3Mthhrf4maarPwp6GsXkM5wSo1AWXGu/e+I5TRewEKwYYSLzNJtrYb+4TPMHSjtb9+fJTzKbFZNQFwwUK5o3H7piT9ULWAgePTngVvOAzS2q3YG1a+D6JpPCQgcymRSj0CAYANUfoyJaVb2AXPD93+6uNpI4OFyJmjSs2g+NzMmyP3tkVLLGH0IkQItgjaAWPC40q42buMynoaTh0Oq3UM7aHHu0s3zOiuXMJcYslAUvr+4CYsE5KSDX2tpWtPrb3Z9723xfkJxz0WxO+62MJcYwNMwmVQNiwZP2XHX0Oe5YJ9rD4/LnPne1rYGsqZWLbOPnPsRiiBW8ZHW3Fjbe/OlLvF7U8MmCVoHN/norfMVIWoxDquAzEsNaPQ7VDQq231fjZ6Vg+ygw/HcGskIAoYKfCR4eMzDk/Xkv9I+aPxz2XOmrSroFPysUECp411QwZmOUw/TICVp8+JeYB+mj20HPCQ2ECt45vdS3vDBox22tPn2graWBpI/6fAkJkCn4eU9D546qoWtr/uR7hs7xLd9K5qQSmYLDfz5iYOc4XuvPS8HgwyCkGGJGyCBS8J1OYO+gx/7a3yAFuc5Ss5kkblQiUnBuP5CUp8tyycRj3YLm+g+dBi8lZBAp+C3vpQDs76P9DYVBjRIMzqtIXEFL5sL3Q62bdw2t6Q3WxzyIvr1wCZFLpAld+L4w85puNwgfF0aebw8nGaQQufD93nLBKR1vOcUfZi26CyUbtJC48H2fdLmj8JiON5X85usRdxhKPkghceG74OXNpCI/HW9SRU7tUhChtrqb9RC48P1lBFiUDeTlut2V1w3wyovUftGwHgIXvqv4b8QvFEId7/otCww6qpJByQgl5JVJxSmODa0PJmbreNszieLYgAN9oaSEEvLKpAHrgSjQ6qDO963xn24pqnafBmshr0ySgnuhIOGB7jc+HG0iE0ymPyG0kFcmScEPy0D8Q91vvOu3b7Cq3xb6M0IKeWXS0OzUS2eC9LhxzVIlT3WzG+0JoYW8Mqm0r6EsVp9WshsWgh7nrvagPSG0EFgmvQjT76C9R6LCn7M6HaA5G9SQVyaBvWP0vPGEv8RwJa2pYABxZdKJlKZpercJ3mHKl+ymMxv0EFYmLXU3adN6XrSea29ywxctfBl0ld6UEAOv26y5nFpm+nCgc8Lt50Z30i7qd/u4A4XhYCdZC3fIKpP6nZcvlCZkT1Lbja4dow8Daclust51kFUm9T2fKHm58odAPd5jVXAyTpG5OzqX3pwQQ1SZ9LKbffOWw5v5L9A3wHJfidU6OjNCD0llUrFk4+i6xq4jqGwjuye6o6GfFoshqUxaOweM37R2OpUQqgyTON9f6EoIB0gqkybtBTzFeUpTuitGz15XLNPzVziWkDSb9EtGbiqYpetM/yckPLjcFczbRFdGGEBSmaRK4mcMj9LvRfQHut4CwuJpJDUYJqhMUq3vbzFuH7UNZHs7lk7p6kRSoURQmdRp/D5f3zsUg6zxrV9vYvgyWhLCAno2nxU9V/8e04LPpYKF2ccpr5pL+jPhcrmYnN3+lAWf7QwuuX39vzZqe4GYFrxpcrH/o5JgqmGk4EAGGEbOWTuUBbdYDmQzystnSqpeYFjwhTZ2rVxKDlHuKhpeVC4BMffpSAkLKAuuqwB27x5sVMZVLzArWCko6GU1IIRfSDXQbwHnw0am0JESHlAWLJ2mSvkVgKVeVS8wKzgvDfzYeWzT19QjnU337qbjthecoSz4tshGVrulnbNa33dmBecng8hCmnaezLMKW0bMC2kanqJvbV2+5Yz6/yAM/w6Wn5KoJk6lI9LsPt7PRw+nIxIOENOj466frXAMLX/vxMoBJ4CElJ/SpAguv9pnLz2R3v2c35MFYtRmx1gKIYKvi9MsxX/REyvkdqn4iYieWOghRHDwjRchl2javX2JP97di5jmwmQILgkGW6YAf0oTSf/yZl9km5669vjAFTIEK/1A7A0goSvcqF6yi9GEbDMkQ/DLoESJatZgmqKViYH/s2JCujkQITiPP821ruv46g/P0Y3bncH3m0hpe0eE4OibgP/Ul6bfwO9+4PuWnk+9G0ZXOLQQIVgKTvUEaXm0xfspYGUTfjVnbrINIgTL32TtA/40TDT8ze1lYl17IeIKEQvft0e3yc+i9XzJfB9H8Tw6AyKDjIXvP9slrKGzXftb8dFwRb9VNEZEBgkL35eJmnvepDXimYFAXPqaiMcsAha+b0x922GfSO9d/ZrIyQQZR8oC6AyJCgIWvsc8LReBkTl0hnwteLZr4kIiNgoTsPA9tPjwUDCh5iMKdSGHH2/ag7bCGiUELHyfPqfPqVfCZzRHvcsnY8qfgDLpVpRBMu9X2sP23k/Xq0+ksL5MUqW3TbYZSf8aue1OXoKJtEdlHtaXSWvHgG9zEs7RHTY/4Fa8qj8BB7+zvkzqeUXFU/40h+6wK5YDAbiWSndY5mF9mTTyaE4/sHgN3WF/XAy6XLugw+FpuML6MumyLPn4BQHtayALRQ9XzIv7je6wzMP2MulEsOf/3DvT+6KyklPBvIbr6Q/LOCwvkwrED/YMDqX9Ees9viQcN8vyMmnKt8eTz+eMhBL7iqNPNyott/CA3WXSfJeURLM35wbAiH1XOGX+OcFjGKGZBEKZtKN9Jc46nMCtJ3eCz0f4LMpK17P36Of5fse99mD1EhihmQRCmVT2tJL0OGqZacG2mSCwaRdjOLM+Q04BgerAOCixGYTVZdKJQcD37cFOcIJvHQ66Xe11CE5w5mB1mVQeOKfrzQBYy+N6x7f3ZP82YXaXSS8C3Tv+AS36jex20GIzBqvLpNxpLeAesEDvQiAksLlMWhC7pRlPz+MZtGKaiVR2HGJ8JmDxbFKxQHk44w7EI503DJi7rkj4BN4ATMDi2aQLfUDWfph7xBLvn+8Jpu+ANwATsLhMeiUHIS+fUO5dWD0d7qt8VV+8YIRlUmZ/zyN+R+HF39Bf2f8XEcu7sdCz+axAwytbBp6il7oMh/oU/T3f0yiV5d06KAtuD8AV16//y7td9QIDgn+EvT/soq/TweRJkAeBC2XBtQCQjlSUjFDbyMOA4IEnIQ/Q9q9vjwI/Dc2w2QMdgs1KAVCYVr0AXfDFCY5nIQ8hBcf7Vcw6sBha/gZfA+CGTdULsAVvDt3dIgTyutaoAhVPGcDqxyzKgr8ydbBIAxetp1S9AFuwoOR8erEQ7hhnhb8ktM2COwZkKAsueXD1xCVwT/1YZciCS4LB7FUgqBTqIODOOA8zaSab30iztkfHThO53fMSAdQx3pExzUf1Qw/Yo0CErYLXdVzqLg4O3wxzjAokoO8JIGfxRlK2CpY/V3iMbErrrm9NKP3Ar8NB+CvY48CDpYLHGwY5DQHplyEO8Z6YC2/FtwKhDwMPdgrOHt5hq0hQSF9zu2q5I+vexIe+FmvMw07B0c//Muwd1IKJg36V1wd1Xkd3+wAGYafgqOf7Mg4lMbN7d5OkzUKo60bgwk7B2cO75z4UMnLAoNK3VPj2djgTQ0GBlYLfrpPXDvY/Ay3+x9zqCob/yuLewmwUXOw3faK4LUNb/0ql4GSfN37MDAYBNgqemw1Szo9k4gmrghFD8z0itzM0GP2wceF7zysKPthFyylnWqDakGjWNEJtyRlbYOPC92kbDg0DExnrgFMeOK3/o6gDTA1HM2xc+P5cELp2YRBj74f/7KPiKwvZuouFjQvfj80yT19cBiu6GvvHg95nFHLGxqMXFi58T+s11SOcwTaDRVLFxplb2brPkH0L338d8O7374JFcIJrZJ00wz6UxhMhGIV9C99n7AB+r/PS4ATXzJP9rVjbcId9ZdLmaYWRYMcEOMGro9s2ts43sK9MKpX2mLaPXwgneDWscJYEZDA6Im2wr0zKX2kXNeIhnNjVcCXirwQwfCOjY9IF68qkeeEzbOWMzCP9y+J17G09y7YyqSBYtW3CDoZrlnWLQeqF07SevMUYbCuTdk4DqeffMnzgTZHg1pYxYaeZHZQm2FYmnekL+KqbSTBCf4YLbYUm9B7rwhhsK5OUoUsirvrDa51UHY/sY8e/YHxUGmBdmfS6Y9MuzKzl+JgiXtflm0VvGB+XOuwrkzJQtGGfvjG3B5i3GsHIVIFQJv02rBKvIGqZaUZ5Q4zi71H/P1XCimkl9gGhTHp0upIOballppFzwq5mUQh+Fy6fCUJeZuxifmDKsKtMUgkenhywN53+wDWhiBoWGdeFjTMO7CqTbnUDszYgWcOqOpA8FMGw1GFXmVQYB0Ieq2T0B9aCXHYeosSyMiliW6hy9DQIgWvmLfwjCmDArjLp+Uo3R/H39J9EqRUiNv4KZtds0inhzBZtUPX3VQUtUWv2xgJYNZvk9+iNvAjRNqE3wcK08NloxqYCm8qktwHgpykgpJjuuFoxdv2eCSCcfQdlsapMEirTclWQW2NVR9jr++3BfPat6qBp89lO9W9BEDyjt9fDfkztSapC9ytAVDYMetcX2qEs+EolpleuVL0A4yl6nl3cJvqjasVZv2ujsgLY10+JeitDc9d3/J+ra9UL9At++kP8RLpjas+ZpJaNejO1Z5U+dBFcpOkTh1pmlQBgrn6BdsFXeD/4Bi+lOagOnBC7nu4H56BTiGgj+IrMusM9UNlYVgOlo932MyI4Or9cqBSg6xsZVjjsEAhgW/NobQQLRp2d4nC9OsEA5Pl1UusWDUGwFJzsB/rl0hxVl/GPfwsGMr+ahBraCK6vBGB30wfVCgZgRbT692gXHPx09B4QgG5hVMxfCqlKwrauhtoIrmwJudY9v3rBmqBd8GG5Z+7gYTQH1YGLvHVesbPQja8f2gjeYlBRm2yx/o9OkWkXrPjZMY2xvg2aeLTAfwHK8fVCq6fouwUVX59k6xSZZsGqkQJPR9TvGQ6OQpyA7mgjWLhMn71ANAteNhq0Py9Cu4lTtdg0ZEk50hR0RhvBi4V1Ox3QeTKUZsExTxVCMFn95AAmmThE/DwrE2kKOqPdi4788c1sM6/rFplmwe0f5/YG49G2qxKpBh8DYnbN+2v9JuuPgQ66zePQLHhDnxVL7gqQFinlcrBjEtvav2st+ET/xnydItP9FD3Hsk3IBXpD6krQ3ZfBhWiW/OmNdoJvjHGyGXFNt8i0l0kByLcGXeCPteexbBepNoLn8+ok7tN5pRvtgsU0x9OD4gOd2HacsFbvopfo836QbsF3EumNpxdvutoErkSdhE6wZuG7crXa4XkISMyWlPRlsgkbZdiy8H2+r5sL+gVRzyNAxKNyVj1msWTh+6/JyvZ3wtSWBTHNzW5g5kZ29fdnycL3IaeACGxBvixZxXt+rtfpTqjT0AWWLHwf8fu1RLB+IY0R9eOoYLiljNkubBRhycL3P6Iydzz1+4vGiHry5pjsPuocdIIdC98fDWzxtbffYfoCUmBIymbIhxbTCivKpNeig6djBPm0xaPChJCA6cJHqLPQHlaUSZtmgWEHD2Jx1PqdcJUQ/NYXdRraw4oyac5PgF9+vTtt8SiwayqIvadk0TlZrCiTTnS70hXMWEZbPApcSgEr5l9jupUiBdhRJo3wSElLwGOtTPysi74ixNOWusCKMunkDOdlZ2mLRg3Fkngj8YA7qNPQGjaUSaO7rm/Ox+btQom419wcXgHqNLSFBWXSX1Fg/YwcbJoY/TStMALsGYM6DW1hQZn07sG1cx4+D66ztwD/V3g80msDC8qky8lKEbiUTFM0yuSkg4k75rNmThhCmXR+cSVy2o5F79Qr+aAA+VThP/TvOrZlFINH61EDQpmUt7GSsEhqmf2D8npQ6EicWlSdzXa9jUfNpgX4l0m5wm6mEXhtu57eMNqXLc3B8S+ThPfvtd/P6FGFNbE/dd1sRdRF1GloB/Zl0u0uYHE2XqtkBp1+Eg62smSnMPZlUlE70O6uCivBmTlArFjHkudo/MukqD0i5Rgclsz+w9mwl/32S1nythL/2aQnba3FUxF1EK6GXX7NXY6hTkJL8J9NejUWad8GzdyLQ52BtuBeJj2KCW0YjOEyN588lnQ1xL1MSsgpk52OoSMSnSh7WiXw2dHWkJ5us6oH6tveaRGslIPDGSAIt2WMc2YuWlUseYw6DW2gLDjP37Z/kfv/M1drgEOLYJUMZO4HAbi9+Y18caEHmKahhzJ+UBYsHXK8i2iGYoV31Qv0/IhO3SQt/hm7JVAJ91W+qu8OoU5DGygLNiwHT78qASrjqhfoEfyqWwNpKl5vot/xc1JJ+mYRmrMFdISy4CYXARgMwHX7qhfoEVy8ahIdYehmjbSF035WPEdTFrzaZM+7rwcsZ1S9QIfgkmR/Ry8slzC+CDdLF5xCnYUWUH+Kvpr/7ss59Q5ldAgethbw/hLi9RrrPb32xt8s4uOYWRVoOpRDA3QIloKCdqAnPos5/kUKfpyBZ2ZVwFuwn2ryetDtJvVAtONX/loMKvvgYw7WgovCRc4lZ/1w7B04b6iiY5dW2CzWrh6cBd/12Rhi0Cj+LtU4MFDN9axnNUeMx5blz4Gz4H6/g+hbqeeohoGE9KnozTMMmrPVAM6CQ4ofhoGF66mGgcPbADBjHaqDFHUAZ8EDDy9dApLPUw0DCWF5XrJSgDqLGsFZ8ENeq5+G9KAaBRYLk294psxFnUWNYCxYtXeyzag9FINAZHeKE8Jz2LQFX8GKiOHD+Ek4lkj/MD9tH24z1WrgKzh7Fkj7c8TP1IJAZYfAY6wApz01msBXcL8/lTzV/vHUgsCk3KdEWH4R99V3+O5smLHxyEAwdw21IDC52gP0/QOvPRcawHdnwxNe+K6NEozrzBehYOvkEjnqNGoA250Np0P5tcQT9TmRizHSZtyTxaM6kVxbcN3Z8FB47+DAGLy3DyjmRZjh/BBYCa47G9YsBT3OnBtAJQQT9J90ButCDt+dDUvXqPgVjzF4k+3l1ycU+Wk/nwXXnQ3XQ073BAO2UQkBn/sBxTKwCYsmqdWCbZm0wdZPNI5SBPj8/D0IeVoWiDqNz4JrmXRzredZ7F8D/vEtWLT0dkfUaXwWTMukeRFzHCRq/5/BjfKAnUUCEdojb2sCzzLpQYBq2+QD2D9Dg2cZzb4OCRyJOo3PgWeZtHci6JqLT/fC6jnfbus4MATDHer/gGWZVNTOoo2z4g7dp1tCYO7mMj64gE2jVA1gWSZF7o/2Tu/rf1LvAIyxMht0vZIzAnUanwHHMulJDPjdM9IC1+WUH/NImL9pTCCWu6c+gGOZVNARdLmI/Tzcey60FZpivTgaQpl0d38lbaP1Tkp8UwT2pep9O7McNhW0xXXlJ4BSJp2YXIkwRN+cFJsc3aLbYrfrWzMXAsdtyBcWok6jWjAsk24IB7i7HdXzZsbJPJrXGaxYjjqNasGwTAq/Vs4vwH2hxD/0ywWSa3Ono06jWvArk5R+4GAGiMB6LcdHbB9U7u9o3xLbplkYlkkS0P0c8GNNS/WRzjLDaSWSItR5VAOGZVLWd77KOf30vBkBgSd+HAmm4PpXGLvZJNUUoUU9weQyvW5GQtxDpeDdsxbqNKoBu9mk2ZnK+M1yzBc6fcLOpOKknWJcJ6+xK5Pk5a9lNWWFGev8XLwwOH1eM9iVSTKwYjboi/GrIU0csEteh+kPHezKpMHLAh4V8lnRRO4f9oR6Xc0ciDoLzdC2+UytTNBP8GiexdcBwjP63IqOgJff7Xn3BXUaGqEs+IrMurJdVK2qF/QR/LanaNOIVT6saMT8EVLwZ6frHfGcNKQsWDDq7BSH6/QILg1ouXWMcdksnJfAaCLm5i2z6IaSfNR5aIKy4PpKAHY3fUCL4KULOp2Vdtn0PfYbfqqQx3MbZL7mcgTqPDRBWXDTilbva93z6RA88MxW81OHBglZMlP4L49bLGmfAwJxfDtDWfAWg4oNlFus/1P1gh6CF/5wUiZw9PhT5xtRo5KBG7FAimOlRP0p+m7lcfZPsqt+Xw/Bb8QRixYE4Pj3oCb6LFa4NW+G49odrHp0XF1t1WcpG/2C0iyLRoKrbbejzkMdnARnxY13jMHtgBUteRpW7vv2TQDqNNTBSPCZJDB/5cJ5+o+Ikou9wbhtOK4ExUjw4nUg9NEd7I7Q0Y4yXtmtuEsY7sXASPCOyQUR4GiG/iMiZbN8oZXtdPx2+2MkeKdtmyV/CnHs368V98Pd+i3iYbd+Fh/BA/ruMjZMuKr/gIjJi38UBg5jt+UVF8F35ObS0L4Xu+g/Hmq2zAaRd99i9xyNi+CgJeMeuXTC8TFUW3JTwfZROdj1BcJEcFG7O3LfY9KLnfUfDzmJk+bUNm61DHUaVcBE8JN2wLnL7648Vq3FqoJymq18vqLjEdR5fAouC99Djsj2Roof6XILdszb9NivbOcg1Gl8CiYL3w8kmllK0kt0uANDFmwAwmZhlj+izuMT8Fj4vjLxmn/nydp/Hk9u+m/swYs6EoZVqYfHwnfx26eBKgELzvL8PHutfJv7gw0LUOfxMVgsfFfJwOJFoO0TrW/AlSHHgP+zFWpT4yjBY+F7QKH02Sv8D5mqkZORrxZOE99HncbHYLHwfRu/vtUiGcZHJGnNDlnj/3mFXESdxkfgUCZt7vi8T4wnC87i1YIVg2LybwswmlTCoUwKfqPwVSZhu31LJ6Je/DQRZP2GOo1/waBM2ttA5jMSDDml5cfxJvx1qW9J1kHUafwL+jIpJzJlcyvhEz7GB+jowILxJT5OxkG3UOfxD+jLpMQ1l81j5W67tPs07qgyrFoaHF7TGpv94MjLpKGWQzx9zg7EqnakhORZatMRrm2uoM7jA6jLpF0D52cJUg8G4X7Go/ZIgYR3qOPJYNR5fICxMmlDNeOP3GfpH2+H8RGFupKxIPS5bxsQgMkCb8bKJInGu4d6DQhwvjVhRo3DsIfSvoYBQYFAjDqPD0Aok17drKRr7Cef0/im+UzyHW+vZcOED7RJlTUMG/vIwaoJJk8VEMqk3WmVuH+6/qxfroabl64BAYG8xljNr1FHMcfMMaC4Nx6Ldxgrk6ZqqoN2Z12LBUeG1TgIy7jXQcXbN9kVi8qesTJp3WL1ey8HNTCOz+Hn1zgIy7jYGwR77HIS4NBrhLEy6cgotU+UCf56aOsiI+Ml9McoeKcSU4b3PDAYdSKAwTLpVle1T2yRL528uDRIiyHYxm/N5F2MEv1t81EnwmCZVKp2huMGfuTM+jmYn+2oJyfanfb0PNqej37un7nZpKqbFlS+LwWy/UFTZtY4BBuJjx+TYpe/YwrqPBicTZJUuV7QEaQ6hxlmsH6pnUbK48LGtEkJ4CFvf8fcbFJUleZICuHi7qCIxZuRPs+vfQs8eCP6y1BPKzE3m9S76kqltMYXTofvqHEAtjLAvbebVDkZ9SHmzM0mTaq6qC5odc9vT9ccn7W0nfO7rbe762W0WdC2+exh1W9UFbxq6af/fjJF69jsZN7i7PjEnpsFaDfk0Ca4xlaGh7779N9j8rSOzU7KwhyXNncS20UgPZCFsuDHH6hR8PXun/zraQw70tCMymdhqsPxUSP8URYKlAX/57/vUftkVcHFn7zRKBPe0SY9djN3svSxe71ujVH+t1IWPPjDT96au836f1wnZeIxmQYX5RDDKEPnPzy9rNRqDMagLFgR9f4g55oFz13575+Xx7KivJkAAAfKSURBVOLYmZV+UteGFjSwKRW1Q9bFksEzGx782zB7TDdMVizB5kWMWVsTyQzJa2RTKkz26Aj6sGpH1W/El/H3t4LkgQPK+Y5hFpcQjc+k4EXvK+HXSazfy68DJRkGIgfn7eFex9GMz6Tgp74Vb0NO8dl25gZFtnjKEg14iVYTkIzOaBulk/5litER6OdIGWZp+iLHsS/De+5DMTizfbLWCv0WfTm/fv/mRvSUXeNcGnvao3jzjkkjNLKZ0iq1Y/30pa3MJjNfPHCCmeBos7A/I+psmRruz/jef04wI5yx87bwT3T2d3Wcw/BqaU4wM2wd4KcS1J9u19GGz+wEMSeYIcZZBBssMXK80VpgeYjJcTnBTFHQxn6RRZyp7V5D0+YM9sLjBDNGWaq/153Wto19fzGxcB7L1H4OTjBzqLKbWPBX2Vk7+4TbN5VNZGZQTjCTlM8xbt15amy9+Ph1fqYOsp5q20HohxPMLD+KHCRuMd5NTJqHyp08TCP+gD0gJ5hhHox1sLRIT2htOM6Dt7JBpKXdGLj9ZzjBzDOnpXm9uU0MrB2NAsMdUo2MLP0mnID2+oMTjADFHntv143hwmDjxk3Cec0jG7qaGNo09fYfDKGXBScYCfm+C3zqNG40ubutW900uzFyUXsbvnuou5FL9x9O0rtfjROMhmeLxq8MMnNyzvRrk17frXaUZSJfzvOUNzNrYm3RqvPMNUfegIcadtXrDicYIYUDLaycwprJQzwDnJo4GNnUT3UaFOPc20hgYNuobl0n80bWbQSjZvSK6rLsyOb9Jx8/e622P6hmOMFo2dvRycjcwynWVSx0+d4usrZn7Q4W052iPd14QTYJjgPrJ9Q2dXdxNDGxsDExNDZyNmom7fHdvE1HcvMvvHm0a+fNV29qWj/BCUbOmw2d3I3tGpk2qRdunzTUoX+DxoYWNhZxDbubrrUZ4DLCXejdooNHUExgkCip8ZCGQuuG9sYWhiaWdes2sTUyMG/YsKGTqUsroacHr8PI2ctHDUzos/5j6ZxgLFDlHdkwN87F2sXCt7Wnr7RVd7vOJmPqBdeN/MbeqGGjuoJ68tpJ9fqZtLONF7aJaxUrign3biNo2drb28XfOcl0trGfZUuewMHMvnYrw05+Y7t8FJkTjBHPDo6JcDJo7G4ZbN7HIknetpVvLzd/T2lK8x72SxtNbdjPMsLewrpus28CvulaN9E0yMLG2zau4XKzUVbjHYd6eAd7dGs6tMWgH7J+7vrRXnscDuXg+JjS7Zkd+c1NmxnXa9jUtL6td4y5Mc9gjHmcRW9biaebT/sW4xzmWE1pNMIixlse3ai/YXSd/vXkdWzNLW3quNWS/M/LosfUWR8tTMbhUA4ODbwuB5e3ndzUUy5smyb3tXO0a2xiY1jP3MEosn43o8FmnR16Wfq6WXX26uwRG9sqgC9o3bJDi9Gu41ymjx23Pe7av3EwOJSDQ2sUZ1eNbh/W2tXa0czcrJm5hb2xsXEzs3rGDeytLc3qB9UbbNClbqR3+scd9iC0UdrsX4lDO73+Gzi0paziy9uC3ZcLbuyaOWH5upldIr0E4qSU8Z+cYAyvjdKG+TokywELeG2UOMFYAK+NEicYC+DVwZxgLOAEEw4nmHA4wYTDCSYcTjDhwBO8x8Pf319QpwE86teHGLyOMcTg30CMbWzq/zFNP3+yOgXBlVRpUEkvy5dDDF7D//OpAbMNemGcLp/mBEOBE6wVnGBNcIK1gxOsFZxgjXCCtYITrAlmBd9Moxjgc6xaBTF46i2Iwf0hxn6coMunqQoGMA+GKiuDGBzqiVb4BKcsmANvOMGEwwkmHE4w4XCCCYcTTDicYMLhBBMORcG73RoEPaAnk08o/apWrVrtYcRXDPnPY/BPYJrjvw8OKfntzQwleTpnTk3wc9NjipGxlEJo5oEprPhRo//7+J/AdMd/HxxO8ncNc5QjZTpnTk3wxqB3/x21IBxyftUBVvxzoMLBh8B0x38fHE7ydzcCcNZK58ypCR7f990XcwjnBJ+wkJn558GJX+HgQ2D641cEh5f8lHidM6cmeMTQd1+anKMUQyOX0q6UZLjDiV/h4ENg+uNXBIeW/J4md3XOnJrgCb3ffTG7TilGtbz9+h6U+BUOPgSmP35F8ApgJL+26XXdM6cm+CcxAAV1aGnY9in3L717GP3vYyjxKxx8CEx//IrgkJLf5l7x1Kxr5tQEvzQ9qEjvTCmEZn6xzS/P9IETv8LBh8D0x68IDif5p9b5Ff/QNXOKdfC+5g3C1Q4WpoNJViZB+RDiF9Wq9a5Kffh3YHrj/x0cSvLZ/3lXXdcq0jVz7k0W4XCCCYcTTDicYMLhBBMOJ5hwOMGEwwkmHE4w4XCCCYcTTDicYMLhBBMOJ5hwOMGEwwkmHE4w4XCCCYcTTDicYML5UgXzF1R8XST8ezMasXypgpfwKr4Kl33YL0YuX6rgF3VuAJBf9+WH/WLk8qUKBoljABjfqeJPnGAi2ecEQLMDFX/iBBOJ0ubEH7bKij9xgslkRP+BmZV/4ASTybUmDu93X3KCCUUgBP/uFyOWL1jwlwEnmHA4wYTDCSYcTjDhcIIJhxNMOJxgwuEEEw4nmHA4wYTDCSYcTjDhcIIJhxNMOJxgwuEEEw4nmHA4wYTz/wHaCL7onVHZ5gAAAABJRU5ErkJggg==\n"
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%R\n",
        "sum(as.numeric(mer_counts[16:1130,1]*mer_counts[16:1130,2]))\n",
        "\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "SaDOOLPdZqlf",
        "outputId": "39cb2682-7936-49cb-bdcf-f39295a7dec5"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[1] 419940727\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%R\n",
        "mer_counts[65:85,]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "tLcUTLbBbG46",
        "outputId": "e9f22054-9536-496f-e8bd-9bf87414b381"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "   V1    V2\n",
            "65 65 64206\n",
            "66 66 65108\n",
            "67 67 66027\n",
            "68 68 65706\n",
            "69 69 66165\n",
            "70 70 66607\n",
            "71 71 67096\n",
            "72 72 67648\n",
            "73 73 67841\n",
            "74 74 68492\n",
            "75 75 68619\n",
            "76 76 68949\n",
            "77 77 69577\n",
            "78 78 69925\n",
            "79 79 69894\n",
            "80 80 70222\n",
            "81 81 69926\n",
            "82 82 70279\n",
            "83 83 70337\n",
            "84 84 69762\n",
            "85 85 69835\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%R\n",
        "sum(as.numeric(mer_counts[16:1134,1]*mer_counts[16:1134,2]))/80\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "RURoFK4qcBQq",
        "outputId": "17d2dd2e-e1f9-43e9-bbfd-c4570954db4e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[1] 5249408\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%R\n",
        "sum(as.numeric(mer_counts[10:150,1]*mer_counts[10:150,2]))/80\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4UPmhh5JdAN2",
        "outputId": "486e0ba3-82bf-4356-9959-b6aa286bdbbf"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[1] 4926079\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%R\n",
        "sum(as.numeric(mer_counts[10:150,1]*mer_counts[10:150,2]))/sum(as.numeric(mer_counts[16:1134,1]*mer_counts[16:1134,2]))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "oDQGzrkEdmu9",
        "outputId": "84b26fa9-9944-4be1-b8f1-7adc133c9fe4"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[1] 0.9384066\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%R\n",
        "singleC <- sum(as.numeric(mer_counts[10:150,1]*mer_counts[10:150,2]))/80\n",
        "plot(1:200,dpois(1:200, 80)*singleC, type = \"l\", col=3, lty=2)\n",
        "lines(mer_counts[1:200,],type=\"l\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 497
        },
        "id": "RSgLstmHd6xq",
        "outputId": "2dddecda-10a0-4db6-b9b7-7fc9d4fe9bc5"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAIAAADytinCAAAgAElEQVR4nO3dd1zT1/4/8E8WGQSSsAmgKCKKo7hXRaUi2FYtFnGjto6qVWttrbdLvR36vVZrrXXUUeuq4ka0aItatbbXW6p1C7JlyiYJCZDk90f6o4BRgXxWktfzjz7ih+Scd9P05eHkfM7hGI1GAgAA2IfLdAEAAGAeAhoAgKUQ0AAALIWABgBgKQQ0AABLIaABAFgKAQ0AwFIIaAAAlkJAAwCwFAIaAIClENAAACyFgAYAYCkENAAASyGgAQBYCgENAMBSCGgAAJZCQAMAsBQCGgCApRDQAAAshYAGAGApBDQAAEshoAEAWAoBDQDAUghoAACWQkADALAUAhoAgKUQ0AAALIWABgBgKQQ0AABLIaABAFgKAQ0AwFIIaAAAlkJAAwCwFAIaAIClENAAACyFgAYAYCkENAAASyGgAQBYCgENAMBSCGgAAJZCQAMAsBQCGgCApRDQAAAshYAGAGApBDQAAEshoAEAWAoBDQDAUghoAACWQkADALAUAhoAgKUQ0AAALIWABgBgKQQ0AABLIaABAFgKAQ0AwFIIaAAAlkJAAwCwFJ/pAlqmuLj4/PnzTFcBAPA3Lpc7evRogUBAReNWFtDnzp07ceLEkCFDmv+SH3/8MSgoqH379tRVBQB2a/fu3SEhIQEBAVQ0bmUBTRDEoEGDZs+e3fznP3z4MDw8fPDgwdSVBAB26+rVq9Q1bvtz0BwOx2g0Ml0FAECLIaABAFjKLgLaYDAwXQUAQIvZfkBzuVyMoAHAGtl+QGOKAwCsFAIaAICl7CKgMQcNANbI9gMac9AAYKVsP6AxxQEAVsouAhpTHABgjazvVu+WwhQHy+mN+p+LE/RGfWdpt3aSQKbLAWARuxhBI6DZLLnit7Lakrbi9nXGOqZrAWAX2x9BI6BZ7jnn3r1lA7gcHtOFALCOXYygMQfNZkKuqD6d/1t+idliAFjF9gMac9Bstj93u5H457/OzcrkHG0mc+UAsIvtBzSmOFirxqDL0+VwCE79lb7ywX9V/sFgSQCsYhcBjSkOdsrQPGgnbrRso4NjpzxtNlP1ALCN7Qc0pjhYq9qg7u7cq+EVCc9xdpu3maoHgG3oW8Wh0+mysrI0Go1UKvX39+fzaeoaUxysFeLcl+kSAFiNjhF0fn5+dHS0TCYLDQ0dO3bswIEDZTLZjBkzysrKaOgdUxzWJas67c+K/zJdBQAr0BHQsbGxgYGBhYWFBQUF6enpRUVFGRkZYrF4xowZNPSOKQ52ytc9PJj33ePXpTznPyp+pb8eABaiY54hOTk5MTGRx/vnTgQPD4/169crlUoaescUBzulqO74ito+ft3Vwb24poj+egBYiI4RtKen55UrV5pcvHjxIgLanj3Q3Atw7GT2RwGSIJ1BS3M9ACxExwh63bp1UVFRISEhwcHBYrFYrVbfvXv35s2b+/fvp6F3zEGzk784wFPobfZH45V0zH0BsB8dAT1y5Mi0tLQzZ86kpqZqNBqlUhkWFhYeHu7k5ERD75iDZqcX3F5iugQAtqNprZtMJhszZgyW2YGJkTDWGHRCrsjsT6vqKn96FD/WewrNVQGwDZbZAQPuVt1IKj71pJ868hwfaO7TWQ8AO2GZHTAgT5fj7uD1pJ9yOTyDUU9nPQDshGV2wIACXV4nabenPKGTtFutsUbAcaCtJAAWwjI7YECQYxcv4dP+67/iNRHpDIBldsCAPvJBTJcAYAWwzA7oZiSMqroqJ77zU55TUvPoXMnpcd7TaKsKgIWwzA7olql58L+KKzFPDV+ZQJ5VnU5bSQDshGV2QLd8Xa6ng/l7COvxOYI6Qy099QCwFpbZAd0KdXneIp9nPq2rc8+GxxUC2CEsswO6dZZ29xO1e+bTXvaIpqEYADbDMjugWydpVzFPwnQVAFYAy+yAbsU1RW4OHs982o3K5JLaR8NcI2koCYCd6BhBm5bZzZ4929PTk8/nK5XK+fPnp6WlDR8+nIbeMQfNKnqj/vuHm5rzTJlAXqjLo7oeADZj7zK7I0eObNmypcnFtLS0Ll26zJs3r/ldY4qDVcpqi+UCl+Y8083BE0ergJ2jI6Dz8/MXLFiQkJAgl8slEolKpVKr1TExMevWrVMoFE961auvvvrqq682ubh48eL8/PwW9Y4pDlYpqSl2Fbg355mOPGmQYxeq6wFgMyyzA1q5Cz2bf593uPsoSosBYDksswNauQjcCEFzn1xnrONzaJqFA2Ahu1hmhykO9ijU5TX/9pPjBfuzqzMorQeAzWx/mR2mOFhlb+63b7X7iMfhPfupBCEXuDyqKWgjfvZdLQA2yfZ3s8MUB6vUGWubmc4EQbg5eBTqWvadMIAtYe8yO7IgoNnDSBg5LZlV8xd3aNHzAWwMe5fZkQVz0KzyitfE5j9ZLnBp5qJpAJuEZXZAHw7B6egY3KKX6AxaiooBYD8sswP6lNeWcjk8Z76s+S/ZmLl6SfsVlFUEwGpYZgf0uVp+OV1zv0Uv4XF4NQYdRfUAsByW2QF9ympLOkpbdve2QuBaWlvy9CPAAWwVltkBfcprS+X8ln0t3FPWX8wVU1QPAMvRt8wuJiamycWSkhJXV1equ0ZAs8dw95dlgpYFdDennhQVA8B+dMxB37t3LywszM/Pb9KkSXl5/+zw6+Pz7IPpLIeAZo8ASRCH4LToJUbCWK3XUFQPAMvREdAzZ858/vnn4+PjQ0JCQkNDHzx4QEOn9bhcLr4kZInM6rSWvqRQl3ekYA8VxQCwHx1THDdu3Lh48SKXy+3Ro0f37t1ffvnlCxcueHl50dA1gRE0a2j06oTCuDf9/9WiV8n4itKaEopKAmA5OkbQSqXyt99+Mz2OjIz8+OOPR4wYkZmZSUPXBAKaNcpqS+SCFn/lIOZJtAZMcYCdomMEvWrVqhdffHHHjh3R0dEEQUyaNEksFg8ePLimpoaG3jHFwRIVtWUtXcJhMsQ1gvRiAKwCHQEdFRXVp08fDofT8MqQIUNOnDhBQ+8YQbNEG3F7z1YtZ+4nH0x6MQBWgaZldr6+vk2uuLi40LMXBwKaJaR8JynRmpXvqrpKCU/K5WBbO7A7tv+hxxQHS2RoUuuMda14YVLx6bQW3iAOYBtsP6AxgmaJU0VHWrc1nUygKK0tJr0eAPZDQANNNHqVI0/aihe6CFzLahDQYI9s/8hkBLS185cEingSpqsAYIDtBzTmoFliks+s1r3QmS9r0RbSADYDUxxAE19R21a/tqwWNxOCPUJAAx3UelVOdWarX74t+0vyagGwGrYf0JjiYIN0TcrNquRWv9yAv2LBLtl+QGMEzQaVteXOfHmrXy7mibHpKNghBDTQobKuwpKADncbxefa/hfaAE3Y/oceAc0G/RWhEp5jq18e7PQcicUAWAvbD2jMQbOBouUbjTZUrdfojXopn45DLAHYA1McQIdbVdcsefk91c0rZefJKgbAWiCggQ4JhYcteblMoCivKyOrGABrYfsBjSkOGyAXuFTUljJdBQDdbD+gMYJmnMGo53N4lrQg57sMVAwjqx4Aa4GABspxObxYv3mWtcDt4hRCVj0A1sL2AxpTHGzg4WDpIe6FujxSKgGwIrYf0BhBM66k5lGuNtvCRvbmfltrpOOUYQD2QEAD5e6qbmRqHljYiIyvqKgtJ6UeAGuBgAbKVdaVOwtaf5+3iVygqMBKO7AzuJMQKFdVV2n5jvsDFEMt2c0DwBrZfkBjBM24cPdRcr6LhY34iNqQUgyAFcEUB1DOReDG5Vj6Sas11mAhB9gb2w9oTHEw7nrl/yxvpLim6MyjE5a3A2BFbD+gMYJmXGLRccsbcebLK+qwigPsCwIaqGX5fd4mEp6jRq+yvB0AK4KABmqp9FVOFi/hIAiCQ3CivCZb3g6AFbH9gMYcNLOc+LIJPq+R0lRHx2BS2gGwFrYf0BhBM4tDcGR8BSlN5WqzDUb8XQt2BAEN1MrVZudrH5LS1NlH8cU1haQ0BWAV6LtRRafTZWVlaTQaqVTq7+/P59PUNaY4mHWn6i9XB3dvka/lTcn48oq6cg+ht+VNAVgFOkbQ+fn50dHRMpksNDR07NixAwcOlMlkM2bMKCujY2sFjKCZVVVXQcqXhARBOAvklVhpB/aEjoCOjY0NDAwsLCwsKChIT08vKirKyMgQi8UzZsygoXcENLPIWsVBEEQP537tJR1JaQrAKtAxz5CcnJyYmMjj/bMY1sPDY/369UqlkobeEdDMesVrIlkB7ergTko7ANaCjhG0p6fnlStXmly8ePEiPQGNOWhmOfPlHIJDSlMGoz5Hm0lKUwBWgY4R9Lp166KiokJCQoKDg8VisVqtvnv37s2bN/fv309D7xhBM+tq+eW+8udJaaraUH2i4MCb/stIaQ2A/egI6JEjR6alpZ05cyY1NVWj0SiVyrCwsPDwcCcnJxp6R0Az61zxj2QFtITnqNZXkdIUgFWgaa2bTCYbM2YMltnZmzpjrYArIKs1DsHBX7VgV7DMDihEylkqDU0k6a5xAKtgJqDr6urqH6vVasv7YHaZHTBILnCZqHydxAbbigNIbA2A5RoFtEqlCg8P37FjR/2Vt99+Ozo6uqbGouPuk5OTP/30U5nsn5GUaZnd5cuXLWkW2I9DcKR8ZxIbzKnOrDHoSGwQgM0aBfR7773n4OAwefI/mzp+9dVXKpXqk08+saQPZpfZAYOyqzPytDkkNvjf8kvZ1RkkNgjAZo2+qTt27Nj58+elUmn9FZFI9NVXX40cOdKSjGZ2mR0w6I7qL2+hr1LkR1aDMj7u9gY70iigi4uLfX2bbmrTtm3bvDyLDutkdpkdMEhVVyl1JPO/shNfVllXQWKDAGzWKKDbtWt39erVYcOGNbyYlJTUrl07C7thcJkdMEhVV+XMl5PYYDfnnjqDlsQGAdis0Rz0nDlzZs2a9b///XMGc1JS0uuvvz537lxL+mB2mR0waLxyhruDJ4kNOvKkLgI3EhsEYLNGw9jFixdXVVUNHTrU3d3dy8srOztbpVItXbp04cKFlvQRGxvbu3fvwsLC+oUcRUVFK1asmDFjxvHjJJz3DKwl5knIbdBIGB+o7wU6dia3WQB2ahTQHA5n+fLl8+fPv3LlSkVFhZ+fX0hIiFxu6a+ozO5mBwy6XJr0vMsLJDZoNBqPFexfGmDRsiIAa2HmRhU3N7fRo0dPmDAhKCjI8nQmsMzOjv1ado7cBrkcLo4lBPth5pu6oqKiOXPmnDp1ytHRsaysbMGCBRMnThw4cGCr+8AyO/tUa6wRcBxIb5ZDzt6lAFbATEBPmDDhpZde2rNnT9euXQmCiI2NnTdvXsNvDluqdcvszp49e+TIkSYXf/vtN29vHElnHdR1KnI34jCZ4jOH9DYB2MlMQGdlZS1ZsqT+j3369Ckvt/TWgFYssxswYECHDh2aXPz000+rqrDhpHWQC1ym+JIfpj6iNqS3CcBOZlJSKpXeuHGje/fupj/eu3dPJBJZ0kd+fv6CBQsSEhLkcrlEIlGpVGq1OiYmZt26dQqF4kmvcnJyenyILZPJNBqNJcUAnURcMelt5lRnuji4OfKkz34qgJUz8yXhZ599NmzYsBEjRhQXF7/yyiuhoaGrV6+2pA/sZmefMqvTsqvTSW/2r6o/MjSppDcLwEJmRtAvv/zyjRs3Tp06NXz4cG9v761bt3p6WnSvAZbZ2acU1W03B8824vbkNovtOMB+mJ8I9vHxmT17Nll9mJbZDR48uOFFLLOzeSp9VTt+IOnNOvFlhTqLNocBsBaNAvrxL+XqPXjwoNV9YJmdfdLoVU4UrOLoLO3WluxROQA7NQrow4cPU9EHdrOzT+O8pwm5Fn29bJaQK6KiWQAWahTQISEhFHUjk8liYmJMj3Nzc4VCIdLZ5lEXo7eqrnV16kFR4wDsYWYVh7+/f4fGAgMD+/Tps2jRopKSklb0UR/N9+7d69q1a/v27b28vAYOHJidnW1R7cBuv5ScpajlhEJKftUDYBszAb1s2bKOHTv+5z//2bdv3xdffBEUFLR48eI1a9aoVKrXX2/NAaDx8fGmB3Pnzn3llVfUarVKpRo2bNi8efMsqh3Y7beyXyhqGXd7g50ws4pjy5YtV69edXD4exeFkSNHDhky5Pfffx8yZIi/v78lnd2+fTsxMZHP5/P5/JUrV+KmbRtmMBp4HN6znwcAT2ZmBF1WVtZwKqOysvLhw4cEQdy5c0cgEFjSWZcuXeqnNbKyssRi8m8zA5aoNmicSD3Pu6EZfgsoahmAVcyMoGfPnt25c+dhw4a5ublVVlYmJSVNnz7dYDD079//yy+/bEUfOp3O3d3ddJf2F198sXXr1tu3b0dGRi5YgP/NbJYjTzqzzVsUNe7h4EVRywCsYiagP/jggzFjxly6dKmsrMzZ2XnJkiV9+/YlCCI9Pd3d3b0VfVRXV5eXl1dUVJSXl5sWbygUim3btkVGRlpYPbAZn0PVsZPZ1ekygULGf+JGLgC2wfz/Ql27djXtNdpQ69KZIAiRSOTl5eXl9c+oR6lUKpXKwsJCC28iB9bKqk6rMdRQdDbVffVtF4FbL9kAKhoHYA8zAX369OlPPvkkLy9Pr9fXXzRNQ5Orbdu2Wi1OaLZN6ZpU6jacc+bLq+oqKGocgD3MBPSSJUs++eSTwMDAhtsbWaK4uJiUdsCKqOoqPYVU7bXixJcV6fIpahyAPcwEtJ+fX3R0NIl9eHh4cLlmlos0HKGDjdHo1TI+CQdamtXRsbNS6EtR4wDsYSY3+/btm5SURGIfS5Ys+eijj+oeIxQKSewFWOVV76nUHX3C5wjkAheKGgdgDzMBnZCQMGLECLlc7tuAJX2sWrXq2rVr169ft6QRsC7ULeEwuV7Z+kMyAayFmf+Ldu3a9cwDA1vWB59//Pjxx6/n5uaS2Auwyk/FJ8PdRlHXfmLR8RDnPtS1D8AGZoKYuj3tmnB1daWnI6BfcvnvlAY0gD1oumH/77//3r9//8efZ8mG/WCHqN7PiM/hGYwGLsfMHB2AzWgU0AcOHJDL5QcOHGCqGrANNYYaRx61+33PbPMW0hlsXqOPeO/evfl8fpcuXXQ6Xe/evTt37nz+/PkLFy506tSJqfrAGjlwHea2fZfSLrCKA+yBmTHIzJkzTd/pzZ8//+zZs7du3Zo5cybthYF1o3qv0UzNg+KaIkq7AGCcmS8Jr1y5kpaWptVqjx07lpmZKZfLAwPJP5sZbFiONrOqtiLY6TnqusiqThdyhW4OHtR1AcA4MwHN4/E4HE5SUlKvXr0UCoXRaNTpdPRXBtYrW5NuJIyUduHElxXXFFLaBQDjzAT0gAEDIiIi7ty5s2HDBoIgli9fTtvCO7ANKn0VdRtxmMj48kxNKqVdADDOTEDv2LEjPj7e29t70KBBBEG4ubnt3LmT9sLAimn11XKKN2tuJwn0EOLINLBxZgLawcGh4WZJCxcupLEesAVjvCZQ3QWXw6XuSC0AlsBKUrBWV8svM10CALUQ0EC+xEdmtl4h3bniH2noBYBBTQP6/PnzFRU4qwIs8he2mgMgQ9M56N9++62wsPDevXs9evQYM2YMIzWBtTNSu8Tub2KeuMagc+BiV3GwWY0CesSIEbW1tb/99tuaNWtGjx7NVE1g1YyEUcJzpKGj1/wWIp3BtjUK6Pj4+BMnTgQHB8fFxQUHB7/wwgtMlQXWi0NwFrZ7n4aOsIoDbF6jgBaJRBkZGYsXL/bz86urq2OqJoDmyK5OF3AdvHE4IdiupnPQy5YtMz2wpQMDORyO0WjkUL1FMRAEQRB52pwCXW5PmZldxUnvSGfQIaDBhtnFMjtTQDNdhb3I1WaX15bS0JEzX15ZV05DRwBMsYuA5nK5BoOB6SrshUpfJaVldthZIFfVVdHQEQBTnng4bGZm5qNHj4xGo7u7e7t27eisiXQYQdNJZ9D6itrS0JGPqM0rXhNp6AiAKU0DOi0t7Ysvvjh69GhNTY27uztBEI8ePXJwcIiKinr33XcDAgKYKNJSCGg6Rbq/Qk9HHIIj5kno6QuAEY2mONauXRseHt62bduLFy+WlZWlpKSkpKSUlZVdunTJ398/PDx83bp1TBVqCUxx2KrLpUlMlwBAoUYBnZGRcevWrWXLlgUFBTW83rFjx2XLlt26dSs9PZ3e8siBETSdThbG0dbXr2XnaOsLgH6NAnrjxo0SiUSn06WkpFy/fv3BgwcNV0NLJJKNGzfSXiEJENB0uqP6i7a+eARPb9TT1h0AzRoFdH5+fnR0tEwmCw0NHTt27MCBA2Uy2YwZM8rKypiqjxSY4qATnX8VOvFlar2Kvv4A6NUooGNjYwMDAwsLCwsKCtLT04uKijIyMsRi8YwZM5iqjxQYQdOJzjuwp/nNc+bLaOsOgGaNVnEkJycnJibyeLz6Kx4eHuvXr1cqqT1fjmoIaDrN93+Ptr5EXDFtfQHQr9EI2tPT88qVK02ecfHiRQQ0sFN2dUam5gHTVQBQpdEIet26dVFRUSEhIcHBwWKxWK1W37179+bNm/v372eqPlJgDpo2BbrcrOr0fvLB9HRXWltcqMvzl3SgpzsAmjUK6JEjR6alpZ05cyY1NVWj0SiVyrCwsPDwcCcnJ6bqIwVG0LTJ0z6sqKXvK2VnvixVfYe27gBo1vROQplMFhMTY3psMBjy8vLEYnKm+XQ6XVZWlkajkUql/v7+fP4T7zInHQKaNip9JT0bcZjI+ApsxwE2rNEcdGpqalhYmJeX19SpU9PS0jp16hQQEODt7X3p0iVL+mB89R6mOGhjMBpcBe60defq4D7Vdw5t3QHQrFFAz5kzZ/To0T///HPHjh2HDh36+eef63S6vXv3vvPOO5b0wfjqPYygaTPUNSJI2oXOHvkcAZ3dAdCpUUDn5OS89dZbXbt2/fDDD9VqdXR0NEEQERERpaUWbe+bnJz86aefymT/rFc1rd67fPmyJc02HwLahl0oOcN0CQBUaRTQQqEwKyuLIIhr165VVVWZHj969IjLtWjbaMZX7yGgaXO0YJ+RoPWt/m/5RZp7BKBNo2/qPv7445CQEF9f3+Li4u+++27YsGEDBgz49ddf582bZ0kfjK/ewxw0bVLVdzgErUeLOfKcNHq1I09KZ6cA9GgU0DExMYMGDUpLSwsJCXF2du7evfu5c+emT58eHh5uSR+Mr97DCJo29L/NcoFCXVeFgAab1HStm4+Pj4+Pj+lx9+7du3fvTko3MplszJgxWGZn81wd3GjucYLydT6Hvs8SAJ3oOJMQy+zsx6w2i2nuEekMNoyOgMYyO6BOvvbh7arrTFcBQIlnjz7y8/NTU1M9PDw6derUuj4Y3yQPAU2PQl3effXtUBeLvrFoKZW+6p7qVhenEDo7BaDHs0fQFy5cmDRp0pEjR1q96ILxZXaY4qBHUU2Bqq6S5k5lfHllXTnNnQLQ49kj6IkTJ06caNHh9owvs8MImh5VdRVOtG+fLxMoqvUamjsFoIeZgCZ9VyMss7MTRqPR3cGT5k6FXNGctkto7hSAHo3CNz8/f8GCBQkJCXK5XCKRqFQqtVodExOzbt06hUJhSTdYZmcPBrmEMdIvj8N79pMArBAdZxJimR1Q6uyjeNztDTapUUBTtKsRltnZicP5uw1GPf39pqhva/Rq+vsFoFqjeQbTcovBgxudV2T5cgsss7MTD9T3uUzMNjjz5ZV15bjbG2wPHWcSUpT7zYcpDtumELhiBA02iY4zCbHMzk54i3wY6XeUZwwj/QJQzcyZhKQvt2hd7ufk5Ny/f7/Jxezs7FaMhRHQ9Jjma9G2tADQBHuX2eXn5ycnJze5WFRU5Ozc4jNJEdC2raTmUbompY98ENOFAJCsUUrGxsb27t27sLCwfiFHUVHRihUrZsyYcfz48Vb30brc79u3b9++fZtcLCgoyM/Pb2kBmIOmQb7uYYrqzhDXEfR3rTfW3aq6hoAG24NldkCOQl1+tYGZW65lAkVFHU1r6gHohGV2QA5VXaUTr8WzT6QQckVcAjcTgg3CMjsgB4/Dc6N9I456C9u9z1TXANTBMjsgxwDFUKZLALA1ZpbZxcQ0XVVaUlLi6ura6j6wmx1Q7efiUwMVQyU8R6YLASBTo4C+d+/evHnzUlNTBw8e/MUXX9RPQfj4+Gi1Wku6wW52Nm9f7rZJPjM5BIeR3ot0+cU1RW3E7RjpHYAijVZxzJw58/nnn4+Pjw8JCQkNDX3w4AEpfWA3O3uQU53JVDoTBOEskGMhB9ieRsPYGzduXLx4kcvl9ujRo3v37i+//PKFCxe8vLws7IOi5dXNhxG0zXMTeOgMFv2SB8BCjQJaqVT+9ttvgwYNIggiMjLy448/HjFiRHx8vIV9YJmdPfAVt2Gw9/6KUAZ7B6BIoymOVatWvfjii4cPHzb9cdKkSStXrhw8eHBNTY0lfeDQWHswxWcO0yUA2JpGI+ioqKg+ffpwOJyGV4YMGXLixAlL+sAyO6BajaHm5+KEFz3GMl0IAJm4Tf7s6+ubkZFRWlra8GJgYKAlfZiW2c2ePdvT05PP5yuVyvnz56elpQ0fPtySZpsPAU217OqM8yWJDBbA5/BvVV1jsAAAKphZ6xYdHb1r167IyEjTH69evTp9+vSCggJLujG7vLqwsNDTk457zzgcDqY4KFVUk69n4rCrelwO12DEf2KwNWYCukkWR0ZGWpjOT9K2bVsLl1c3E5fLxQiaUpW1Fc582bOfRyWFwIXZAgBI16y7RSwc6hYXF7f6taTAFAfVRDyxl5CZ41TqzWm7hNkCAEjXrIC2cKjr4eHB5Tad7CYIQq+n6ZdiBDTVBmIjDgAKNApoioa6S5YskUqly5cvb3JdJBJR0d3jMAdtD66UXfAXByhFfkwXAkCaRgNbDw8PL3N0Op0lfaxateratWvXr1+3rNTWwxw01XbkbGC6BEKjVwhwwYQAACAASURBVOfpcpiuAoBMjUbQFA11+Xy+2Vu6c3NzLWm2+TDFQbVHukKmSyDkAkV5bemznwdgPZreSUjnUNeSLUxbBFMc9sDDwZvZpX4ApGs0gmZ8qEsRTHFQykgYOzgGMV0F0UbcDtuNgo1pNIJeuHCh2dUapqGuVqtdtGgRTXWRClMclOIQnGjvWKarALBBjQLaz8+va9eua9eubbITdFpa2tq1a7t27err60tveeRAQNuJg3nfMV0CAJkaBfS77757+vTpe/fu9e/fX6FQBAUFBQUFubi49O3b9+7du6dPn3733XeZKtQSmIOm1O2q61fKzjNdBUEQRIaGnCMmAFii6Y0qHTt23LZt27fffpuWlmZaFu3q6tqhQ4eGW9xZHcxBU+pRTYGYHYcBCrlCnUEr5NK0vh6AamZu8Kuurr5y5UqHDh26det26dKlEydOqNVq+isjEaY4KFVZV+HE9EYcJv6SDjhXBWyJmYCeOXOmaS3H/Pnzz549e+vWrZkzZ9JeGJkwxUEpZ77M08Gb6SoIgiCivCY58+VMVwFAGjN7cVy5ciUtLU2r1R47diwzM1Mul1u4HzTjMMVBqaGukUyXAGCbzIygeTweh8NJSkrq1auXQqEgCMLCW70ZhykOO5GhSf219BzTVQCQxkxADxgwICIiYs6cOW+++SZBEMuXLw8JCaG9MDJhioNSW7PWMl3C3zgcbo42k+kqAEhjZopjx44d8fHx3t7epuO93dzcdu7cSXthZMIUB6XKWLMDhoLvgu04wJaYCWgHB4fo6Oj6Py5cuJDGeiiBKQ7q6AxaIVfIdBV/c+LLWLKeBIAUjQK6Q4cOv//+e//+/R9/XpN7C60LApo6dca6rs49mK7ib1wOd7LPLKarACBNo4A+cOCAXC4/cOAAU9VQBHPQ1HHkScPdRjFdBYBtahTQvXv3rv9nZWVlUVGRp6enk5MTM6WRB3PQ9iMu//tRHuPEPAnThQCQwMwqjnv37vXr10+hUPTo0UMulw8dOjQzM5P2wsiEKQ7qXCg5c6vqGtNV/KPGoC2tZfiQYgCymAnoOXPmTJ8+Xa1WV1VVqVSq0aNHz549m/7KSIQpDuoU1xSyarjqInBDQIPNMBPQRUVFc+fONR1zJRaL33777Zwc6z7qDVMc1CmvLZOx6e5qP3E7rrlPNYA1MvNR5vP5DddspKamCgQCGksiH6Y4qOMraiMX0HR0WXN0c+rZxcm676sCqGdmHfTHH3/cu3fv0NBQNze34uLiS5cuff/99/RXRiIENHUiPaKYLgHAZpkZQY8bN+7GjRtjxozp2LFjVFTUrVu3Ro8eTX9lJMIctP0wEsYdORuYrgKAHGZG0ARBtGnTZvTo0WVlZS4uLm5ubjTXRDrMQVOkzli3NWvtfP/3Wvdyg8Fw584dvV5fXV197ty5srKyQYMGVVZWJiYmFhQUGAyGmJiYGTNmmHbvauY8G4fgFNcUtq4eALYxE9AXL16cNWvWgwcPxGKxWq0ODg7evn37gAED6C+OLFwuFyNoKlTUljlwHVrxwpSUlNWrV6ekpDz33HMGg0EqlQ4ZMsTd3f3y5cuOjo6rVq1q27ZtdXX1999/HxERIRKJKisr16xZM3jwYNL/FQDYzExAL1q0aP369eHh4Xw+X6vVHjp0aOLEiVa9FBpz0BSprCuX8RXNfHJmZmZubu7169fPnj3L4XA++uijXr16NXlOv3796h+LxeI33njjjTfeIAiiuLh45syZV65cee+9Z4/WOzp2MRj1XA6v2f8eACxlZg5aoVCMHDmSz+cTBCESiaZOnerg0JpREnsgoCki5Io6Sbs9/Tk1NTXnzp0bPnz4Bx988OOPP8rl8q1btx4/fvzxdH4KNze3Y8eOaTSa2NjYZ+5OHuU1CekMtsHMCLpLly6JiYmRkZEEQRiNxkOHDuFLQjBLKfJTivzM/kij0Xz//fc//PCDUCjs3r37d9995+dn/pnNxOFwVq5cuX///ujo6Li4OLFYbElrAFbBTECfP39+48aNcrlcJpOVlJRotVp/f3/TKYWEdW5rhy8J6XTt2rWdO3deu3ZtypQpiYmJEgmZ9xlOmjRJJBJFRUXNnDmze/fuHTt2fPw5NyqTNXpVf8UQEvsFYISZgP7uu++s/c6UJjDFQZG9uVvDnF86uf9Uly5dlErlgQMHTp8+3aVLl2nTpn399dcUdTp27Fh/f//Lly/v2bMnKCjo008/bTIFJ+KJ0jT3KeodgE5mArpPnz7010EpBDRF7qfc37p8b9QrUSdPnszKyoqJiXnnnXdo+MaiZ8+ePXv2XLhw4c6dO0eOHHno0CEXF5f6n7oI3MpqS6iuAYAGTTfsN/skvV6fkZFBSz2UwDI7KuzevXvfxiNn9v3M4KHvr732WmBg4JgxYw4dOuTl5WW6KBe4egq9mSoJgESNAvrw4cMEQZw6der+/fuTJ092cXEpLCzcsWNHaGgoQ+WRAyNo0r333ntarXbj8S8DlYyls8ngwYO/+uqr6Ojo0aNHL1y4UCQS8Tn8lzyin/1KANZrFNCm07tfe+215ORkDodjuhgREdG9e/fFixczUB1JENDkiouLKykp2b59O9OF/K1nz56//PLLtm3bxo0bd+TIEWtfFQpQz8w66OLi4sLCf26WLSoqKi8vp7Ek8mGZHYmuX7++ceNG03eAdcY6psv5G4/He+ONNyZMmDB16lS9Xv9dzka1XsV0UQCWMvMl4RtvvNG5c+fQ0FCFQlFeXv7LL79Y+8HeWGZHlt27d+/atWv//v1isThNff+26vpoz/FMF/WPyZMnazSaqVOnvrz6hUc1hY5iKdMVAVjETEC///77UVFRv/zyS2lpqUKhWLly5XPPPWd5TzqdLisrS6PRSKVSf39/052K9MAUByn27dt37ty5xMRE0xxCcW2RM5u26jeZNWsWQRDb39oduL2TvziA6XIALGI+JTt37ty5c2ey+sjPz1+wYEFCQoJcLpdIJCqVSq1Wx8TErFu3TqFo7k4OlsAUh+UePny4efPmn3/+uX6Gt7y21Evow2xVZs2aNauwOv/zhf85uud4/VcpANaIjsOBYmNjAwMDCwsLCwoK0tPTi4qKMjIyxGLxjBkzaOidwBSHZdLS0uLj46dPn75hwwbTQWgm3kKfNuJ2DBb2FB8u/HhA90Fvv/0204UAWISOgE5OTv70009lMln9FQ8Pj/Xr11++fJmG3glMcbSW0Whcs2bN/Pnz09LSPv744549ezb8aXfn3go2HXbVxNKlS41G4+bNm5kuBKD16JgI9vT0vHLlSpPNfC9evKhUKmnonUBAt4per582bVrnzp1//PFHsxMFOoNWyBU9fp0l1md8unbt2qioqA4dOoSHhzNdDkBr0BHQ69ati4qKCgkJCQ4ONh0CcPfu3Zs3b+7fv5+G3gncSdhyRqNx9uzZAwcOnDdv3pOeszFz9ZL2K2gsqmV0hmoej7dnz56YmJi8vLxp06YxXRFAi9ExxTFy5Mi0tLTZs2d7enry+XylUmn6rXn48OE09E5gBN1Cubm548aN69Sp01PS2WA0GI2s/jtPwpNq9GqZTJaQkPDrr7+uXr2a6YoAWoymtW4ymWzMmDFYZsd+Z8+e/eSTT9asWdO/f/+nPK2yrtyJL3vKExjXSzbAYDQQBCEQCLZu3Tp16tQffvhh4sSJTNcF0AJ0pCSW2VmLe/fuffLJJ6dPn3Zycnr6M/kcQR/58/RU1ToDFUPrH3M4nB07drz88stKpXLIEOwTDVYDy+zgb7m5ua+99tru3bufmc4EQUj5Tj1l/Z75NGY1vBNdKBTGxcV99NFHf/zxB4MlAbQIltkBQRDEH3/8MW7cuC1btrRr16ylzTqD1sDuOehcbfbh/N0NrygUikOHDi1cuDA1NZWpqgBahI6ANi2za3KRzmV2WMXxdH/88cfbb7997Nix7t27N/MlJwsPZVenU1qVheQCl+KawiYXPT099+zZM3369JIS7OgPVsAultmJRCKtVktPX1YnPT19wYIFx44d8/T0bP6rimsK2XyXCkEQjjypqs7MhnYBAQH/+c9/Jk+enJCQQOc31QCtYBfL7MRicXV1NT19WZecnJwpU6bs2rWr/jiSZqqsK3cWsG6npCYGPOHc2EGDBo0aNWrlypU01wPQUnaxzA4BbVZOTs748eO//fbboKCglr52iGsEh2D7PkRDXEc86Ufz5s2Ljo6+cOHC0KFDaawIoGXoGEHn5+dHR0fLZLLQ0NCxY8cOHDhQJpPNmDGjrKyMht4JBLQ5qamp48eP37ZtW9euXVvx8n7ywc9+EtNqjTV6o97sjzgczrfffvuvf/0rPz+f5qoAms8ultmJxWKNRkNPX+xnNBpPnTo1bdq0vXv3dunSpRUtGIx6VV0V6YWR7n/lv/63/OKTfurq6vr1119PmzattraWzqoAmo+OeYbk5OTExEQej1d/xbTM7umrOI4cObJly5YmF1NSUp509PhTYARd7969e7NmzRo0aFB8fLybm1srG1HdSq9OfZn1B7O6CjzuqP56yhN69+4dExOzbNmytWvX0lYVQPOxdze7V1999dVXX21ycfHixa34nZTP5+v15n/VtSulpaWvvfba3r1727dvb0k7xbVFroJWhjudPIRev5SeefpzZs6cGRsbe+rUqZdeeomeqgCazy6W2QFBEHq9fvLkyf/3f/9nYToTBFFS86iLEwmnoFFNLnB5zrnPM5/2zTffREZGdu/e3c/Pj4aqAJrPLpbZAUEQ33zzTVhYWJPfY1qno2Own4ilZ6k0xCE4zfky08nJaevWrePHj8/KyqKhKoDms4tldpCdnX3kyJGkpCRSWuviFEJKOzSoqquU8p2euSKwa9euO3bsmDBhwr59+yz/DQOALHaxzI4gCHvei6OoqOi111778ssvyfpLsazWau6T/vHRsSxNWnOe2blz5927d0+ZMuXBgwdUVwXQTHaxzI4gCPs83bmuru6777575ZVXli9f3uREwVYzGPU7c74mpSkauDt4FtY091vlwMDAPXv2xMbGZmZmUlkUQHPZxW529ikhIWHw4MFFRUVJSUmkTD2blNaWyAUuZLVGNQ8Hr0e6guY/PyAgYPv27ZMnTy4tLaWuKoBmYu8yO3IJBILa2lqBQEBbj8yKi4vbvXt3UlKSRCIht+WSmkeuAndy26SOv6QDj8N79vMaCA4OXr169bhx444fP96crbEBqGMvy+xM96rYSUAfPHjwhx9+OHLkiFAoJL1xNwePPvJBpDdLEUeetJO0W0tfNXjw4KVLl0ZFRR05cqThb34ANKMjoE3L7M6cOZOamqrRaJRKZVhYWHh4OJ3DE1NAOzs709YjU06dOrV79+6jR49Skc4EQbg6WM3w2eRRTaG7Qwt2UjWJiIjg8/kTJkxISEhoeBMsAJ3oW2YXExNjepybmysUCmn+5VEkEtnD3d5Xrlz58ssvjx8/TlE6EwRRoMvzEtI3N2W5o/l7xytntGLe/IUXXrh169a///1vbEwKTKHjS8L6aL53717Xrl3bt2/v5eU1cODA7OxsGno3kUgkNr9f0v/+979ly5YdPnxYKpVS18v3D7+hrnEqeAl98nQPW/fahQsX3rlz58yZZ9wvDkAROgI6Pj7e9GDu3LmvvPKKWq1WqVTDhg2bN28eDb2b2Px+Sfn5+QsXLjx8+LBczvZ99GnmJfIp0Oa27rWm48A/++yzmzdvklsVQHPQejvf7du3ExMT+Xw+n89fuXKlt7c3bV3bdkAbjcY5c+asX7/ew8OD0o6q6iqlPCubx+8ifa68rvW3RDk7O//www/R0dHHjh1r6bkzABaiYwRdr0uXLvXTGllZWWKxmLaubTugN27c2KtXr379+lHdkZAretFjLNW9kEvKd/YVtbWkBR8fn9WrV3/wwQdklQTQTHQEtE6nc3d379Chw/3797/44guCIG7fvj106NA333yTht5NbDig4+Lizp07R098OHAd2kkCaeiIXDnVmRa2MGTIkPLy8hs3bpBRDkBz0THFUV1dXV5eXlFRUV5eblq8oVAotm3bFhkZSUPvJrYa0MePH9+3b19cXBw9m0/lax+6CT0EHAca+iLR0YK9s9u8LeZZdM/O559/vmTJkoSEBLKqAngmOkbQIpHIy8srKCioX79+wcHBBEEolUo605kgCIlEYnsB/eeff27YsOHgwYPULapr4mRRnFUcdtWEl9CnQNfK7wnrBQUF9e7d+8MPPySlJIDmoHUOmkG2N4LOz8+fP3/+vn37RCIRbZ2WWdVGHPWUIr88bY7l7axYsUKn0y1fvtzypgCaAwFtlaqqqiZMmLBp0yY6V8IQBGE0Es/cW5mFejj3a8UN32atWbMmJyfn+PHjpLQG8HT2smu+LQV0bW3thAkTPvzwwx49etDc9USf12jukRRSvpOUIO3O1a+//joiIqJXr144IguohhG09dm4cWNERER4eDj9XbcVB9DfKSlS1LfJasrR0XHTpk3Tp0/X6XRktQlgFgLaypSWlh49epTOmzDrFehyK+sq6O+XFOeKf6yoJe0En+7du8+dO3fOnDlkNQhglh0FtG3sxfHhhx9+9NFHjJzo+EvJ2ZKaIvr7JYWf2D9bm0Fig9HR0Uqlcv369SS2CdCEHQW0DYygN2/erNFoRowYwUjvedocpchaZ139RP4Pq0k+tPvTTz9NSkq6dOkSuc0C1ENAW43Vq1cnJyfv2LGDqQJ0Bp2QS9+SPnIFOz3XXxFKbptcLnfXrl1Lly7Nz2/usYcALYKAtg6bNm3KyMjYtm0bg5vHz/Cbz1TXluNzBAqBK+nNurq6btq0ady4cYWFhaQ3DoCAtgKHDh06f/78pk2bmD2b3NOq9ul/3PXK/9UZa0lvtkePHuvXr3/11VeLi4tJbxzsnL0EtEAgqKurY7qK1khKStq+ffuePXuYPXgpR5tZUvOIwQIsd091k/RpaJPevXuvWbNmwoQJWq2WivbBbtlLQBMEYTQamS6hxa5du7Zy5cq4uDg67+c260rp+cq6cmZrsJC/uEO6JpWixgcMGDBv3rzp06db48cMWMuOAprZ+YFWKCwsnDdv3sGDB9lwsHS+7qG3yJfpKiwS4BiUVZ1GXftjx47t3Lnz119/TV0XYG/sKKCtS21t7dSpU7/66iuad9t4Ep1BK+LSd8ACFdwdPCcoqb1V/cMPPzx58iS2jQayIKBZasGCBZMnT+7bty/ThfztjbbvMF0CCSzcEvqZeDze9u3b586d++iRdc/XA0sgoNlo06ZNUql02rRpTBfyDxlfwXQJJEhR37b8dJWna9u27TfffDNu3LjS0lJKOwJ7gIBmnc2bN//000//93//x3Qh/7hVda281hbipsZQc73yKtW9hISEfPLJJ2PGjKk/gROgdewooPl8fm0t+ctgSaTX62fPnp2Wlnbo0CFmF9U1can0Z2vcBvpx7SUdMyhbyNHQ4MGDN2/ePH78+D///JOG7sBW2VFAs/xeFaPROG/evOeee+6LL75gZC+kpyivLZUJbGGKQ8JzNBI0LYPr2rXrsWPH5s+fjxvBodUQ0GyxdOlSPz+/+fNZdzt1jaHGkUfabveMW9SOvkMFvby8vvnmmylTpmDnaGgdBDTzjEbjokWLHB0d2XkgqQPXYZ7/UqarsFY9e/Z8/fXXGdm/G2wAApp5Cxcu9PHxWbFiBdOFPBGfw64pF0vojfoj+Xvp7HHSpElubm64gQVawY4CWiKRsDCg165dK5FIli5l7xD1p+KTeqOe6SpIw+PwUtS3aZuJNvn888/Pnz///fff09kp2AA7Cmi2jaBrampWrVqVnJy8atUqpmt5mj/Kr/A4LFpSYjkfUZuHWkp2TXoSHo938ODBc+fOrV69ms5+wdohoJlx9+7dsLAwhUKxd+9eLpe9/xXKakuo2EaZWT1k/Ug8n7CZBALBrl27Hj58uGHDBpq7ButlO3OLz8SegL5z587rr79+4MCBtm3bMl3LM+Rpc7o69WC6CpJ1c+rJSL8cDmfDhg2TJ0+Wy+WxsbGM1ADWhb1jN9I5OjqqVCqmqyD++uuvmTNnHjx4kP3pTBBEF6eQ511eYLoK22E6Jeu3336bPHkyNviHZ7KjgPb398/IIPNc51b49ddf33zzzcOHD7dp04bZSuzcmUcn7qqY2XNOKBRu3rx51qxZUVFRFRUVjNQA1sKOArpDhw6pqXTc5vskcXFxK1asOHHihFJpHWdHVdVVHivYz3QVlOjoGPxnxX8ZLGDo0KErV66cOHFiTU0Ng2UAy9nRHHTbtm2zsmj97t6kpqbml19+OXbsWFVVVUJCglAopL+G1klR35bwHJmughL+kg4/5O0wEkYG9xgJCwsrKSkZPnz46tWrBw4cyFQZwGZ2NILm8/l6Pd3reXU6XVRU1IULFyZPnrxnzx4rSmeCIO6rbgU5dmG6CkpwCM5Q10jG13ePGzfuwIED69evf//99+n/cAL72VFAEwTh5ORUVVVFW3d6vX7q1KnTpk377LPPBg0aRFu/ZJHwpG0lAUxXQZWBiqFsuENSqVTGxcV5e3uPHj0aU9LQhH0FdIcOHR48eEBPX5WVlWPHjo2IiIiJiaGnR9K94jXRNnYZfZJCXR7TJfxtwYIFS5YsGTVqVEFBAdO1AIvYV0AHBgbS8D1hWlrapk2bRo4cOW/evNdff53q7ihSZ6xjugTK/VR8kp7toZsjLCzsyy+/jIqKwhbSUM++ApqGEfQPP/wwZ84cZ2fn48ePR0REUNoXpdan/5vmDSvoN1Ax7HJpEtNV/KNXr15HjhxZunTphg0bsLoDCHsLaKpH0Fu2bDl69OipU6emTJni7u5OXUdUq6wrd+AKbXt+gyCI9pKOOdpMxr8qbEipVP74448EQQwZMuT06dNMlwMMs6+AbtOmDUUr7QwGw3vvvffXX38dOHDAupZqmHW98n/POfdhugo6zG7zNpfDrv8LBALBwoULf/75523btm3fvp3pcoBJ7PpoUo2ilXZXrlyJjIz09vbevHkzq84SbDVPB2Vf+fNMV0EHNwcPdv6i4OjoeOjQoeTk5Pnz52s0GqbLAWYwv8yIZqaVdk5OJJzhZDQaz5w5s2HDBqVSuW3bNqvYW6OZgqS2ufzZrGsV/3V1cG8jbs90IU3x+fzNmzcfPnz4hRdeGDt27Kuvvtq+PeuKBErZ1wiaIIjnnnvu6tWrlrdz6tSpYcOGJSUlmX4PtaV0TlHfMbBpWpZq3iK/k4WHmK7iiaKjo3/66ad27drNnTt33bp1RqONf3MLDdldQL/66quHDln0f2N1dfWsWbNOnTp16NChNWvW+Pj4kFUbGxiM+kN533NYNi1LKS+hkkNw8nUPmS7kiaRSaXR09I8//lhZWTlq1KikpCTEtJ2wo/8PTXr27Hnjxo26utYs8q2pqfn+++9feOGFiIiITZs2WfU6jSf5s/K/3Zx7snNaljqjvcY/rGZgn5YW4XK5K1as+PLLL0+fPj148OB9+/bh7nCbZ3cBTRBEWFjYuXPnWvSS1NTURYsWDR069NGjRz/++GN0dDRFtTHuSun5F9xeYroKuvmK2vaRW8e9+IGBgWvXrj116lRmZmZoaOixY8cwmrZh9hjQ48ePP3jwYDOfbDQaN27c+MYbb4wfP/7XX3995513ZDIZpeUx603/ZY48KdNVMCNVfddgNDBdRbPIZLIPPvggISHh6tWroaGh+/bt02q1TBcF5LPHgO7WrVtKSkp6evrTn6bVag8cOBAaGqpSqc6cOTNw4EAOx/Z/8efa1vmwLZJTnZFQxN5vCx+nUChWrVp18uTJvLy84cOHz5w58+TJk1iTZ0vsMaAJgti5c+eUKVPu3r37+I+ys7O3bNkyatSoF198MTMz8+TJk8uWLePzbX894vmSRFbd90y/YW4js6rTU9VmPhVsJpfL33333cuXLy9atOjmzZujR48ODw/funVrbW0t06WBpWw/d8wKDAz84YcfXn/99Xbt2o0ePVqr1ebn5//++++ZmZlt2rSJjIzcuXOnTX4H+CTFNUX/Lbu0NODfTBfCJA7Bme47/0JJYqBjZ6ZraY1u3bp169bt/fffV6vV27ZtCw0Nffnll8PDw/v06WMPv/zZJI51fcOwePHi/Pz8AwcOkNXgn3/+mZSUxOFw2rRpM2DAAD8/P7Jati4bM1eP8ZzgJ/ZnuhAgjUajOX/+fGJi4p9//vniiy8OHTq0V69eIpGI6bpszcyZM//1r38FBFCyc7qdjqDr9ezZs2fPnkxXwbwpPrPlAhemq2CRH/J2jPIYJ+U7M11I60kkkpdeeumll16qqak5c+bMiRMnli9fThBEr169goOD+/fvHxQUxHSN8Az2HtCQq832EbVBOjfRV/b8V5mfTfed7yOy+vPXHRwcRo0aNWrUKIIgNBrNX3/9dfv27VWrVt2/f9/d3T0gIKBLly7dunVr06aNp6cnl2un30uxEwLaftUZa3c/3OLh4GUDGUS6AMegN9q880Pejllt3hJybWdaQCKRDBgwYMCAATNnziQIoqKiIj09/ebNm0ePHs3JycnKyvL39+/YsWNlZaWrq2tQUFCHDh0CAgKkUjtdeck4+gJap9NlZWVpNBqpVOrv728P6yLYrLimaFv2l2FuL/aTD2a6FpZydXB/03+Z6fGNyuQuTiE8m1uDKJPJevTo0aNHj/oraWlp2dnZzs7OxcXF9+7d++WXXzIyMqqqqoxGo0gk8vDwcHV1dXV1VSgUCoVCLpebNtdVKBRCobB9+/YSiYS5fxsbREdK5ufnL1iwICEhQS6XSyQSlUqlVqtjYmLWrVunUChoKADq6Y36iroyF4GbM1+2wP9fVj3HSqc8Xc7Jorh24sAwt5FeQpvafaWJgICA+u+7mhwJpNVqHz16VFJSUlJSkpR7zAAACitJREFUUlZWVlZWlpeXZzAYysvLjUajSqVKS0urv19Gr9crFAoPDw+ZTMbn82UyGY/Hk8vlpp+KRCKxWEwQBIfDcXd3N43QjUZjeXm5RCJxd3fn8XgGg8E031L/ZIIgnJyc7GpsR8e/amxsbO/evQsLC+vvwSsqKlqxYsWMGTOOHz9OQwH2TG/Um8Z9F0oSb1VdU9WpBrkMG+wy3IErdOBa/cECtIl0fyXCfUy6JsV0VGOBLu+nR/FeQh93oVegY2c7ufdSJBL5+fk1f6VTeXl5UVFRVVWVRqPR6XQ6nU6tVpsW/FVWVpaVlQmFQr1ef+vWrfpYl0qlarW6tLS0vhG9Xl9ZWWl6bDAYKioqdDodj8erra0Vi8UGg8H0S7npCTqdzmg0CgQCtVrt7OxcUVHh4OAgFoslEgmXy1WpVARBODs7m/7CKC8vJwiCx+M5Ozs7ODg4OjoSBFFTU6NWqwmCCA0NnTx5MinvmyXoCOjk5OTExMSGO9l7eHisX79eqVQ+5VXx8fF79+5tcvGvv/7y9/dvZr9H8vcW1eQTBOHq4BHjPc1mrhzN31tYk08QhJuDxzjvaQRBHMjb+VCbRRCEt9B3ss8sgiD2527Prs7gcAg/UbtJPjMJgmgnCewpG+DMt+X71CnFITgBkr+XPXgKvcPcXizQ5Rbq8twdPB150hxt5r6H2wjCyOcKojwnBTgG5Wgz9z38ls8VEARRf+Vg3ncEQYi5kpEeUe0lHXOqM+vvXYx0f6WdJLDhlRHuowMkQdZ9RUiM8B3dS9LrSc/hEoQn4TjCfWL9FVdC2IZwIbee2wU3EouOCx0d9DX6wbIR3hzf9LLUS+qfBEKBVqUdJAuT17rlaXIulf4sJAiZVDbEfUSIT+/WfU7IRcc66M6dO3/77beDBzea6/z555/ffvvtGzduPOlVtbW1pr/xGjp69KharV64cGFz+q0x6GqNtQRB8Dl80/c8NnZFwHFw4DoQBFFjqBFwBfa2BZ01Uuv//khLeI4cgmMkjBq92nRFzJVwOVxcYc+VZ//nJAjCBtZBr1u3LioqKiQkJDg4WCwWq9Xqu3fv3rx5c//+/U95lUAgeHyG2snJSafTNbNfB67QgRDaxxUHAqxBk8kQDsHBFdZeYQM6AnrkyJFpaWlnzpxJTU3VaDRKpTIsLCw8PJyUc6cAAGwVTd+HymSymJgYevoCALANuGsIAIClENAAACyFgAYAYCkENAAASyGgAQBYCgENAMBSCGgAAJZCQAMAsJSVbdwnk8k+//zzY8eOPf1p165ds66zFqmm0WiwUW9DeEOawBvSkNFo5PF4ISEhzXlyTk5O/W6opLOyQ2ObaezYsTt27MBm0/WGDRt2/vx5pqtgEbwhTeANaaioqGjBggUHDx5kuhBMcQAAsBUCGgCApRDQAAAshYAGAGApBDQAAEvZZkBzuVzTecBgYlcHITcH3pAm8IY0xJ4Asc1ldlVVVTiupSG8IU3gDWkCb0gTLHlDbDOgAQBsACuG8QAA8DgENAAASyGgAQBYCgENAMBSCGgAAJZCQAMAsBQCGgCApWwtoBMTE7t16+bq6hoZGVlQUMB0OczQ6XQcDkf0/8XExJiu29ubU1dXt3TpUi6XW1xcXH/R7JtgJ+/M42+InX9U4uPjO3fuLJfLhw4dmpKSYrrIrk+I0YaUl5e7ublduXKltrb2gw8+iI6OZroiZuTn57u5uTW5aIdvzpgxY5YvX87j8R49emS6YvZNsJ935vE3xJ4/Kjk5OTKZ7PLly3q9/oMPPhg2bJiRfZ8QmwrouLi4iIgI0+Py8nKhUKjVapktiRH37t0LCAhoctEO3xzTyWcN88jsm2A/78zjb4g9f1RycnLi4uJMj//8808fHx8j+z4hNjXFkZKS0rFjR9NjmUwml8uzsrKYLYkR5eXlGo0mLCzMw8MjPDzc9LubHb45j58pZ/ZNsJ935vE3xJ4/Kr6+vuPGjTM9/umnn55//nmCfZ8QmwpojUbT8PRGiUSi0WgYrIcpTk5Oo0aN2rRpU3Z2du/evaOjowm8OQRBPOFNsOd3Bh8VgiDOnDmzZcuWL774gmDfJ8SmAtrR0VGtVtf/UaVSSaVSButhSnBw8NatWzt16iQSif7973/fv38/Ly8Pbw7xhE+IPb8z+Kjs379/0aJFZ8+e9fX1Jdj3CbGpgO7UqdPNmzdNj3Nzc9Vqtb+/P6MVMSM/P//OnTumxwaDQa/XOzg44M0hnvAJsed3xs4/KidOnFi9evWFCxc6dOhgusK6Twg9U930qKysdHNzS0pKqq2tnTt3bmxsLNMVMeP06dNt2rTJyMioq6v78MMP+/XrZ7TjN6fhd2Jm3wR7e2caviH2/FEpLS319fXNyMhoeJFtnxCbCmij0Xj27Nng4GAXF5eXX365uLiY6XIYs2rVKh8fH1dX14iIiPqPoF29OcXFxUKhUCgUEgRhelBQUGB8wptgD+/Mk94Qu/2o7Ny5k8PhCBsw/Zuy6hOCDfsBAFjKpuagAQBsCQIaAIClENAAACyFgAYAYCkENAAASyGgAQBYCgENAMBSCGgAAJZCQAMAsBQCGgCApRDQAAAshYAGAGApBDQAAEshoAEAWAoBDQDAUghoAACWQkADALAUAhoAgKUQ0AAALIWABptSV1e3dOlSLpdbXFz8+E/j4+M7d+4sl8uHDh2akpJiupiYmNitWzdXV9fIyMiCgoKnXASgGQIabEp0dLREIuFyzXywHz58GBsbu3379tLS0ueff/6NN94gCKKiomLq1KnffvttYWFh7969FyxY8KSLAAyg7fxwABpcu3bNaDTyeLxHjx7VX/T09Lx06VJOTk5cXJzpyp9//unj42M0GuPi4iIiIkwXy8vLhUKhVqs1e5HWfw0Ao9FoNGIEDTYlJCTk8Yu7du0KDg729fUdN26c6cpPP/30/PPPEwSRkpLSsWNH00WZTCaXy7OyssxepKV8gEb4TBcAQLnIyMiGfzxz5syWLVsuXrxIEIRGoxGLxfU/kkgkGo3G7EXaqgWohxE02Jf9+/cvWrTo7Nmzvr6+BEE4Ojqq1er6n6pUKqlUavYiA7WC3UNAgx05ceLE6tWrL1y40KFDB9OVTp063bx50/Q4NzdXrVb7+/ubvchIwWDnENBg+xITE0tLS8vKyt588834+HgvL6/6H4WHh9+5c+fcuXN1dXWfffZZdHQ0n883e5HB+sFucYxGI9M1AJCjpKTEx8eHIAidTicUCgmCyMrK8vT09PLyOnz4cGpq6uuvv+7g4FD//NzcXFdX159++umtt94qKCgYOHDgrl27XF1dCYIwexGAZghoAACWwhQHAABLIaABAFgKAQ0AwFIIaAAAlkJAAwCwFAIaAIClENAAACyFgAYAYCkENAAASyGgAQBYCgENAMBSCGgAAJZCQAMAsBQCGgCApRDQAAAshYAGAGApBDQAAEshoAEAWOr/AV4E5le/T+luAAAAAElFTkSuQmCC\n"
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "V6e_ohwElCfr"
      },
      "source": [
        "# 3. Assembling E. coli X genome from paired reads"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bpe7qcdzlBgF",
        "outputId": "7abf8799-6f84-41b5-d21a-4b3940a121aa"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  bamtools bwa cwltool fonts-glyphicons-halflings libbamtools2.5.1 libc-ares2\n",
            "  libhat-trie0 libjs-bootstrap libnlopt0 libnode64 libssw0 nodejs nodejs-doc\n",
            "  python3-atomicwrites python3-attr python3-bs4 python3-cachecontrol\n",
            "  python3-coloredlogs python3-dateutil python3-decorator python3-html5lib\n",
            "  python3-humanfriendly python3-importlib-metadata python3-isodate\n",
            "  python3-joblib python3-lockfile python3-lxml python3-mistune\n",
            "  python3-more-itertools python3-msgpack python3-mypy-extensions\n",
            "  python3-networkx python3-packaging python3-pluggy python3-prov\n",
            "  python3-psutil python3-py python3-pyparsing python3-pytest python3-rdflib\n",
            "  python3-rdflib-jsonld python3-ruamel.yaml python3-schema-salad\n",
            "  python3-shellescape python3-simplejson python3-soupsieve\n",
            "  python3-sparqlwrapper python3-typing-extensions python3-wcwidth\n",
            "  python3-webencodings python3-yaml python3-zipp samtools\n",
            "Suggested packages:\n",
            "  docker.io | singularity-container npm python-attr-doc python3-genshi\n",
            "  python-lockfile-doc python3-lxml-dbg python-lxml-doc python-networkx-doc\n",
            "  python3-matplotlib python3-pygraphviz | python3-pydot python3-scipy\n",
            "  python-prov-doc python3-pydot python-psutil-doc subversion\n",
            "  python-pyparsing-doc python-rdflib-doc\n",
            "The following NEW packages will be installed:\n",
            "  bamtools bwa cwltool fonts-glyphicons-halflings libbamtools2.5.1 libc-ares2\n",
            "  libhat-trie0 libjs-bootstrap libnlopt0 libnode64 libssw0 nodejs nodejs-doc\n",
            "  python3-atomicwrites python3-attr python3-bs4 python3-cachecontrol\n",
            "  python3-coloredlogs python3-dateutil python3-decorator python3-html5lib\n",
            "  python3-humanfriendly python3-importlib-metadata python3-isodate\n",
            "  python3-joblib python3-lockfile python3-lxml python3-mistune\n",
            "  python3-more-itertools python3-msgpack python3-mypy-extensions\n",
            "  python3-networkx python3-packaging python3-pluggy python3-prov\n",
            "  python3-psutil python3-py python3-pyparsing python3-pytest python3-rdflib\n",
            "  python3-rdflib-jsonld python3-ruamel.yaml python3-schema-salad\n",
            "  python3-shellescape python3-simplejson python3-soupsieve\n",
            "  python3-sparqlwrapper python3-typing-extensions python3-wcwidth\n",
            "  python3-webencodings python3-yaml python3-zipp samtools spades\n",
            "0 upgraded, 54 newly installed, 0 to remove and 22 not upgraded.\n",
            "Need to get 38.2 kB/18.1 MB of archives.\n",
            "After this operation, 86.7 MB of additional disk space will be used.\n",
            "Ign:1 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libc-ares2 amd64 1.15.0-1ubuntu0.1\n",
            "Err:1 http://security.ubuntu.com/ubuntu focal-updates/main amd64 libc-ares2 amd64 1.15.0-1ubuntu0.1\n",
            "  404  Not Found [IP: 91.189.91.39 80]\n",
            "\u001b[1;31mE: \u001b[0mFailed to fetch http://security.ubuntu.com/ubuntu/pool/main/c/c-ares/libc-ares2_1.15.0-1ubuntu0.1_amd64.deb  404  Not Found [IP: 91.189.91.39 80]\u001b[0m\n",
            "\u001b[1;31mE: \u001b[0mUnable to fetch some archives, maybe run apt-get update or try with --fix-missing?\u001b[0m\n"
          ]
        }
      ],
      "source": [
        "!apt -y install spades"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "-js_Ijqyl19V",
        "outputId": "36cc0da1-2505-4aaf-eeb8-ce996b556185"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/bin/bash: spades: command not found\n"
          ]
        }
      ],
      "source": [
        "!spades -1 SRR292678_1.fastq -2 SRR292678_2.fastq -o output"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "LHxlvoZnnnw2",
        "outputId": "fbaa984c-6b4e-4cda-cbd0-e6ffa1e3cdeb"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Collecting quast\n",
            "  Downloading quast-5.2.0.tar.gz (34.2 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m34.2/34.2 MB\u001b[0m \u001b[31m24.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: joblib in /usr/local/lib/python3.8/dist-packages (from quast) (1.2.0)\n",
            "Collecting simplejson\n",
            "  Downloading simplejson-3.18.3-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (135 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m135.5/135.5 KB\u001b[0m \u001b[31m14.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hBuilding wheels for collected packages: quast\n",
            "  Building wheel for quast (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for quast: filename=quast-5.2.0-py3-none-any.whl size=31206447 sha256=281f73ed867a5bfd27bbaa73b629be2644c3b00567c0bd0ef551a5930cb65a97\n",
            "  Stored in directory: /root/.cache/pip/wheels/79/fe/84/429b22fc6184f0807dfdcb67bdf0d95441b3415bce83219ad2\n",
            "Successfully built quast\n",
            "Installing collected packages: simplejson, quast\n",
            "Successfully installed quast-5.2.0 simplejson-3.18.3\n"
          ]
        }
      ],
      "source": [
        "!pip install quast"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "FVmXaOYhq6vp",
        "outputId": "b9f867dd-bfab-4f95-9004-64335ececf94"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/usr/local/bin/quast.py output/contigs.fasta\n",
            "\n",
            "Version: 5.2.0\n",
            "\n",
            "System information:\n",
            "  OS: Linux-5.10.147+-x86_64-with-glibc2.29 (linux_64)\n",
            "  Python version: 3.8.10\n",
            "  CPUs number: 2\n",
            "\n",
            "Started: 2023-03-03 11:27:07\n",
            "\n",
            "Logging to /content/quast_results/results_2023_03_03_11_27_07/quast.log\n",
            "NOTICE: Maximum number of threads is set to 1 (use --threads option to set it manually)\n",
            "\n",
            "CWD: /content\n",
            "Main parameters: \n",
            "  MODE: default, threads: 1, min contig length: 500, min alignment length: 65, min alignment IDY: 95.0, \\\n",
            "  ambiguity: one, min local misassembly length: 200, min extensive misassembly length: 1000\n",
            "\n",
            "Contigs:\n",
            "  Pre-processing...\n",
            "  output/contigs.fasta ==> contigs\n",
            "\n",
            "2023-03-03 11:27:08\n",
            "Running Basic statistics processor...\n",
            "  Contig files: \n",
            "    contigs\n",
            "  Calculating N50 and L50...\n",
            "    contigs, N50 = 105346, L50 = 15, auN = 127832.1, Total length = 5245723, GC % = 50.53, # N's per 100 kbp =  0.00\n",
            "  Drawing Nx plot...\n",
            "    saved to /content/quast_results/results_2023_03_03_11_27_07/basic_stats/Nx_plot.pdf\n",
            "  Drawing cumulative plot...\n",
            "    saved to /content/quast_results/results_2023_03_03_11_27_07/basic_stats/cumulative_plot.pdf\n",
            "  Drawing GC content plot...\n",
            "    saved to /content/quast_results/results_2023_03_03_11_27_07/basic_stats/GC_content_plot.pdf\n",
            "  Drawing contigs GC content plot...\n",
            "    saved to /content/quast_results/results_2023_03_03_11_27_07/basic_stats/contigs_GC_content_plot.pdf\n",
            "  Drawing Coverage histogram (bin size: 1x)...\n",
            "    saved to /content/quast_results/results_2023_03_03_11_27_07/basic_stats/coverage_histogram.pdf\n",
            "  Drawing contigs coverage histogram (bin size: 1x)...\n",
            "    saved to /content/quast_results/results_2023_03_03_11_27_07/basic_stats/contigs_coverage_histogram.pdf\n",
            "Done.\n",
            "\n",
            "NOTICE: Genes are not predicted by default. Use --gene-finding or --glimmer option to enable it.\n",
            "\n",
            "2023-03-03 11:27:11\n",
            "Creating large visual summaries...\n",
            "This may take a while: press Ctrl-C to skip this step..\n",
            "  1 of 2: Creating PDF with all tables and plots...\n",
            "  2 of 2: Creating Icarus viewers...\n",
            "Done\n",
            "\n",
            "2023-03-03 11:27:11\n",
            "RESULTS:\n",
            "  Text versions of total report are saved to /content/quast_results/results_2023_03_03_11_27_07/report.txt, report.tsv, and report.tex\n",
            "  Text versions of transposed total report are saved to /content/quast_results/results_2023_03_03_11_27_07/transposed_report.txt, transposed_report.tsv, and transposed_report.tex\n",
            "  HTML version (interactive tables and plots) is saved to /content/quast_results/results_2023_03_03_11_27_07/report.html\n",
            "  PDF version (tables and plots) is saved to /content/quast_results/results_2023_03_03_11_27_07/report.pdf\n",
            "  Icarus (contig browser) is saved to /content/quast_results/results_2023_03_03_11_27_07/icarus.html\n",
            "  Log is saved to /content/quast_results/results_2023_03_03_11_27_07/quast.log\n",
            "\n",
            "Finished: 2023-03-03 11:27:11\n",
            "Elapsed time: 0:00:04.517275\n",
            "NOTICEs: 2; WARNINGs: 0; non-fatal ERRORs: 0\n",
            "\n",
            "Thank you for using QUAST!\n"
          ]
        }
      ],
      "source": [
        "!quast.py output/contigs.fasta"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "03mxoOrlriNm",
        "outputId": "64b3c9e1-3b6c-4af5-c49e-193e9aeba1d3"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "All statistics are based on contigs of size >= 500 bp, unless otherwise noted (e.g., \"# contigs (>= 0 bp)\" and \"Total length (>= 0 bp)\" include all contigs).\n",
            "\n",
            "Assembly                    contigs \n",
            "# contigs (>= 0 bp)         510     \n",
            "# contigs (>= 1000 bp)      147     \n",
            "# contigs (>= 5000 bp)      82      \n",
            "# contigs (>= 10000 bp)     69      \n",
            "# contigs (>= 25000 bp)     51      \n",
            "# contigs (>= 50000 bp)     33      \n",
            "Total length (>= 0 bp)      5316686 \n",
            "Total length (>= 1000 bp)   5206945 \n",
            "Total length (>= 5000 bp)   5043449 \n",
            "Total length (>= 10000 bp)  4953165 \n",
            "Total length (>= 25000 bp)  4664641 \n",
            "Total length (>= 50000 bp)  4002075 \n",
            "# contigs                   201     \n",
            "Largest contig              300784  \n",
            "Total length                5245723 \n",
            "GC (%)                      50.53   \n",
            "N50                         105346  \n",
            "N90                         21421   \n",
            "auN                         127832.1\n",
            "L50                         15      \n",
            "L90                         54      \n",
            "# N's per 100 kbp           0.00    \n"
          ]
        }
      ],
      "source": [
        "!cat quast_results/latest/report.txt"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "tGI7s-Ssr4Vk"
      },
      "source": [
        "# 3a. Effect of read correction."
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!unzip spades_output.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Q5k1UPoBIA2i",
        "outputId": "66b60c82-b1a3-4f1e-dc98-972e1847b823"
      },
      "execution_count": 14,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  spades_output.zip\n",
            "  End-of-central-directory signature not found.  Either this file is not\n",
            "  a zipfile, or it constitutes one disk of a multi-part archive.  In the\n",
            "  latter case the central directory and zipfile comment will be found on\n",
            "  the last disk(s) of this archive.\n",
            "unzip:  cannot find zipfile directory in one of spades_output.zip or\n",
            "        spades_output.zip.zip, and cannot find spades_output.zip.ZIP, period.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "maRo7YOpsCLw"
      },
      "source": [
        "# 4. Impact of reads with large insert size"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true,
          "base_uri": "https://localhost:8080/"
        },
        "id": "09KFbVsVsskd",
        "outputId": "8ea86f80-f8c2-48bd-a98f-27ad3488b959"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Command line: /usr/lib/spades/bin/spades.py\t--pe1-1\t/content/SRR292678_1.fastq\t--pe1-2\t/content/SRR292678_2.fastq\t--mp1-1\t/content/SRR292770_1.fastq\t--mp1-2\t/content/SRR292770_2.fastq\t--mp2-1\t/content/SRR292862_1.fastq\t--mp2-2\t/content/SRR292862_2.fastq\t-o\t/content/spades_output\t\n",
            "\n",
            "System information:\n",
            "  SPAdes version: 3.13.1\n",
            "  Python version: 3.8.10\n",
            "  OS: Linux-5.10.147+-x86_64-with-glibc2.29\n",
            "\n",
            "Output dir: /content/spades_output\n",
            "Mode: read error correction and assembling\n",
            "Debug mode is turned OFF\n",
            "\n",
            "Dataset parameters:\n",
            "  Multi-cell mode (you should set '--sc' flag if input data was obtained with MDA (single-cell) technology or --meta flag if processing metagenomic dataset)\n",
            "  Reads:\n",
            "    Library number: 1, library type: mate-pairs\n",
            "      orientation: rf\n",
            "      left reads: ['/content/SRR292770_1.fastq']\n",
            "      right reads: ['/content/SRR292770_2.fastq']\n",
            "      interlaced reads: not specified\n",
            "      single reads: not specified\n",
            "      merged reads: not specified\n",
            "    Library number: 2, library type: mate-pairs\n",
            "      orientation: rf\n",
            "      left reads: ['/content/SRR292862_1.fastq']\n",
            "      right reads: ['/content/SRR292862_2.fastq']\n",
            "      interlaced reads: not specified\n",
            "      single reads: not specified\n",
            "      merged reads: not specified\n",
            "    Library number: 3, library type: paired-end\n",
            "      orientation: fr\n",
            "      left reads: ['/content/SRR292678_1.fastq']\n",
            "      right reads: ['/content/SRR292678_2.fastq']\n",
            "      interlaced reads: not specified\n",
            "      single reads: not specified\n",
            "      merged reads: not specified\n",
            "Read error correction parameters:\n",
            "  Iterations: 1\n",
            "  PHRED offset will be auto-detected\n",
            "  Corrected reads will be compressed\n",
            "Assembly parameters:\n",
            "  k: automatic selection based on read length\n",
            "  Repeat resolution is enabled\n",
            "  Mismatch careful mode is turned OFF\n",
            "  MismatchCorrector will be SKIPPED\n",
            "  Coverage cutoff is turned OFF\n",
            "Other parameters:\n",
            "  Dir for temp files: /content/spades_output/tmp\n",
            "  Threads: 16\n",
            "  Memory limit (in Gb): 12\n",
            "\n",
            "\n",
            "======= SPAdes pipeline started. Log can be found here: /content/spades_output/spades.log\n",
            "\n",
            "\n",
            "===== Read error correction started. \n",
            "\n",
            "\n",
            "== Running read error correction tool: /usr/lib/spades/bin/spades-hammer /content/spades_output/corrected/configs/config.info\n",
            "\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from N/A, git revision N/A\n",
            "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /content/spades_output/corrected/configs/config.info\n",
            "  0:00:00.002     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 2\n",
            "  0:00:00.002     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 12 Gb\n",
            "  0:00:00.002     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
            "  0:00:00.005     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
            "  0:00:00.005     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
            "  0:00:00.005     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
            "     === ITERATION 0 begins ===\n",
            "  0:00:00.005     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
            "  0:00:00.005     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 32 files using 2 threads. This might take a while.\n",
            "  0:00:00.005     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1048576\n",
            "  0:00:00.005     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 1.99935 Gb\n",
            "  0:00:00.005     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2097152\n",
            "  0:00:00.006     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /content/SRR292770_1.fastq\n",
            "  0:00:22.571     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1151751 reads\n",
            "  0:00:41.437     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2301589 reads\n",
            "  0:01:05.532     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 3452535 reads\n",
            "  0:01:24.956     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 4604364 reads\n",
            "  0:01:35.054     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 5102041 reads\n",
            "  0:01:35.054     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /content/SRR292770_2.fastq\n",
            "  0:01:55.863     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 6252803 reads\n",
            "  0:02:17.402     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 7404737 reads\n",
            "  0:02:37.461     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 8556627 reads\n",
            "  0:02:58.909     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 9708905 reads\n",
            "  0:03:06.393     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /content/SRR292862_1.fastq\n",
            "  0:04:46.591     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /content/SRR292862_2.fastq\n",
            "  0:05:26.496     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 17608756 reads\n",
            "  0:06:15.778     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /content/SRR292678_1.fastq\n",
            "  0:10:10.165     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /content/SRR292678_2.fastq\n",
            "  0:14:12.877     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 32630388 reads processed\n",
            "  0:14:12.949     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
            "  0:15:34.821     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 172000368 kmers in total.\n",
            "  0:15:34.822     8M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
            "  0:15:39.921     8M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
            "  0:16:40.580    84M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
            "  0:16:45.554    84M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 79767094 bytes occupied (3.71009 bits per kmer).\n",
            "  0:16:45.555    84M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
            "  0:17:27.026     2G / 2G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
            "  1:07:23.394     2G / 2G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
            "  1:10:46.529     2G / 8G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 112913170\n",
            "  1:10:46.634     1G / 8G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
            "  1:10:53.021     5G / 8G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /content/SRR292770_1.fastq\n",
            "  1:19:52.417     5G / 8G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /content/SRR292770_2.fastq\n",
            "  1:28:47.778     5G / 8G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /content/SRR292862_1.fastq\n",
            "  1:37:48.343     5G / 8G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /content/SRR292862_2.fastq\n",
            "  1:46:49.766     5G / 8G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /content/SRR292678_1.fastq\n",
            "  2:13:12.421     5G / 8G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /content/SRR292678_2.fastq\n",
            "  2:38:25.024     5G / 8G    INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
            "  2:38:25.993     5G / 8G    INFO   K-mer Counting           (kmer_data.cpp             : 402)   There are 172000368 kmers in total. Among them 150225724 (87.3404%) are singletons.\n",
            "  2:38:25.993     5G / 8G    INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
            "  2:40:54.537     5G / 8G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 192 non-read kmers were generated.\n",
            "  2:40:54.538     5G / 8G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
            "  2:40:54.538     5G / 8G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 100688108. Among them 58768832 (58.3672%) are good\n",
            "  2:40:54.538     5G / 8G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 21708. Among them 21426 (98.7009%) are good\n",
            "  2:40:54.538     5G / 8G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 12483111. Among them 11379382 (91.1582%) are good\n",
            "  2:40:54.538     5G / 8G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 5.71271 kmers\n",
            "  2:40:54.538     5G / 8G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.02288\n",
            "  2:40:54.538     5G / 8G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 70169640\n",
            "  2:40:54.538     5G / 8G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.935262,0.0322471,0.0205622,0.0119286),(0.0180181,0.955737,0.011651,0.0145935),(0.0136698,0.0107818,0.956539,0.019009),(0.0109792,0.019632,0.0313045,0.938084))\n",
            "  2:40:54.653     5G / 8G    INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
            "  2:40:54.654     5G / 8G    INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 2 threads.\n",
            "  3:15:14.384     5G / 8G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 11952901 new k-mers.\n",
            "  3:47:55.976     5G / 8G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 185270 new k-mers.\n",
            "  4:23:33.899     5G / 8G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 838 new k-mers.\n"
          ]
        }
      ],
      "source": [
        "!spades.py --pe1-1 SRR292678_1.fastq --pe1-2 SRR292678_2.fastq \\\n",
        "    --mp1-1 SRR292770_1.fastq --mp1-2 SRR292770_2.fastq \\\n",
        "    --mp2-1 SRR292862_1.fastq --mp2-2 SRR292862_2.fastq \\\n",
        "    -o spades_output"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!unzip three_libs_spades_out.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "h3vGkvejYu2H",
        "outputId": "144c4dbc-56ed-46e5-87c2-39a2f7340a68"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  three_libs_spades_out.zip\n",
            "   creating: three_libs_spades_out/\n",
            "  inflating: three_libs_spades_out/PROKKA_03032023.zip  \n",
            "  inflating: three_libs_spades_out/contigs.fasta.fai  \n",
            "  inflating: three_libs_spades_out/contigs.fasta  \n",
            "  inflating: three_libs_spades_out/scaffolds.fasta  \n",
            "  inflating: three_libs_spades_out/spades.log  \n",
            "   creating: three_libs_spades_out/PROKKA_03032023/\n",
            "  inflating: three_libs_spades_out/PROKKA_03032023/PROKKA_03032023.fna.fai  \n",
            "  inflating: three_libs_spades_out/PROKKA_03032023/PROKKA_03032023.log  \n",
            "  inflating: three_libs_spades_out/PROKKA_03032023/PROKKA_03032023.fna  \n",
            "  inflating: three_libs_spades_out/PROKKA_03032023/PROKKA_03032023.gff  \n",
            "  inflating: three_libs_spades_out/PROKKA_03032023/PROKKA_03032023.faa  \n",
            "  inflating: three_libs_spades_out/PROKKA_03032023/PROKKA_03032023.ffn  \n",
            "  inflating: three_libs_spades_out/PROKKA_03032023/PROKKA_03032023.tbl  \n",
            "  inflating: three_libs_spades_out/PROKKA_03032023/PROKKA_03032023.fsa  \n",
            "  inflating: three_libs_spades_out/PROKKA_03032023/PROKKA_03032023.tsv  \n",
            "  inflating: three_libs_spades_out/PROKKA_03032023/PROKKA_03032023.txt  \n",
            "  inflating: three_libs_spades_out/PROKKA_03032023/PROKKA_03032023.gbk  \n",
            "  inflating: three_libs_spades_out/PROKKA_03032023/PROKKA_03032023.sqn  \n",
            "  inflating: three_libs_spades_out/PROKKA_03032023/PROKKA_03032023.err  \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "G-AC5tVWtGvb",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "501fe9b6-f5e1-4030-a92f-69cc9f6d1ef0"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/usr/local/bin/quast.py three_libs_spades_out/contigs.fasta\n",
            "\n",
            "Version: 5.2.0\n",
            "\n",
            "System information:\n",
            "  OS: Linux-5.10.147+-x86_64-with-glibc2.29 (linux_64)\n",
            "  Python version: 3.8.10\n",
            "  CPUs number: 2\n",
            "\n",
            "Started: 2023-03-04 06:54:21\n",
            "\n",
            "Logging to /content/quast_results/results_2023_03_04_06_54_21/quast.log\n",
            "NOTICE: Maximum number of threads is set to 1 (use --threads option to set it manually)\n",
            "\n",
            "CWD: /content\n",
            "Main parameters: \n",
            "  MODE: default, threads: 1, min contig length: 500, min alignment length: 65, min alignment IDY: 95.0, \\\n",
            "  ambiguity: one, min local misassembly length: 200, min extensive misassembly length: 1000\n",
            "\n",
            "Contigs:\n",
            "  Pre-processing...\n",
            "  three_libs_spades_out/contigs.fasta ==> contigs\n",
            "\n",
            "2023-03-04 06:54:22\n",
            "Running Basic statistics processor...\n",
            "  Contig files: \n",
            "    contigs\n",
            "  Calculating N50 and L50...\n",
            "    contigs, N50 = 335515, L50 = 6, auN = 319603.4, Total length = 5350156, GC % = 50.59, # N's per 100 kbp =  0.00\n",
            "  Drawing Nx plot...\n",
            "    saved to /content/quast_results/results_2023_03_04_06_54_21/basic_stats/Nx_plot.pdf\n",
            "  Drawing cumulative plot...\n",
            "    saved to /content/quast_results/results_2023_03_04_06_54_21/basic_stats/cumulative_plot.pdf\n",
            "  Drawing GC content plot...\n",
            "    saved to /content/quast_results/results_2023_03_04_06_54_21/basic_stats/GC_content_plot.pdf\n",
            "  Drawing contigs GC content plot...\n",
            "    saved to /content/quast_results/results_2023_03_04_06_54_21/basic_stats/contigs_GC_content_plot.pdf\n",
            "  Drawing Coverage histogram (bin size: 3x)...\n",
            "    saved to /content/quast_results/results_2023_03_04_06_54_21/basic_stats/coverage_histogram.pdf\n",
            "  Drawing contigs coverage histogram (bin size: 3x)...\n",
            "    saved to /content/quast_results/results_2023_03_04_06_54_21/basic_stats/contigs_coverage_histogram.pdf\n",
            "Done.\n",
            "\n",
            "NOTICE: Genes are not predicted by default. Use --gene-finding or --glimmer option to enable it.\n",
            "\n",
            "2023-03-04 06:54:24\n",
            "Creating large visual summaries...\n",
            "This may take a while: press Ctrl-C to skip this step..\n",
            "  1 of 2: Creating PDF with all tables and plots...\n",
            "  2 of 2: Creating Icarus viewers...\n",
            "Done\n",
            "\n",
            "2023-03-04 06:54:25\n",
            "RESULTS:\n",
            "  Text versions of total report are saved to /content/quast_results/results_2023_03_04_06_54_21/report.txt, report.tsv, and report.tex\n",
            "  Text versions of transposed total report are saved to /content/quast_results/results_2023_03_04_06_54_21/transposed_report.txt, transposed_report.tsv, and transposed_report.tex\n",
            "  HTML version (interactive tables and plots) is saved to /content/quast_results/results_2023_03_04_06_54_21/report.html\n",
            "  PDF version (tables and plots) is saved to /content/quast_results/results_2023_03_04_06_54_21/report.pdf\n",
            "  Icarus (contig browser) is saved to /content/quast_results/results_2023_03_04_06_54_21/icarus.html\n",
            "  Log is saved to /content/quast_results/results_2023_03_04_06_54_21/quast.log\n",
            "\n",
            "Finished: 2023-03-04 06:54:25\n",
            "Elapsed time: 0:00:04.123320\n",
            "NOTICEs: 2; WARNINGs: 0; non-fatal ERRORs: 0\n",
            "\n",
            "Thank you for using QUAST!\n"
          ]
        }
      ],
      "source": [
        "!quast.py three_libs_spades_out/contigs.fasta"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "R2AaRaiQ52pj",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "953c6df4-66cd-480b-ba7b-59879596a44d"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "All statistics are based on contigs of size >= 500 bp, unless otherwise noted (e.g., \"# contigs (>= 0 bp)\" and \"Total length (>= 0 bp)\" include all contigs).\n",
            "\n",
            "Assembly                    contigs \n",
            "# contigs (>= 0 bp)         369     \n",
            "# contigs (>= 1000 bp)      79      \n",
            "# contigs (>= 5000 bp)      33      \n",
            "# contigs (>= 10000 bp)     30      \n",
            "# contigs (>= 25000 bp)     26      \n",
            "# contigs (>= 50000 bp)     22      \n",
            "Total length (>= 0 bp)      5403327 \n",
            "Total length (>= 1000 bp)   5331230 \n",
            "Total length (>= 5000 bp)   5202939 \n",
            "Total length (>= 10000 bp)  5183802 \n",
            "Total length (>= 25000 bp)  5133691 \n",
            "Total length (>= 50000 bp)  4975501 \n",
            "# contigs                   105     \n",
            "Largest contig              698474  \n",
            "Total length                5350156 \n",
            "GC (%)                      50.59   \n",
            "N50                         335515  \n",
            "N90                         79998   \n",
            "auN                         319603.4\n",
            "L50                         6       \n",
            "L90                         20      \n",
            "# N's per 100 kbp           0.00    \n"
          ]
        }
      ],
      "source": [
        "!cat quast_results/latest/report.txt"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "baQ4j_uutLc4"
      },
      "source": [
        "# 5. Genome Annotation"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ot13p0ZatM93",
        "outputId": "b7699fe3-545e-4095-ebef-38303dfb1dc9"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[1;30;43mВыходные данные были обрезаны до нескольких последних строк (5000).\u001b[0m\n",
            "t/00_diag.t ............................. ok\n",
            "t/alien_build_plugin_download_gitlab.t .. ok\n",
            "All tests successful.\n",
            "Files=2, Tests=2,  0 wallclock secs ( 0.03 usr  0.01 sys +  0.37 cusr  0.04 csys =  0.45 CPU)\n",
            "Result: PASS\n",
            "  PLICEASE/Alien-Build-Plugin-Download-GitLab-0.01.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for PLICEASE/Alien-Build-Plugin-Download-GitLab-0.01.tar.gz\n",
            "Manifying 1 pod document\n",
            "Installing /usr/local/share/perl/5.30.0/Alien/Build/Plugin/Download/GitLab.pm\n",
            "Installing /usr/local/man/man3/Alien::Build::Plugin::Download::GitLab.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  PLICEASE/Alien-Build-Plugin-Download-GitLab-0.01.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "  PLICEASE/Alien-Libxml2-0.19.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/Alien-Libxml2-0.19-0\n",
            "Configuring P/PL/PLICEASE/Alien-Libxml2-0.19.tar.gz with Makefile.PL\n",
            "Alien::Build::Plugin::PkgConfig::Negotiate> Using PkgConfig plugin: PkgConfig::CommandLine\n",
            "Alien::Build::CommandSequence> + xml2-config --version\n",
            "Alien::Build::CommandSequence> [output consumed by Alien::Build recipe]\n",
            "Alien::Build::CommandSequence> + xml2-config --cflags\n",
            "Alien::Build::CommandSequence> [output consumed by Alien::Build recipe]\n",
            "Alien::Build::CommandSequence> + xml2-config --libs\n",
            "Alien::Build::CommandSequence> [output consumed by Alien::Build recipe]\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for Alien::Libxml2\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  PLICEASE/Alien-Libxml2-0.19.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for P/PL/PLICEASE/Alien-Libxml2-0.19.tar.gz\n",
            "cp lib/Alien/Libxml2.pm blib/lib/Alien/Libxml2.pm\n",
            "\"/usr/bin/perl\" -MAlien::Build::MM=cmd -e prefix site /usr/lib/x86_64-linux-gnu/perl/5.30 /usr/local/lib/x86_64-linux-gnu/perl/5.30.0 /usr/lib/x86_64-linux-gnu/perl5/5.30\n",
            "Alien::Build::Plugin::PkgConfig::Negotiate> Using PkgConfig plugin: PkgConfig::CommandLine\n",
            "main> prefix /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/share/dist/Alien-Libxml2\n",
            "\"/usr/bin/perl\" -MAlien::Build::MM=cmd -e version 0.19\n",
            "Alien::Build::Plugin::PkgConfig::Negotiate> Using PkgConfig plugin: PkgConfig::CommandLine\n",
            "\"/usr/bin/perl\" -MAlien::Build::MM=cmd -e download\n",
            "Alien::Build::Plugin::PkgConfig::Negotiate> Using PkgConfig plugin: PkgConfig::CommandLine\n",
            "\"/usr/bin/perl\" -MAlien::Build::MM=cmd -e build\n",
            "Alien::Build::Plugin::PkgConfig::Negotiate> Using PkgConfig plugin: PkgConfig::CommandLine\n",
            "Alien::Build::Plugin::Core::Legacy> adding legacy hash to config\n",
            "Alien::Build::Plugin::Core::Gather> mkdir -p /root/.cpan/build/Alien-Libxml2-0.19-0/blib/lib/auto/share/dist/Alien-Libxml2/_alien\n",
            "Manifying 1 pod document\n",
            "  PLICEASE/Alien-Libxml2-0.19.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for PLICEASE/Alien-Libxml2-0.19.tar.gz\n",
            "\"/usr/bin/perl\" -MAlien::Build::MM=cmd -e test\n",
            "Alien::Build::Plugin::PkgConfig::Negotiate> Using PkgConfig plugin: PkgConfig::CommandLine\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/00_diag.t ........ 1/? # \n",
            "# \n",
            "# \n",
            "# HARNESS_ACTIVE=1\n",
            "# HARNESS_VERSION=3.42\n",
            "# LANG=en_US.UTF-8\n",
            "# PERL5LIB=/root/.cpan/build/Alien-Libxml2-0.19-0/blib/lib:/root/.cpan/build/Alien-Libxml2-0.19-0/blib/arch:\n",
            "# PERL5OPT=\n",
            "# PERL5_CPANPLUS_IS_RUNNING=3407\n",
            "# PERL5_CPAN_IS_RUNNING=3407\n",
            "# PERL_DL_NONLAZY=1\n",
            "# PERL_MM_USE_DEFAULT=1\n",
            "# PERL_USE_UNSAFE_INC=1\n",
            "# SHELL=/bin/bash\n",
            "# \n",
            "# \n",
            "# \n",
            "# PERL5LIB path\n",
            "# /root/.cpan/build/Alien-Libxml2-0.19-0/blib/lib\n",
            "# /root/.cpan/build/Alien-Libxml2-0.19-0/blib/arch\n",
            "# \n",
            "# \n",
            "# \n",
            "# perl                                     5.030000 linux x86_64-linux-gnu-thread-multi\n",
            "# Alien::Base                              2.77\n",
            "# Alien::Build                             2.77\n",
            "# Alien::Build::MM                         2.77\n",
            "# Alien::Build::Plugin::Build::SearchDep   2.77\n",
            "# Alien::Build::Plugin::Download::GitLab   0.01\n",
            "# Alien::Build::Plugin::Prefer::BadVersion 2.77\n",
            "# Alien::Build::Plugin::Probe::Vcpkg       2.77\n",
            "# ExtUtils::CBuilder                       0.280231\n",
            "# ExtUtils::MakeMaker                      7.34\n",
            "# Test2::V0                                0.000145\n",
            "# Test::Alien                              2.77\n",
            "# \n",
            "# \n",
            "# \n",
            "# version        = 2.9.10\n",
            "# cflags         = -I/usr/include/libxml2\n",
            "# cflags_static  = -I/usr/include/libxml2\n",
            "# libs           = -lxml2\n",
            "# libs_static    = -lxml2\n",
            "# \n",
            "# \n",
            "# \n",
            "t/00_diag.t ........ ok\n",
            "t/alien_libxml2.t .. ok\n",
            "All tests successful.\n",
            "Files=2, Tests=4,  1 wallclock secs ( 0.03 usr  0.01 sys +  1.31 cusr  0.23 csys =  1.58 CPU)\n",
            "Result: PASS\n",
            "  PLICEASE/Alien-Libxml2-0.19.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for PLICEASE/Alien-Libxml2-0.19.tar.gz\n",
            "\"/usr/bin/perl\" -MAlien::Build::MM=cmd -e clean_install\n",
            "Alien::Build::Plugin::PkgConfig::Negotiate> Using PkgConfig plugin: PkgConfig::CommandLine\n",
            "Manifying 1 pod document\n",
            "Files found in blib/arch: installing files in blib/lib into architecture dependent library tree\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Alien/Libxml2/Libxml2.txt\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/share/dist/Alien-Libxml2/_alien/alienfile\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/share/dist/Alien-Libxml2/_alien/alien.json\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/Alien/Libxml2.pm\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/Alien/Libxml2/Install/Files.pm\n",
            "Installing /usr/local/man/man3/Alien::Libxml2.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  PLICEASE/Alien-Libxml2-0.19.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "  SHLOMIF/XML-LibXML-2.0208.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/XML-LibXML-2.0208-0\n",
            "Configuring S/SH/SHLOMIF/XML-LibXML-2.0208.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Warning: prerequisite XML::NamespaceSupport 1.07 not found.\n",
            "Warning: prerequisite XML::SAX 0.11 not found.\n",
            "Warning: prerequisite XML::SAX::Base 0 not found.\n",
            "Warning: prerequisite XML::SAX::DocumentLocator 0 not found.\n",
            "Warning: prerequisite XML::SAX::Exception 0 not found.\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for XML::LibXML\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  SHLOMIF/XML-LibXML-2.0208.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for S/SH/SHLOMIF/XML-LibXML-2.0208.tar.gz\n",
            "---- Unsatisfied dependencies detected during ----\n",
            "----     SHLOMIF/XML-LibXML-2.0208.tar.gz     ----\n",
            "    XML::NamespaceSupport [requires]\n",
            "    XML::SAX [requires]\n",
            "    XML::SAX::Base [requires]\n",
            "    XML::SAX::DocumentLocator [requires]\n",
            "    XML::SAX::Exception [requires]\n",
            "Running install for module 'XML::NamespaceSupport'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/P/PE/PERIGRIN/XML-NamespaceSupport-1.12.tar.gz\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/P/PE/PERIGRIN/CHECKSUMS\n",
            "Checksum for /root/.cpan/sources/authors/id/P/PE/PERIGRIN/XML-NamespaceSupport-1.12.tar.gz ok\n",
            "Configuring P/PE/PERIGRIN/XML-NamespaceSupport-1.12.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for XML::NamespaceSupport\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  PERIGRIN/XML-NamespaceSupport-1.12.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for P/PE/PERIGRIN/XML-NamespaceSupport-1.12.tar.gz\n",
            "cp lib/XML/NamespaceSupport.pm blib/lib/XML/NamespaceSupport.pm\n",
            "Manifying 1 pod document\n",
            "  PERIGRIN/XML-NamespaceSupport-1.12.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for PERIGRIN/XML-NamespaceSupport-1.12.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/00base.t .. ok\n",
            "All tests successful.\n",
            "Files=1, Tests=49,  0 wallclock secs ( 0.02 usr  0.01 sys +  0.06 cusr  0.01 csys =  0.10 CPU)\n",
            "Result: PASS\n",
            "  PERIGRIN/XML-NamespaceSupport-1.12.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for PERIGRIN/XML-NamespaceSupport-1.12.tar.gz\n",
            "Manifying 1 pod document\n",
            "Installing /usr/local/share/perl/5.30.0/XML/NamespaceSupport.pm\n",
            "Installing /usr/local/man/man3/XML::NamespaceSupport.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  PERIGRIN/XML-NamespaceSupport-1.12.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "Running install for module 'XML::SAX'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/G/GR/GRANTM/XML-SAX-1.02.tar.gz\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/G/GR/GRANTM/CHECKSUMS\n",
            "Checksum for /root/.cpan/sources/authors/id/G/GR/GRANTM/XML-SAX-1.02.tar.gz ok\n",
            "Configuring G/GR/GRANTM/XML-SAX-1.02.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Warning: prerequisite XML::SAX::Base 1.05 not found.\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for XML::SAX\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  GRANTM/XML-SAX-1.02.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for G/GR/GRANTM/XML-SAX-1.02.tar.gz\n",
            "---- Unsatisfied dependencies detected during ----\n",
            "----        GRANTM/XML-SAX-1.02.tar.gz        ----\n",
            "    XML::SAX::Base [requires]\n",
            "Running install for module 'XML::SAX::Base'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/G/GR/GRANTM/XML-SAX-Base-1.09.tar.gz\n",
            "Checksum for /root/.cpan/sources/authors/id/G/GR/GRANTM/XML-SAX-Base-1.09.tar.gz ok\n",
            "Configuring G/GR/GRANTM/XML-SAX-Base-1.09.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for XML::SAX::Base\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  GRANTM/XML-SAX-Base-1.09.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for G/GR/GRANTM/XML-SAX-Base-1.09.tar.gz\n",
            "cp BuildSAXBase.pl blib/lib/XML/SAX/BuildSAXBase.pl\n",
            "cp lib/XML/SAX/Base.pm blib/lib/XML/SAX/Base.pm\n",
            "cp lib/XML/SAX/Exception.pm blib/lib/XML/SAX/Exception.pm\n",
            "Manifying 3 pod documents\n",
            "  GRANTM/XML-SAX-Base-1.09.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for GRANTM/XML-SAX-Base-1.09.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/00basic.t ............. ok\n",
            "t/01exception.t ......... ok\n",
            "t/01simpledriver.t ...... ok\n",
            "t/02simplefilter.t ...... ok\n",
            "t/03chdriver.t .......... ok\n",
            "t/04chfilter.t .......... ok\n",
            "t/05dtdhdriver.t ........ ok\n",
            "t/06lexhdriver.t ........ ok\n",
            "t/07declhdriver.t ....... ok\n",
            "t/08errorhdriver.t ...... ok\n",
            "t/09resoldriver.t ....... ok\n",
            "t/10dochdriver.t ........ ok\n",
            "t/11sax1multiclass.t .... ok\n",
            "t/12sax2multiclass.t .... ok\n",
            "t/13handlerswitch.t ..... ok\n",
            "t/14downstreamswitch.t .. ok\n",
            "t/15parentswitch.t ...... ok\n",
            "t/16gethandlers.t ....... ok\n",
            "t/release-pod-syntax.t .. skipped: these tests are for release candidate testing\n",
            "All tests successful.\n",
            "Files=19, Tests=137,  0 wallclock secs ( 0.07 usr  0.04 sys +  0.50 cusr  0.11 csys =  0.72 CPU)\n",
            "Result: PASS\n",
            "  GRANTM/XML-SAX-Base-1.09.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for GRANTM/XML-SAX-Base-1.09.tar.gz\n",
            "Manifying 3 pod documents\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/BuildSAXBase.pl\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/Base.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/Exception.pm\n",
            "Installing /usr/local/man/man3/XML::SAX::Base.3pm\n",
            "Installing /usr/local/man/man3/XML::SAX::BuildSAXBase.3pm\n",
            "Installing /usr/local/man/man3/XML::SAX::Exception.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  GRANTM/XML-SAX-Base-1.09.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "  GRANTM/XML-SAX-1.02.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/XML-SAX-1.02-0\n",
            "  GRANTM/XML-SAX-1.02.tar.gz\n",
            "  Has already been prepared\n",
            "Running make for G/GR/GRANTM/XML-SAX-1.02.tar.gz\n",
            "cp lib/XML/SAX/PurePerl/Productions.pm blib/lib/XML/SAX/PurePerl/Productions.pm\n",
            "cp lib/XML/SAX/PurePerl/EncodingDetect.pm blib/lib/XML/SAX/PurePerl/EncodingDetect.pm\n",
            "cp lib/XML/SAX/PurePerl/DebugHandler.pm blib/lib/XML/SAX/PurePerl/DebugHandler.pm\n",
            "cp lib/XML/SAX.pm blib/lib/XML/SAX.pm\n",
            "cp lib/XML/SAX/PurePerl/NoUnicodeExt.pm blib/lib/XML/SAX/PurePerl/NoUnicodeExt.pm\n",
            "cp lib/XML/SAX/PurePerl/DocType.pm blib/lib/XML/SAX/PurePerl/DocType.pm\n",
            "cp lib/XML/SAX/PurePerl/Reader.pm blib/lib/XML/SAX/PurePerl/Reader.pm\n",
            "cp lib/XML/SAX/PurePerl/XMLDecl.pm blib/lib/XML/SAX/PurePerl/XMLDecl.pm\n",
            "cp lib/XML/SAX/PurePerl/DTDDecls.pm blib/lib/XML/SAX/PurePerl/DTDDecls.pm\n",
            "cp lib/XML/SAX/PurePerl/UnicodeExt.pm blib/lib/XML/SAX/PurePerl/UnicodeExt.pm\n",
            "cp lib/XML/SAX/PurePerl/Reader/URI.pm blib/lib/XML/SAX/PurePerl/Reader/URI.pm\n",
            "cp lib/XML/SAX/PurePerl/Reader/UnicodeExt.pm blib/lib/XML/SAX/PurePerl/Reader/UnicodeExt.pm\n",
            "cp lib/XML/SAX/PurePerl/Reader/String.pm blib/lib/XML/SAX/PurePerl/Reader/String.pm\n",
            "cp lib/XML/SAX/PurePerl.pm blib/lib/XML/SAX/PurePerl.pm\n",
            "cp lib/XML/SAX/DocumentLocator.pm blib/lib/XML/SAX/DocumentLocator.pm\n",
            "cp lib/XML/SAX/PurePerl/Exception.pm blib/lib/XML/SAX/PurePerl/Exception.pm\n",
            "cp lib/XML/SAX/PurePerl/Reader/NoUnicodeExt.pm blib/lib/XML/SAX/PurePerl/Reader/NoUnicodeExt.pm\n",
            "cp lib/XML/SAX/Intro.pod blib/lib/XML/SAX/Intro.pod\n",
            "cp lib/XML/SAX/PurePerl/Reader/Stream.pm blib/lib/XML/SAX/PurePerl/Reader/Stream.pm\n",
            "cp lib/XML/SAX/ParserFactory.pm blib/lib/XML/SAX/ParserFactory.pm\n",
            "Manifying 6 pod documents\n",
            "  GRANTM/XML-SAX-1.02.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for GRANTM/XML-SAX-1.02.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/00basic.t ....... ok\n",
            "t/01known.t ....... ok\n",
            "t/10xmldecl1.t .... ok\n",
            "t/11xmldecl2.t .... ok\n",
            "t/12miscstart.t ... ok\n",
            "t/13int_ent.t ..... ok\n",
            "t/14encoding.t .... ok\n",
            "t/15element.t ..... ok\n",
            "t/16large.t ....... 1/3 parsed 80085 bytes in 0 seconds\n",
            "t/16large.t ....... ok\n",
            "t/19pi.t .......... ok\n",
            "t/20factory.t ..... ok\n",
            "t/21saxini.t ...... ok\n",
            "t/30parse_file.t .. ok\n",
            "t/40cdata.t ....... ok\n",
            "t/42entities.t .... ok\n",
            "t/99cleanup.t ..... ok\n",
            "All tests successful.\n",
            "Files=16, Tests=113,  2 wallclock secs ( 0.08 usr  0.04 sys +  1.51 cusr  0.32 csys =  1.95 CPU)\n",
            "Result: PASS\n",
            "  GRANTM/XML-SAX-1.02.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for GRANTM/XML-SAX-1.02.tar.gz\n",
            "Manifying 6 pod documents\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/Intro.pod\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/PurePerl.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/DocumentLocator.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/ParserFactory.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/PurePerl/Reader.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/PurePerl/EncodingDetect.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/PurePerl/UnicodeExt.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/PurePerl/Productions.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/PurePerl/DTDDecls.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/PurePerl/XMLDecl.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/PurePerl/Exception.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/PurePerl/DebugHandler.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/PurePerl/NoUnicodeExt.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/PurePerl/DocType.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/PurePerl/Reader/Stream.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/PurePerl/Reader/UnicodeExt.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/PurePerl/Reader/String.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/PurePerl/Reader/URI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/PurePerl/Reader/NoUnicodeExt.pm\n",
            "Installing /usr/local/man/man3/XML::SAX::PurePerl::Reader.3pm\n",
            "Installing /usr/local/man/man3/XML::SAX.3pm\n",
            "Installing /usr/local/man/man3/XML::SAX::DocumentLocator.3pm\n",
            "Installing /usr/local/man/man3/XML::SAX::PurePerl.3pm\n",
            "Installing /usr/local/man/man3/XML::SAX::ParserFactory.3pm\n",
            "Installing /usr/local/man/man3/XML::SAX::Intro.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "could not find ParserDetails.ini in /root/.cpan/build/XML-SAX-1.02-0/blib/lib/XML/SAX\n",
            "  GRANTM/XML-SAX-1.02.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "XML::SAX::DocumentLocator is up to date (undef).\n",
            "XML::SAX::Exception is up to date (1.09).\n",
            "  SHLOMIF/XML-LibXML-2.0208.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/XML-LibXML-2.0208-0\n",
            "  SHLOMIF/XML-LibXML-2.0208.tar.gz\n",
            "  Has already been prepared\n",
            "Running make for S/SH/SHLOMIF/XML-LibXML-2.0208.tar.gz\n",
            "cp lib/XML/LibXML/RegExp.pod blib/lib/XML/LibXML/RegExp.pod\n",
            "cp lib/XML/LibXML/SAX/Builder.pm blib/lib/XML/LibXML/SAX/Builder.pm\n",
            "cp lib/XML/LibXML/XPathContext.pm blib/lib/XML/LibXML/XPathContext.pm\n",
            "cp LibXML.pod blib/lib/XML/LibXML.pod\n",
            "cp lib/XML/LibXML/PI.pod blib/lib/XML/LibXML/PI.pod\n",
            "cp lib/XML/LibXML/Element.pod blib/lib/XML/LibXML/Element.pod\n",
            "cp lib/XML/LibXML/Reader.pod blib/lib/XML/LibXML/Reader.pod\n",
            "cp lib/XML/LibXML/Parser.pod blib/lib/XML/LibXML/Parser.pod\n",
            "cp lib/XML/LibXML/SAX.pod blib/lib/XML/LibXML/SAX.pod\n",
            "cp lib/XML/LibXML/Boolean.pm blib/lib/XML/LibXML/Boolean.pm\n",
            "cp lib/XML/LibXML/SAX.pm blib/lib/XML/LibXML/SAX.pm\n",
            "cp lib/XML/LibXML/ErrNo.pod blib/lib/XML/LibXML/ErrNo.pod\n",
            "cp lib/XML/LibXML/Number.pm blib/lib/XML/LibXML/Number.pm\n",
            "cp lib/XML/LibXML/RelaxNG.pod blib/lib/XML/LibXML/RelaxNG.pod\n",
            "cp lib/XML/LibXML/ErrNo.pm blib/lib/XML/LibXML/ErrNo.pm\n",
            "cp lib/XML/LibXML/Node.pod blib/lib/XML/LibXML/Node.pod\n",
            "cp lib/XML/LibXML/Text.pod blib/lib/XML/LibXML/Text.pod\n",
            "cp lib/XML/LibXML/NodeList.pm blib/lib/XML/LibXML/NodeList.pm\n",
            "cp lib/XML/LibXML/InputCallback.pod blib/lib/XML/LibXML/InputCallback.pod\n",
            "cp lib/XML/LibXML/SAX/Parser.pm blib/lib/XML/LibXML/SAX/Parser.pm\n",
            "cp lib/XML/LibXML/AttributeHash.pm blib/lib/XML/LibXML/AttributeHash.pm\n",
            "cp lib/XML/LibXML/Document.pod blib/lib/XML/LibXML/Document.pod\n",
            "cp lib/XML/LibXML/SAX/Generator.pm blib/lib/XML/LibXML/SAX/Generator.pm\n",
            "cp LibXML.pm blib/lib/XML/LibXML.pm\n",
            "cp lib/XML/LibXML/Pattern.pod blib/lib/XML/LibXML/Pattern.pod\n",
            "cp lib/XML/LibXML/Namespace.pod blib/lib/XML/LibXML/Namespace.pod\n",
            "cp lib/XML/LibXML/SAX/Builder.pod blib/lib/XML/LibXML/SAX/Builder.pod\n",
            "cp lib/XML/LibXML/Dtd.pod blib/lib/XML/LibXML/Dtd.pod\n",
            "cp lib/XML/LibXML/Comment.pod blib/lib/XML/LibXML/Comment.pod\n",
            "cp lib/XML/LibXML/Schema.pod blib/lib/XML/LibXML/Schema.pod\n",
            "cp lib/XML/LibXML/Reader.pm blib/lib/XML/LibXML/Reader.pm\n",
            "cp lib/XML/LibXML/Common.pod blib/lib/XML/LibXML/Common.pod\n",
            "cp lib/XML/LibXML/DOM.pod blib/lib/XML/LibXML/DOM.pod\n",
            "cp lib/XML/LibXML/DocumentFragment.pod blib/lib/XML/LibXML/DocumentFragment.pod\n",
            "cp lib/XML/LibXML/Error.pod blib/lib/XML/LibXML/Error.pod\n",
            "cp lib/XML/LibXML/Devel.pm blib/lib/XML/LibXML/Devel.pm\n",
            "cp lib/XML/LibXML/CDATASection.pod blib/lib/XML/LibXML/CDATASection.pod\n",
            "cp lib/XML/LibXML/Attr.pod blib/lib/XML/LibXML/Attr.pod\n",
            "cp lib/XML/LibXML/Literal.pm blib/lib/XML/LibXML/Literal.pm\n",
            "cp lib/XML/LibXML/Error.pm blib/lib/XML/LibXML/Error.pm\n",
            "cp lib/XML/LibXML/Common.pm blib/lib/XML/LibXML/Common.pm\n",
            "cp lib/XML/LibXML/XPathContext.pod blib/lib/XML/LibXML/XPathContext.pod\n",
            "cp lib/XML/LibXML/XPathExpression.pod blib/lib/XML/LibXML/XPathExpression.pod\n",
            "Running Mkbootstrap for LibXML ()\n",
            "chmod 644 \"LibXML.bs\"\n",
            "\"/usr/bin/perl\" -MExtUtils::Command::MM -e 'cp_nonempty' -- LibXML.bs blib/arch/auto/XML/LibXML/LibXML.bs 644\n",
            "x86_64-linux-gnu-gcc -c  -I/usr/include/libxml2 -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\\\"2.0208\\\" -DXS_VERSION=\\\"2.0208\\\" -fPIC \"-I/usr/lib/x86_64-linux-gnu/perl/5.30/CORE\"  -DHAVE_UTF8 Av_CharPtrPtr.c\n",
            "\"/usr/bin/perl\" \"/usr/share/perl/5.30/ExtUtils/xsubpp\"  -typemap '/usr/share/perl/5.30/ExtUtils/typemap' -typemap '/root/.cpan/build/XML-LibXML-2.0208-0/typemap'  Devel.xs > Devel.xsc\n",
            "mv Devel.xsc Devel.c\n",
            "x86_64-linux-gnu-gcc -c  -I/usr/include/libxml2 -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\\\"2.0208\\\" -DXS_VERSION=\\\"2.0208\\\" -fPIC \"-I/usr/lib/x86_64-linux-gnu/perl/5.30/CORE\"  -DHAVE_UTF8 Devel.c\n",
            "\"/usr/bin/perl\" \"/usr/share/perl/5.30/ExtUtils/xsubpp\"  -typemap '/usr/share/perl/5.30/ExtUtils/typemap' -typemap '/root/.cpan/build/XML-LibXML-2.0208-0/typemap'  LibXML.xs > LibXML.xsc\n",
            "mv LibXML.xsc LibXML.c\n",
            "x86_64-linux-gnu-gcc -c  -I/usr/include/libxml2 -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\\\"2.0208\\\" -DXS_VERSION=\\\"2.0208\\\" -fPIC \"-I/usr/lib/x86_64-linux-gnu/perl/5.30/CORE\"  -DHAVE_UTF8 LibXML.c\n",
            "In file included from \u001b[01m\u001b[K/usr/include/string.h:495\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/lib/x86_64-linux-gnu/perl/5.30/CORE/perl.h:1100\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KLibXML.xs:20\u001b[m\u001b[K:\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[KLibXML_input_read\u001b[m\u001b[K’ at \u001b[01m\u001b[KLibXML.xs:718:17\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:106:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound depends on the length of the source argument [\u001b[01;35m\u001b[K-Wstringop-overflow=\u001b[m\u001b[K]\n",
            "  106 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len, __bos (__dest))\u001b[m\u001b[K;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KLibXML.xs:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KLibXML_input_read\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KLibXML.xs:716:23:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Klength computed here\n",
            "  716 |             res_len = \u001b[01;36m\u001b[Kstrlen(output)\u001b[m\u001b[K;\n",
            "      |                       \u001b[01;36m\u001b[K^~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "x86_64-linux-gnu-gcc -c  -I/usr/include/libxml2 -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\\\"2.0208\\\" -DXS_VERSION=\\\"2.0208\\\" -fPIC \"-I/usr/lib/x86_64-linux-gnu/perl/5.30/CORE\"  -DHAVE_UTF8 dom.c\n",
            "x86_64-linux-gnu-gcc -c  -I/usr/include/libxml2 -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\\\"2.0208\\\" -DXS_VERSION=\\\"2.0208\\\" -fPIC \"-I/usr/lib/x86_64-linux-gnu/perl/5.30/CORE\"  -DHAVE_UTF8 perl-libxml-mm.c\n",
            "\u001b[01m\u001b[Kperl-libxml-mm.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KPmmDumpRegistry\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kperl-libxml-mm.c:142:18:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kpassing argument 2 of ‘\u001b[01m\u001b[KxmlHashScan\u001b[m\u001b[K’ from incompatible pointer type [\u001b[01;35m\u001b[K-Wincompatible-pointer-types\u001b[m\u001b[K]\n",
            "  142 |   xmlHashScan(r, \u001b[01;35m\u001b[KPmmRegistryDumpHashScanner\u001b[m\u001b[K, NULL);\n",
            "      |                  \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "      |                  \u001b[01;35m\u001b[K|\u001b[m\u001b[K\n",
            "      |                  \u001b[01;35m\u001b[Kvoid (*)(void *, void *, xmlChar *) {aka void (*)(void *, void *, unsigned char *)}\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/libxml2/libxml/parser.h:18\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kperl-libxml-mm.h:20\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kperl-libxml-mm.c:26\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/libxml2/libxml/hash.h:213:22:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kexpected ‘\u001b[01m\u001b[KxmlHashScanner\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Kvoid (*)(void *, void *, const unsigned char *)\u001b[m\u001b[K’} but argument is of type ‘\u001b[01m\u001b[Kvoid (*)(void *, void *, xmlChar *)\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Kvoid (*)(void *, void *, unsigned char *)\u001b[m\u001b[K’}\n",
            "  213 |       \u001b[01;36m\u001b[KxmlHashScanner f\u001b[m\u001b[K,\n",
            "      |       \u001b[01;36m\u001b[K~~~~~~~~~~~~~~~^\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kperl-libxml-mm.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KPmmUnregisterProxyNode\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kperl-libxml-mm.c:234:44:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kpassing argument 3 of ‘\u001b[01m\u001b[KxmlHashRemoveEntry\u001b[m\u001b[K’ from incompatible pointer type [\u001b[01;35m\u001b[K-Wincompatible-pointer-types\u001b[m\u001b[K]\n",
            "  234 |  if( xmlHashRemoveEntry(PmmREGISTRY, name, \u001b[01;35m\u001b[KPmmRegistryHashDeallocator\u001b[m\u001b[K) )\n",
            "      |                                            \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "      |                                            \u001b[01;35m\u001b[K|\u001b[m\u001b[K\n",
            "      |                                            \u001b[01;35m\u001b[Kvoid (*)(void *, xmlChar *) {aka void (*)(void *, unsigned char *)}\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/libxml2/libxml/parser.h:18\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kperl-libxml-mm.h:20\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kperl-libxml-mm.c:26\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/libxml2/libxml/hash.h:160:47:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kexpected ‘\u001b[01m\u001b[KxmlHashDeallocator\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Kvoid (*)(void *, const unsigned char *)\u001b[m\u001b[K’} but argument is of type ‘\u001b[01m\u001b[Kvoid (*)(void *, xmlChar *)\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Kvoid (*)(void *, unsigned char *)\u001b[m\u001b[K’}\n",
            "  160 |                            \u001b[01;36m\u001b[KxmlHashDeallocator f\u001b[m\u001b[K);\n",
            "      |                            \u001b[01;36m\u001b[K~~~~~~~~~~~~~~~~~~~^\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kperl-libxml-mm.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KPmmCloneProxyNodes\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kperl-libxml-mm.c:301:38:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kpassing argument 2 of ‘\u001b[01m\u001b[KxmlHashCopy\u001b[m\u001b[K’ from incompatible pointer type [\u001b[01;35m\u001b[K-Wincompatible-pointer-types\u001b[m\u001b[K]\n",
            "  301 |  reg_copy = xmlHashCopy(PmmREGISTRY, \u001b[01;35m\u001b[KPmmRegistryHashCopier\u001b[m\u001b[K);\n",
            "      |                                      \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "      |                                      \u001b[01;35m\u001b[K|\u001b[m\u001b[K\n",
            "      |                                      \u001b[01;35m\u001b[Kvoid * (*)(void *, xmlChar *) {aka void * (*)(void *, unsigned char *)}\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/libxml2/libxml/parser.h:18\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kperl-libxml-mm.h:20\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kperl-libxml-mm.c:26\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/libxml2/libxml/hash.h:208:21:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kexpected ‘\u001b[01m\u001b[KxmlHashCopier\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Kvoid * (*)(void *, const unsigned char *)\u001b[m\u001b[K’} but argument is of type ‘\u001b[01m\u001b[Kvoid * (*)(void *, xmlChar *)\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Kvoid * (*)(void *, unsigned char *)\u001b[m\u001b[K’}\n",
            "  208 |       \u001b[01;36m\u001b[KxmlHashCopier f\u001b[m\u001b[K);\n",
            "      |       \u001b[01;36m\u001b[K~~~~~~~~~~~~~~^\u001b[m\u001b[K\n",
            "x86_64-linux-gnu-gcc -c  -I/usr/include/libxml2 -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\\\"2.0208\\\" -DXS_VERSION=\\\"2.0208\\\" -fPIC \"-I/usr/lib/x86_64-linux-gnu/perl/5.30/CORE\"  -DHAVE_UTF8 perl-libxml-sax.c\n",
            "x86_64-linux-gnu-gcc -c  -I/usr/include/libxml2 -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\\\"2.0208\\\" -DXS_VERSION=\\\"2.0208\\\" -fPIC \"-I/usr/lib/x86_64-linux-gnu/perl/5.30/CORE\"  -DHAVE_UTF8 xpath.c\n",
            "rm -f blib/arch/auto/XML/LibXML/LibXML.so\n",
            "x86_64-linux-gnu-gcc  -shared -L/usr/local/lib -fstack-protector-strong  Av_CharPtrPtr.o Devel.o LibXML.o dom.o perl-libxml-mm.o perl-libxml-sax.o xpath.o  -o blib/arch/auto/XML/LibXML/LibXML.so  \\\n",
            "   -lxml2   \\\n",
            "  \n",
            "chmod 755 blib/arch/auto/XML/LibXML/LibXML.so\n",
            "Manifying 34 pod documents\n",
            "  SHLOMIF/XML-LibXML-2.0208.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for SHLOMIF/XML-LibXML-2.0208.tar.gz\n",
            "\"/usr/bin/perl\" -MExtUtils::Command::MM -e 'cp_nonempty' -- LibXML.bs blib/arch/auto/XML/LibXML/LibXML.bs 644\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/00-report-prereqs.t .............................. # \n",
            "# Versions for all modules listed in MYMETA.json (including optional ones):\n",
            "# \n",
            "# === Configure Requires ===\n",
            "# \n",
            "#     Module               Want     Have\n",
            "#     -------------------- ---- --------\n",
            "#     Alien::Base::Wrapper  any     2.77\n",
            "#     Alien::Libxml2       0.14     0.19\n",
            "#     Config                any 5.030000\n",
            "#     ExtUtils::MakeMaker   any     7.34\n",
            "# \n",
            "# === Build Requires ===\n",
            "# \n",
            "#     Module              Want Have\n",
            "#     ------------------- ---- ----\n",
            "#     ExtUtils::MakeMaker  any 7.34\n",
            "# \n",
            "# === Test Requires ===\n",
            "# \n",
            "#     Module       Want     Have\n",
            "#     ------------ ---- --------\n",
            "#     Config        any 5.030000\n",
            "#     Errno         any     1.30\n",
            "#     IO::File      any     1.40\n",
            "#     IO::Handle    any     1.40\n",
            "#     POSIX         any     1.88\n",
            "#     Scalar::Util  any     1.50\n",
            "#     Test::More    any 1.302192\n",
            "#     locale        any     1.09\n",
            "#     utf8          any     1.22\n",
            "# \n",
            "# === Runtime Requires ===\n",
            "# \n",
            "#     Module                    Want  Have\n",
            "#     ------------------------- ---- -----\n",
            "#     Carp                       any  1.50\n",
            "#     DynaLoader                 any  1.45\n",
            "#     Encode                     any  3.01\n",
            "#     Exporter                  5.57  5.73\n",
            "#     IO::Handle                 any  1.40\n",
            "#     Scalar::Util               any  1.50\n",
            "#     Tie::Hash                  any  1.05\n",
            "#     XML::NamespaceSupport     1.07  1.12\n",
            "#     XML::SAX                  0.11  1.02\n",
            "#     XML::SAX::Base             any  1.09\n",
            "#     XML::SAX::DocumentLocator  any undef\n",
            "#     XML::SAX::Exception        any  1.09\n",
            "#     base                       any  2.27\n",
            "#     constant                   any  1.33\n",
            "#     overload                   any  1.30\n",
            "#     parent                     any 0.237\n",
            "#     strict                     any  1.11\n",
            "#     vars                       any  1.05\n",
            "#     warnings                   any  1.44\n",
            "# \n",
            "t/00-report-prereqs.t .............................. ok\n",
            "t/01basic.t ........................................ 1/3 # \n",
            "# \n",
            "# Compiled against libxml2 version: 20910\n",
            "# Running libxml2 version:          20910\n",
            "# \n",
            "t/01basic.t ........................................ ok\n",
            "t/02parse.t ........................................ ok\n",
            "t/03doc.t .......................................... ok\n",
            "t/04node.t ......................................... ok\n",
            "t/05text.t ......................................... ok\n",
            "t/06elements.t ..................................... ok\n",
            "t/07dtd.t .......................................... ok\n",
            "t/08findnodes.t .................................... ok\n",
            "t/09xpath.t ........................................ ok\n",
            "t/10ns.t ........................................... ok\n",
            "t/11memory.t ....................................... skipped: These tests are for authors only!\n",
            "t/12html.t ......................................... ok\n",
            "t/13dtd.t .......................................... ok\n",
            "t/14sax.t .......................................... ok\n",
            "t/15nodelist.t ..................................... ok\n",
            "t/16docnodes.t ..................................... ok\n",
            "t/17callbacks.t .................................... ok\n",
            "t/18docfree.t ...................................... ok\n",
            "t/19die_on_invalid_utf8_rt_58848.t ................. ok\n",
            "t/19encoding.t ..................................... ok\n",
            "t/20extras.t ....................................... ok\n",
            "t/21catalog.t ...................................... ok\n",
            "t/23rawfunctions.t ................................. ok\n",
            "t/24c14n.t ......................................... ok\n",
            "t/25relaxng.t ...................................... ok\n",
            "t/26schema.t ....................................... ok\n",
            "t/27new_callbacks_simple.t ......................... ok\n",
            "t/28new_callbacks_multiple.t ....................... ok\n",
            "t/29id.t ........................................... ok\n",
            "t/30keep_blanks.t .................................. ok\n",
            "t/30xpathcontext.t ................................. ok\n",
            "t/31xpc_functions.t ................................ ok\n",
            "t/32xpc_variables.t ................................ ok\n",
            "t/35huge_mode.t .................................... ok\n",
            "t/40reader.t ....................................... ok\n",
            "t/40reader_mem_error.t ............................. ok\n",
            "t/41xinclude.t ..................................... ok\n",
            "t/42common.t ....................................... ok\n",
            "t/43options.t ...................................... ok\n",
            "t/44extent.t ....................................... ok\n",
            "t/45regex.t ........................................ ok\n",
            "t/46err_column.t ................................... ok\n",
            "t/47load_xml_callbacks.t ........................... ok\n",
            "t/48_gh_pr63_detect_undef_values.t ................. ok\n",
            "t/48_memleak_rt_83744.t ............................ ok\n",
            "t/48_reader_undef_warning_on_empty_str_rt106830.t .. ok\n",
            "t/48_removeChild_crashes_rt_80395.t ................ ok\n",
            "t/48_replaceNode_DTD_nodes_rT_80521.t .............. ok\n",
            "t/48_RH5_double_free_rt83779.t ..................... ok\n",
            "t/48_rt123379_setNamespace.t ....................... ok\n",
            "t/48_rt55000.t ..................................... ok\n",
            "t/48_rt93429_recover_2_in_html_parsing.t ........... ok\n",
            "t/48_SAX_Builder_rt_91433.t ........................ ok\n",
            "t/48importing_nodes_IDs_rt_69520.t ................. ok\n",
            "t/49_load_html.t ................................... ok\n",
            "t/49callbacks_returning_undef.t .................... ok\n",
            "t/49global_extent.t ................................ ok\n",
            "t/50devel.t ........................................ ok\n",
            "t/51_parse_html_string_rt87089.t ................... ok\n",
            "t/60error_prev_chain.t ............................. ok\n",
            "t/60struct_error.t ................................. ok\n",
            "t/61error.t ........................................ ok\n",
            "t/62overload.t ..................................... ok\n",
            "t/71overloads.t .................................... ok\n",
            "t/72destruction.t .................................. ok\n",
            "t/80registryleak.t ................................. ok\n",
            "t/90shared_clone_failed_rt_91800.t ................. skipped: optional (set THREAD_TEST=1 to run these tests)\n",
            "t/90stack.t ........................................ ok\n",
            "t/90threads.t ...................................... skipped: optional (set THREAD_TEST=1 to run these tests)\n",
            "t/91unique_key.t ................................... ok\n",
            "t/cpan-changes.t ................................... skipped: These tests are for authors only!\n",
            "t/namespace-magic-scalar-rt144415.t ................ ok\n",
            "t/pod-files-presence.t ............................. skipped: only for AUTHORS\n",
            "t/pod.t ............................................ skipped: These tests are for authors only!\n",
            "t/release-kwalitee.t ............................... skipped: These tests are for authors only!\n",
            "t/style-trailing-space.t ........................... skipped: These tests are for authors only!\n",
            "All tests successful.\n",
            "Files=77, Tests=2547,  9 wallclock secs ( 0.44 usr  0.13 sys +  7.31 cusr  1.32 csys =  9.20 CPU)\n",
            "Result: PASS\n",
            "  SHLOMIF/XML-LibXML-2.0208.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for SHLOMIF/XML-LibXML-2.0208.tar.gz\n",
            "\"/usr/bin/perl\" -MExtUtils::Command::MM -e 'cp_nonempty' -- LibXML.bs blib/arch/auto/XML/LibXML/LibXML.bs 644\n",
            "Manifying 34 pod documents\n",
            "Files found in blib/arch: installing files in blib/lib into architecture dependent library tree\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/XML/LibXML/LibXML.so\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML.pm\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/SAX.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Element.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Parser.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Reader.pm\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Common.pm\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Schema.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/SAX.pm\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Error.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Document.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Devel.pm\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/AttributeHash.pm\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/DocumentFragment.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/XPathContext.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Pattern.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Node.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Boolean.pm\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Error.pm\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/XPathExpression.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/InputCallback.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/ErrNo.pm\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/RelaxNG.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/RegExp.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Number.pm\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Namespace.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Comment.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Attr.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Dtd.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Text.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/NodeList.pm\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Common.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Literal.pm\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/Reader.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/PI.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/XPathContext.pm\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/ErrNo.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/DOM.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/CDATASection.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/SAX/Builder.pm\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/SAX/Generator.pm\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/SAX/Builder.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/XML/LibXML/SAX/Parser.pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::NodeList.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::Comment.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::Dtd.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::RelaxNG.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::Error.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::Document.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::SAX::Generator.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::SAX::Builder.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::CDATASection.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::Parser.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::Devel.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::Common.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::DocumentFragment.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::Text.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::XPathExpression.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::Namespace.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::RegExp.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::Attr.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::SAX.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::Schema.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::Node.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::Literal.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::Number.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::PI.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::AttributeHash.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::InputCallback.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::XPathContext.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::DOM.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::Element.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::Boolean.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::Reader.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::Pattern.3pm\n",
            "Installing /usr/local/man/man3/XML::LibXML::ErrNo.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "could not find ParserDetails.ini in /usr/local/share/perl/5.30.0/XML/SAX\n",
            "  SHLOMIF/XML-LibXML-2.0208.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "XML::LibXML::Reader is up to date (2.0208).\n",
            "XML::SAX is up to date (1.02).\n",
            "Running install for module 'XML::SAX::Writer'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/P/PE/PERIGRIN/XML-SAX-Writer-0.57.tar.gz\n",
            "Checksum for /root/.cpan/sources/authors/id/P/PE/PERIGRIN/XML-SAX-Writer-0.57.tar.gz ok\n",
            "Configuring P/PE/PERIGRIN/XML-SAX-Writer-0.57.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Warning: prerequisite XML::Filter::BufferText 1.00 not found.\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for XML::SAX::Writer\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  PERIGRIN/XML-SAX-Writer-0.57.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for P/PE/PERIGRIN/XML-SAX-Writer-0.57.tar.gz\n",
            "---- Unsatisfied dependencies detected during ----\n",
            "----    PERIGRIN/XML-SAX-Writer-0.57.tar.gz   ----\n",
            "    XML::Filter::BufferText [requires]\n",
            "Running install for module 'XML::Filter::BufferText'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/R/RB/RBERJON/XML-Filter-BufferText-1.01.tar.gz\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/R/RB/RBERJON/CHECKSUMS\n",
            "Checksum for /root/.cpan/sources/authors/id/R/RB/RBERJON/XML-Filter-BufferText-1.01.tar.gz ok\n",
            "Configuring R/RB/RBERJON/XML-Filter-BufferText-1.01.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for XML::Filter::BufferText\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  RBERJON/XML-Filter-BufferText-1.01.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for R/RB/RBERJON/XML-Filter-BufferText-1.01.tar.gz\n",
            "cp BufferText.pm blib/lib/XML/Filter/BufferText.pm\n",
            "Manifying 1 pod document\n",
            "  RBERJON/XML-Filter-BufferText-1.01.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for RBERJON/XML-Filter-BufferText-1.01.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/05basic.t .. ok\n",
            "All tests successful.\n",
            "Files=1, Tests=4,  0 wallclock secs ( 0.03 usr  0.00 sys +  0.10 cusr  0.01 csys =  0.14 CPU)\n",
            "Result: PASS\n",
            "  RBERJON/XML-Filter-BufferText-1.01.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for RBERJON/XML-Filter-BufferText-1.01.tar.gz\n",
            "Manifying 1 pod document\n",
            "Installing /usr/local/share/perl/5.30.0/XML/Filter/BufferText.pm\n",
            "Installing /usr/local/man/man3/XML::Filter::BufferText.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  RBERJON/XML-Filter-BufferText-1.01.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "  PERIGRIN/XML-SAX-Writer-0.57.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/XML-SAX-Writer-0.57-0\n",
            "  PERIGRIN/XML-SAX-Writer-0.57.tar.gz\n",
            "  Has already been prepared\n",
            "Running make for P/PE/PERIGRIN/XML-SAX-Writer-0.57.tar.gz\n",
            "cp lib/XML/SAX/Writer.pm blib/lib/XML/SAX/Writer.pm\n",
            "cp lib/XML/SAX/Writer/XML.pm blib/lib/XML/SAX/Writer/XML.pm\n",
            "Manifying 2 pod documents\n",
            "  PERIGRIN/XML-SAX-Writer-0.57.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for PERIGRIN/XML-SAX-Writer-0.57.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/00-compile.t ... ok\n",
            "t/05basic.t ...... ok\n",
            "t/10consumers.t .. ok\n",
            "t/20bugs.t ....... ok\n",
            "All tests successful.\n",
            "Files=4, Tests=50,  0 wallclock secs ( 0.05 usr  0.01 sys +  0.51 cusr  0.10 csys =  0.67 CPU)\n",
            "Result: PASS\n",
            "  PERIGRIN/XML-SAX-Writer-0.57.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for PERIGRIN/XML-SAX-Writer-0.57.tar.gz\n",
            "Manifying 2 pod documents\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/Writer.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/SAX/Writer/XML.pm\n",
            "Installing /usr/local/man/man3/XML::SAX::Writer::XML.3pm\n",
            "Installing /usr/local/man/man3/XML::SAX::Writer.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  PERIGRIN/XML-SAX-Writer-0.57.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "Running install for module 'XML::Twig'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/M/MI/MIROD/XML-Twig-3.52.tar.gz\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/M/MI/MIROD/CHECKSUMS\n",
            "Checksum for /root/.cpan/sources/authors/id/M/MI/MIROD/XML-Twig-3.52.tar.gz ok\n",
            "Configuring M/MI/MIROD/XML-Twig-3.52.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for XML::Twig\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  MIROD/XML-Twig-3.52.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for M/MI/MIROD/XML-Twig-3.52.tar.gz\n",
            "\"/usr/bin/perl\" speedup Twig_pm.slow > Twig.pm\n",
            "perl version is 5.030000\n",
            "\"/usr/bin/perl\" -i_bak -p filter_for_5.005 Twig.pm Twig/XPath.pm\n",
            "\"/usr/bin/perl\" check_optional_modules\n",
            "cp Twig.pm blib/lib/XML/Twig.pm\n",
            "cp Twig/XPath.pm blib/lib/XML/Twig/XPath.pm\n",
            "Manifying 1 pod document\n",
            "  MIROD/XML-Twig-3.52.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for MIROD/XML-Twig-3.52.tar.gz\n",
            "\"/usr/bin/perl\" speedup Twig_pm.slow > Twig.pm\n",
            "perl version is 5.030000\n",
            "\"/usr/bin/perl\" -i_bak -p filter_for_5.005 Twig.pm Twig/XPath.pm\n",
            "\"/usr/bin/perl\" check_optional_modules\n",
            "Skip blib/lib/XML/Twig/XPath.pm (unchanged)\n",
            "Skip blib/lib/XML/Twig.pm (unchanged)\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/is_field.t ........................ ok\n",
            "t/pod.t ............................. 1/1 Author test. Set $ENV{TEST_AUTHOR} to a true value to run.\n",
            "t/pod.t ............................. ok\n",
            "t/pod_coverage.t .................... 1/1 Author test. Set $ENV{TEST_AUTHOR} to a true value to run.\n",
            "t/pod_coverage.t .................... ok\n",
            "t/test1.t ........................... ok\n",
            "t/test2.t ........................... ok\n",
            "t/test3.t ........................... ok\n",
            "t/test4.t ........................... ok\n",
            "t/test5.t ........................... ok\n",
            "t/test_3_24.t ....................... ok\n",
            "t/test_3_26.t ....................... 1/18 \n",
            "need HTML::TreeBuilder 3.13+ to use set_inner_html method: skipping 4 tests\n",
            "\n",
            "HTML::TreeBuilder not available: skipping 5 tests\n",
            "t/test_3_26.t ....................... ok\n",
            "t/test_3_27.t ....................... 1/82 \n",
            "need HTML::TreeBuilder > 4.00 to test error display with HTML data: skipping 4 tests\n",
            "t/test_3_27.t ....................... ok\n",
            "t/test_3_30.t ....................... \n",
            "Tie::IxHash not available, cannot test the cvs pretty_print option: skipping 1 tests\n",
            "t/test_3_30.t ....................... 1/87 \n",
            "HTML::TreeBuilder not available, cannot test safe_parse.*_html methods: skipping 3 tests\n",
            "t/test_3_30.t ....................... ok\n",
            "t/test_3_32.t ....................... ok\n",
            "t/test_3_35.t ....................... ok\n",
            "t/test_3_36.t ....................... 1/67 \n",
            "need HTML::Tidy to use the use_tidy method method: skipping 4 tests\n",
            "\n",
            "HTML::Tidy not available: skipping 5 tests\n",
            "\n",
            "HTML::Tidy not available, cannot test safe_parse.*_html methods with the use_tidy option: skipping 3 tests\n",
            "\n",
            "need HTML::Tidy and LWP to test parse_html with the use_tidy option: skipping 3 tests\n",
            "\n",
            "need HTML::TreeBuilder for additional HTML tests: skipping 3 tests\n",
            "\n",
            ": skipping 1 tests\n",
            "t/test_3_36.t ....................... ok\n",
            "t/test_3_38.t ....................... 1/21 \n",
            "LWP and/or HTML::TreeBuilder not available, cannot test safe_parseurl_html: skipping 1 tests\n",
            "t/test_3_38.t ....................... ok\n",
            "t/test_3_39.t ....................... 1/12 \n",
            ": skipping 3 tests\n",
            "t/test_3_39.t ....................... ok\n",
            "t/test_3_40.t ....................... ok\n",
            "t/test_3_41.t ....................... ok\n",
            "t/test_3_42.t ....................... ok\n",
            "t/test_3_44.t ....................... ok\n",
            "t/test_3_45.t ....................... ok\n",
            "t/test_3_47.t ....................... ok\n",
            "t/test_3_48.t ....................... ok\n",
            "t/test_3_50.t ....................... ok\n",
            "t/test_additional.t ................. 8/663 \n",
            "Tie::IxHash not available: skipping 7 tests\n",
            "\n",
            "need Unicode::Map8 and Unicode::String for those tests: skipping 3 tests\n",
            "\n",
            "require XML::Handler::YAWriter: skipping 3 tests\n",
            "t/test_additional.t ................. ok\n",
            "t/test_attregexp_cond.t ............. ok\n",
            "t/test_autoencoding_conversion.t .... ok\n",
            "t/test_bugs_3_15.t .................. \n",
            "option do_not_escape_amp_in_atts not available (it's only available in an old version of expat), no worries: skipping 1 tests\n",
            "t/test_bugs_3_15.t .................. ok\n",
            "t/test_bugs_3_18.t .................. ok\n",
            "t/test_bugs_3_19.t .................. ok\n",
            "t/test_bugs_3_21.t .................. ok\n",
            "t/test_bugs_3_22.t .................. 1/181 \n",
            "need HTML::TreeBuilder 3.13+ and HTML::Entities::Numbered for this test: skipping 1 tests\n",
            "t/test_bugs_3_22.t .................. ok\n",
            "t/test_cdata.t ...................... ok\n",
            "t/test_changes.t .................... skipped: Test::CPAN::Changes required for this test\n",
            "t/test_class_methods.t .............. ok\n",
            "t/test_class_selector.t ............. ok\n",
            "t/test_comment_handler.t ............ ok\n",
            "t/test_drop_comments.t .............. ok\n",
            "t/test_entities.t ................... ok\n",
            "t/test_erase.t ...................... ok\n",
            "t/test_errors.t ..................... ok\n",
            "t/test_even_more_coverage.t ......... ok\n",
            "t/test_expand_external_entities.t ... ok\n",
            "t/test_ignore_elts.t ................ ok\n",
            "t/test_keep_atts_order.t ............ Tie::IxHash not available, option keep_atts_order not allowed\n",
            "t/test_keep_atts_order.t ............ ok\n",
            "t/test_kwalitee.t ................... Author test. Set $ENV{TEST_AUTHOR} to a true value to run.\n",
            "t/test_kwalitee.t ................... ok\n",
            "t/test_mark.t ....................... ok\n",
            "t/test_memory.t ..................... 1/6 \n",
            "need HTML::TreeBuilder 3.13+: skipping 1 tests\n",
            "\n",
            "need HTML::Tidy: skipping 1 tests\n",
            "test 4: possible memory leak parsing xml (31060 > 27172) (no weak references, disregard unless running with a _really_ old perl, like pre 5.8) at t/test_memory.t line 68, <STATUS> line 280.\n",
            "\n",
            "note that memory leaks can happen even if the module itself doesn't leak, if running\n",
            "tests under Devel::Cover for example. So do not panic if you get a warning here.\n",
            "t/test_memory.t ..................... ok\n",
            "t/test_meta_json.t .................. skipped: Test::CPAN::Meta::JSON required for testing META.json\n",
            "t/test_nav.t ........................ ok\n",
            "t/test_need_3_args_open.t ........... ok\n",
            "t/test_need_io_scalar.t ............. ok\n",
            "t/test_need_use_bytes.t ............. ok\n",
            "t/test_new_features_3_15.t .......... ok\n",
            "t/test_new_features_3_16.t .......... ok\n",
            "t/test_new_features_3_18.t .......... ok\n",
            "t/test_new_features_3_22.t .......... \n",
            "need  HTML::TreeBuilder 3.13+ and LWP to test parse_html: skipping 3 tests\n",
            "\n",
            "need HTML::TreeBuilder 3.13+: skipping 3 tests\n",
            "\n",
            "cannot test html url parsing without LWP::Simple and HTML::TreeBuilder 3.13+: skipping 1 tests\n",
            "t/test_new_features_3_22.t .......... ok\n",
            "t/test_pi_handler.t ................. ok\n",
            "t/test_pos.t ........................ ok\n",
            "t/test_safe_encode.t ................ ok\n",
            "t/test_simplify.t ................... skipping: XML::Simple is not installed\n",
            "t/test_simplify.t ................... ok\n",
            "t/test_spaces.t ..................... ok\n",
            "t/test_twig_roots.t ................. ok\n",
            "t/test_unique_xpath.t ............... ok\n",
            "t/test_variables.t .................. ok\n",
            "t/test_with_lwp.t ................... 1/13 \n",
            "\n",
            "### warning is normal here ###\n",
            "\n",
            "file:test_with_lwp_no_file.xml File `test_with_lwp_no_file.xml' does not exist at t/test_with_lwp.t line 54.\n",
            "t/test_with_lwp.t ................... ok\n",
            "t/test_wrapped.t .................... ok\n",
            "t/test_xml_split.t .................. 42/59 \n",
            "need IO::CaptureOutput to test tool options: skipping 15 tests\n",
            "t/test_xml_split.t .................. ok\n",
            "t/test_xml_split_g.t ................ ok\n",
            "t/test_xpath_cond.t ................. ok\n",
            "t/tests_3_23.t ...................... \n",
            ": skipping 3 tests\n",
            "t/tests_3_23.t ...................... ok\n",
            "t/xmlxpath_01basic.t ................ cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_01basic.t ................ ok\n",
            "t/xmlxpath_02descendant.t ........... cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_02descendant.t ........... ok\n",
            "t/xmlxpath_03star.t ................. cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_03star.t ................. ok\n",
            "t/xmlxpath_04pos.t .................. cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_04pos.t .................. ok\n",
            "t/xmlxpath_05attrib.t ............... 1/1 cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_05attrib.t ............... ok\n",
            "t/xmlxpath_06attrib_val.t ........... 1/1 cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_06attrib_val.t ........... ok\n",
            "t/xmlxpath_07count.t ................ cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_07count.t ................ ok\n",
            "t/xmlxpath_08name.t ................. cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_08name.t ................. ok\n",
            "t/xmlxpath_09a_string_length.t ...... cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_09a_string_length.t ...... ok\n",
            "t/xmlxpath_09string_length.t ........ cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_09string_length.t ........ ok\n",
            "t/xmlxpath_10pipe.t ................. cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_10pipe.t ................. ok\n",
            "t/xmlxpath_12axisdescendant.t ....... cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_12axisdescendant.t ....... ok\n",
            "t/xmlxpath_13axisparent.t ........... cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_13axisparent.t ........... ok\n",
            "t/xmlxpath_14axisancestor.t ......... cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_14axisancestor.t ......... ok\n",
            "t/xmlxpath_15axisfol_sib.t .......... cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_15axisfol_sib.t .......... ok\n",
            "t/xmlxpath_16axisprec_sib.t ......... cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_16axisprec_sib.t ......... ok\n",
            "t/xmlxpath_17axisfollowing.t ........ cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_17axisfollowing.t ........ ok\n",
            "t/xmlxpath_18axispreceding.t ........ cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_18axispreceding.t ........ ok\n",
            "t/xmlxpath_19axisd_or_s.t ........... cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_19axisd_or_s.t ........... ok\n",
            "t/xmlxpath_20axisa_or_s.t ........... cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_20axisa_or_s.t ........... ok\n",
            "t/xmlxpath_21allnodes.t ............. cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_21allnodes.t ............. ok\n",
            "t/xmlxpath_22name_select.t .......... cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_22name_select.t .......... ok\n",
            "t/xmlxpath_23func.t ................. cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_23func.t ................. ok\n",
            "t/xmlxpath_24namespaces.t ........... cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_24namespaces.t ........... ok\n",
            "t/xmlxpath_25scope.t ................ cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_25scope.t ................ ok\n",
            "t/xmlxpath_26predicate.t ............ cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_26predicate.t ............ ok\n",
            "t/xmlxpath_28ancestor2.t ............ cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_28ancestor2.t ............ ok\n",
            "t/xmlxpath_29desc_with_predicate.t .. cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_29desc_with_predicate.t .. ok\n",
            "t/xmlxpath_30lang.t ................. cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_30lang.t ................. ok\n",
            "t/xmlxpath_31vars.t ................. cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_31vars.t ................. ok\n",
            "t/xmlxpath_additional.t ............. cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_additional.t ............. ok\n",
            "t/xmlxpath_nav.t .................... cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_nav.t .................... ok\n",
            "t/xmlxpath_test1.t .................. cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_test1.t .................. ok\n",
            "t/xmlxpath_test_twig_roots.t ........ cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_test_twig_roots.t ........ ok\n",
            "t/xmlxpath_test_with_handlers.t ..... cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_test_with_handlers.t ..... ok\n",
            "t/xmlxpath_xpath_cond.t ............. cannot use XML::Twig::XPath: neither XML::XPathEngine 0.09+ nor XML::XPath are available\n",
            "t/xmlxpath_xpath_cond.t ............. ok\n",
            "t/zz_dump_config.t .................. \n",
            "\n",
            "Configuration:\n",
            "\n",
            "perl: 5.030000\n",
            "OS: linux - x86_64-linux\n",
            "required\n",
            "  XML::Parser              :             2.46 \n",
            "  expat                    : <no version information found> \n",
            "Strongly Recommended\n",
            "  Scalar::Util             :              1.5 (for improved memory management)\n",
            "  Encode                   :             3.01 (for encoding conversions)\n",
            "Modules providing additional features\n",
            "  XML::XPathEngine         :  <not available> (to use XML::Twig::XPath)\n",
            "  XML::XPath               :  <not available> (to use XML::Twig::XPath if Tree::XPathEngine not available)\n",
            "  LWP                      :             6.68 (for the parseurl method)\n",
            "  HTML::TreeBuilder        :  <not available> (to use parse_html and parsefile_html)\n",
            "  HTML::Entities::Numbered :  <not available> (to allow parsing of HTML containing named entities)\n",
            "  HTML::Tidy               :  <not available> (to use parse_html and parsefile_html with the use_tidy option)\n",
            "  HTML::Entities           :             3.81 (for the html_encode filter)\n",
            "  Tie::IxHash              :  <not available> (for the keep_atts_order option)\n",
            "  Text::Wrap               :        2013.0523 (to use the \"wrapped\" option for pretty_print)\n",
            "Modules used only by the auto tests\n",
            "  Test                     :             1.31 \n",
            "  Test::Pod                :  <not available> \n",
            "  XML::Simple              :  <not available> \n",
            "  XML::Handler::YAWriter   :  <not available> \n",
            "  XML::SAX::Writer         :             0.57 \n",
            "  XML::Filter::BufferText  :             1.01 \n",
            "  IO::Scalar               :            2.113 \n",
            "  IO::CaptureOutput        :  <not available> \n",
            "\n",
            "\n",
            "Please add this information to bug reports (you can run t/zz_dump_config.t to get it)\n",
            "\n",
            "if you are upgrading the module from a previous version, make sure you read the\n",
            "Changes file for bug fixes, new features and the occasional COMPATIBILITY WARNING\n",
            "\n",
            "t/zz_dump_config.t .................. ok\n",
            "All tests successful.\n",
            "Files=109, Tests=4481, 25 wallclock secs ( 0.68 usr  0.17 sys + 20.34 cusr  3.08 csys = 24.27 CPU)\n",
            "Result: PASS\n",
            "  MIROD/XML-Twig-3.52.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for MIROD/XML-Twig-3.52.tar.gz\n",
            "\"/usr/bin/perl\" speedup Twig_pm.slow > Twig.pm\n",
            "perl version is 5.030000\n",
            "\"/usr/bin/perl\" -i_bak -p filter_for_5.005 Twig.pm Twig/XPath.pm\n",
            "\"/usr/bin/perl\" check_optional_modules\n",
            "Skip blib/lib/XML/Twig.pm (unchanged)\n",
            "Skip blib/lib/XML/Twig/XPath.pm (unchanged)\n",
            "Manifying 1 pod document\n",
            "Installing /usr/local/share/perl/5.30.0/XML/Twig.pm\n",
            "Installing /usr/local/share/perl/5.30.0/XML/Twig/XPath.pm\n",
            "Installing /usr/local/man/man3/XML::Twig.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  MIROD/XML-Twig-3.52.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "Running install for module 'XML::Writer'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/J/JO/JOSEPHW/XML-Writer-0.900.tar.gz\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/J/JO/JOSEPHW/CHECKSUMS\n",
            "Checksum for /root/.cpan/sources/authors/id/J/JO/JOSEPHW/XML-Writer-0.900.tar.gz ok\n",
            "Configuring J/JO/JOSEPHW/XML-Writer-0.900.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for XML::Writer\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  JOSEPHW/XML-Writer-0.900.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for J/JO/JOSEPHW/XML-Writer-0.900.tar.gz\n",
            "cp Writer.pm blib/lib/XML/Writer.pm\n",
            "Manifying 1 pod document\n",
            "  JOSEPHW/XML-Writer-0.900.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for JOSEPHW/XML-Writer-0.900.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/01_main.t ............... ok\n",
            "t/pod-coverage.t .......... skipped: Test::Pod::Coverage required for testing pod coverage\n",
            "t/pod.t ................... skipped: Test::Pod 1.00 required for testing POD\n",
            "t/selfcontained_output.t .. ok\n",
            "All tests successful.\n",
            "Files=4, Tests=272,  1 wallclock secs ( 0.06 usr  0.01 sys +  0.35 cusr  0.05 csys =  0.47 CPU)\n",
            "Result: PASS\n",
            "  JOSEPHW/XML-Writer-0.900.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for JOSEPHW/XML-Writer-0.900.tar.gz\n",
            "Manifying 1 pod document\n",
            "Installing /usr/local/share/perl/5.30.0/XML/Writer.pm\n",
            "Installing /usr/local/man/man3/XML::Writer.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  JOSEPHW/XML-Writer-0.900.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "Running install for module 'YAML'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/T/TI/TINITA/YAML-1.30.tar.gz\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/T/TI/TINITA/CHECKSUMS\n",
            "Checksum for /root/.cpan/sources/authors/id/T/TI/TINITA/YAML-1.30.tar.gz ok\n",
            "Configuring T/TI/TINITA/YAML-1.30.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Warning: prerequisite Test::YAML 1.05 not found.\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for YAML\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  TINITA/YAML-1.30.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for T/TI/TINITA/YAML-1.30.tar.gz\n",
            "---- Unsatisfied dependencies detected during ----\n",
            "----          TINITA/YAML-1.30.tar.gz         ----\n",
            "    Test::YAML [build_requires]\n",
            "Running install for module 'Test::YAML'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/T/TI/TINITA/Test-YAML-1.07.tar.gz\n",
            "Checksum for /root/.cpan/sources/authors/id/T/TI/TINITA/Test-YAML-1.07.tar.gz ok\n",
            "Configuring T/TI/TINITA/Test-YAML-1.07.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Warning: prerequisite Test::Base 0.89 not found.\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for Test::YAML\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  TINITA/Test-YAML-1.07.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for T/TI/TINITA/Test-YAML-1.07.tar.gz\n",
            "---- Unsatisfied dependencies detected during ----\n",
            "----       TINITA/Test-YAML-1.07.tar.gz       ----\n",
            "    Test::Base [requires]\n",
            "Running install for module 'Test::Base'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/I/IN/INGY/Test-Base-0.89.tar.gz\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/I/IN/INGY/CHECKSUMS\n",
            "Checksum for /root/.cpan/sources/authors/id/I/IN/INGY/Test-Base-0.89.tar.gz ok\n",
            "Configuring I/IN/INGY/Test-Base-0.89.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Warning: prerequisite Spiffy 0.40 not found.\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for Test::Base\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  INGY/Test-Base-0.89.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for I/IN/INGY/Test-Base-0.89.tar.gz\n",
            "---- Unsatisfied dependencies detected during ----\n",
            "----        INGY/Test-Base-0.89.tar.gz        ----\n",
            "    Spiffy [requires]\n",
            "Running install for module 'Spiffy'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/I/IN/INGY/Spiffy-0.46.tar.gz\n",
            "Checksum for /root/.cpan/sources/authors/id/I/IN/INGY/Spiffy-0.46.tar.gz ok\n",
            "Configuring I/IN/INGY/Spiffy-0.46.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for Spiffy\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  INGY/Spiffy-0.46.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for I/IN/INGY/Spiffy-0.46.tar.gz\n",
            "cp lib/Spiffy.pm blib/lib/Spiffy.pm\n",
            "cp lib/Spiffy.pod blib/lib/Spiffy.pod\n",
            "cp lib/Spiffy/mixin.pm blib/lib/Spiffy/mixin.pm\n",
            "Manifying 1 pod document\n",
            "  INGY/Spiffy-0.46.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for INGY/Spiffy-0.46.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/autoload.t ............ ok\n",
            "t/base.t ................ ok\n",
            "t/base2.t ............... ok\n",
            "t/cascade.t ............. ok\n",
            "t/const.t ............... ok\n",
            "t/early.t ............... ok\n",
            "t/export1.t ............. ok\n",
            "t/export2.t ............. ok\n",
            "t/export3.t ............. ok\n",
            "t/export4.t ............. ok\n",
            "t/export5.t ............. ok\n",
            "t/export6.t ............. ok\n",
            "t/export7.t ............. ok\n",
            "t/exporter.t ............ ok\n",
            "t/field.t ............... ok\n",
            "t/field2.t .............. ok\n",
            "t/field3.t .............. ok\n",
            "t/filter.t .............. ok\n",
            "t/filter2.t ............. ok\n",
            "t/filter3.t ............. ok\n",
            "t/filter4.t ............. ok\n",
            "t/filter5.t ............. ok\n",
            "t/mixin.t ............... ok\n",
            "t/mixin2.t .............. ok\n",
            "t/mixin3.t .............. ok\n",
            "t/new.t ................. ok\n",
            "t/package.t ............. ok\n",
            "t/parse.t ............... ok\n",
            "t/release-pod-syntax.t .. skipped: these tests are for release candidate testing\n",
            "t/stub.t ................ ok\n",
            "t/super.t ............... ok\n",
            "t/super2.t .............. ok\n",
            "All tests successful.\n",
            "Files=32, Tests=198,  3 wallclock secs ( 0.11 usr  0.05 sys +  1.99 cusr  0.39 csys =  2.54 CPU)\n",
            "Result: PASS\n",
            "  INGY/Spiffy-0.46.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for INGY/Spiffy-0.46.tar.gz\n",
            "Manifying 1 pod document\n",
            "Installing /usr/local/share/perl/5.30.0/Spiffy.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Spiffy.pod\n",
            "Installing /usr/local/share/perl/5.30.0/Spiffy/mixin.pm\n",
            "Installing /usr/local/man/man3/Spiffy.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  INGY/Spiffy-0.46.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "  INGY/Test-Base-0.89.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/Test-Base-0.89-0\n",
            "  INGY/Test-Base-0.89.tar.gz\n",
            "  Has already been prepared\n",
            "Running make for I/IN/INGY/Test-Base-0.89.tar.gz\n",
            "cp lib/Test/Base.pod blib/lib/Test/Base.pod\n",
            "cp lib/Test/Base/Filter.pm blib/lib/Test/Base/Filter.pm\n",
            "cp lib/Test/Base/Filter.pod blib/lib/Test/Base/Filter.pod\n",
            "cp lib/Test/Base.pm blib/lib/Test/Base.pm\n",
            "Manifying 2 pod documents\n",
            "  INGY/Test-Base-0.89.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for INGY/Test-Base-0.89.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/000-require-modules.t .. ok\n",
            "t/append.t ............... ok\n",
            "t/arguments.t ............ ok\n",
            "t/array.t ................ ok\n",
            "t/author-pod-syntax.t .... skipped: these tests are for testing by the author\n",
            "t/autoload.t ............. ok\n",
            "t/base64.t ............... ok\n",
            "t/blocks-scalar.t ........ ok\n",
            "t/blocks_grep.t .......... ok\n",
            "t/chomp.t ................ ok\n",
            "t/chop.t ................. ok\n",
            "t/compact.t .............. ok\n",
            "t/compile.t .............. ok\n",
            "t/deep.t ................. ok\n",
            "t/delimiters.t ........... ok\n",
            "t/description.t .......... ok\n",
            "t/diff_is.t .............. ok\n",
            "t/dumper.t ............... ok\n",
            "t/embed_perl.t ........... ok\n",
            "t/escape.t ............... ok\n",
            "t/eval.t ................. ok\n",
            "t/eval_all.t ............. ok\n",
            "t/eval_stderr.t .......... ok\n",
            "t/eval_stdout.t .......... ok\n",
            "t/expected-zero.t ........ ok\n",
            "t/export.t ............... ok\n",
            "t/exported_func.t ........ ok\n",
            "t/filter_arguments.t ..... ok\n",
            "t/filter_delay.t ......... ok\n",
            "t/filter_functions.t ..... ok\n",
            "t/filters-append.t ....... ok\n",
            "t/filters.t .............. ok\n",
            "t/filters_map.t .......... ok\n",
            "t/first_block.t .......... ok\n",
            "t/flatten.t .............. ok\n",
            "t/get_url.t .............. skipped: Need to figure out network testing\n",
            "t/hash.t ................. ok\n",
            "t/head.t ................. ok\n",
            "t/internals.t ............ ok\n",
            "t/is.t ................... ok\n",
            "t/jit-run.t .............. ok\n",
            "t/join-deep.t ............ ok\n",
            "t/join.t ................. ok\n",
            "t/last.t ................. ok\n",
            "t/late.t ................. ok\n",
            "t/lazy-filters.t ......... ok\n",
            "t/lines.t ................ ok\n",
            "t/list.t ................. ok\n",
            "t/main_filters.t ......... 1/6 # Filters returning numbers are supposed to do munging $_: your filter 'main::mike5' apparently doesn't.\n",
            "t/main_filters.t ......... ok\n",
            "t/multi-level-inherit.t .. ok\n",
            "t/name.t ................. ok\n",
            "t/next.t ................. ok\n",
            "t/no_diff.t .............. ok\n",
            "t/no_plan.t .............. ok\n",
            "t/normalize.t ............ ok\n",
            "t/only-with-implicit.t ... ok\n",
            "t/only.t ................. ok\n",
            "t/oo.t ................... ok\n",
            "t/oo_run.t ............... ok\n",
            "t/parentheses.t .......... ok\n",
            "t/prepend.t .............. ok\n",
            "t/preserve-order.t ....... ok\n",
            "t/prototypes.t ........... ok\n",
            "t/quick-plan.t ........... ok\n",
            "t/quick_test.t ........... ok\n",
            "t/read_file.t ............ ok\n",
            "t/regexp.t ............... ok\n",
            "t/repeated-filters.t ..... ok\n",
            "t/require.t .............. ok\n",
            "t/reserved_names.t ....... ok\n",
            "t/reverse-deep.t ......... ok\n",
            "t/reverse.t .............. ok\n",
            "t/run-args.t ............. ok\n",
            "t/run_compare.t .......... ok\n",
            "t/run_is.t ............... ok\n",
            "t/run_is_deep.t .......... ok\n",
            "t/run_is_deeply.t ........ ok\n",
            "t/run_like.t ............. ok\n",
            "t/run_unlike.t ........... ok\n",
            "t/simple.t ............... ok\n",
            "t/skip.t ................. ok\n",
            "t/slice.t ................ ok\n",
            "t/sort-deep.t ............ ok\n",
            "t/sort.t ................. ok\n",
            "t/spec_file.t ............ ok\n",
            "t/spec_string.t .......... ok\n",
            "t/split-deep.t ........... ok\n",
            "t/split-regexp.t ......... ok\n",
            "t/split.t ................ ok\n",
            "t/strict-warnings.t ...... ok\n",
            "t/strict.t ............... ok\n",
            "t/subclass-autoclass.t ... ok\n",
            "t/subclass-import.t ...... ok\n",
            "t/subclass.t ............. ok\n",
            "t/subclass_late.t ........ ok\n",
            "t/tail.t ................. ok\n",
            "t/tie_output.t ........... ok\n",
            "t/trim.t ................. ok\n",
            "t/unchomp.t .............. ok\n",
            "t/undef.t ................ ok\n",
            "t/use-test-more.t ........ ok\n",
            "t/use.t .................. ok\n",
            "t/write_file.t ........... ok\n",
            "t/xxx.t .................. skipped: Requires YAML\n",
            "t/yaml.t ................. skipped: Requires YAML\n",
            "t/zero-blocks.t .......... ok\n",
            "All tests successful.\n",
            "Files=106, Tests=431, 12 wallclock secs ( 0.37 usr  0.15 sys +  8.84 cusr  1.59 csys = 10.95 CPU)\n",
            "Result: PASS\n",
            "  INGY/Test-Base-0.89.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for INGY/Test-Base-0.89.tar.gz\n",
            "Manifying 2 pod documents\n",
            "Installing /usr/local/share/perl/5.30.0/Test/Base.pod\n",
            "Installing /usr/local/share/perl/5.30.0/Test/Base.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Test/Base/Filter.pod\n",
            "Installing /usr/local/share/perl/5.30.0/Test/Base/Filter.pm\n",
            "Installing /usr/local/man/man3/Test::Base.3pm\n",
            "Installing /usr/local/man/man3/Test::Base::Filter.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  INGY/Test-Base-0.89.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "  TINITA/Test-YAML-1.07.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/Test-YAML-1.07-0\n",
            "  TINITA/Test-YAML-1.07.tar.gz\n",
            "  Has already been prepared\n",
            "Running make for T/TI/TINITA/Test-YAML-1.07.tar.gz\n",
            "cp lib/Test/YAML.pod blib/lib/Test/YAML.pod\n",
            "cp lib/Test/YAML.pm blib/lib/Test/YAML.pm\n",
            "cp bin/test-yaml blib/script/test-yaml\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/test-yaml\n",
            "Manifying 1 pod document\n",
            "  TINITA/Test-YAML-1.07.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for TINITA/Test-YAML-1.07.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/author-pod-syntax.t .. skipped: these tests are for testing by the author\n",
            "t/load.t ............... ok\n",
            "All tests successful.\n",
            "Files=2, Tests=1,  1 wallclock secs ( 0.03 usr  0.00 sys +  0.07 cusr  0.01 csys =  0.11 CPU)\n",
            "Result: PASS\n",
            "  TINITA/Test-YAML-1.07.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for TINITA/Test-YAML-1.07.tar.gz\n",
            "Manifying 1 pod document\n",
            "Installing /usr/local/share/perl/5.30.0/Test/YAML.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Test/YAML.pod\n",
            "Installing /usr/local/man/man3/Test::YAML.3pm\n",
            "Installing /usr/local/bin/test-yaml\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  TINITA/Test-YAML-1.07.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "  TINITA/YAML-1.30.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/YAML-1.30-0\n",
            "  TINITA/YAML-1.30.tar.gz\n",
            "  Has already been prepared\n",
            "Running make for T/TI/TINITA/YAML-1.30.tar.gz\n",
            "cp lib/YAML/Any.pm blib/lib/YAML/Any.pm\n",
            "cp lib/YAML/Dumper.pm blib/lib/YAML/Dumper.pm\n",
            "cp lib/YAML/Dumper/Base.pod blib/lib/YAML/Dumper/Base.pod\n",
            "cp lib/YAML/Tag.pm blib/lib/YAML/Tag.pm\n",
            "cp lib/YAML.pm blib/lib/YAML.pm\n",
            "cp lib/YAML/Loader.pod blib/lib/YAML/Loader.pod\n",
            "cp lib/YAML/Tag.pod blib/lib/YAML/Tag.pod\n",
            "cp lib/YAML/Dumper/Base.pm blib/lib/YAML/Dumper/Base.pm\n",
            "cp lib/YAML/Mo.pm blib/lib/YAML/Mo.pm\n",
            "cp lib/YAML/Any.pod blib/lib/YAML/Any.pod\n",
            "cp lib/YAML/Node.pod blib/lib/YAML/Node.pod\n",
            "cp lib/YAML/Error.pod blib/lib/YAML/Error.pod\n",
            "cp lib/YAML/Loader/Base.pm blib/lib/YAML/Loader/Base.pm\n",
            "cp lib/YAML/Marshall.pod blib/lib/YAML/Marshall.pod\n",
            "cp lib/YAML/Error.pm blib/lib/YAML/Error.pm\n",
            "cp lib/YAML/Dumper.pod blib/lib/YAML/Dumper.pod\n",
            "cp lib/YAML/Types.pm blib/lib/YAML/Types.pm\n",
            "cp lib/YAML/Node.pm blib/lib/YAML/Node.pm\n",
            "cp lib/YAML/Marshall.pm blib/lib/YAML/Marshall.pm\n",
            "cp lib/YAML/Loader/Base.pod blib/lib/YAML/Loader/Base.pod\n",
            "cp lib/YAML/Loader.pm blib/lib/YAML/Loader.pm\n",
            "cp lib/YAML.pod blib/lib/YAML.pod\n",
            "cp lib/YAML/Types.pod blib/lib/YAML/Types.pod\n",
            "Manifying 11 pod documents\n",
            "  TINITA/YAML-1.30.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for TINITA/YAML-1.30.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/000-compile-modules.t ............ ok\n",
            "t/2-scalars.t ...................... ok\n",
            "t/author-pod-syntax.t .............. skipped: these tests are for testing by the author\n",
            "t/basic-tests.t .................... ok\n",
            "t/bugs-emailed.t ................... ok\n",
            "t/bugs-rt.t ........................ ok\n",
            "t/changes.t ........................ ok\n",
            "t/dump-basics.t .................... ok\n",
            "t/dump-blessed-glob.t .............. ok\n",
            "t/dump-blessed.t ................... ok\n",
            "t/dump-code.t ...................... ok\n",
            "t/dump-file-utf8.t ................. ok\n",
            "t/dump-file.t ...................... ok\n",
            "t/dump-nested.t .................... ok\n",
            "t/dump-opts.t ...................... ok\n",
            "t/dump-perl-types-512.t ............ skipped: test only for perls before v5.13.5-11-gfb85c04\n",
            "t/dump-perl-types-514.t ............ ok\n",
            "t/dump-perl-types.t ................ ok\n",
            "t/dump-stringify.t ................. ok\n",
            "t/dump-stringy-numbers.t ........... ok\n",
            "t/dump-synopsis.t .................. ok\n",
            "t/dump-tests-512.t ................. skipped: test only for perls before v5.13.5-11-gfb85c04\n",
            "t/dump-tests-514.t ................. ok\n",
            "t/dump-tests.t ..................... ok\n",
            "t/dump-works.t ..................... ok\n",
            "t/errors.t ......................... ok\n",
            "t/export.t ......................... ok\n",
            "t/freeze-thaw.t .................... ok\n",
            "t/global-api.t ..................... ok\n",
            "t/inbox.t .......................... ok\n",
            "t/io-handle.t ...................... ok\n",
            "t/issue-149.t ...................... ok\n",
            "t/issue-69.t ....................... ok\n",
            "t/load-code.t ...................... ok\n",
            "t/load-fails.t ..................... ok\n",
            "t/load-passes.t .................... ok\n",
            "t/load-slides.t .................... ok\n",
            "t/load-spec.t ...................... ok\n",
            "t/load-tests.t ..................... ok\n",
            "t/load-works.t ..................... ok\n",
            "t/marshall.t ....................... ok\n",
            "t/no-load-blessed.t ................ ok\n",
            "t/node-info.t ...................... ok\n",
            "t/numify.t ......................... ok\n",
            "t/preserve.t ....................... ok\n",
            "t/pugs-objects.t ................... ok\n",
            "t/references.t ..................... ok\n",
            "t/regexp.t ......................... ok\n",
            "t/roundtrip.t ...................... ok\n",
            "t/rt-90593.t ....................... ok\n",
            "t/svk.t ............................ ok\n",
            "t/test.t ........................... ok\n",
            "t/trailing-comments-content.t ...... ok\n",
            "t/trailing-comments-non-content.t .. ok\n",
            "All tests successful.\n",
            "Files=54, Tests=542, 10 wallclock secs ( 0.27 usr  0.07 sys +  8.46 cusr  1.02 csys =  9.82 CPU)\n",
            "Result: PASS\n",
            "  TINITA/YAML-1.30.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for TINITA/YAML-1.30.tar.gz\n",
            "Manifying 11 pod documents\n",
            "Installing /usr/local/share/perl/5.30.0/YAML.pm\n",
            "Installing /usr/local/share/perl/5.30.0/YAML.pod\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Dumper.pod\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Node.pm\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Any.pod\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Error.pod\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Node.pod\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Mo.pm\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Error.pm\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Types.pm\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Types.pod\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Loader.pod\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Dumper.pm\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Marshall.pm\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Tag.pm\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Any.pm\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Loader.pm\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Marshall.pod\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Tag.pod\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Dumper/Base.pod\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Dumper/Base.pm\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Loader/Base.pod\n",
            "Installing /usr/local/share/perl/5.30.0/YAML/Loader/Base.pm\n",
            "Installing /usr/local/man/man3/YAML::Node.3pm\n",
            "Installing /usr/local/man/man3/YAML::Loader.3pm\n",
            "Installing /usr/local/man/man3/YAML::Dumper::Base.3pm\n",
            "Installing /usr/local/man/man3/YAML::Any.3pm\n",
            "Installing /usr/local/man/man3/YAML::Types.3pm\n",
            "Installing /usr/local/man/man3/YAML::Loader::Base.3pm\n",
            "Installing /usr/local/man/man3/YAML::Tag.3pm\n",
            "Installing /usr/local/man/man3/YAML::Error.3pm\n",
            "Installing /usr/local/man/man3/YAML::Dumper.3pm\n",
            "Installing /usr/local/man/man3/YAML::Marshall.3pm\n",
            "Installing /usr/local/man/man3/YAML.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  TINITA/YAML-1.30.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "  CJFIELDS/BioPerl-1.7.8.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/BioPerl-1.7.8-0\n",
            "  CJFIELDS/BioPerl-1.7.8.tar.gz\n",
            "  Has already been prepared\n",
            "Running make for C/CJ/CJFIELDS/BioPerl-1.7.8.tar.gz\n",
            "cp lib/Bio/AnalysisParserI.pm blib/lib/Bio/AnalysisParserI.pm\n",
            "cp lib/Bio/AlignIO/xmfa.pm blib/lib/Bio/AlignIO/xmfa.pm\n",
            "cp lib/Bio/AlignIO/mase.pm blib/lib/Bio/AlignIO/mase.pm\n",
            "cp lib/Bio/Align/AlignI.pm blib/lib/Bio/Align/AlignI.pm\n",
            "cp lib/Bio/AlignIO/largemultifasta.pm blib/lib/Bio/AlignIO/largemultifasta.pm\n",
            "cp lib/Bio/AlignIO/Handler/GenericAlignHandler.pm blib/lib/Bio/AlignIO/Handler/GenericAlignHandler.pm\n",
            "cp lib/Bio/Annotation/Reference.pm blib/lib/Bio/Annotation/Reference.pm\n",
            "cp lib/Bio/Align/ProteinStatistics.pm blib/lib/Bio/Align/ProteinStatistics.pm\n",
            "cp lib/Bio/AlignIO/prodom.pm blib/lib/Bio/AlignIO/prodom.pm\n",
            "cp lib/Bio/AlignIO/selex.pm blib/lib/Bio/AlignIO/selex.pm\n",
            "cp lib/Bio/AlignIO/clustalw.pm blib/lib/Bio/AlignIO/clustalw.pm\n",
            "cp lib/Bio/AlignIO/psi.pm blib/lib/Bio/AlignIO/psi.pm\n",
            "cp lib/Bio/AlignIO/meme.pm blib/lib/Bio/AlignIO/meme.pm\n",
            "cp lib/Bio/Annotation/Comment.pm blib/lib/Bio/Annotation/Comment.pm\n",
            "cp lib/Bio/AlignIO/emboss.pm blib/lib/Bio/AlignIO/emboss.pm\n",
            "cp lib/Bio/AlignIO/phylip.pm blib/lib/Bio/AlignIO/phylip.pm\n",
            "cp lib/Bio/Annotation/OntologyTerm.pm blib/lib/Bio/Annotation/OntologyTerm.pm\n",
            "cp lib/Bio/AlignIO/po.pm blib/lib/Bio/AlignIO/po.pm\n",
            "cp lib/Bio/AnalysisI.pm blib/lib/Bio/AnalysisI.pm\n",
            "cp lib/Bio/AlignIO/metafasta.pm blib/lib/Bio/AlignIO/metafasta.pm\n",
            "cp lib/Bio/AlignIO/maf.pm blib/lib/Bio/AlignIO/maf.pm\n",
            "cp lib/Bio/Align/PairwiseStatistics.pm blib/lib/Bio/Align/PairwiseStatistics.pm\n",
            "cp lib/Bio/AlignIO.pm blib/lib/Bio/AlignIO.pm\n",
            "cp lib/Bio/Annotation/Collection.pm blib/lib/Bio/Annotation/Collection.pm\n",
            "cp lib/Bio/Align/StatisticsI.pm blib/lib/Bio/Align/StatisticsI.pm\n",
            "cp lib/Bio/AlignIO/pfam.pm blib/lib/Bio/AlignIO/pfam.pm\n",
            "cp lib/Bio/AlignIO/arp.pm blib/lib/Bio/AlignIO/arp.pm\n",
            "cp lib/Bio/AlignIO/bl2seq.pm blib/lib/Bio/AlignIO/bl2seq.pm\n",
            "cp lib/Bio/AnalysisResultI.pm blib/lib/Bio/AnalysisResultI.pm\n",
            "cp lib/Bio/Align/DNAStatistics.pm blib/lib/Bio/Align/DNAStatistics.pm\n",
            "cp lib/Bio/Annotation/Relation.pm blib/lib/Bio/Annotation/Relation.pm\n",
            "cp lib/Bio/Annotation/DBLink.pm blib/lib/Bio/Annotation/DBLink.pm\n",
            "cp lib/Bio/Annotation/AnnotationFactory.pm blib/lib/Bio/Annotation/AnnotationFactory.pm\n",
            "cp lib/Bio/AlignIO/mega.pm blib/lib/Bio/AlignIO/mega.pm\n",
            "cp lib/Bio/AlignIO/msf.pm blib/lib/Bio/AlignIO/msf.pm\n",
            "cp lib/Bio/Align/Utilities.pm blib/lib/Bio/Align/Utilities.pm\n",
            "cp lib/Bio/AlignIO/proda.pm blib/lib/Bio/AlignIO/proda.pm\n",
            "cp lib/Bio/AlignIO/fasta.pm blib/lib/Bio/AlignIO/fasta.pm\n",
            "cp lib/Bio/AnnotatableI.pm blib/lib/Bio/AnnotatableI.pm\n",
            "cp lib/Bio/AlignIO/nexus.pm blib/lib/Bio/AlignIO/nexus.pm\n",
            "cp lib/Bio/DB/Taxonomy/list.pm blib/lib/Bio/DB/Taxonomy/list.pm\n",
            "cp lib/Bio/Annotation/Target.pm blib/lib/Bio/Annotation/Target.pm\n",
            "cp lib/Bio/DB/SeqI.pm blib/lib/Bio/DB/SeqI.pm\n",
            "cp lib/Bio/DB/Failover.pm blib/lib/Bio/DB/Failover.pm\n",
            "cp lib/Bio/DB/Qual.pm blib/lib/Bio/DB/Qual.pm\n",
            "cp lib/Bio/CodonUsage/IO.pm blib/lib/Bio/CodonUsage/IO.pm\n",
            "cp lib/Bio/DB/GFF/Util/Binning.pm blib/lib/Bio/DB/GFF/Util/Binning.pm\n",
            "cp lib/Bio/DB/Taxonomy.pm blib/lib/Bio/DB/Taxonomy.pm\n",
            "cp lib/Bio/Annotation/TagTree.pm blib/lib/Bio/Annotation/TagTree.pm\n",
            "cp lib/Bio/AnnotationCollectionI.pm blib/lib/Bio/AnnotationCollectionI.pm\n",
            "cp lib/Bio/DB/Query/WebQuery.pm blib/lib/Bio/DB/Query/WebQuery.pm\n",
            "cp lib/Bio/CodonUsage/Table.pm blib/lib/Bio/CodonUsage/Table.pm\n",
            "cp lib/Bio/DB/Flat/BDB/fasta.pm blib/lib/Bio/DB/Flat/BDB/fasta.pm\n",
            "cp lib/Bio/AnnotationI.pm blib/lib/Bio/AnnotationI.pm\n",
            "cp lib/Bio/DB/QueryI.pm blib/lib/Bio/DB/QueryI.pm\n",
            "cp lib/Bio/Annotation/SimpleValue.pm blib/lib/Bio/Annotation/SimpleValue.pm\n",
            "cp lib/Bio/DB/ReferenceI.pm blib/lib/Bio/DB/ReferenceI.pm\n",
            "cp lib/Bio/DB/FileCache.pm blib/lib/Bio/DB/FileCache.pm\n",
            "cp lib/Bio/DB/Flat.pm blib/lib/Bio/DB/Flat.pm\n",
            "cp lib/Bio/DB/Flat/BDB/embl.pm blib/lib/Bio/DB/Flat/BDB/embl.pm\n",
            "cp lib/Bio/DB/Fasta.pm blib/lib/Bio/DB/Fasta.pm\n",
            "cp lib/Bio/DB/LocationI.pm blib/lib/Bio/DB/LocationI.pm\n",
            "cp lib/Bio/DB/GFF/Util/Rearrange.pm blib/lib/Bio/DB/GFF/Util/Rearrange.pm\n",
            "cp lib/Bio/DB/Flat/BDB.pm blib/lib/Bio/DB/Flat/BDB.pm\n",
            "cp lib/Bio/DB/Taxonomy/silva.pm blib/lib/Bio/DB/Taxonomy/silva.pm\n",
            "cp lib/Bio/DB/IndexedBase.pm blib/lib/Bio/DB/IndexedBase.pm\n",
            "cp lib/Bio/DB/Flat/BDB/swiss.pm blib/lib/Bio/DB/Flat/BDB/swiss.pm\n",
            "cp lib/Bio/DB/DBFetch.pm blib/lib/Bio/DB/DBFetch.pm\n",
            "cp lib/Bio/DB/RandomAccessI.pm blib/lib/Bio/DB/RandomAccessI.pm\n",
            "cp lib/Bio/Annotation/TypeManager.pm blib/lib/Bio/Annotation/TypeManager.pm\n",
            "cp lib/Bio/DB/Flat/BDB/genbank.pm blib/lib/Bio/DB/Flat/BDB/genbank.pm\n",
            "cp lib/Bio/DB/Registry.pm blib/lib/Bio/DB/Registry.pm\n",
            "cp lib/Bio/Annotation/Tree.pm blib/lib/Bio/Annotation/Tree.pm\n",
            "cp lib/Bio/DB/InMemoryCache.pm blib/lib/Bio/DB/InMemoryCache.pm\n",
            "cp lib/Bio/DB/Flat/BinarySearch.pm blib/lib/Bio/DB/Flat/BinarySearch.pm\n",
            "cp lib/Bio/DB/Taxonomy/greengenes.pm blib/lib/Bio/DB/Taxonomy/greengenes.pm\n",
            "cp lib/Bio/Annotation/StructuredValue.pm blib/lib/Bio/Annotation/StructuredValue.pm\n",
            "cp lib/Bio/DB/Taxonomy/flatfile.pm blib/lib/Bio/DB/Taxonomy/flatfile.pm\n",
            "cp lib/Bio/DB/GenericWebAgent.pm blib/lib/Bio/DB/GenericWebAgent.pm\n",
            "cp lib/Bio/DB/UpdateableSeqI.pm blib/lib/Bio/DB/UpdateableSeqI.pm\n",
            "cp lib/Bio/Factory/DriverFactory.pm blib/lib/Bio/Factory/DriverFactory.pm\n",
            "cp lib/Bio/Factory/TreeFactoryI.pm blib/lib/Bio/Factory/TreeFactoryI.pm\n",
            "cp lib/Bio/Factory/ObjectFactory.pm blib/lib/Bio/Factory/ObjectFactory.pm\n",
            "cp lib/Bio/Index/SwissPfam.pm blib/lib/Bio/Index/SwissPfam.pm\n",
            "cp lib/Bio/Index/AbstractSeq.pm blib/lib/Bio/Index/AbstractSeq.pm\n",
            "cp lib/Bio/Index/Fasta.pm blib/lib/Bio/Index/Fasta.pm\n",
            "cp lib/Bio/Index/Qual.pm blib/lib/Bio/Index/Qual.pm\n",
            "cp lib/Bio/Index/Fastq.pm blib/lib/Bio/Index/Fastq.pm\n",
            "cp lib/Bio/Factory/SeqAnalysisParserFactory.pm blib/lib/Bio/Factory/SeqAnalysisParserFactory.pm\n",
            "cp lib/Bio/Index/Swissprot.pm blib/lib/Bio/Index/Swissprot.pm\n",
            "cp lib/Bio/FeatureHolderI.pm blib/lib/Bio/FeatureHolderI.pm\n",
            "cp lib/Bio/HandlerBaseI.pm blib/lib/Bio/HandlerBaseI.pm\n",
            "cp lib/Bio/LocatableSeq.pm blib/lib/Bio/LocatableSeq.pm\n",
            "cp lib/Bio/Factory/SequenceStreamI.pm blib/lib/Bio/Factory/SequenceStreamI.pm\n",
            "cp lib/Bio/Event/EventGeneratorI.pm blib/lib/Bio/Event/EventGeneratorI.pm\n",
            "cp lib/Bio/Factory/SequenceProcessorI.pm blib/lib/Bio/Factory/SequenceProcessorI.pm\n",
            "cp lib/Bio/Factory/FTLocationFactory.pm blib/lib/Bio/Factory/FTLocationFactory.pm\n",
            "cp lib/Bio/Index/EMBL.pm blib/lib/Bio/Index/EMBL.pm\n",
            "cp lib/Bio/Factory/ObjectFactoryI.pm blib/lib/Bio/Factory/ObjectFactoryI.pm\n",
            "cp lib/Bio/DBLinkContainerI.pm blib/lib/Bio/DBLinkContainerI.pm\n",
            "cp lib/Bio/Factory/LocationFactoryI.pm blib/lib/Bio/Factory/LocationFactoryI.pm\n",
            "cp lib/Bio/Factory/SequenceFactoryI.pm blib/lib/Bio/Factory/SequenceFactoryI.pm\n",
            "cp lib/Bio/DasI.pm blib/lib/Bio/DasI.pm\n",
            "cp lib/Bio/DescribableI.pm blib/lib/Bio/DescribableI.pm\n",
            "cp lib/Bio/Das/SegmentI.pm blib/lib/Bio/Das/SegmentI.pm\n",
            "cp lib/Bio/Event/EventHandlerI.pm blib/lib/Bio/Event/EventHandlerI.pm\n",
            "cp lib/Bio/Das/FeatureTypeI.pm blib/lib/Bio/Das/FeatureTypeI.pm\n",
            "cp lib/Bio/Index/Blast.pm blib/lib/Bio/Index/Blast.pm\n",
            "cp lib/Bio/Factory/ObjectBuilderI.pm blib/lib/Bio/Factory/ObjectBuilderI.pm\n",
            "cp lib/Bio/IdCollectionI.pm blib/lib/Bio/IdCollectionI.pm\n",
            "cp lib/Bio/IdentifiableI.pm blib/lib/Bio/IdentifiableI.pm\n",
            "cp lib/Bio/Index/BlastTable.pm blib/lib/Bio/Index/BlastTable.pm\n",
            "cp lib/Bio/Factory/ApplicationFactoryI.pm blib/lib/Bio/Factory/ApplicationFactoryI.pm\n",
            "cp lib/Bio/Factory/AnalysisI.pm blib/lib/Bio/Factory/AnalysisI.pm\n",
            "cp lib/Bio/DB/WebDBSeqI.pm blib/lib/Bio/DB/WebDBSeqI.pm\n",
            "cp lib/Bio/Index/Abstract.pm blib/lib/Bio/Index/Abstract.pm\n",
            "cp lib/Bio/Index/GenBank.pm blib/lib/Bio/Index/GenBank.pm\n",
            "cp lib/Bio/Factory/SeqAnalysisParserFactoryI.pm blib/lib/Bio/Factory/SeqAnalysisParserFactoryI.pm\n",
            "cp lib/Bio/Location/SplitLocationI.pm blib/lib/Bio/Location/SplitLocationI.pm\n",
            "cp lib/Bio/Location/Split.pm blib/lib/Bio/Location/Split.pm\n",
            "cp lib/Bio/Matrix/MatrixI.pm blib/lib/Bio/Matrix/MatrixI.pm\n",
            "cp lib/Bio/Location/AvWithinCoordPolicy.pm blib/lib/Bio/Location/AvWithinCoordPolicy.pm\n",
            "cp lib/Bio/Location/NarrowestCoordPolicy.pm blib/lib/Bio/Location/NarrowestCoordPolicy.pm\n",
            "cp lib/Bio/Matrix/IO/phylip.pm blib/lib/Bio/Matrix/IO/phylip.pm\n",
            "cp lib/Bio/Matrix/PSM/InstanceSite.pm blib/lib/Bio/Matrix/PSM/InstanceSite.pm\n",
            "cp lib/Bio/Matrix/PSM/PsmHeaderI.pm blib/lib/Bio/Matrix/PSM/PsmHeaderI.pm\n",
            "cp lib/Bio/Matrix/PSM/SiteMatrix.pm blib/lib/Bio/Matrix/PSM/SiteMatrix.pm\n",
            "cp lib/Bio/LocationI.pm blib/lib/Bio/LocationI.pm\n",
            "cp lib/Bio/Matrix/PSM/SiteMatrixI.pm blib/lib/Bio/Matrix/PSM/SiteMatrixI.pm\n",
            "cp lib/Bio/Matrix/Scoring.pm blib/lib/Bio/Matrix/Scoring.pm\n",
            "cp lib/Bio/Location/WidestCoordPolicy.pm blib/lib/Bio/Location/WidestCoordPolicy.pm\n",
            "cp lib/Bio/Matrix/PSM/PsmHeader.pm blib/lib/Bio/Matrix/PSM/PsmHeader.pm\n",
            "cp lib/Bio/Matrix/PSM/Psm.pm blib/lib/Bio/Matrix/PSM/Psm.pm\n",
            "cp lib/Bio/Matrix/PSM/IO/transfac.pm blib/lib/Bio/Matrix/PSM/IO/transfac.pm\n",
            "cp lib/Bio/Matrix/PSM/IO.pm blib/lib/Bio/Matrix/PSM/IO.pm\n",
            "cp lib/Bio/Matrix/PhylipDist.pm blib/lib/Bio/Matrix/PhylipDist.pm\n",
            "cp lib/Bio/Matrix/PSM/IO/mast.pm blib/lib/Bio/Matrix/PSM/IO/mast.pm\n",
            "cp lib/Bio/Matrix/Mlagan.pm blib/lib/Bio/Matrix/Mlagan.pm\n",
            "cp lib/Bio/Location/Fuzzy.pm blib/lib/Bio/Location/Fuzzy.pm\n",
            "cp lib/Bio/Matrix/PSM/IO/psiblast.pm blib/lib/Bio/Matrix/PSM/IO/psiblast.pm\n",
            "cp lib/Bio/Ontology/DocumentRegistry.pm blib/lib/Bio/Ontology/DocumentRegistry.pm\n",
            "cp lib/Bio/Matrix/PSM/IO/masta.pm blib/lib/Bio/Matrix/PSM/IO/masta.pm\n",
            "cp lib/Bio/Location/Atomic.pm blib/lib/Bio/Location/Atomic.pm\n",
            "cp lib/Bio/Matrix/PSM/ProtMatrix.pm blib/lib/Bio/Matrix/PSM/ProtMatrix.pm\n",
            "cp lib/Bio/Matrix/PSM/ProtPsm.pm blib/lib/Bio/Matrix/PSM/ProtPsm.pm\n",
            "cp lib/Bio/Location/FuzzyLocationI.pm blib/lib/Bio/Location/FuzzyLocationI.pm\n",
            "cp lib/Bio/Matrix/Generic.pm blib/lib/Bio/Matrix/Generic.pm\n",
            "cp lib/Bio/Matrix/PSM/InstanceSiteI.pm blib/lib/Bio/Matrix/PSM/InstanceSiteI.pm\n",
            "cp lib/Bio/Location/Simple.pm blib/lib/Bio/Location/Simple.pm\n",
            "cp lib/Bio/Matrix/PSM/IO/meme.pm blib/lib/Bio/Matrix/PSM/IO/meme.pm\n",
            "cp lib/Bio/Matrix/PSM/PsmI.pm blib/lib/Bio/Matrix/PSM/PsmI.pm\n",
            "cp lib/Bio/Matrix/IO/mlagan.pm blib/lib/Bio/Matrix/IO/mlagan.pm\n",
            "cp lib/Bio/Matrix/IO.pm blib/lib/Bio/Matrix/IO.pm\n",
            "cp lib/Bio/Location/CoordinatePolicyI.pm blib/lib/Bio/Location/CoordinatePolicyI.pm\n",
            "cp lib/Bio/Matrix/IO/scoring.pm blib/lib/Bio/Matrix/IO/scoring.pm\n",
            "cp lib/Bio/Range.pm blib/lib/Bio/Range.pm\n",
            "cp lib/Bio/Ontology/SimpleOntologyEngine.pm blib/lib/Bio/Ontology/SimpleOntologyEngine.pm\n",
            "cp lib/Bio/Ontology/OBOterm.pm blib/lib/Bio/Ontology/OBOterm.pm\n",
            "cp lib/Bio/Ontology/Ontology.pm blib/lib/Bio/Ontology/Ontology.pm\n",
            "cp lib/Bio/Ontology/OntologyStore.pm blib/lib/Bio/Ontology/OntologyStore.pm\n",
            "cp lib/Bio/Ontology/Term.pm blib/lib/Bio/Ontology/Term.pm\n",
            "cp lib/Bio/PrimarySeqI.pm blib/lib/Bio/PrimarySeqI.pm\n",
            "cp lib/Bio/RangeI.pm blib/lib/Bio/RangeI.pm\n",
            "cp lib/Bio/Ontology/OntologyI.pm blib/lib/Bio/Ontology/OntologyI.pm\n",
            "cp lib/Bio/ParameterBaseI.pm blib/lib/Bio/ParameterBaseI.pm\n",
            "cp lib/Bio/PrimarySeq.pm blib/lib/Bio/PrimarySeq.pm\n",
            "cp lib/Bio/Ontology/OntologyEngineI.pm blib/lib/Bio/Ontology/OntologyEngineI.pm\n",
            "cp lib/Bio/OntologyIO/Handlers/InterProHandler.pm blib/lib/Bio/OntologyIO/Handlers/InterProHandler.pm\n",
            "cp lib/Bio/OntologyIO/soflat.pm blib/lib/Bio/OntologyIO/soflat.pm\n",
            "cp lib/Bio/Root/Exception.pm blib/lib/Bio/Root/Exception.pm\n",
            "cp lib/Bio/Ontology/PathI.pm blib/lib/Bio/Ontology/PathI.pm\n",
            "cp lib/Bio/Ontology/Relationship.pm blib/lib/Bio/Ontology/Relationship.pm\n",
            "cp lib/Bio/OntologyIO/Handlers/BaseSAXHandler.pm blib/lib/Bio/OntologyIO/Handlers/BaseSAXHandler.pm\n",
            "cp lib/Bio/Ontology/TermI.pm blib/lib/Bio/Ontology/TermI.pm\n",
            "cp lib/Bio/Ontology/Path.pm blib/lib/Bio/Ontology/Path.pm\n",
            "cp lib/Bio/Ontology/RelationshipI.pm blib/lib/Bio/Ontology/RelationshipI.pm\n",
            "cp lib/Bio/Ontology/SimpleGOEngine/GraphAdaptor.pm blib/lib/Bio/Ontology/SimpleGOEngine/GraphAdaptor.pm\n",
            "cp lib/Bio/PullParserI.pm blib/lib/Bio/PullParserI.pm\n",
            "cp lib/Bio/OntologyIO.pm blib/lib/Bio/OntologyIO.pm\n",
            "cp lib/Bio/OntologyIO/obo.pm blib/lib/Bio/OntologyIO/obo.pm\n",
            "cp lib/Bio/OntologyIO/simplehierarchy.pm blib/lib/Bio/OntologyIO/simplehierarchy.pm\n",
            "cp lib/Bio/Ontology/InterProTerm.pm blib/lib/Bio/Ontology/InterProTerm.pm\n",
            "cp lib/Bio/OntologyIO/Handlers/InterPro_BioSQL_Handler.pm blib/lib/Bio/OntologyIO/Handlers/InterPro_BioSQL_Handler.pm\n",
            "cp lib/Bio/Ontology/GOterm.pm blib/lib/Bio/Ontology/GOterm.pm\n",
            "cp lib/Bio/Ontology/TermFactory.pm blib/lib/Bio/Ontology/TermFactory.pm\n",
            "cp lib/Bio/Ontology/OBOEngine.pm blib/lib/Bio/Ontology/OBOEngine.pm\n",
            "cp lib/Bio/Ontology/RelationshipType.pm blib/lib/Bio/Ontology/RelationshipType.pm\n",
            "cp lib/Bio/OntologyIO/goflat.pm blib/lib/Bio/OntologyIO/goflat.pm\n",
            "cp lib/Bio/OntologyIO/dagflat.pm blib/lib/Bio/OntologyIO/dagflat.pm\n",
            "cp lib/Bio/OntologyIO/InterProParser.pm blib/lib/Bio/OntologyIO/InterProParser.pm\n",
            "cp lib/Bio/Ontology/RelationshipFactory.pm blib/lib/Bio/Ontology/RelationshipFactory.pm\n",
            "cp lib/Bio/Search/BlastStatistics.pm blib/lib/Bio/Search/BlastStatistics.pm\n",
            "cp lib/Bio/Search/HSP/BlastPullHSP.pm blib/lib/Bio/Search/HSP/BlastPullHSP.pm\n",
            "cp lib/Bio/Root/Test.pm blib/lib/Bio/Root/Test.pm\n",
            "cp lib/Bio/Search/GenericStatistics.pm blib/lib/Bio/Search/GenericStatistics.pm\n",
            "cp lib/Bio/Root/Storable.pm blib/lib/Bio/Root/Storable.pm\n",
            "cp lib/Bio/Search/Hit/BlastPullHit.pm blib/lib/Bio/Search/Hit/BlastPullHit.pm\n",
            "cp lib/Bio/Search/BlastUtils.pm blib/lib/Bio/Search/BlastUtils.pm\n",
            "cp lib/Bio/Search/Hit/HitFactory.pm blib/lib/Bio/Search/Hit/HitFactory.pm\n",
            "cp lib/Bio/Search/Iteration/GenericIteration.pm blib/lib/Bio/Search/Iteration/GenericIteration.pm\n",
            "cp lib/Bio/Search/Hit/ModelHit.pm blib/lib/Bio/Search/Hit/ModelHit.pm\n",
            "cp lib/Bio/Search/HSP/PSLHSP.pm blib/lib/Bio/Search/HSP/PSLHSP.pm\n",
            "cp lib/Bio/Root/HTTPget.pm blib/lib/Bio/Root/HTTPget.pm\n",
            "cp lib/Bio/Search/HSP/HSPI.pm blib/lib/Bio/Search/HSP/HSPI.pm\n",
            "cp lib/Bio/Search/Hit/GenericHit.pm blib/lib/Bio/Search/Hit/GenericHit.pm\n",
            "cp lib/Bio/Search/HSP/ModelHSP.pm blib/lib/Bio/Search/HSP/ModelHSP.pm\n",
            "cp lib/Bio/Root/IO.pm blib/lib/Bio/Root/IO.pm\n",
            "cp lib/Bio/Search/HSP/GenericHSP.pm blib/lib/Bio/Search/HSP/GenericHSP.pm\n",
            "cp lib/Bio/Root/Root.pm blib/lib/Bio/Root/Root.pm\n",
            "cp lib/Bio/Search/Hit/BlastHit.pm blib/lib/Bio/Search/Hit/BlastHit.pm\n",
            "cp lib/Bio/Search/Processor.pm blib/lib/Bio/Search/Processor.pm\n",
            "cp lib/Bio/Search/Hit/PullHitI.pm blib/lib/Bio/Search/Hit/PullHitI.pm\n",
            "cp lib/Bio/Search/HSP/PullHSPI.pm blib/lib/Bio/Search/HSP/PullHSPI.pm\n",
            "cp lib/Bio/Search/GenericDatabase.pm blib/lib/Bio/Search/GenericDatabase.pm\n",
            "cp lib/Bio/Search/HSP/FastaHSP.pm blib/lib/Bio/Search/HSP/FastaHSP.pm\n",
            "cp lib/Bio/Search/DatabaseI.pm blib/lib/Bio/Search/DatabaseI.pm\n",
            "cp lib/Bio/Search/Hit/Fasta.pm blib/lib/Bio/Search/Hit/Fasta.pm\n",
            "cp lib/Bio/Root/Utilities.pm blib/lib/Bio/Root/Utilities.pm\n",
            "cp lib/Bio/Search/Hit/PsiBlastHit.pm blib/lib/Bio/Search/Hit/PsiBlastHit.pm\n",
            "cp lib/Bio/Root/RootI.pm blib/lib/Bio/Root/RootI.pm\n",
            "cp lib/Bio/Search/HSP/WABAHSP.pm blib/lib/Bio/Search/HSP/WABAHSP.pm\n",
            "cp lib/Bio/Search/Iteration/IterationI.pm blib/lib/Bio/Search/Iteration/IterationI.pm\n",
            "cp lib/Bio/Search/HSP/PsiBlastHSP.pm blib/lib/Bio/Search/HSP/PsiBlastHSP.pm\n",
            "cp lib/Bio/Search/HSP/HSPFactory.pm blib/lib/Bio/Search/HSP/HSPFactory.pm\n",
            "cp lib/Bio/Search/HSP/BlastHSP.pm blib/lib/Bio/Search/HSP/BlastHSP.pm\n",
            "cp lib/Bio/Root/TestObject.pm blib/lib/Bio/Root/TestObject.pm\n",
            "cp lib/Bio/Search/Hit/HitI.pm blib/lib/Bio/Search/Hit/HitI.pm\n",
            "cp lib/Bio/Root/Version.pm blib/lib/Bio/Root/Version.pm\n",
            "cp lib/Bio/SearchIO/EventHandlerI.pm blib/lib/Bio/SearchIO/EventHandlerI.pm\n",
            "cp lib/Bio/Search/Result/GenericResult.pm blib/lib/Bio/Search/Result/GenericResult.pm\n",
            "cp lib/Bio/Search/Result/ResultFactory.pm blib/lib/Bio/Search/Result/ResultFactory.pm\n",
            "cp lib/Bio/Search/Result/PullResultI.pm blib/lib/Bio/Search/Result/PullResultI.pm\n",
            "cp lib/Bio/SearchIO/SearchWriterI.pm blib/lib/Bio/SearchIO/SearchWriterI.pm\n",
            "cp lib/Bio/SearchIO/Writer/HTMLResultWriter.pm blib/lib/Bio/SearchIO/Writer/HTMLResultWriter.pm\n",
            "cp lib/Bio/SearchIO/Writer/GbrowseGFF.pm blib/lib/Bio/SearchIO/Writer/GbrowseGFF.pm\n",
            "cp lib/Bio/Search/Tiling/TilingI.pm blib/lib/Bio/Search/Tiling/TilingI.pm\n",
            "cp lib/Bio/Search/Result/CrossMatchResult.pm blib/lib/Bio/Search/Result/CrossMatchResult.pm\n",
            "cp lib/Bio/SearchIO/blast_pull.pm blib/lib/Bio/SearchIO/blast_pull.pm\n",
            "cp lib/Bio/Search/StatisticsI.pm blib/lib/Bio/Search/StatisticsI.pm\n",
            "cp lib/Bio/SearchIO/axt.pm blib/lib/Bio/SearchIO/axt.pm\n",
            "cp lib/Bio/SearchIO.pm blib/lib/Bio/SearchIO.pm\n",
            "cp lib/Bio/SearchIO/Writer/ResultTableWriter.pm blib/lib/Bio/SearchIO/Writer/ResultTableWriter.pm\n",
            "cp lib/Bio/SearchIO/Writer/TextResultWriter.pm blib/lib/Bio/SearchIO/Writer/TextResultWriter.pm\n",
            "cp lib/Bio/SearchIO/IteratedSearchResultEventBuilder.pm blib/lib/Bio/SearchIO/IteratedSearchResultEventBuilder.pm\n",
            "cp lib/Bio/Search/Result/WABAResult.pm blib/lib/Bio/Search/Result/WABAResult.pm\n",
            "cp lib/Bio/Search/Result/BlastPullResult.pm blib/lib/Bio/Search/Result/BlastPullResult.pm\n",
            "cp lib/Bio/Search/Result/ResultI.pm blib/lib/Bio/Search/Result/ResultI.pm\n",
            "cp lib/Bio/SearchIO/SearchResultEventBuilder.pm blib/lib/Bio/SearchIO/SearchResultEventBuilder.pm\n",
            "cp lib/Bio/SearchIO/blasttable.pm blib/lib/Bio/SearchIO/blasttable.pm\n",
            "cp lib/Bio/Search/Result/BlastResult.pm blib/lib/Bio/Search/Result/BlastResult.pm\n",
            "cp lib/Bio/SearchIO/Writer/HSPTableWriter.pm blib/lib/Bio/SearchIO/Writer/HSPTableWriter.pm\n",
            "cp lib/Bio/Search/Result/INFERNALResult.pm blib/lib/Bio/Search/Result/INFERNALResult.pm\n",
            "cp lib/Bio/SearchIO/blast.pm blib/lib/Bio/SearchIO/blast.pm\n",
            "cp lib/Bio/SearchIO/Writer/HitTableWriter.pm blib/lib/Bio/SearchIO/Writer/HitTableWriter.pm\n",
            "cp lib/Bio/Search/SearchUtils.pm blib/lib/Bio/Search/SearchUtils.pm\n",
            "cp lib/Bio/SearchIO/FastHitEventBuilder.pm blib/lib/Bio/SearchIO/FastHitEventBuilder.pm\n",
            "cp lib/Bio/SearchIO/cross_match.pm blib/lib/Bio/SearchIO/cross_match.pm\n",
            "cp lib/Bio/Search/Tiling/MapTiling.pm blib/lib/Bio/Search/Tiling/MapTiling.pm\n",
            "cp lib/Bio/SearchIO/erpin.pm blib/lib/Bio/SearchIO/erpin.pm\n",
            "cp lib/Bio/Search/Tiling/MapTileUtils.pm blib/lib/Bio/Search/Tiling/MapTileUtils.pm\n",
            "cp lib/Bio/Seq/PrimedSeq.pm blib/lib/Bio/Seq/PrimedSeq.pm\n",
            "cp lib/Bio/Seq.pm blib/lib/Bio/Seq.pm\n",
            "cp lib/Bio/SeqFeature/FeaturePair.pm blib/lib/Bio/SeqFeature/FeaturePair.pm\n",
            "cp lib/Bio/SearchIO/wise.pm blib/lib/Bio/SearchIO/wise.pm\n",
            "cp lib/Bio/Seq/SequenceTrace.pm blib/lib/Bio/Seq/SequenceTrace.pm\n",
            "cp lib/Bio/SearchIO/psl.pm blib/lib/Bio/SearchIO/psl.pm\n",
            "cp lib/Bio/SearchIO/waba.pm blib/lib/Bio/SearchIO/waba.pm\n",
            "cp lib/Bio/Seq/RichSeq.pm blib/lib/Bio/Seq/RichSeq.pm\n",
            "cp lib/Bio/SeqFeature/CollectionI.pm blib/lib/Bio/SeqFeature/CollectionI.pm\n",
            "cp lib/Bio/SearchIO/sim4.pm blib/lib/Bio/SearchIO/sim4.pm\n",
            "cp lib/Bio/SearchIO/rnamotif.pm blib/lib/Bio/SearchIO/rnamotif.pm\n",
            "cp lib/Bio/Seq/SeqFastaSpeedFactory.pm blib/lib/Bio/Seq/SeqFastaSpeedFactory.pm\n",
            "cp lib/Bio/Seq/MetaI.pm blib/lib/Bio/Seq/MetaI.pm\n",
            "cp lib/Bio/SearchIO/fasta.pm blib/lib/Bio/SearchIO/fasta.pm\n",
            "cp lib/Bio/Seq/Meta.pm blib/lib/Bio/Seq/Meta.pm\n",
            "cp lib/Bio/Seq/QualI.pm blib/lib/Bio/Seq/QualI.pm\n",
            "cp lib/Bio/SearchIO/megablast.pm blib/lib/Bio/SearchIO/megablast.pm\n",
            "cp lib/Bio/Seq/EncodedSeq.pm blib/lib/Bio/Seq/EncodedSeq.pm\n",
            "cp lib/Bio/Seq/PrimaryQual.pm blib/lib/Bio/Seq/PrimaryQual.pm\n",
            "cp lib/Bio/Seq/Meta/Array.pm blib/lib/Bio/Seq/Meta/Array.pm\n",
            "cp lib/Bio/Seq/LargeSeq.pm blib/lib/Bio/Seq/LargeSeq.pm\n",
            "cp lib/Bio/Seq/SimulatedRead.pm blib/lib/Bio/Seq/SimulatedRead.pm\n",
            "cp lib/Bio/Seq/RichSeqI.pm blib/lib/Bio/Seq/RichSeqI.pm\n",
            "cp lib/Bio/SearchIO/infernal.pm blib/lib/Bio/SearchIO/infernal.pm\n",
            "cp lib/Bio/SeqAnalysisParserI.pm blib/lib/Bio/SeqAnalysisParserI.pm\n",
            "cp lib/Bio/Seq/BaseSeqProcessor.pm blib/lib/Bio/Seq/BaseSeqProcessor.pm\n",
            "cp lib/Bio/Seq/LargeLocatableSeq.pm blib/lib/Bio/Seq/LargeLocatableSeq.pm\n",
            "cp lib/Bio/SeqFeature/AnnotationAdaptor.pm blib/lib/Bio/SeqFeature/AnnotationAdaptor.pm\n",
            "cp lib/Bio/SeqFeature/Gene/Exon.pm blib/lib/Bio/SeqFeature/Gene/Exon.pm\n",
            "cp lib/Bio/SeqFeature/Amplicon.pm blib/lib/Bio/SeqFeature/Amplicon.pm\n",
            "cp lib/Bio/Seq/LargePrimarySeq.pm blib/lib/Bio/Seq/LargePrimarySeq.pm\n",
            "cp lib/Bio/Seq/SeqFactory.pm blib/lib/Bio/Seq/SeqFactory.pm\n",
            "cp lib/Bio/Seq/Quality.pm blib/lib/Bio/Seq/Quality.pm\n",
            "cp lib/Bio/SearchIO/gmap_f9.pm blib/lib/Bio/SearchIO/gmap_f9.pm\n",
            "cp lib/Bio/SeqFeature/Computation.pm blib/lib/Bio/SeqFeature/Computation.pm\n",
            "cp lib/Bio/SeqFeature/Collection.pm blib/lib/Bio/SeqFeature/Collection.pm\n",
            "cp lib/Bio/Seq/TraceI.pm blib/lib/Bio/Seq/TraceI.pm\n",
            "cp lib/Bio/Seq/SeqBuilder.pm blib/lib/Bio/Seq/SeqBuilder.pm\n",
            "cp lib/Bio/SearchIO/exonerate.pm blib/lib/Bio/SearchIO/exonerate.pm\n",
            "cp lib/Bio/Seq/LargeSeqI.pm blib/lib/Bio/Seq/LargeSeqI.pm\n",
            "cp lib/Bio/SeqIO/fastq.pm blib/lib/Bio/SeqIO/fastq.pm\n",
            "cp lib/Bio/SeqFeature/Gene/ExonI.pm blib/lib/Bio/SeqFeature/Gene/ExonI.pm\n",
            "cp lib/Bio/SeqIO/asciitree.pm blib/lib/Bio/SeqIO/asciitree.pm\n",
            "cp lib/Bio/SeqFeature/Tools/FeatureNamer.pm blib/lib/Bio/SeqFeature/Tools/FeatureNamer.pm\n",
            "cp lib/Bio/SeqFeature/Tools/Unflattener.pm blib/lib/Bio/SeqFeature/Tools/Unflattener.pm\n",
            "cp lib/Bio/SeqFeature/Gene/GeneStructureI.pm blib/lib/Bio/SeqFeature/Gene/GeneStructureI.pm\n",
            "cp lib/Bio/SeqFeatureI.pm blib/lib/Bio/SeqFeatureI.pm\n",
            "cp lib/Bio/SeqFeature/Gene/Transcript.pm blib/lib/Bio/SeqFeature/Gene/Transcript.pm\n",
            "cp lib/Bio/SeqFeature/Gene/TranscriptI.pm blib/lib/Bio/SeqFeature/Gene/TranscriptI.pm\n",
            "cp lib/Bio/SeqFeature/Lite.pm blib/lib/Bio/SeqFeature/Lite.pm\n",
            "cp lib/Bio/SeqFeature/Gene/GeneStructure.pm blib/lib/Bio/SeqFeature/Gene/GeneStructure.pm\n",
            "cp lib/Bio/SeqFeature/SubSeq.pm blib/lib/Bio/SeqFeature/SubSeq.pm\n",
            "cp lib/Bio/SeqFeature/PositionProxy.pm blib/lib/Bio/SeqFeature/PositionProxy.pm\n",
            "cp lib/Bio/SeqFeature/Gene/Promoter.pm blib/lib/Bio/SeqFeature/Gene/Promoter.pm\n",
            "cp lib/Bio/SeqFeature/Gene/Intron.pm blib/lib/Bio/SeqFeature/Gene/Intron.pm\n",
            "cp lib/Bio/SeqIO/fasta.pm blib/lib/Bio/SeqIO/fasta.pm\n",
            "cp lib/Bio/SeqFeature/Gene/NC_Feature.pm blib/lib/Bio/SeqFeature/Gene/NC_Feature.pm\n",
            "cp lib/Bio/SeqFeature/Primer.pm blib/lib/Bio/SeqFeature/Primer.pm\n",
            "cp lib/Bio/SeqFeature/Tools/IDHandler.pm blib/lib/Bio/SeqFeature/Tools/IDHandler.pm\n",
            "cp lib/Bio/SeqIO/bsml_sax.pm blib/lib/Bio/SeqIO/bsml_sax.pm\n",
            "cp lib/Bio/SeqIO.pm blib/lib/Bio/SeqIO.pm\n",
            "cp lib/Bio/SeqFeature/Similarity.pm blib/lib/Bio/SeqFeature/Similarity.pm\n",
            "cp lib/Bio/SeqIO/MultiFile.pm blib/lib/Bio/SeqIO/MultiFile.pm\n",
            "cp lib/Bio/SeqIO/embldriver.pm blib/lib/Bio/SeqIO/embldriver.pm\n",
            "cp lib/Bio/SeqIO/embl.pm blib/lib/Bio/SeqIO/embl.pm\n",
            "cp lib/Bio/SeqIO/bsml.pm blib/lib/Bio/SeqIO/bsml.pm\n",
            "cp lib/Bio/SeqFeature/Gene/UTR.pm blib/lib/Bio/SeqFeature/Gene/UTR.pm\n",
            "cp lib/Bio/SeqIO/FTHelper.pm blib/lib/Bio/SeqIO/FTHelper.pm\n",
            "cp lib/Bio/SeqFeature/Gene/Poly_A_site.pm blib/lib/Bio/SeqFeature/Gene/Poly_A_site.pm\n",
            "cp lib/Bio/SeqFeature/Tools/TypeMapper.pm blib/lib/Bio/SeqFeature/Tools/TypeMapper.pm\n",
            "cp lib/Bio/SeqFeature/Generic.pm blib/lib/Bio/SeqFeature/Generic.pm\n",
            "cp lib/Bio/SeqIO/ace.pm blib/lib/Bio/SeqIO/ace.pm\n",
            "cp lib/Bio/SeqFeature/TypedSeqFeatureI.pm blib/lib/Bio/SeqFeature/TypedSeqFeatureI.pm\n",
            "cp lib/Bio/SeqIO/Handler/GenericRichSeqHandler.pm blib/lib/Bio/SeqIO/Handler/GenericRichSeqHandler.pm\n",
            "cp lib/Bio/SeqFeature/SimilarityPair.pm blib/lib/Bio/SeqFeature/SimilarityPair.pm\n",
            "cp lib/Bio/SeqI.pm blib/lib/Bio/SeqI.pm\n",
            "cp lib/Bio/Tools/Alignment/Consed.pm blib/lib/Bio/Tools/Alignment/Consed.pm\n",
            "cp lib/Bio/SeqIO/raw.pm blib/lib/Bio/SeqIO/raw.pm\n",
            "cp lib/Bio/SeqIO/tigr.pm blib/lib/Bio/SeqIO/tigr.pm\n",
            "cp lib/Bio/SeqIO/qual.pm blib/lib/Bio/SeqIO/qual.pm\n",
            "cp lib/Bio/SeqIO/game/seqHandler.pm blib/lib/Bio/SeqIO/game/seqHandler.pm\n",
            "cp lib/Bio/SeqIO/msout.pm blib/lib/Bio/SeqIO/msout.pm\n",
            "cp lib/Bio/Tools/AnalysisResult.pm blib/lib/Bio/Tools/AnalysisResult.pm\n",
            "cp lib/Bio/SeqIO/locuslink.pm blib/lib/Bio/SeqIO/locuslink.pm\n",
            "cp lib/Bio/SeqIO/game/featHandler.pm blib/lib/Bio/SeqIO/game/featHandler.pm\n",
            "cp lib/Bio/SeqIO/swiss.pm blib/lib/Bio/SeqIO/swiss.pm\n",
            "cp lib/Bio/SeqIO/gbxml.pm blib/lib/Bio/SeqIO/gbxml.pm\n",
            "cp lib/Bio/SeqIO/swissdriver.pm blib/lib/Bio/SeqIO/swissdriver.pm\n",
            "cp lib/Bio/SeqIO/tinyseq.pm blib/lib/Bio/SeqIO/tinyseq.pm\n",
            "cp lib/Bio/SeqIO/scf.pm blib/lib/Bio/SeqIO/scf.pm\n",
            "cp lib/Bio/SeqIO/game/gameWriter.pm blib/lib/Bio/SeqIO/game/gameWriter.pm\n",
            "cp lib/Bio/SeqIO/tab.pm blib/lib/Bio/SeqIO/tab.pm\n",
            "cp lib/Bio/SeqIO/mbsout.pm blib/lib/Bio/SeqIO/mbsout.pm\n",
            "cp lib/Bio/SeqIO/gcg.pm blib/lib/Bio/SeqIO/gcg.pm\n",
            "cp lib/Bio/Tools/Blat.pm blib/lib/Bio/Tools/Blat.pm\n",
            "cp lib/Bio/Tools/Analysis/SimpleAnalysisBase.pm blib/lib/Bio/Tools/Analysis/SimpleAnalysisBase.pm\n",
            "cp lib/Bio/SeqIO/table.pm blib/lib/Bio/SeqIO/table.pm\n",
            "cp lib/Bio/SeqIO/phd.pm blib/lib/Bio/SeqIO/phd.pm\n",
            "cp lib/Bio/SeqIO/largefasta.pm blib/lib/Bio/SeqIO/largefasta.pm\n",
            "cp lib/Bio/SeqIO/seqxml.pm blib/lib/Bio/SeqIO/seqxml.pm\n",
            "cp lib/Bio/SimpleAnalysisI.pm blib/lib/Bio/SimpleAnalysisI.pm\n",
            "cp lib/Bio/SeqIO/pir.pm blib/lib/Bio/SeqIO/pir.pm\n",
            "cp lib/Bio/SeqIO/genbank.pm blib/lib/Bio/SeqIO/genbank.pm\n",
            "cp lib/Bio/SeqIO/game/gameHandler.pm blib/lib/Bio/SeqIO/game/gameHandler.pm\n",
            "cp lib/Bio/SeqIO/kegg.pm blib/lib/Bio/SeqIO/kegg.pm\n",
            "cp lib/Bio/SimpleAlign.pm blib/lib/Bio/SimpleAlign.pm\n",
            "cp lib/Bio/SeqIO/tigrxml.pm blib/lib/Bio/SeqIO/tigrxml.pm\n",
            "cp lib/Bio/Species.pm blib/lib/Bio/Species.pm\n",
            "cp lib/Bio/Taxon.pm blib/lib/Bio/Taxon.pm\n",
            "cp lib/Bio/SeqIO/game.pm blib/lib/Bio/SeqIO/game.pm\n",
            "cp lib/Bio/SeqIO/tinyseq/tinyseqHandler.pm blib/lib/Bio/SeqIO/tinyseq/tinyseqHandler.pm\n",
            "cp lib/Bio/SeqIO/game/gameSubs.pm blib/lib/Bio/SeqIO/game/gameSubs.pm\n",
            "cp lib/Bio/SeqIO/metafasta.pm blib/lib/Bio/SeqIO/metafasta.pm\n",
            "cp lib/Bio/Tools/AmpliconSearch.pm blib/lib/Bio/Tools/AmpliconSearch.pm\n",
            "cp lib/Bio/SeqUtils.pm blib/lib/Bio/SeqUtils.pm\n",
            "cp lib/Bio/Tools/Alignment/Trim.pm blib/lib/Bio/Tools/Alignment/Trim.pm\n",
            "cp lib/Bio/SeqIO/gbdriver.pm blib/lib/Bio/SeqIO/gbdriver.pm\n",
            "cp lib/Bio/Tools/Coil.pm blib/lib/Bio/Tools/Coil.pm\n",
            "cp lib/Bio/Tools/Lucy.pm blib/lib/Bio/Tools/Lucy.pm\n",
            "cp lib/Bio/Tools/Promoterwise.pm blib/lib/Bio/Tools/Promoterwise.pm\n",
            "cp lib/Bio/Tools/Profile.pm blib/lib/Bio/Tools/Profile.pm\n",
            "cp lib/Bio/Tools/Glimmer.pm blib/lib/Bio/Tools/Glimmer.pm\n",
            "cp lib/Bio/Tools/Genewise.pm blib/lib/Bio/Tools/Genewise.pm\n",
            "cp lib/Bio/Tools/Phylo/Phylip/ProtDist.pm blib/lib/Bio/Tools/Phylo/Phylip/ProtDist.pm\n",
            "cp lib/Bio/Tools/GFF.pm blib/lib/Bio/Tools/GFF.pm\n",
            "cp lib/Bio/Tools/Genscan.pm blib/lib/Bio/Tools/Genscan.pm\n",
            "cp lib/Bio/Tools/Primer/Assessor/Base.pm blib/lib/Bio/Tools/Primer/Assessor/Base.pm\n",
            "cp lib/Bio/Tools/Primer3.pm blib/lib/Bio/Tools/Primer3.pm\n",
            "cp lib/Bio/Tools/ESTScan.pm blib/lib/Bio/Tools/ESTScan.pm\n",
            "cp lib/Bio/Tools/Primer/AssessorI.pm blib/lib/Bio/Tools/Primer/AssessorI.pm\n",
            "cp lib/Bio/Tools/Prediction/Exon.pm blib/lib/Bio/Tools/Prediction/Exon.pm\n",
            "cp lib/Bio/Tools/Genemark.pm blib/lib/Bio/Tools/Genemark.pm\n",
            "cp lib/Bio/Tools/Phylo/Molphy/Result.pm blib/lib/Bio/Tools/Phylo/Molphy/Result.pm\n",
            "cp lib/Bio/Tools/Est2Genome.pm blib/lib/Bio/Tools/Est2Genome.pm\n",
            "cp lib/Bio/Tools/EMBOSS/Palindrome.pm blib/lib/Bio/Tools/EMBOSS/Palindrome.pm\n",
            "cp lib/Bio/Tools/IUPAC.pm blib/lib/Bio/Tools/IUPAC.pm\n",
            "cp lib/Bio/Tools/ECnumber.pm blib/lib/Bio/Tools/ECnumber.pm\n",
            "cp lib/Bio/Tools/MZEF.pm blib/lib/Bio/Tools/MZEF.pm\n",
            "cp lib/Bio/Tools/Pseudowise.pm blib/lib/Bio/Tools/Pseudowise.pm\n",
            "cp lib/Bio/Tools/CodonTable.pm blib/lib/Bio/Tools/CodonTable.pm\n",
            "cp lib/Bio/Tools/Eponine.pm blib/lib/Bio/Tools/Eponine.pm\n",
            "cp lib/Bio/Tools/Match.pm blib/lib/Bio/Tools/Match.pm\n",
            "cp lib/Bio/Tools/OddCodes.pm blib/lib/Bio/Tools/OddCodes.pm\n",
            "cp lib/Bio/Tools/Genomewise.pm blib/lib/Bio/Tools/Genomewise.pm\n",
            "cp lib/Bio/Tools/Primer/Pair.pm blib/lib/Bio/Tools/Primer/Pair.pm\n",
            "cp lib/Bio/Tools/FootPrinter.pm blib/lib/Bio/Tools/FootPrinter.pm\n",
            "cp lib/Bio/Tools/EPCR.pm blib/lib/Bio/Tools/EPCR.pm\n",
            "cp lib/Bio/Tools/Grail.pm blib/lib/Bio/Tools/Grail.pm\n",
            "cp lib/Bio/Tools/Primer/Feature.pm blib/lib/Bio/Tools/Primer/Feature.pm\n",
            "cp lib/Bio/Tools/Prints.pm blib/lib/Bio/Tools/Prints.pm\n",
            "cp lib/Bio/Tools/Fgenesh.pm blib/lib/Bio/Tools/Fgenesh.pm\n",
            "cp lib/Bio/Tools/Phylo/Gerp.pm blib/lib/Bio/Tools/Phylo/Gerp.pm\n",
            "cp lib/Bio/Tools/Prediction/Gene.pm blib/lib/Bio/Tools/Prediction/Gene.pm\n",
            "cp lib/Bio/Tools/Phylo/Molphy.pm blib/lib/Bio/Tools/Phylo/Molphy.pm\n",
            "cp lib/Bio/Tools/PrositeScan.pm blib/lib/Bio/Tools/PrositeScan.pm\n",
            "cp lib/Bio/Tools/Geneid.pm blib/lib/Bio/Tools/Geneid.pm\n",
            "cp lib/Bio/Tools/GuessSeqFormat.pm blib/lib/Bio/Tools/GuessSeqFormat.pm\n",
            "cp lib/Bio/Tools/Sim4/Exon.pm blib/lib/Bio/Tools/Sim4/Exon.pm\n",
            "cp lib/Bio/Tools/SeqPattern/Backtranslate.pm blib/lib/Bio/Tools/SeqPattern/Backtranslate.pm\n",
            "cp lib/Bio/Tools/Sigcleave.pm blib/lib/Bio/Tools/Sigcleave.pm\n",
            "cp lib/Bio/Tools/tRNAscanSE.pm blib/lib/Bio/Tools/tRNAscanSE.pm\n",
            "cp lib/Bio/Tools/TargetP.pm blib/lib/Bio/Tools/TargetP.pm\n",
            "cp lib/Bio/Tree/RandomFactory.pm blib/lib/Bio/Tree/RandomFactory.pm\n",
            "cp lib/Bio/Tools/Run/Analysis.pm blib/lib/Bio/Tools/Run/Analysis.pm\n",
            "cp lib/Bio/Tools/Signalp/ExtendedSignalp.pm blib/lib/Bio/Tools/Signalp/ExtendedSignalp.pm\n",
            "cp lib/Bio/Tree/NodeI.pm blib/lib/Bio/Tree/NodeI.pm\n",
            "cp lib/Bio/Tools/Tmhmm.pm blib/lib/Bio/Tools/Tmhmm.pm\n",
            "cp lib/Bio/Tools/RandomDistFunctions.pm blib/lib/Bio/Tools/RandomDistFunctions.pm\n",
            "cp lib/Bio/Tree/DistanceFactory.pm blib/lib/Bio/Tree/DistanceFactory.pm\n",
            "cp lib/Bio/Tools/Signalp.pm blib/lib/Bio/Tools/Signalp.pm\n",
            "cp lib/Bio/Tools/Spidey/Exon.pm blib/lib/Bio/Tools/Spidey/Exon.pm\n",
            "cp lib/Bio/Tools/Run/AnalysisFactory.pm blib/lib/Bio/Tools/Run/AnalysisFactory.pm\n",
            "cp lib/Bio/Tree/NodeNHX.pm blib/lib/Bio/Tree/NodeNHX.pm\n",
            "cp lib/Bio/Tree/Compatible.pm blib/lib/Bio/Tree/Compatible.pm\n",
            "cp lib/Bio/Tools/Run/WrapperBase.pm blib/lib/Bio/Tools/Run/WrapperBase.pm\n",
            "cp lib/Bio/Tools/isPcr.pm blib/lib/Bio/Tools/isPcr.pm\n",
            "cp lib/Bio/Tools/Run/WrapperBase/CommandExts.pm blib/lib/Bio/Tools/Run/WrapperBase/CommandExts.pm\n",
            "cp lib/Bio/Tools/SeqWords.pm blib/lib/Bio/Tools/SeqWords.pm\n",
            "cp lib/Bio/Tools/Run/ParametersI.pm blib/lib/Bio/Tools/Run/ParametersI.pm\n",
            "cp lib/Bio/Tools/Run/Phylo/PhyloBase.pm blib/lib/Bio/Tools/Run/Phylo/PhyloBase.pm\n",
            "cp lib/Bio/Tools/pICalculator.pm blib/lib/Bio/Tools/pICalculator.pm\n",
            "cp lib/Bio/Tools/TandemRepeatsFinder.pm blib/lib/Bio/Tools/TandemRepeatsFinder.pm\n",
            "cp lib/Bio/Tools/Run/GenericParameters.pm blib/lib/Bio/Tools/Run/GenericParameters.pm\n",
            "cp lib/Bio/Tree/Statistics.pm blib/lib/Bio/Tree/Statistics.pm\n",
            "cp lib/Bio/Tools/SeqStats.pm blib/lib/Bio/Tools/SeqStats.pm\n",
            "cp lib/Bio/Tree/Node.pm blib/lib/Bio/Tree/Node.pm\n",
            "cp lib/Bio/Tools/ipcress.pm blib/lib/Bio/Tools/ipcress.pm\n",
            "cp lib/Bio/Tree/AnnotatableNode.pm blib/lib/Bio/Tree/AnnotatableNode.pm\n",
            "cp lib/Bio/Tree/Tree.pm blib/lib/Bio/Tree/Tree.pm\n",
            "cp lib/Bio/Tools/Sim4/Results.pm blib/lib/Bio/Tools/Sim4/Results.pm\n",
            "cp lib/Bio/Tools/QRNA.pm blib/lib/Bio/Tools/QRNA.pm\n",
            "cp lib/Bio/Tools/RepeatMasker.pm blib/lib/Bio/Tools/RepeatMasker.pm\n",
            "cp lib/Bio/Tools/Spidey/Results.pm blib/lib/Bio/Tools/Spidey/Results.pm\n",
            "cp lib/Bio/Tools/Seg.pm blib/lib/Bio/Tools/Seg.pm\n",
            "cp lib/Bio/Tools/SeqPattern.pm blib/lib/Bio/Tools/SeqPattern.pm\n",
            "cp lib/Bio/Tree/TreeI.pm blib/lib/Bio/Tree/TreeI.pm\n",
            "cp lib/Bio/TreeIO/pag.pm blib/lib/Bio/TreeIO/pag.pm\n",
            "cp lib/Bio/TreeIO/tabtree.pm blib/lib/Bio/TreeIO/tabtree.pm\n",
            "cp lib/Bio/TreeIO/newick.pm blib/lib/Bio/TreeIO/newick.pm\n",
            "cp lib/Bio/TreeIO/nexus.pm blib/lib/Bio/TreeIO/nexus.pm\n",
            "cp lib/Bio/TreeIO/TreeEventBuilder.pm blib/lib/Bio/TreeIO/TreeEventBuilder.pm\n",
            "cp lib/Bio/TreeIO/cluster.pm blib/lib/Bio/TreeIO/cluster.pm\n",
            "cp lib/BioPerl.pm blib/lib/BioPerl.pm\n",
            "cp lib/Bio/UpdateableSeqI.pm blib/lib/Bio/UpdateableSeqI.pm\n",
            "cp lib/Bio/TreeIO/NewickParser.pm blib/lib/Bio/TreeIO/NewickParser.pm\n",
            "cp lib/Bio/TreeIO/phyloxml.pm blib/lib/Bio/TreeIO/phyloxml.pm\n",
            "cp lib/Bio/TreeIO.pm blib/lib/Bio/TreeIO.pm\n",
            "cp lib/Bio/WebAgent.pm blib/lib/Bio/WebAgent.pm\n",
            "cp lib/Bio/TreeIO/nhx.pm blib/lib/Bio/TreeIO/nhx.pm\n",
            "cp lib/Bio/TreeIO/lintree.pm blib/lib/Bio/TreeIO/lintree.pm\n",
            "cp lib/Bio/Tree/TreeFunctionsI.pm blib/lib/Bio/Tree/TreeFunctionsI.pm\n",
            "cp bin/bp_aacomp blib/script/bp_aacomp\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_aacomp\n",
            "cp bin/bp_bioflat_index blib/script/bp_bioflat_index\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_bioflat_index\n",
            "cp bin/bp_biogetseq blib/script/bp_biogetseq\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_biogetseq\n",
            "cp bin/bp_dbsplit blib/script/bp_dbsplit\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_dbsplit\n",
            "cp bin/bp_extract_feature_seq blib/script/bp_extract_feature_seq\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_extract_feature_seq\n",
            "cp bin/bp_fastam9_to_table blib/script/bp_fastam9_to_table\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_fastam9_to_table\n",
            "cp bin/bp_fetch blib/script/bp_fetch\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_fetch\n",
            "cp bin/bp_filter_search blib/script/bp_filter_search\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_filter_search\n",
            "cp bin/bp_find-blast-matches blib/script/bp_find-blast-matches\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_find-blast-matches\n",
            "cp bin/bp_gccalc blib/script/bp_gccalc\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_gccalc\n",
            "cp bin/bp_genbank2gff3 blib/script/bp_genbank2gff3\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_genbank2gff3\n",
            "cp bin/bp_index blib/script/bp_index\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_index\n",
            "cp bin/bp_local_taxonomydb_query blib/script/bp_local_taxonomydb_query\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_local_taxonomydb_query\n",
            "cp bin/bp_make_mrna_protein blib/script/bp_make_mrna_protein\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_make_mrna_protein\n",
            "cp bin/bp_mask_by_search blib/script/bp_mask_by_search\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_mask_by_search\n",
            "cp bin/bp_mrtrans blib/script/bp_mrtrans\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_mrtrans\n",
            "cp bin/bp_mutate blib/script/bp_mutate\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_mutate\n",
            "cp bin/bp_nexus2nh blib/script/bp_nexus2nh\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_nexus2nh\n",
            "cp bin/bp_nrdb blib/script/bp_nrdb\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_nrdb\n",
            "cp bin/bp_oligo_count blib/script/bp_oligo_count\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_oligo_count\n",
            "cp bin/bp_process_gadfly blib/script/bp_process_gadfly\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_process_gadfly\n",
            "cp bin/bp_process_sgd blib/script/bp_process_sgd\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_process_sgd\n",
            "cp bin/bp_revtrans-motif blib/script/bp_revtrans-motif\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_revtrans-motif\n",
            "cp bin/bp_search2alnblocks blib/script/bp_search2alnblocks\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_search2alnblocks\n",
            "cp bin/bp_search2gff blib/script/bp_search2gff\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_search2gff\n",
            "cp bin/bp_search2table blib/script/bp_search2table\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_search2table\n",
            "cp bin/bp_search2tribe blib/script/bp_search2tribe\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_search2tribe\n",
            "cp bin/bp_seq_length blib/script/bp_seq_length\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_seq_length\n",
            "cp bin/bp_seqconvert blib/script/bp_seqconvert\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_seqconvert\n",
            "cp bin/bp_seqcut blib/script/bp_seqcut\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_seqcut\n",
            "cp bin/bp_seqpart blib/script/bp_seqpart\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_seqpart\n",
            "cp bin/bp_seqret blib/script/bp_seqret\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_seqret\n",
            "cp bin/bp_seqretsplit blib/script/bp_seqretsplit\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_seqretsplit\n",
            "cp bin/bp_split_seq blib/script/bp_split_seq\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_split_seq\n",
            "cp bin/bp_sreformat blib/script/bp_sreformat\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_sreformat\n",
            "cp bin/bp_taxid4species blib/script/bp_taxid4species\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_taxid4species\n",
            "cp bin/bp_taxonomy2tree blib/script/bp_taxonomy2tree\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_taxonomy2tree\n",
            "cp bin/bp_translate_seq blib/script/bp_translate_seq\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_translate_seq\n",
            "cp bin/bp_tree2pag blib/script/bp_tree2pag\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_tree2pag\n",
            "cp bin/bp_unflatten_seq blib/script/bp_unflatten_seq\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_unflatten_seq\n",
            "Manifying 40 pod documents\n",
            "Manifying 35 pod documents\n",
            "Manifying 33 pod documents\n",
            "Manifying 32 pod documents\n",
            "Manifying 34 pod documents\n",
            "Manifying 32 pod documents\n",
            "Manifying 32 pod documents\n",
            "Manifying 30 pod documents\n",
            "Manifying 27 pod documents\n",
            "Manifying 35 pod documents\n",
            "Manifying 29 pod documents\n",
            "Manifying 35 pod documents\n",
            "Manifying 36 pod documents\n",
            "Manifying 33 pod documents\n",
            "Manifying 33 pod documents\n",
            "Manifying 15 pod documents\n",
            "  CJFIELDS/BioPerl-1.7.8.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for CJFIELDS/BioPerl-1.7.8.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t t/Align/*.t t/AlignIO/*.t t/Annotation/*.t t/LocalDB/*.t t/LocalDB/Index/*.t t/LocalDB/Taxonomy/*.t t/Matrix/*.t t/Matrix/IO/*.t t/Ontology/*.t t/Ontology/IO/*.t t/RemoteDB/*.t t/Root/*.t t/SearchIO/*.t t/SearchIO/Writer/*.t t/Seq/*.t t/SeqFeature/*.t t/SeqIO/*.t t/SeqTools/*.t t/Tools/*.t t/Tools/Alignment/*.t t/Tools/EMBOSS/*.t t/Tools/Phylo/*.t t/Tools/Phylo/Phylip/*.t t/Tools/Signalp/*.t t/Tools/Spidey/*.t t/Tree/*.t t/Tree/TreeIO/*.t\n",
            "t/00-compile.t ...................... ok\n",
            "t/Align/AlignStats.t ................ ok\n",
            "t/Align/AlignUtil.t ................. ok\n",
            "t/Align/SimpleAlign.t ............... ok\n",
            "t/Align/TreeBuild.t ................. ok\n",
            "t/Align/Utilities.t ................. ok\n",
            "t/AlignIO/AlignIO.t ................. ok\n",
            "t/AlignIO/arp.t ..................... ok\n",
            "t/AlignIO/bl2seq.t .................. ok\n",
            "t/AlignIO/clustalw.t ................ ok\n",
            "t/AlignIO/emboss.t .................. ok\n",
            "t/AlignIO/fasta.t ................... ok\n",
            "t/AlignIO/largemultifasta.t ......... ok\n",
            "t/AlignIO/maf.t ..................... ok\n",
            "t/AlignIO/mase.t .................... ok\n",
            "t/AlignIO/mega.t .................... ok\n",
            "t/AlignIO/meme.t .................... ok\n",
            "t/AlignIO/metafasta.t ............... ok\n",
            "t/AlignIO/msf.t ..................... ok\n",
            "t/AlignIO/nexus.t ................... ok\n",
            "t/AlignIO/pfam.t .................... ok\n",
            "t/AlignIO/phylip.t .................. ok\n",
            "t/AlignIO/po.t ...................... ok\n",
            "t/AlignIO/prodom.t .................. ok\n",
            "t/AlignIO/psi.t ..................... ok\n",
            "t/AlignIO/selex.t ................... ok\n",
            "t/AlignIO/xmfa.t .................... ok\n",
            "t/Annotation/Annotation.t ........... ok\n",
            "t/Annotation/AnnotationAdaptor.t .... ok\n",
            "t/author-mojibake.t ................. skipped: these tests are for testing by the author\n",
            "t/author-pod-syntax.t ............... skipped: these tests are for testing by the author\n",
            "t/LocalDB/Fasta.t ................... ok\n",
            "t/LocalDB/Flat.t .................... ok\n",
            "t/LocalDB/Index/Blast.t ............. ok\n",
            "t/LocalDB/Index/BlastTable.t ........ ok\n",
            "t/LocalDB/Index/Index.t ............. ok\n",
            "t/LocalDB/Qual.t .................... ok\n",
            "t/LocalDB/Registry.t ................ 1/14 Use of uninitialized value $proto in socket at /usr/local/share/perl/5.30.0/Test/RequiresInternet.pm line 55.\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: Couldn't load Bio::DB::BioFetch\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: Couldn't load Bio::DB::BioFetch\n",
            "---------------------------------------------------\n",
            "t/LocalDB/Registry.t ................ ok\n",
            "t/LocalDB/Taxonomy/greengenes.t ..... ok\n",
            "t/LocalDB/Taxonomy/silva.t .......... ok\n",
            "t/Matrix/InstanceSite.t ............. ok\n",
            "t/Matrix/IO/masta.t ................. ok\n",
            "t/Matrix/IO/psm.t ................... ok\n",
            "t/Matrix/Matrix.t ................... ok\n",
            "t/Matrix/ProtMatrix.t ............... ok\n",
            "t/Matrix/ProtPsm.t .................. ok\n",
            "t/Matrix/SiteMatrix.t ............... ok\n",
            "t/Ontology/GOterm.t ................. ok\n",
            "t/Ontology/GraphAdaptor.t ........... ok\n",
            "t/Ontology/IO/go.t .................. ok\n",
            "t/Ontology/IO/interpro.t ............ ok\n",
            "t/Ontology/IO/obo.t ................. ok\n",
            "t/Ontology/Ontology.t ............... ok\n",
            "t/Ontology/OntologyEngine.t ......... ok\n",
            "t/Ontology/OntologyStore.t .......... Use of uninitialized value $proto in socket at /usr/local/share/perl/5.30.0/Test/RequiresInternet.pm line 55.\n",
            "t/Ontology/OntologyStore.t .......... ok\n",
            "t/Ontology/Relationship.t ........... ok\n",
            "t/Ontology/RelationshipType.t ....... ok\n",
            "t/Ontology/Term.t ................... ok\n",
            "t/RemoteDB/SeqRead_fail.t ........... Use of uninitialized value $proto in socket at /usr/local/share/perl/5.30.0/Test/RequiresInternet.pm line 55.\n",
            "t/RemoteDB/SeqRead_fail.t ........... ok\n",
            "t/RemoteDB/Taxonomy.t ............... ok\n",
            "t/Root/Exception.t .................. ok\n",
            "t/Root/HTTPget.t .................... Use of uninitialized value $proto in socket at /usr/local/share/perl/5.30.0/Test/RequiresInternet.pm line 55.\n",
            "t/Root/HTTPget.t .................... ok\n",
            "t/Root/IO.t ......................... ok\n",
            "t/Root/RootI.t ...................... ok\n",
            "t/Root/RootIO.t ..................... Use of uninitialized value $proto in socket at /usr/local/share/perl/5.30.0/Test/RequiresInternet.pm line 55.\n",
            "t/Root/RootIO.t ..................... ok\n",
            "t/Root/Storable.t ................... ok\n",
            "t/Root/Utilities.t .................. 1/56 \n",
            "--------------------- WARNING ---------------------\n",
            "MSG: find_exe: Multiple paths to 'gzip' found. Using /usr/bin/gzip.\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: find_exe: Multiple paths to 'gzip' found. Using /usr/bin/gzip.\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: Not owner of file t/data/test.txt. Compressing to temp file /tmp/ao1kO4O8Th.tmp.bioperl.gz.\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: find_exe: Multiple paths to 'gunzip' found. Using /usr/bin/gunzip.\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: find_exe: Multiple paths to 'gzip' found. Using /usr/bin/gzip.\n",
            "---------------------------------------------------\n",
            "\n",
            "--------------------- WARNING ---------------------\n",
            "MSG: find_exe: Multiple paths to 'gunzip' found. Using /usr/bin/gunzip.\n",
            "---------------------------------------------------\n",
            "t/Root/Utilities.t .................. ok\n",
            "t/SearchIO/axt.t .................... ok\n",
            "t/SearchIO/blast.t .................. ok\n",
            "t/SearchIO/blast_pull.t ............. ok\n",
            "t/SearchIO/blasttable.t ............. ok\n",
            "t/SearchIO/CigarString.t ............ ok\n",
            "t/SearchIO/cross_match.t ............ ok\n",
            "t/SearchIO/erpin.t .................. ok\n",
            "t/SearchIO/exonerate.t .............. ok\n",
            "t/SearchIO/fasta.t .................. ok\n",
            "t/SearchIO/gmap_f9.t ................ ok\n",
            "t/SearchIO/infernal.t ............... ok\n",
            "t/SearchIO/megablast.t .............. ok\n",
            "t/SearchIO/psl.t .................... ok\n",
            "t/SearchIO/rnamotif.t ............... ok\n",
            "t/SearchIO/SearchIO.t ............... ok\n",
            "t/SearchIO/sim4.t ................... ok\n",
            "t/SearchIO/SimilarityPair.t ......... ok\n",
            "t/SearchIO/Tiling.t ................. ok\n",
            "t/SearchIO/waba.t ................... ok\n",
            "t/SearchIO/wise.t ................... ok\n",
            "t/SearchIO/Writer/GbrowseGFF.t ...... ok\n",
            "t/SearchIO/Writer/HitTableWriter.t .. ok\n",
            "t/SearchIO/Writer/HSPTableWriter.t .. ok\n",
            "t/SearchIO/Writer/HTMLWriter.t ...... ok\n",
            "t/SearchIO/Writer/TextWriter.t ...... ok\n",
            "t/Seq/DBLink.t ...................... ok\n",
            "t/Seq/EncodedSeq.t .................. ok\n",
            "t/Seq/LargeLocatableSeq.t ........... ok\n",
            "t/Seq/LargePSeq.t ................... ok\n",
            "t/Seq/LocatableSeq.t ................ ok\n",
            "t/Seq/MetaSeq.t ..................... ok\n",
            "t/Seq/PrimaryQual.t ................. ok\n",
            "t/Seq/PrimarySeq.t .................. ok\n",
            "t/Seq/PrimedSeq.t ................... ok\n",
            "t/Seq/Quality.t ..................... ok\n",
            "t/Seq/Seq.t ......................... ok\n",
            "t/Seq/SimulatedRead.t ............... ok\n",
            "t/SeqFeature/Amplicon.t ............. ok\n",
            "t/SeqFeature/Clone.t ................ ok\n",
            "t/SeqFeature/Collection.t ........... ok\n",
            "t/SeqFeature/Computation.t .......... ok\n",
            "t/SeqFeature/FeaturePair.t .......... ok\n",
            "t/SeqFeature/Gene.t ................. ok\n",
            "t/SeqFeature/Generic.t .............. ok\n",
            "t/SeqFeature/Location.t ............. ok\n",
            "t/SeqFeature/LocationFactory.t ...... ok\n",
            "t/SeqFeature/Primer.t ............... ok\n",
            "t/SeqFeature/Range.t ................ ok\n",
            "t/SeqFeature/RangeI.t ............... ok\n",
            "t/SeqFeature/SeqAnalysisParser.t .... ok\n",
            "t/SeqFeature/SubSeq.t ............... ok\n",
            "t/SeqFeature/Unflattener.t .......... ok\n",
            "t/SeqIO/ace.t ....................... ok\n",
            "t/SeqIO/asciitree.t ................. ok\n",
            "t/SeqIO/bsml.t ...................... ok\n",
            "t/SeqIO/bsml_sax.t .................. ok\n",
            "t/SeqIO/embl.t ...................... ok\n",
            "t/SeqIO/fasta.t ..................... ok\n",
            "t/SeqIO/fastq.t ..................... ok\n",
            "t/SeqIO/game.t ...................... ok\n",
            "t/SeqIO/gbxml.t ..................... ok\n",
            "t/SeqIO/gcg.t ....................... ok\n",
            "t/SeqIO/genbank.t ................... ok\n",
            "t/SeqIO/Handler.t ................... ok\n",
            "t/SeqIO/kegg.t ...................... ok\n",
            "t/SeqIO/largefasta.t ................ ok\n",
            "t/SeqIO/locuslink.t ................. ok\n",
            "t/SeqIO/mbsout.t .................... ok\n",
            "t/SeqIO/metafasta.t ................. ok\n",
            "t/SeqIO/msout.t ..................... ok\n",
            "t/SeqIO/MultiFile.t ................. ok\n",
            "t/SeqIO/Multiple_fasta.t ............ ok\n",
            "t/SeqIO/phd.t ....................... ok\n",
            "t/SeqIO/pir.t ....................... ok\n",
            "t/SeqIO/qual.t ...................... ok\n",
            "t/SeqIO/raw.t ....................... ok\n",
            "t/SeqIO/scf.t ....................... ok\n",
            "t/SeqIO/SeqBuilder.t ................ ok\n",
            "t/SeqIO/SeqIO.t ..................... ok\n",
            "t/SeqIO/seqxml.t .................... ok\n",
            "t/SeqIO/Splicedseq.t ................ ok\n",
            "t/SeqIO/swiss.t ..................... ok\n",
            "t/SeqIO/tab.t ....................... ok\n",
            "t/SeqIO/table.t ..................... ok\n",
            "t/SeqIO/tigr.t ...................... ok\n",
            "t/SeqIO/tigrxml.t ................... ok\n",
            "t/SeqIO/tinyseq.t ................... ok\n",
            "t/SeqTools/Backtranslate.t .......... ok\n",
            "t/SeqTools/CodonTable.t ............. ok\n",
            "t/SeqTools/ECnumber.t ............... ok\n",
            "t/SeqTools/GuessSeqFormat.t ......... ok\n",
            "t/SeqTools/OddCodes.t ............... ok\n",
            "t/SeqTools/SeqPattern.t ............. ok\n",
            "t/SeqTools/SeqStats.t ............... ok\n",
            "t/SeqTools/SeqUtils.t ............... ok\n",
            "t/SeqTools/SeqWords.t ............... ok\n",
            "t/Species.t ......................... ok\n",
            "t/Tools/Alignment/Consed.t .......... ok\n",
            "t/Tools/AmpliconSearch.t ............ ok\n",
            "t/Tools/EMBOSS/Palindrome.t ......... ok\n",
            "t/Tools/ePCR.t ...................... ok\n",
            "t/Tools/Est2Genome.t ................ ok\n",
            "t/Tools/FootPrinter.t ............... ok\n",
            "t/Tools/Geneid.t .................... ok\n",
            "t/Tools/Genewise.t .................. ok\n",
            "t/Tools/Genomewise.t ................ ok\n",
            "t/Tools/Genpred.t ................... ok\n",
            "t/Tools/GFF.t ....................... ok\n",
            "t/Tools/IUPAC.t ..................... ok\n",
            "t/Tools/Lucy.t ...................... ok\n",
            "t/Tools/Match.t ..................... ok\n",
            "t/Tools/Phylo/Gerp.t ................ ok\n",
            "t/Tools/Phylo/Molphy.t .............. ok\n",
            "t/Tools/Phylo/Phylip/ProtDist.t ..... ok\n",
            "t/Tools/pICalculator.t .............. ok\n",
            "t/Tools/Primer3.t ................... ok\n",
            "t/Tools/Promoterwise.t .............. ok\n",
            "t/Tools/PrositeScan.t ............... ok\n",
            "t/Tools/Pseudowise.t ................ ok\n",
            "t/Tools/QRNA.t ...................... ok\n",
            "t/Tools/RandDistFunctions.t ......... ok\n",
            "t/Tools/RepeatMasker.t .............. ok\n",
            "t/Tools/Seg.t ....................... ok\n",
            "t/Tools/Sigcleave.t ................. ok\n",
            "t/Tools/Signalp.t ................... ok\n",
            "t/Tools/Signalp/ExtendedSignalp.t ... ok\n",
            "t/Tools/Sim4.t ...................... ok\n",
            "t/Tools/Spidey/Spidey.t ............. ok\n",
            "t/Tools/TandemRepeatsFinder.t ....... ok\n",
            "t/Tools/TargetP.t ................... ok\n",
            "t/Tools/Tmhmm.t ..................... ok\n",
            "t/Tools/tRNAscanSE.t ................ ok\n",
            "t/Tree/Compatible.t ................. ok\n",
            "t/Tree/Node.t ....................... ok\n",
            "t/Tree/RandomTreeFactory.t .......... ok\n",
            "t/Tree/Tree.t ....................... ok\n",
            "t/Tree/TreeIO.t ..................... ok\n",
            "t/Tree/TreeIO/lintree.t ............. ok\n",
            "t/Tree/TreeIO/newick.t .............. ok\n",
            "t/Tree/TreeIO/nexus.t ............... ok\n",
            "t/Tree/TreeIO/nhx.t ................. ok\n",
            "t/Tree/TreeIO/phyloxml.t ............ ok\n",
            "t/Tree/TreeIO/tabtree.t ............. ok\n",
            "t/Tree/TreeStatistics.t ............. ok\n",
            "All tests successful.\n",
            "Files=211, Tests=14841, 114 wallclock secs ( 2.68 usr  0.49 sys + 94.84 cusr 12.90 csys = 110.91 CPU)\n",
            "Result: PASS\n",
            "  CJFIELDS/BioPerl-1.7.8.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for CJFIELDS/BioPerl-1.7.8.tar.gz\n",
            "Manifying 40 pod documents\n",
            "Manifying 35 pod documents\n",
            "Manifying 33 pod documents\n",
            "Manifying 32 pod documents\n",
            "Manifying 34 pod documents\n",
            "Manifying 32 pod documents\n",
            "Manifying 32 pod documents\n",
            "Manifying 30 pod documents\n",
            "Manifying 27 pod documents\n",
            "Manifying 35 pod documents\n",
            "Manifying 29 pod documents\n",
            "Manifying 35 pod documents\n",
            "Manifying 36 pod documents\n",
            "Manifying 33 pod documents\n",
            "Manifying 33 pod documents\n",
            "Manifying 15 pod documents\n",
            "Installing /usr/local/share/perl/5.30.0/BioPerl.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/TreeIO.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/OntologyIO.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AnnotationI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AnnotationCollectionI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/LocatableSeq.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DBLinkContainerI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/PrimarySeq.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AnalysisI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AnnotatableI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AnalysisParserI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/IdentifiableI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SimpleAnalysisI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/PrimarySeqI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/RangeI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DasI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Range.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/ParameterBaseI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqUtils.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeatureI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqAnalysisParserI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Taxon.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Species.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/IdCollectionI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/FeatureHolderI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SimpleAlign.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/WebAgent.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AnalysisResultI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/PullParserI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/HandlerBaseI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DescribableI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/UpdateableSeqI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/LocationI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/Ontology.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/OBOterm.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/PathI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/GOterm.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/TermI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/RelationshipFactory.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/OntologyI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/Term.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/OntologyStore.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/OBOEngine.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/RelationshipI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/RelationshipType.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/Relationship.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/InterProTerm.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/TermFactory.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/SimpleOntologyEngine.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/OntologyEngineI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/Path.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/DocumentRegistry.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Ontology/SimpleGOEngine/GraphAdaptor.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/EncodedSeq.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/LargeSeq.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/SeqBuilder.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/QualI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/MetaI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/SeqFastaSpeedFactory.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/LargeSeqI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/TraceI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/LargePrimarySeq.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/PrimedSeq.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/SimulatedRead.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/LargeLocatableSeq.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/PrimaryQual.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/RichSeq.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/SequenceTrace.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/Quality.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/SeqFactory.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/RichSeqI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/BaseSeqProcessor.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/Meta.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Seq/Meta/Array.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Align/ProteinStatistics.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Align/DNAStatistics.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Align/PairwiseStatistics.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Align/AlignI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Align/Utilities.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Align/StatisticsI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Factory/ObjectFactoryI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Factory/SeqAnalysisParserFactory.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Factory/AnalysisI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Factory/TreeFactoryI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Factory/DriverFactory.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Factory/ApplicationFactoryI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Factory/SequenceProcessorI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Factory/SequenceStreamI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Factory/SeqAnalysisParserFactoryI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Factory/ObjectBuilderI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Factory/LocationFactoryI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Factory/ObjectFactory.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Factory/FTLocationFactory.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Factory/SequenceFactoryI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/raw.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/phd.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/game.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/tab.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/genbank.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/FTHelper.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/qual.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/swiss.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/largefasta.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/fastq.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/kegg.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/embl.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/gcg.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/metafasta.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/locuslink.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/swissdriver.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/MultiFile.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/gbxml.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/tigrxml.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/msout.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/ace.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/tinyseq.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/bsml.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/table.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/gbdriver.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/asciitree.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/seqxml.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/bsml_sax.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/mbsout.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/fasta.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/embldriver.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/scf.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/pir.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/tigr.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/Handler/GenericRichSeqHandler.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/tinyseq/tinyseqHandler.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/game/featHandler.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/game/gameWriter.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/game/seqHandler.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/game/gameSubs.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/game/gameHandler.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/TreeIO/phyloxml.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/TreeIO/lintree.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/TreeIO/NewickParser.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/TreeIO/nexus.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/TreeIO/cluster.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/TreeIO/newick.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/TreeIO/pag.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/TreeIO/nhx.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/TreeIO/tabtree.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/TreeIO/TreeEventBuilder.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Annotation/SimpleValue.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Annotation/Target.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Annotation/Comment.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Annotation/Collection.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Annotation/OntologyTerm.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Annotation/AnnotationFactory.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Annotation/Tree.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Annotation/Relation.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Annotation/TypeManager.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Annotation/TagTree.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Annotation/StructuredValue.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Annotation/Reference.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Annotation/DBLink.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/IO.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/PhylipDist.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/Generic.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/Mlagan.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/Scoring.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/MatrixI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/IO/mlagan.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/IO/phylip.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/IO/scoring.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/PSM/IO.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/PSM/Psm.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/PSM/PsmI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/PSM/ProtPsm.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/PSM/InstanceSite.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/PSM/SiteMatrix.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/PSM/PsmHeader.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/PSM/PsmHeaderI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/PSM/InstanceSiteI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/PSM/SiteMatrixI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/PSM/ProtMatrix.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/PSM/IO/transfac.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/PSM/IO/masta.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/PSM/IO/psiblast.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/PSM/IO/mast.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Matrix/PSM/IO/meme.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Fgenesh.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Genscan.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Promoterwise.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/GuessSeqFormat.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Primer3.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Est2Genome.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/ECnumber.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Prints.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/isPcr.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Grail.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/SeqWords.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Profile.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/AmpliconSearch.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/RandomDistFunctions.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/ESTScan.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/IUPAC.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/SeqPattern.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Sigcleave.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/SeqStats.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/MZEF.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/QRNA.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/TargetP.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Coil.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Seg.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/OddCodes.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Tmhmm.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/TandemRepeatsFinder.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Match.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Genomewise.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Pseudowise.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Eponine.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/GFF.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/AnalysisResult.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Genewise.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/EPCR.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/CodonTable.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Geneid.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/pICalculator.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/ipcress.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Lucy.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/FootPrinter.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/PrositeScan.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/tRNAscanSE.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Genemark.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/RepeatMasker.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Glimmer.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Blat.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Signalp.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Signalp/ExtendedSignalp.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Analysis/SimpleAnalysisBase.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Run/WrapperBase.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Run/Analysis.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Run/AnalysisFactory.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Run/GenericParameters.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Run/ParametersI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Run/Phylo/PhyloBase.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Run/WrapperBase/CommandExts.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Phylo/Gerp.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Phylo/Molphy.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Phylo/Molphy/Result.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Phylo/Phylip/ProtDist.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Alignment/Consed.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Alignment/Trim.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Prediction/Gene.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Prediction/Exon.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/EMBOSS/Palindrome.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/SeqPattern/Backtranslate.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Spidey/Results.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Spidey/Exon.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Primer/AssessorI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Primer/Feature.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Primer/Pair.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Primer/Assessor/Base.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Sim4/Results.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Sim4/Exon.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/GenericDatabase.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/GenericStatistics.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/BlastUtils.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/BlastStatistics.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/SearchUtils.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/DatabaseI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/StatisticsI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Processor.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/HSP/PSLHSP.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/HSP/GenericHSP.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/HSP/FastaHSP.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/HSP/WABAHSP.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/HSP/HSPFactory.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/HSP/PsiBlastHSP.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/HSP/PullHSPI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/HSP/BlastHSP.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/HSP/ModelHSP.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/HSP/BlastPullHSP.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/HSP/HSPI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Result/WABAResult.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Result/GenericResult.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Result/ResultI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Result/ResultFactory.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Result/BlastPullResult.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Result/BlastResult.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Result/PullResultI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Result/INFERNALResult.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Result/CrossMatchResult.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Tiling/TilingI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Tiling/MapTileUtils.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Tiling/MapTiling.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Hit/BlastHit.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Hit/PsiBlastHit.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Hit/BlastPullHit.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Hit/HitFactory.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Hit/Fasta.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Hit/ModelHit.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Hit/PullHitI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Hit/HitI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Hit/GenericHit.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Iteration/IterationI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Search/Iteration/GenericIteration.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Event/EventHandlerI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Event/EventGeneratorI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Primer.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/PositionProxy.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Collection.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/SubSeq.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/CollectionI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Generic.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/AnnotationAdaptor.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Lite.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Amplicon.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Computation.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Similarity.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/TypedSeqFeatureI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/FeaturePair.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/SimilarityPair.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Tools/FeatureNamer.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Tools/Unflattener.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Tools/IDHandler.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Tools/TypeMapper.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Gene/NC_Feature.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Gene/Intron.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Gene/GeneStructureI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Gene/Promoter.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Gene/UTR.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Gene/GeneStructure.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Gene/TranscriptI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Gene/ExonI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Gene/Transcript.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Gene/Poly_A_site.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqFeature/Gene/Exon.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/Qual.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/ReferenceI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/RandomAccessI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/Failover.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/GenericWebAgent.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/SeqI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/Fasta.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/WebDBSeqI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/QueryI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/Registry.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/Taxonomy.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/InMemoryCache.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/IndexedBase.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/FileCache.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/Flat.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/DBFetch.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/UpdateableSeqI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/LocationI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/Query/WebQuery.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/Taxonomy/flatfile.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/Taxonomy/list.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/Taxonomy/silva.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/Taxonomy/greengenes.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/Flat/BinarySearch.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/Flat/BDB.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/Flat/BDB/genbank.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/Flat/BDB/swiss.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/Flat/BDB/embl.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/Flat/BDB/fasta.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/GFF/Util/Binning.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/GFF/Util/Rearrange.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/gmap_f9.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/wise.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/EventHandlerI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/SearchWriterI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/SearchResultEventBuilder.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/axt.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/psl.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/FastHitEventBuilder.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/exonerate.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/waba.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/cross_match.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/blast_pull.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/megablast.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/infernal.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/blasttable.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/sim4.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/blast.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/erpin.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/fasta.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/rnamotif.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/IteratedSearchResultEventBuilder.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/Writer/GbrowseGFF.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/Writer/HSPTableWriter.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/Writer/TextResultWriter.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/Writer/ResultTableWriter.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/Writer/HTMLResultWriter.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SearchIO/Writer/HitTableWriter.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/OntologyIO/goflat.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/OntologyIO/simplehierarchy.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/OntologyIO/InterProParser.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/OntologyIO/soflat.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/OntologyIO/dagflat.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/OntologyIO/obo.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/OntologyIO/Handlers/InterPro_BioSQL_Handler.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/OntologyIO/Handlers/BaseSAXHandler.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/OntologyIO/Handlers/InterProHandler.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tree/Node.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tree/NodeI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tree/Compatible.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tree/Tree.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tree/AnnotatableNode.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tree/RandomFactory.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tree/TreeI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tree/NodeNHX.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tree/DistanceFactory.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tree/Statistics.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tree/TreeFunctionsI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Index/Qual.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Index/Abstract.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Index/AbstractSeq.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Index/SwissPfam.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Index/Fasta.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Index/Blast.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Index/EMBL.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Index/Swissprot.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Index/Fastq.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Index/BlastTable.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Index/GenBank.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Das/FeatureTypeI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Das/SegmentI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Root/IO.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Root/Test.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Root/TestObject.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Root/Exception.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Root/Version.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Root/Root.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Root/HTTPget.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Root/Utilities.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Root/RootI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Root/Storable.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/CodonUsage/IO.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/CodonUsage/Table.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/clustalw.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/largemultifasta.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/selex.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/bl2seq.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/mase.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/nexus.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/pfam.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/psi.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/meme.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/mega.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/metafasta.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/prodom.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/xmfa.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/phylip.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/emboss.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/proda.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/po.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/msf.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/arp.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/fasta.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/maf.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/AlignIO/Handler/GenericAlignHandler.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Location/FuzzyLocationI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Location/CoordinatePolicyI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Location/WidestCoordPolicy.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Location/Atomic.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Location/Split.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Location/SplitLocationI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Location/NarrowestCoordPolicy.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Location/Simple.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Location/AvWithinCoordPolicy.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Location/Fuzzy.pm\n",
            "Installing /usr/local/man/man1/bp_split_seq.1p\n",
            "Installing /usr/local/man/man1/bp_seqret.1p\n",
            "Installing /usr/local/man/man1/bp_seqcut.1p\n",
            "Installing /usr/local/man/man1/bp_fetch.1p\n",
            "Installing /usr/local/man/man1/bp_mutate.1p\n",
            "Installing /usr/local/man/man1/bp_taxid4species.1p\n",
            "Installing /usr/local/man/man1/bp_process_sgd.1p\n",
            "Installing /usr/local/man/man1/bp_revtrans-motif.1p\n",
            "Installing /usr/local/man/man1/bp_filter_search.1p\n",
            "Installing /usr/local/man/man1/bp_search2tribe.1p\n",
            "Installing /usr/local/man/man1/bp_seqconvert.1p\n",
            "Installing /usr/local/man/man1/bp_nexus2nh.1p\n",
            "Installing /usr/local/man/man1/bp_make_mrna_protein.1p\n",
            "Installing /usr/local/man/man1/bp_search2gff.1p\n",
            "Installing /usr/local/man/man1/bp_nrdb.1p\n",
            "Installing /usr/local/man/man1/bp_translate_seq.1p\n",
            "Installing /usr/local/man/man1/bp_seqretsplit.1p\n",
            "Installing /usr/local/man/man1/bp_search2alnblocks.1p\n",
            "Installing /usr/local/man/man1/bp_gccalc.1p\n",
            "Installing /usr/local/man/man1/bp_sreformat.1p\n",
            "Installing /usr/local/man/man1/bp_fastam9_to_table.1p\n",
            "Installing /usr/local/man/man1/bp_find-blast-matches.1p\n",
            "Installing /usr/local/man/man1/bp_bioflat_index.1p\n",
            "Installing /usr/local/man/man1/bp_genbank2gff3.1p\n",
            "Installing /usr/local/man/man1/bp_mask_by_search.1p\n",
            "Installing /usr/local/man/man1/bp_aacomp.1p\n",
            "Installing /usr/local/man/man1/bp_local_taxonomydb_query.1p\n",
            "Installing /usr/local/man/man1/bp_index.1p\n",
            "Installing /usr/local/man/man1/bp_dbsplit.1p\n",
            "Installing /usr/local/man/man1/bp_oligo_count.1p\n",
            "Installing /usr/local/man/man1/bp_unflatten_seq.1p\n",
            "Installing /usr/local/man/man1/bp_seq_length.1p\n",
            "Installing /usr/local/man/man1/bp_extract_feature_seq.1p\n",
            "Installing /usr/local/man/man1/bp_taxonomy2tree.1p\n",
            "Installing /usr/local/man/man1/bp_tree2pag.1p\n",
            "Installing /usr/local/man/man1/bp_search2table.1p\n",
            "Installing /usr/local/man/man1/bp_mrtrans.1p\n",
            "Installing /usr/local/man/man1/bp_process_gadfly.1p\n",
            "Installing /usr/local/man/man1/bp_biogetseq.1p\n",
            "Installing /usr/local/man/man1/bp_seqpart.1p\n",
            "Installing /usr/local/man/man3/Bio::Annotation::StructuredValue.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqUtils.3pm\n",
            "Installing /usr/local/man/man3/Bio::OntologyIO::soflat.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::PSM::ProtMatrix.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tree::Statistics.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::SequenceTrace.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Result::CrossMatchResult.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::HSP::GenericHSP.3pm\n",
            "Installing /usr/local/man/man3/Bio::Root::HTTPget.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::prodom.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Amplicon.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::PSM::PsmI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::QRNA.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::HSP::WABAHSP.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::Qual.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::clustalw.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::SeqPattern::Backtranslate.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::RelationshipI.3pm\n",
            "Installing /usr/local/man/man3/Bio::TreeIO::tabtree.3pm\n",
            "Installing /usr/local/man/man3/Bio::Index::Fastq.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Gene::TranscriptI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Location::CoordinatePolicyI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::DocumentRegistry.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Lucy.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::bl2seq.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::IO::mlagan.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Result::GenericResult.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::BlastUtils.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::FeaturePair.3pm\n",
            "Installing /usr/local/man/man3/Bio::AnalysisParserI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Root::Root.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::IteratedSearchResultEventBuilder.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::InMemoryCache.3pm\n",
            "Installing /usr/local/man/man3/Bio::TreeIO::cluster.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::axt.3pm\n",
            "Installing /usr/local/man/man3/Bio::Factory::ObjectBuilderI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::SimilarityPair.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Profile.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Hit::GenericHit.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::RepeatMasker.3pm\n",
            "Installing /usr/local/man/man3/Bio::Das::FeatureTypeI.3pm\n",
            "Installing /usr/local/man/man3/Bio::IdCollectionI.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::DBFetch.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::RandomDistFunctions.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::TypedSeqFeatureI.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::mega.3pm\n",
            "Installing /usr/local/man/man3/Bio::Align::Utilities.3pm\n",
            "Installing /usr/local/man/man3/Bio::LocatableSeq.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::PSM::PsmHeader.3pm\n",
            "Installing /usr/local/man/man3/Bio::Location::AvWithinCoordPolicy.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Computation.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::HSP::PsiBlastHSP.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::PhylipDist.3pm\n",
            "Installing /usr/local/man/man3/Bio::TreeIO::NewickParser.3pm\n",
            "Installing /usr/local/man/man3/Bio::DasI.3pm\n",
            "Installing /usr/local/man/man3/Bio::AnnotationCollectionI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::SimpleGOEngine::GraphAdaptor.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeatureI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Factory::SequenceStreamI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tree::NodeNHX.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::locuslink.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Gene::GeneStructure.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Tiling::MapTiling.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Processor.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::game::featHandler.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::tinyseq.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Fgenesh.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::pir.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::infernal.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::AnnotationAdaptor.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::EventHandlerI.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::GFF::Util::Binning.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Generic.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tree::DistanceFactory.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Gene::Transcript.3pm\n",
            "Installing /usr/local/man/man3/Bio::Root::Test.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Primer.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Gene::UTR.3pm\n",
            "Installing /usr/local/man/man3/Bio::Location::Fuzzy.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Prediction::Exon.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqI.3pm\n",
            "Installing /usr/local/man/man3/Bio::OntologyIO::InterProParser.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::SearchWriterI.3pm\n",
            "Installing /usr/local/man/man3/Bio::TreeIO.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Result::ResultI.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::xmfa.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::table.3pm\n",
            "Installing /usr/local/man/man3/Bio::Root::IO.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::gbdriver.3pm\n",
            "Installing /usr/local/man/man3/Bio::LocationI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::PSM::IO::transfac.3pm\n",
            "Installing /usr/local/man/man3/Bio::Index::SwissPfam.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Gene::Exon.3pm\n",
            "Installing /usr/local/man/man3/Bio::Factory::DriverFactory.3pm\n",
            "Installing /usr/local/man/man3/Bio::WebAgent.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tree::Compatible.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Result::BlastPullResult.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::QualI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::largefasta.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::LargeSeqI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::gmap_f9.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::ReferenceI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Run::WrapperBase.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::FTHelper.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Prints.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::LargeSeq.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::BaseSeqProcessor.3pm\n",
            "Installing /usr/local/man/man3/Bio::TreeIO::nhx.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::erpin.3pm\n",
            "Installing /usr/local/man/man3/Bio::Root::Storable.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::SubSeq.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::mase.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::LocationI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Event::EventGeneratorI.3pm\n",
            "Installing /usr/local/man/man3/Bio::HandlerBaseI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::genbank.3pm\n",
            "Installing /usr/local/man/man3/Bio::Annotation::Relation.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Hit::HitI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::PSM::SiteMatrixI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Genomewise.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::ECnumber.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::LargeLocatableSeq.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::Taxonomy::silva.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::AmpliconSearch.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::Writer::ResultTableWriter.3pm\n",
            "Installing /usr/local/man/man3/Bio::Event::EventHandlerI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::kegg.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Gene::ExonI.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::WebDBSeqI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tree::RandomFactory.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tree::TreeFunctionsI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Root::Exception.3pm\n",
            "Installing /usr/local/man/man3/Bio::SimpleAlign.3pm\n",
            "Installing /usr/local/man/man3/Bio::Factory::LocationFactoryI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Annotation::AnnotationFactory.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Hit::BlastHit.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::HSP::PSLHSP.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::tab.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::TraceI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Gene::NC_Feature.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::OntologyStore.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::SimpleOntologyEngine.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::asciitree.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::IUPAC.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Collection.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::PSM::PsmHeaderI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::game::seqHandler.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::FileCache.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::HSP::HSPFactory.3pm\n",
            "Installing /usr/local/man/man3/Bio::TreeIO::newick.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::TermFactory.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Gene::Poly_A_site.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Result::WABAResult.3pm\n",
            "Installing /usr/local/man/man3/Bio::Annotation::Comment.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Alignment::Trim.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Primer::Pair.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Sigcleave.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Signalp::ExtendedSignalp.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::largemultifasta.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tree::Tree.3pm\n",
            "Installing /usr/local/man/man3/Bio::Factory::TreeFactoryI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Prediction::Gene.3pm\n",
            "Installing /usr/local/man/man3/Bio::DBLinkContainerI.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::Flat::BDB::swiss.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::waba.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::Flat::BDB::genbank.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::game::gameSubs.3pm\n",
            "Installing /usr/local/man/man3/Bio::TreeIO::lintree.3pm\n",
            "Installing /usr/local/man/man3/Bio::Index::GenBank.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::PathI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::swissdriver.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::rnamotif.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::ace.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::RandomAccessI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::Writer::HTMLResultWriter.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::SearchResultEventBuilder.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Alignment::Consed.3pm\n",
            "Installing /usr/local/man/man3/Bio::Annotation::TypeManager.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::SeqFastaSpeedFactory.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Primer::AssessorI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::bsml.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Primer3.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tree::AnnotatableNode.3pm\n",
            "Installing /usr/local/man/man3/Bio::OntologyIO::Handlers::InterPro_BioSQL_Handler.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Est2Genome.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::metafasta.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Primer::Assessor::Base.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::Query::WebQuery.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::Taxonomy.3pm\n",
            "Installing /usr/local/man/man3/Bio::Factory::ObjectFactory.3pm\n",
            "Installing /usr/local/man/man3/Bio::Location::Split.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::PrositeScan.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::SeqWords.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::nexus.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::MatrixI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Index::Swissprot.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::Writer::GbrowseGFF.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::msf.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::EncodedSeq.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::GuessSeqFormat.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::Flat::BDB.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::PSM::IO::psiblast.3pm\n",
            "Installing /usr/local/man/man3/Bio::Index::Blast.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Genscan.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::GenericWebAgent.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::Taxonomy::flatfile.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Tmhmm.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Hit::Fasta.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Tools::TypeMapper.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::gbxml.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::HSP::PullHSPI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::TermI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Align::DNAStatistics.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Genewise.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::PrimedSeq.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Eponine.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::blasttable.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tree::TreeI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Location::SplitLocationI.3pm\n",
            "Installing /usr/local/man/man3/Bio::ParameterBaseI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::Relationship.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::SearchUtils.3pm\n",
            "Installing /usr/local/man/man3/Bio::AnalysisI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::psl.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::Flat.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Primer::Feature.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::qual.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::Generic.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Analysis::SimpleAnalysisBase.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::tigr.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::HSP::BlastPullHSP.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::megablast.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::BlastStatistics.3pm\n",
            "Installing /usr/local/man/man3/Bio::Align::ProteinStatistics.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::IO::scoring.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::PSM::InstanceSite.3pm\n",
            "Installing /usr/local/man/man3/Bio::Annotation::SimpleValue.3pm\n",
            "Installing /usr/local/man/man3/Bio::OntologyIO::dagflat.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::PositionProxy.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::Term.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Result::PullResultI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Range.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::PSM::InstanceSiteI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Signalp.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::OddCodes.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Gene::GeneStructureI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::Writer::HSPTableWriter.3pm\n",
            "Installing /usr/local/man/man3/Bio::PullParserI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Gene::Promoter.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::wise.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::IndexedBase.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::embl.3pm\n",
            "Installing /usr/local/man/man3/Bio::Annotation::OntologyTerm.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::meme.3pm\n",
            "Installing /usr/local/man/man3/Bio::OntologyIO::obo.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Geneid.3pm\n",
            "Installing /usr/local/man/man3/Bio::Factory::AnalysisI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Annotation::Target.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::tigrxml.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::blast.3pm\n",
            "Installing /usr/local/man/man3/Bio::IdentifiableI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Factory::SequenceFactoryI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::metafasta.3pm\n",
            "Installing /usr/local/man/man3/Bio::Root::Version.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Lite.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::HSP::FastaHSP.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Phylo::Gerp.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::psi.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::tRNAscanSE.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::PSM::ProtPsm.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::DatabaseI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Index::EMBL.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Pseudowise.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::fasta.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::PrimaryQual.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::CodonTable.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Run::AnalysisFactory.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::IO.3pm\n",
            "Installing /usr/local/man/man3/Bio::Taxon.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::Quality.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::proda.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::PSM::IO.3pm\n",
            "Installing /usr/local/man/man3/Bio::Das::SegmentI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Result::ResultFactory.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::selex.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Hit::BlastPullHit.3pm\n",
            "Installing /usr/local/man/man3/Bio::Location::FuzzyLocationI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::MZEF.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::AnalysisResult.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Tools::IDHandler.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::scf.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::GFF::Util::Rearrange.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::HSP::HSPI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Location::WidestCoordPolicy.3pm\n",
            "Installing /usr/local/man/man3/Bio::Species.3pm\n",
            "Installing /usr/local/man/man3/Bio::DescribableI.3pm\n",
            "Installing /usr/local/man/man3/Bio::OntologyIO::Handlers::InterProHandler.3pm\n",
            "Installing /usr/local/man/man3/Bio::Factory::ApplicationFactoryI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::Writer::TextResultWriter.3pm\n",
            "Installing /usr/local/man/man3/Bio::SimpleAnalysisI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Tools::FeatureNamer.3pm\n",
            "Installing /usr/local/man/man3/Bio::Factory::ObjectFactoryI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::SeqPattern.3pm\n",
            "Installing /usr/local/man/man3/Bio::Annotation::Reference.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::Failover.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::cross_match.3pm\n",
            "Installing /usr/local/man/man3/Bio::Factory::SeqAnalysisParserFactoryI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::tinyseq::tinyseqHandler.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Phylo::Molphy::Result.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Hit::PullHitI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::gcg.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::Ontology.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Grail.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::msout.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::GenericDatabase.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::fasta.3pm\n",
            "Installing /usr/local/man/man3/Bio::Location::Atomic.3pm\n",
            "Installing /usr/local/man/man3/Bio::Factory::SeqAnalysisParserFactory.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::EPCR.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::StatisticsI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::fastq.3pm\n",
            "Installing /usr/local/man/man3/Bio::Annotation::DBLink.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::ipcress.3pm\n",
            "Installing /usr/local/man/man3/Bio::RangeI.3pm\n",
            "Installing /usr/local/man/man3/Bio::OntologyIO::simplehierarchy.3pm\n",
            "Installing /usr/local/man/man3/Bio::PrimarySeqI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Align::AlignI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::GenericStatistics.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::OntologyEngineI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::SeqBuilder.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Run::ParametersI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::ESTScan.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::Registry.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Hit::HitFactory.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::embldriver.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Match.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqAnalysisParserI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::TandemRepeatsFinder.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Similarity.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::raw.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Result::INFERNALResult.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::PSM::IO::masta.3pm\n",
            "Installing /usr/local/man/man3/Bio::Factory::SequenceProcessorI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::MetaI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::FastHitEventBuilder.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::Path.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Spidey::Results.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Genemark.3pm\n",
            "Installing /usr/local/man/man3/Bio::OntologyIO::Handlers::BaseSAXHandler.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::Flat::BDB::embl.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::InterProTerm.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::pfam.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::mbsout.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::SimulatedRead.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::Flat::BDB::fasta.3pm\n",
            "Installing /usr/local/man/man3/Bio::Index::BlastTable.3pm\n",
            "Installing /usr/local/man/man3/Bio::Index::AbstractSeq.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::exonerate.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Hit::PsiBlastHit.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::QueryI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::Meta.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::Handler::GenericRichSeqHandler.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::arp.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::SeqI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tree::Node.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::RichSeq.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::EMBOSS::Palindrome.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::bsml_sax.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::TargetP.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::CollectionI.3pm\n",
            "Installing /usr/local/man/man3/Bio::AnalysisResultI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::GOterm.3pm\n",
            "Installing /usr/local/man/man3/Bio::TreeIO::TreeEventBuilder.3pm\n",
            "Installing /usr/local/man/man3/Bio::FeatureHolderI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::OBOterm.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tree::NodeI.3pm\n",
            "Installing /usr/local/man/man3/Bio::AnnotatableI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::RelationshipFactory.3pm\n",
            "Installing /usr/local/man/man3/Bio::Factory::FTLocationFactory.3pm\n",
            "Installing /usr/local/man/man3/Bio::Root::RootI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Coil.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Glimmer.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Blat.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::SeqStats.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Sim4::Exon.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::PSM::Psm.3pm\n",
            "Installing /usr/local/man/man3/Bio::Index::Qual.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::Flat::BinarySearch.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Gene::Intron.3pm\n",
            "Installing /usr/local/man/man3/Bio::CodonUsage::IO.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::Writer::HitTableWriter.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::sim4.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Spidey::Exon.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::OBOEngine.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::maf.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::RichSeqI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::game::gameWriter.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Phylo::Phylip::ProtDist.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::IO::phylip.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Result::BlastResult.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::HSP::ModelHSP.3pm\n",
            "Installing /usr/local/man/man3/Bio::CodonUsage::Table.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::SeqFactory.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::Handler::GenericAlignHandler.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::PSM::IO::mast.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::OntologyI.3pm\n",
            "Installing /usr/local/man/man3/Bio::TreeIO::nexus.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::emboss.3pm\n",
            "Installing /usr/local/man/man3/Bio::Index::Fasta.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::Taxonomy::list.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Run::WrapperBase::CommandExts.3pm\n",
            "Installing /usr/local/man/man3/Bio::Annotation::Tree.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::FootPrinter.3pm\n",
            "Installing /usr/local/man/man3/Bio::TreeIO::pag.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::isPcr.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::seqxml.3pm\n",
            "Installing /usr/local/man/man3/Bio::Location::NarrowestCoordPolicy.3pm\n",
            "Installing /usr/local/man/man3/Bio::Annotation::TagTree.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::phylip.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::Fasta.3pm\n",
            "Installing /usr/local/man/man3/Bio::Root::Utilities.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::game::gameHandler.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::PSM::SiteMatrix.3pm\n",
            "Installing /usr/local/man/man3/Bio::PrimarySeq.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqFeature::Tools::Unflattener.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Iteration::IterationI.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::UpdateableSeqI.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::MultiFile.3pm\n",
            "Installing /usr/local/man/man3/Bio::TreeIO::phyloxml.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Sim4::Results.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Promoterwise.3pm\n",
            "Installing /usr/local/man/man3/Bio::Ontology::RelationshipType.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::HSP::BlastHSP.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Run::Phylo::PhyloBase.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::fasta.3pm\n",
            "Installing /usr/local/man/man3/Bio::OntologyIO.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Iteration::GenericIteration.3pm\n",
            "Installing /usr/local/man/man3/Bio::DB::Taxonomy::greengenes.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::Scoring.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Run::Analysis.3pm\n",
            "Installing /usr/local/man/man3/Bio::Root::TestObject.3pm\n",
            "Installing /usr/local/man/man3/Bio::Location::Simple.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Phylo::Molphy.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::Mlagan.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::LargePrimarySeq.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Hit::ModelHit.3pm\n",
            "Installing /usr/local/man/man3/Bio::Align::StatisticsI.3pm\n",
            "Installing /usr/local/man/man3/Bio::AnnotationI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Run::GenericParameters.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::game.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::swiss.3pm\n",
            "Installing /usr/local/man/man3/Bio::AlignIO::po.3pm\n",
            "Installing /usr/local/man/man3/Bio::Align::PairwiseStatistics.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::GFF.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Tiling::TilingI.3pm\n",
            "Installing /usr/local/man/man3/Bio::UpdateableSeqI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Annotation::Collection.3pm\n",
            "Installing /usr/local/man/man3/BioPerl.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::pICalculator.3pm\n",
            "Installing /usr/local/man/man3/Bio::Seq::Meta::Array.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::phd.3pm\n",
            "Installing /usr/local/man/man3/Bio::Matrix::PSM::IO::meme.3pm\n",
            "Installing /usr/local/man/man3/Bio::OntologyIO::goflat.3pm\n",
            "Installing /usr/local/man/man3/Bio::Search::Tiling::MapTileUtils.3pm\n",
            "Installing /usr/local/man/man3/Bio::Index::Abstract.3pm\n",
            "Installing /usr/local/man/man3/Bio::SearchIO::blast_pull.3pm\n",
            "Installing /usr/local/man/man3/Bio::Tools::Seg.3pm\n",
            "Installing /usr/local/bin/bp_extract_feature_seq\n",
            "Installing /usr/local/bin/bp_mask_by_search\n",
            "Installing /usr/local/bin/bp_local_taxonomydb_query\n",
            "Installing /usr/local/bin/bp_gccalc\n",
            "Installing /usr/local/bin/bp_seq_length\n",
            "Installing /usr/local/bin/bp_nexus2nh\n",
            "Installing /usr/local/bin/bp_unflatten_seq\n",
            "Installing /usr/local/bin/bp_search2tribe\n",
            "Installing /usr/local/bin/bp_index\n",
            "Installing /usr/local/bin/bp_split_seq\n",
            "Installing /usr/local/bin/bp_biogetseq\n",
            "Installing /usr/local/bin/bp_seqret\n",
            "Installing /usr/local/bin/bp_aacomp\n",
            "Installing /usr/local/bin/bp_seqconvert\n",
            "Installing /usr/local/bin/bp_seqretsplit\n",
            "Installing /usr/local/bin/bp_revtrans-motif\n",
            "Installing /usr/local/bin/bp_taxid4species\n",
            "Installing /usr/local/bin/bp_seqcut\n",
            "Installing /usr/local/bin/bp_fetch\n",
            "Installing /usr/local/bin/bp_search2table\n",
            "Installing /usr/local/bin/bp_tree2pag\n",
            "Installing /usr/local/bin/bp_sreformat\n",
            "Installing /usr/local/bin/bp_process_gadfly\n",
            "Installing /usr/local/bin/bp_find-blast-matches\n",
            "Installing /usr/local/bin/bp_bioflat_index\n",
            "Installing /usr/local/bin/bp_mutate\n",
            "Installing /usr/local/bin/bp_dbsplit\n",
            "Installing /usr/local/bin/bp_filter_search\n",
            "Installing /usr/local/bin/bp_fastam9_to_table\n",
            "Installing /usr/local/bin/bp_search2alnblocks\n",
            "Installing /usr/local/bin/bp_genbank2gff3\n",
            "Installing /usr/local/bin/bp_mrtrans\n",
            "Installing /usr/local/bin/bp_seqpart\n",
            "Installing /usr/local/bin/bp_translate_seq\n",
            "Installing /usr/local/bin/bp_make_mrna_protein\n",
            "Installing /usr/local/bin/bp_nrdb\n",
            "Installing /usr/local/bin/bp_taxonomy2tree\n",
            "Installing /usr/local/bin/bp_process_sgd\n",
            "Installing /usr/local/bin/bp_search2gff\n",
            "Installing /usr/local/bin/bp_oligo_count\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  CJFIELDS/BioPerl-1.7.8.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "Test::Most is up to date (0.38).\n",
            "  CJFIELDS/Bio-DB-EMBL-1.7.4.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/Bio-DB-EMBL-1.7.4-0\n",
            "  CJFIELDS/Bio-DB-EMBL-1.7.4.tar.gz\n",
            "  Has already been prepared\n",
            "Running make for C/CJ/CJFIELDS/Bio-DB-EMBL-1.7.4.tar.gz\n",
            "cp lib/Bio/DB/EMBL.pm blib/lib/Bio/DB/EMBL.pm\n",
            "Manifying 1 pod document\n",
            "  CJFIELDS/Bio-DB-EMBL-1.7.4.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for CJFIELDS/Bio-DB-EMBL-1.7.4.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/00-compile.t ........... ok\n",
            "t/author-mojibake.t ...... skipped: these tests are for testing by the author\n",
            "t/author-pod-coverage.t .. skipped: these tests are for testing by the author\n",
            "t/author-pod-syntax.t .... skipped: these tests are for testing by the author\n",
            "t/EMBL.t ................. Use of uninitialized value $proto in socket at /usr/local/share/perl/5.30.0/Test/RequiresInternet.pm line 55.\n",
            "t/EMBL.t ................. ok\n",
            "All tests successful.\n",
            "Files=5, Tests=17,  3 wallclock secs ( 0.04 usr  0.02 sys +  0.64 cusr  0.09 csys =  0.79 CPU)\n",
            "Result: PASS\n",
            "  CJFIELDS/Bio-DB-EMBL-1.7.4.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for CJFIELDS/Bio-DB-EMBL-1.7.4.tar.gz\n",
            "Manifying 1 pod document\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/EMBL.pm\n",
            "Installing /usr/local/man/man3/Bio::DB::EMBL.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  CJFIELDS/Bio-DB-EMBL-1.7.4.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "Running install for module 'Bio::DB::GenBank'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/C/CJ/CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz\n",
            "Checksum for /root/.cpan/sources/authors/id/C/CJ/CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz ok\n",
            "Configuring C/CJ/CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Warning: prerequisite Bio::SeqIO::entrezgene 0 not found.\n",
            "Warning: prerequisite CGI 0 not found.\n",
            "Warning: prerequisite Cache::FileCache 0 not found.\n",
            "Warning: prerequisite LWP::Protocol::https 0 not found.\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for Bio::DB::NCBIHelper\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for C/CJ/CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz\n",
            "---- Unsatisfied dependencies detected during ----\n",
            "----  CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz ----\n",
            "    Bio::SeqIO::entrezgene [requires]\n",
            "    CGI [requires]\n",
            "    Cache::FileCache [requires]\n",
            "    LWP::Protocol::https [requires]\n",
            "Running install for module 'Bio::SeqIO::entrezgene'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/C/CJ/CJFIELDS/Bio-ASN1-EntrezGene-1.73.tar.gz\n",
            "Checksum for /root/.cpan/sources/authors/id/C/CJ/CJFIELDS/Bio-ASN1-EntrezGene-1.73.tar.gz ok\n",
            "Configuring C/CJ/CJFIELDS/Bio-ASN1-EntrezGene-1.73.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Warning: prerequisite Bio::Cluster::SequenceFamily 0 not found.\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for Bio::ASN1::EntrezGene\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  CJFIELDS/Bio-ASN1-EntrezGene-1.73.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for C/CJ/CJFIELDS/Bio-ASN1-EntrezGene-1.73.tar.gz\n",
            "---- Unsatisfied dependencies detected during ----\n",
            "---- CJFIELDS/Bio-ASN1-EntrezGene-1.73.tar.gz ----\n",
            "    Bio::Cluster::SequenceFamily [requires]\n",
            "Running install for module 'Bio::Cluster::SequenceFamily'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/C/CJ/CJFIELDS/Bio-Cluster-1.7.3.tar.gz\n",
            "Checksum for /root/.cpan/sources/authors/id/C/CJ/CJFIELDS/Bio-Cluster-1.7.3.tar.gz ok\n",
            "Configuring C/CJ/CJFIELDS/Bio-Cluster-1.7.3.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Warning: prerequisite Bio::Variation::SNP 0 not found.\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for Bio::Cluster\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  CJFIELDS/Bio-Cluster-1.7.3.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for C/CJ/CJFIELDS/Bio-Cluster-1.7.3.tar.gz\n",
            "---- Unsatisfied dependencies detected during ----\n",
            "----     CJFIELDS/Bio-Cluster-1.7.3.tar.gz    ----\n",
            "    Bio::Variation::SNP [requires]\n",
            "Running install for module 'Bio::Variation::SNP'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/C/CJ/CJFIELDS/Bio-Variation-1.7.5.tar.gz\n",
            "Checksum for /root/.cpan/sources/authors/id/C/CJ/CJFIELDS/Bio-Variation-1.7.5.tar.gz ok\n",
            "Configuring C/CJ/CJFIELDS/Bio-Variation-1.7.5.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for Bio::Variation\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  CJFIELDS/Bio-Variation-1.7.5.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for C/CJ/CJFIELDS/Bio-Variation-1.7.5.tar.gz\n",
            "cp lib/Bio/Variation/AAReverseMutate.pm blib/lib/Bio/Variation/AAReverseMutate.pm\n",
            "cp lib/Bio/Variation/IO.pm blib/lib/Bio/Variation/IO.pm\n",
            "cp lib/Bio/Variation/SeqDiff.pm blib/lib/Bio/Variation/SeqDiff.pm\n",
            "cp lib/Bio/Variation.pm blib/lib/Bio/Variation.pm\n",
            "cp lib/Bio/Variation/Allele.pm blib/lib/Bio/Variation/Allele.pm\n",
            "cp lib/Bio/Variation/IO/xml.pm blib/lib/Bio/Variation/IO/xml.pm\n",
            "cp lib/Bio/Variation/RNAChange.pm blib/lib/Bio/Variation/RNAChange.pm\n",
            "cp lib/Bio/Variation/AAChange.pm blib/lib/Bio/Variation/AAChange.pm\n",
            "cp lib/Bio/Variation/IO/flat.pm blib/lib/Bio/Variation/IO/flat.pm\n",
            "cp lib/Bio/Variation/SNP.pm blib/lib/Bio/Variation/SNP.pm\n",
            "cp lib/Bio/Variation/VariantI.pm blib/lib/Bio/Variation/VariantI.pm\n",
            "cp lib/Bio/Variation/DNAMutation.pm blib/lib/Bio/Variation/DNAMutation.pm\n",
            "cp bin/bp_flanks blib/script/bp_flanks\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_flanks\n",
            "Manifying 1 pod document\n",
            "Manifying 12 pod documents\n",
            "  CJFIELDS/Bio-Variation-1.7.5.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for CJFIELDS/Bio-Variation-1.7.5.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/00-compile.t ........... ok\n",
            "t/AAChange.t ............. ok\n",
            "t/AAReverseMutate.t ...... ok\n",
            "t/Allele.t ............... ok\n",
            "t/author-mojibake.t ...... skipped: these tests are for testing by the author\n",
            "t/author-pod-coverage.t .. skipped: these tests are for testing by the author\n",
            "t/author-pod-syntax.t .... skipped: these tests are for testing by the author\n",
            "t/DNAMutation.t .......... ok\n",
            "t/RNAChange.t ............ ok\n",
            "t/SeqDiff.t .............. ok\n",
            "t/SNP.t .................. ok\n",
            "t/Variation_IO.t ......... ok\n",
            "All tests successful.\n",
            "Files=12, Tests=221,  4 wallclock secs ( 0.07 usr  0.03 sys +  3.00 cusr  0.37 csys =  3.47 CPU)\n",
            "Result: PASS\n",
            "  CJFIELDS/Bio-Variation-1.7.5.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for CJFIELDS/Bio-Variation-1.7.5.tar.gz\n",
            "Manifying 1 pod document\n",
            "Manifying 12 pod documents\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Variation.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Variation/AAChange.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Variation/SeqDiff.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Variation/IO.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Variation/Allele.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Variation/DNAMutation.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Variation/SNP.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Variation/RNAChange.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Variation/VariantI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Variation/AAReverseMutate.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Variation/IO/xml.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Variation/IO/flat.pm\n",
            "Installing /usr/local/man/man1/bp_flanks.1p\n",
            "Installing /usr/local/man/man3/Bio::Variation.3pm\n",
            "Installing /usr/local/man/man3/Bio::Variation::AAReverseMutate.3pm\n",
            "Installing /usr/local/man/man3/Bio::Variation::DNAMutation.3pm\n",
            "Installing /usr/local/man/man3/Bio::Variation::IO::xml.3pm\n",
            "Installing /usr/local/man/man3/Bio::Variation::VariantI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Variation::AAChange.3pm\n",
            "Installing /usr/local/man/man3/Bio::Variation::RNAChange.3pm\n",
            "Installing /usr/local/man/man3/Bio::Variation::IO.3pm\n",
            "Installing /usr/local/man/man3/Bio::Variation::SeqDiff.3pm\n",
            "Installing /usr/local/man/man3/Bio::Variation::SNP.3pm\n",
            "Installing /usr/local/man/man3/Bio::Variation::IO::flat.3pm\n",
            "Installing /usr/local/man/man3/Bio::Variation::Allele.3pm\n",
            "Installing /usr/local/bin/bp_flanks\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  CJFIELDS/Bio-Variation-1.7.5.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "  CJFIELDS/Bio-Cluster-1.7.3.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/Bio-Cluster-1.7.3-0\n",
            "  CJFIELDS/Bio-Cluster-1.7.3.tar.gz\n",
            "  Has already been prepared\n",
            "Running make for C/CJ/CJFIELDS/Bio-Cluster-1.7.3.tar.gz\n",
            "cp lib/Bio/Cluster/FamilyI.pm blib/lib/Bio/Cluster/FamilyI.pm\n",
            "cp lib/Bio/ClusterIO/unigene.pm blib/lib/Bio/ClusterIO/unigene.pm\n",
            "cp lib/Bio/ClusterIO/dbsnp.pm blib/lib/Bio/ClusterIO/dbsnp.pm\n",
            "cp lib/Bio/ClusterIO.pm blib/lib/Bio/ClusterIO.pm\n",
            "cp lib/Bio/Cluster/SequenceFamily.pm blib/lib/Bio/Cluster/SequenceFamily.pm\n",
            "cp lib/Bio/Cluster/UniGene.pm blib/lib/Bio/Cluster/UniGene.pm\n",
            "cp lib/Bio/Cluster/UniGeneI.pm blib/lib/Bio/Cluster/UniGeneI.pm\n",
            "cp lib/Bio/ClusterI.pm blib/lib/Bio/ClusterI.pm\n",
            "cp lib/Bio/Cluster.pm blib/lib/Bio/Cluster.pm\n",
            "cp lib/Bio/Cluster/ClusterFactory.pm blib/lib/Bio/Cluster/ClusterFactory.pm\n",
            "Manifying 10 pod documents\n",
            "  CJFIELDS/Bio-Cluster-1.7.3.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for CJFIELDS/Bio-Cluster-1.7.3.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/00-compile.t ......... ok\n",
            "t/author-mojibake.t .... skipped: these tests are for testing by the author\n",
            "t/author-pod-syntax.t .. skipped: these tests are for testing by the author\n",
            "t/Cluster-UniGene.t .... ok\n",
            "t/ClusterIO-unigene.t .. ok\n",
            "t/ClusterIO.t .......... ok\n",
            "t/SequenceFamily.t ..... ok\n",
            "All tests successful.\n",
            "Files=7, Tests=114,  2 wallclock secs ( 0.05 usr  0.01 sys +  1.20 cusr  0.23 csys =  1.49 CPU)\n",
            "Result: PASS\n",
            "  CJFIELDS/Bio-Cluster-1.7.3.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for CJFIELDS/Bio-Cluster-1.7.3.tar.gz\n",
            "Manifying 10 pod documents\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/ClusterI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Cluster.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/ClusterIO.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/ClusterIO/unigene.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/ClusterIO/dbsnp.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Cluster/SequenceFamily.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Cluster/UniGene.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Cluster/UniGeneI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Cluster/FamilyI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Cluster/ClusterFactory.pm\n",
            "Installing /usr/local/man/man3/Bio::Cluster.3pm\n",
            "Installing /usr/local/man/man3/Bio::ClusterIO::dbsnp.3pm\n",
            "Installing /usr/local/man/man3/Bio::ClusterI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Cluster::FamilyI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Cluster::UniGeneI.3pm\n",
            "Installing /usr/local/man/man3/Bio::Cluster::SequenceFamily.3pm\n",
            "Installing /usr/local/man/man3/Bio::ClusterIO.3pm\n",
            "Installing /usr/local/man/man3/Bio::Cluster::ClusterFactory.3pm\n",
            "Installing /usr/local/man/man3/Bio::ClusterIO::unigene.3pm\n",
            "Installing /usr/local/man/man3/Bio::Cluster::UniGene.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  CJFIELDS/Bio-Cluster-1.7.3.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "  CJFIELDS/Bio-ASN1-EntrezGene-1.73.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/Bio-ASN1-EntrezGene-1.73-0\n",
            "  CJFIELDS/Bio-ASN1-EntrezGene-1.73.tar.gz\n",
            "  Has already been prepared\n",
            "Running make for C/CJ/CJFIELDS/Bio-ASN1-EntrezGene-1.73.tar.gz\n",
            "cp lib/Bio/ASN1/EntrezGene.pm blib/lib/Bio/ASN1/EntrezGene.pm\n",
            "cp lib/Bio/ASN1/EntrezGene/Indexer.pm blib/lib/Bio/ASN1/EntrezGene/Indexer.pm\n",
            "cp lib/Bio/ASN1/Sequence/Indexer.pm blib/lib/Bio/ASN1/Sequence/Indexer.pm\n",
            "cp lib/Bio/ASN1/Sequence.pm blib/lib/Bio/ASN1/Sequence.pm\n",
            "cp lib/Bio/SeqIO/entrezgene.pm blib/lib/Bio/SeqIO/entrezgene.pm\n",
            "Manifying 5 pod documents\n",
            "  CJFIELDS/Bio-ASN1-EntrezGene-1.73.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for CJFIELDS/Bio-ASN1-EntrezGene-1.73.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/00-compile.t ........... ok\n",
            "t/author-eol.t ........... skipped: these tests are for testing by the author\n",
            "t/author-mojibake.t ...... skipped: these tests are for testing by the author\n",
            "t/author-no-tabs.t ....... skipped: these tests are for testing by the author\n",
            "t/author-pod-coverage.t .. skipped: these tests are for testing by the author\n",
            "t/author-pod-syntax.t .... skipped: these tests are for testing by the author\n",
            "t/entrezgene.t ........... ok\n",
            "t/testindexer.t .......... ok\n",
            "t/testparser.t ........... # \n",
            "# \n",
            "# Test parsers (Bio::ASN1::EntrezGene, Bio::ASN1::Sequence),\n",
            "# parsing and method call:\n",
            "t/testparser.t ........... 1/10 # \n",
            "# \n",
            "# First testing gene parser:\n",
            "# \n",
            "# \n",
            "# Now testing sequence parser:\n",
            "t/testparser.t ........... ok\n",
            "All tests successful.\n",
            "Files=9, Tests=1448,  3 wallclock secs ( 0.22 usr  0.03 sys +  2.38 cusr  0.22 csys =  2.85 CPU)\n",
            "Result: PASS\n",
            "  CJFIELDS/Bio-ASN1-EntrezGene-1.73.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for CJFIELDS/Bio-ASN1-EntrezGene-1.73.tar.gz\n",
            "Manifying 5 pod documents\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/SeqIO/entrezgene.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/ASN1/EntrezGene.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/ASN1/Sequence.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/ASN1/Sequence/Indexer.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/ASN1/EntrezGene/Indexer.pm\n",
            "Installing /usr/local/man/man3/Bio::ASN1::EntrezGene.3pm\n",
            "Installing /usr/local/man/man3/Bio::SeqIO::entrezgene.3pm\n",
            "Installing /usr/local/man/man3/Bio::ASN1::Sequence.3pm\n",
            "Installing /usr/local/man/man3/Bio::ASN1::Sequence::Indexer.3pm\n",
            "Installing /usr/local/man/man3/Bio::ASN1::EntrezGene::Indexer.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  CJFIELDS/Bio-ASN1-EntrezGene-1.73.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "Running install for module 'CGI'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/L/LE/LEEJO/CGI-4.56.tar.gz\n",
            "Checksum for /root/.cpan/sources/authors/id/L/LE/LEEJO/CGI-4.56.tar.gz ok\n",
            "Configuring L/LE/LEEJO/CGI-4.56.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Warning: prerequisite Test::NoWarnings 0 not found.\n",
            "Warning: LINKTYPE set to '', no longer necessary\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for CGI\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  LEEJO/CGI-4.56.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for L/LE/LEEJO/CGI-4.56.tar.gz\n",
            "---- Unsatisfied dependencies detected during ----\n",
            "----           LEEJO/CGI-4.56.tar.gz          ----\n",
            "    Test::NoWarnings [build_requires]\n",
            "Running install for module 'Test::NoWarnings'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/H/HA/HAARG/Test-NoWarnings-1.06.tar.gz\n",
            "Checksum for /root/.cpan/sources/authors/id/H/HA/HAARG/Test-NoWarnings-1.06.tar.gz ok\n",
            "Configuring H/HA/HAARG/Test-NoWarnings-1.06.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for Test::NoWarnings\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  HAARG/Test-NoWarnings-1.06.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for H/HA/HAARG/Test-NoWarnings-1.06.tar.gz\n",
            "cp lib/Test/NoWarnings/Warning.pm blib/lib/Test/NoWarnings/Warning.pm\n",
            "cp lib/Test/NoWarnings.pm blib/lib/Test/NoWarnings.pm\n",
            "Manifying 1 pod document\n",
            "  HAARG/Test-NoWarnings-1.06.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for HAARG/Test-NoWarnings-1.06.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/01_compile.t ....... ok\n",
            "t/02_none.t .......... ok\n",
            "t/03_end.t ........... ok\n",
            "t/04_no_tests.t ...... ok\n",
            "t/05_no_end.t ........ ok\n",
            "t/06_fork.t .......... ok\n",
            "t/07_done_testing.t .. ok\n",
            "All tests successful.\n",
            "Files=7, Tests=48,  1 wallclock secs ( 0.03 usr  0.01 sys +  0.48 cusr  0.07 csys =  0.59 CPU)\n",
            "Result: PASS\n",
            "  HAARG/Test-NoWarnings-1.06.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for HAARG/Test-NoWarnings-1.06.tar.gz\n",
            "Manifying 1 pod document\n",
            "Installing /usr/local/share/perl/5.30.0/Test/NoWarnings.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Test/NoWarnings/Warning.pm\n",
            "Installing /usr/local/man/man3/Test::NoWarnings.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  HAARG/Test-NoWarnings-1.06.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "  LEEJO/CGI-4.56.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/CGI-4.56-0\n",
            "  LEEJO/CGI-4.56.tar.gz\n",
            "  Has already been prepared\n",
            "Running make for L/LE/LEEJO/CGI-4.56.tar.gz\n",
            "cp lib/CGI/File/Temp.pm blib/lib/CGI/File/Temp.pm\n",
            "cp lib/Fh.pm blib/lib/Fh.pm\n",
            "cp lib/CGI/Util.pm blib/lib/CGI/Util.pm\n",
            "cp lib/CGI/Carp.pm blib/lib/CGI/Carp.pm\n",
            "cp lib/CGI/HTML/Functions.pod blib/lib/CGI/HTML/Functions.pod\n",
            "cp lib/CGI/Cookie.pm blib/lib/CGI/Cookie.pm\n",
            "cp lib/CGI/Push.pm blib/lib/CGI/Push.pm\n",
            "cp lib/CGI.pod blib/lib/CGI.pod\n",
            "cp lib/CGI.pm blib/lib/CGI.pm\n",
            "cp lib/CGI/Pretty.pm blib/lib/CGI/Pretty.pm\n",
            "cp lib/CGI/HTML/Functions.pm blib/lib/CGI/HTML/Functions.pm\n",
            "Manifying 7 pod documents\n",
            "  LEEJO/CGI-4.56.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for LEEJO/CGI-4.56.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t t/headers/*.t\n",
            "t/append_query.t ............ ok\n",
            "t/arbitrary_handles.t ....... ok\n",
            "t/autoescape.t .............. ok\n",
            "t/can.t ..................... ok\n",
            "t/carp.t .................... 1/76 [Sat Mar  4 07:16:33 2023] carp.t: foo at /root/.cpan/build/CGI-4.56-0/blib/lib/CGI/Carp.pm line 360.\n",
            "t/carp.t .................... ok\n",
            "t/cgi.t ..................... 1/25 # cgi-lib.pl routines\n",
            "private_tempfiles has been deprecated at /root/.cpan/build/CGI-4.56-0/blib/lib/CGI.pm line 3238.\n",
            "t/cgi.t ..................... ok\n",
            "t/changes.t ................. skipped: Test::CPAN::Changes required for this test\n",
            "t/charset.t ................. ok\n",
            "t/checkbox_group.t .......... ok\n",
            "t/command_line.t ............ ok\n",
            "t/compiles_pod.t ............ ok\n",
            "t/cookie.t .................. ok\n",
            "t/delete.t .................. ok\n",
            "t/Dump.t .................... ok\n",
            "t/end_form.t ................ ok\n",
            "t/form.t .................... ok\n",
            "t/function.t ................ ok\n",
            "t/gh-155.t .................. ok\n",
            "t/headers.t ................. ok\n",
            "t/headers/attachment.t ...... ok\n",
            "t/headers/charset.t ......... ok\n",
            "t/headers/cookie.t .......... ok\n",
            "t/headers/default.t ......... ok\n",
            "t/headers/nph.t ............. ok\n",
            "t/headers/p3p.t ............. ok\n",
            "t/headers/target.t .......... ok\n",
            "t/headers/type.t ............ ok\n",
            "t/hidden.t .................. ok\n",
            "t/html.t .................... ok\n",
            "t/html_functions.t .......... ok\n",
            "t/http.t .................... ok\n",
            "t/init.t .................... ok\n",
            "t/multipart_globals.t ....... ok\n",
            "t/multipart_init.t .......... ok\n",
            "t/multipart_start.t ......... ok\n",
            "t/no_tabindex.t ............. ok\n",
            "t/param_fetch.t ............. ok\n",
            "t/param_list_context.t ...... ok\n",
            "t/popup_menu.t .............. ok\n",
            "t/postdata.t ................ ok\n",
            "t/pretty.t .................. CGI::Pretty is DEPRECATED and will be removed in a future release. Please see https://github.com/leejo/CGI.pm/issues/162 for more information at /root/.cpan/build/CGI-4.56-0/blib/lib/CGI/Pretty.pm line 20.\n",
            "t/pretty.t .................. ok\n",
            "t/push.t .................... ok\n",
            "t/query_string.t ............ ok\n",
            "t/redirect_query_string.t ... ok\n",
            "t/request.t ................. ok\n",
            "t/rt-31107.t ................ ok\n",
            "t/rt-52469.t ................ ok\n",
            "t/rt-57524.t ................ ok\n",
            "t/rt-75628.t ................ ok\n",
            "t/rt-84767.t ................ ok\n",
            "t/save_read_roundtrip.t ..... ok\n",
            "t/sorted.t .................. ok\n",
            "t/start_end_asterisk.t ...... ok\n",
            "t/start_end_end.t ........... ok\n",
            "t/start_end_start.t ......... ok\n",
            "t/unescapeHTML.t ............ ok\n",
            "t/upload.t .................. ok\n",
            "t/upload_quoted_unquoted.t .. ok\n",
            "t/uploadInfo.t .............. ok\n",
            "t/url.t ..................... ok\n",
            "t/user_agent.t .............. ok\n",
            "t/utf8.t .................... ok\n",
            "t/util-58.t ................. ok\n",
            "t/util.t .................... ok\n",
            "All tests successful.\n",
            "\n",
            "Test Summary Report\n",
            "-------------------\n",
            "t/compiles_pod.t          (Wstat: 0 Tests: 18 Failed: 0)\n",
            "  TODO passed:   3, 6\n",
            "Files=64, Tests=1588,  7 wallclock secs ( 0.37 usr  0.08 sys +  5.48 cusr  0.95 csys =  6.88 CPU)\n",
            "Result: PASS\n",
            "  LEEJO/CGI-4.56.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for LEEJO/CGI-4.56.tar.gz\n",
            "Manifying 7 pod documents\n",
            "Installing /usr/local/share/perl/5.30.0/Fh.pm\n",
            "Installing /usr/local/share/perl/5.30.0/CGI.pod\n",
            "Installing /usr/local/share/perl/5.30.0/CGI.pm\n",
            "Installing /usr/local/share/perl/5.30.0/CGI/Carp.pm\n",
            "Installing /usr/local/share/perl/5.30.0/CGI/Push.pm\n",
            "Installing /usr/local/share/perl/5.30.0/CGI/Cookie.pm\n",
            "Installing /usr/local/share/perl/5.30.0/CGI/Pretty.pm\n",
            "Installing /usr/local/share/perl/5.30.0/CGI/Util.pm\n",
            "Installing /usr/local/share/perl/5.30.0/CGI/HTML/Functions.pod\n",
            "Installing /usr/local/share/perl/5.30.0/CGI/HTML/Functions.pm\n",
            "Installing /usr/local/share/perl/5.30.0/CGI/File/Temp.pm\n",
            "Installing /usr/local/man/man3/CGI.3pm\n",
            "Installing /usr/local/man/man3/CGI::Push.3pm\n",
            "Installing /usr/local/man/man3/CGI::HTML::Functions.3pm\n",
            "Installing /usr/local/man/man3/CGI::Cookie.3pm\n",
            "Installing /usr/local/man/man3/CGI::Carp.3pm\n",
            "Installing /usr/local/man/man3/CGI::Pretty.3pm\n",
            "Installing /usr/local/man/man3/CGI::Util.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  LEEJO/CGI-4.56.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "Running install for module 'Cache::FileCache'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/R/RJ/RJBS/Cache-Cache-1.08.tar.gz\n",
            "Checksum for /root/.cpan/sources/authors/id/R/RJ/RJBS/Cache-Cache-1.08.tar.gz ok\n",
            "Configuring R/RJ/RJBS/Cache-Cache-1.08.tar.gz with Makefile.PL\n",
            "Checking for IPC::ShareLite... not found\n",
            "\n",
            "  NOTE: Your system does not seem to have IPC::ShareLite installed.\n",
            "  This module is a prerequisite for the SharedMemoryCache\n",
            "  implementations.  However, since it is not available on all\n",
            "  platforms, the Cache::Cache module does not list it as an explicit\n",
            "  dependency.  If you are able to build IPC::ShareLite (available on\n",
            "  CPAN), please install it now and re run this Makefile.PL.\n",
            "  Automatically disabling the tests for the SharedMemoryCache.\n",
            "\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Warning: prerequisite Digest::SHA1 2.02 not found.\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for Cache::Cache\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "------------------------------------------------------------------\n",
            "Please be patient while running \"make test\" as the full test suite\n",
            "takes roughly two minutes to complete.\n",
            "------------------------------------------------------------------\n",
            "  RJBS/Cache-Cache-1.08.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for R/RJ/RJBS/Cache-Cache-1.08.tar.gz\n",
            "---- Unsatisfied dependencies detected during ----\n",
            "----       RJBS/Cache-Cache-1.08.tar.gz       ----\n",
            "    Digest::SHA1 [requires]\n",
            "Running install for module 'Digest::SHA1'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/G/GA/GAAS/Digest-SHA1-2.13.tar.gz\n",
            "Checksum for /root/.cpan/sources/authors/id/G/GA/GAAS/Digest-SHA1-2.13.tar.gz ok\n",
            "Configuring G/GA/GAAS/Digest-SHA1-2.13.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for Digest::SHA1\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  GAAS/Digest-SHA1-2.13.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for G/GA/GAAS/Digest-SHA1-2.13.tar.gz\n",
            "cp SHA1.pm blib/lib/Digest/SHA1.pm\n",
            "Running Mkbootstrap for SHA1 ()\n",
            "chmod 644 \"SHA1.bs\"\n",
            "\"/usr/bin/perl\" -MExtUtils::Command::MM -e 'cp_nonempty' -- SHA1.bs blib/arch/auto/Digest/SHA1/SHA1.bs 644\n",
            "\"/usr/bin/perl\" \"/usr/share/perl/5.30/ExtUtils/xsubpp\"  -typemap '/usr/share/perl/5.30/ExtUtils/typemap' -typemap '/root/.cpan/build/Digest-SHA1-2.13-0/typemap'  SHA1.xs > SHA1.xsc\n",
            "mv SHA1.xsc SHA1.c\n",
            "x86_64-linux-gnu-gcc -c   -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\\\"2.13\\\" -DXS_VERSION=\\\"2.13\\\" -fPIC \"-I/usr/lib/x86_64-linux-gnu/perl/5.30/CORE\"   SHA1.c\n",
            "In file included from \u001b[01m\u001b[K/usr/lib/x86_64-linux-gnu/perl/5.30/CORE/perl.h:2508\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSHA1.xs:6\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[KSHA1.xs:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Digest__SHA1_sha1\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[K/usr/lib/x86_64-linux-gnu/perl/5.30/CORE/handy.h:501:23:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Knull argument where non-null required (argument 1) [\u001b[01;35m\u001b[K-Wnonnull\u001b[m\u001b[K]\n",
            "  501 | #define strEQ(s1,s2) (\u001b[01;35m\u001b[Kstrcmp\u001b[m\u001b[K(s1,s2) == 0)\n",
            "      |                       \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSHA1.xs:576:27:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kin expansion of macro ‘\u001b[01m\u001b[KstrEQ\u001b[m\u001b[K’\n",
            "  576 |       if (SvOBJECT(sv) && \u001b[01;36m\u001b[KstrEQ\u001b[m\u001b[K(HvNAME(SvSTASH(sv)), \"Digest::SHA1\"))\n",
            "      |                           \u001b[01;36m\u001b[K^~~~~\u001b[m\u001b[K\n",
            "rm -f blib/arch/auto/Digest/SHA1/SHA1.so\n",
            "x86_64-linux-gnu-gcc  -shared -L/usr/local/lib -fstack-protector-strong  SHA1.o  -o blib/arch/auto/Digest/SHA1/SHA1.so  \\\n",
            "      \\\n",
            "  \n",
            "chmod 755 blib/arch/auto/Digest/SHA1/SHA1.so\n",
            "Manifying 1 pod document\n",
            "  GAAS/Digest-SHA1-2.13.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for GAAS/Digest-SHA1-2.13.tar.gz\n",
            "\"/usr/bin/perl\" -MExtUtils::Command::MM -e 'cp_nonempty' -- SHA1.bs blib/arch/auto/Digest/SHA1/SHA1.bs 644\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/badfile.t .. ok\n",
            "t/bits.t ..... ok\n",
            "t/sha1.t ..... ok\n",
            "All tests successful.\n",
            "Files=3, Tests=17,  0 wallclock secs ( 0.02 usr  0.01 sys +  0.02 cusr  0.01 csys =  0.06 CPU)\n",
            "Result: PASS\n",
            "  GAAS/Digest-SHA1-2.13.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for GAAS/Digest-SHA1-2.13.tar.gz\n",
            "\"/usr/bin/perl\" -MExtUtils::Command::MM -e 'cp_nonempty' -- SHA1.bs blib/arch/auto/Digest/SHA1/SHA1.bs 644\n",
            "Manifying 1 pod document\n",
            "Files found in blib/arch: installing files in blib/lib into architecture dependent library tree\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Digest/SHA1/SHA1.so\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/Digest/SHA1.pm\n",
            "Installing /usr/local/man/man3/Digest::SHA1.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  GAAS/Digest-SHA1-2.13.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "  RJBS/Cache-Cache-1.08.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/Cache-Cache-1.08-0\n",
            "  RJBS/Cache-Cache-1.08.tar.gz\n",
            "  Has already been prepared\n",
            "Running make for R/RJ/RJBS/Cache-Cache-1.08.tar.gz\n",
            "cp lib/Cache/CacheUtils.pm blib/lib/Cache/CacheUtils.pm\n",
            "cp lib/Cache/SizeAwareMemoryCache.pm blib/lib/Cache/SizeAwareMemoryCache.pm\n",
            "cp lib/Cache/Object.pm blib/lib/Cache/Object.pm\n",
            "cp lib/Cache/BaseCache.pm blib/lib/Cache/BaseCache.pm\n",
            "cp lib/Cache/BaseCacheTester.pm blib/lib/Cache/BaseCacheTester.pm\n",
            "cp lib/Cache/SizeAwareCache.pm blib/lib/Cache/SizeAwareCache.pm\n",
            "cp lib/Cache/CacheMetaData.pm blib/lib/Cache/CacheMetaData.pm\n",
            "cp lib/Cache/CacheSizer.pm blib/lib/Cache/CacheSizer.pm\n",
            "cp lib/Cache/FileBackend.pm blib/lib/Cache/FileBackend.pm\n",
            "cp lib/Cache/MemoryCache.pm blib/lib/Cache/MemoryCache.pm\n",
            "cp lib/Cache/Cache.pm blib/lib/Cache/Cache.pm\n",
            "cp lib/Cache/FileCache.pm blib/lib/Cache/FileCache.pm\n",
            "cp lib/Cache/NullCache.pm blib/lib/Cache/NullCache.pm\n",
            "cp lib/Cache/SharedMemoryBackend.pm blib/lib/Cache/SharedMemoryBackend.pm\n",
            "cp lib/Cache/SizeAwareFileCache.pm blib/lib/Cache/SizeAwareFileCache.pm\n",
            "cp lib/Cache/MemoryBackend.pm blib/lib/Cache/MemoryBackend.pm\n",
            "cp lib/Cache/CacheTester.pm blib/lib/Cache/CacheTester.pm\n",
            "cp lib/Cache/SharedMemoryCache.pm blib/lib/Cache/SharedMemoryCache.pm\n",
            "cp lib/Cache/SizeAwareCacheTester.pm blib/lib/Cache/SizeAwareCacheTester.pm\n",
            "cp lib/Cache/SizeAwareSharedMemoryCache.pm blib/lib/Cache/SizeAwareSharedMemoryCache.pm\n",
            "Manifying 20 pod documents\n",
            "  RJBS/Cache-Cache-1.08.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for RJBS/Cache-Cache-1.08.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/1_test_cache_interface.t t/2_test_memory_cache.t t/3_test_file_cache.t t/5_test_size_aware_file_cache.t t/6_test_size_aware_memory_cache.t\n",
            "t/1_test_cache_interface.t .......... ok\n",
            "t/2_test_memory_cache.t ............. ok\n",
            "t/3_test_file_cache.t ............... ok\n",
            "t/5_test_size_aware_file_cache.t .... ok\n",
            "t/6_test_size_aware_memory_cache.t .. ok\n",
            "All tests successful.\n",
            "Files=5, Tests=167, 82 wallclock secs ( 0.06 usr  0.02 sys +  0.22 cusr  0.09 csys =  0.39 CPU)\n",
            "Result: PASS\n",
            "  RJBS/Cache-Cache-1.08.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for RJBS/Cache-Cache-1.08.tar.gz\n",
            "Manifying 20 pod documents\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/SizeAwareFileCache.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/NullCache.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/SizeAwareCache.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/BaseCacheTester.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/FileBackend.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/Object.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/CacheTester.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/SharedMemoryCache.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/BaseCache.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/SizeAwareSharedMemoryCache.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/SharedMemoryBackend.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/SizeAwareMemoryCache.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/Cache.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/CacheMetaData.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/CacheUtils.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/SizeAwareCacheTester.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/CacheSizer.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/MemoryCache.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/FileCache.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Cache/MemoryBackend.pm\n",
            "Installing /usr/local/man/man3/Cache::MemoryCache.3pm\n",
            "Installing /usr/local/man/man3/Cache::FileBackend.3pm\n",
            "Installing /usr/local/man/man3/Cache::CacheSizer.3pm\n",
            "Installing /usr/local/man/man3/Cache::CacheMetaData.3pm\n",
            "Installing /usr/local/man/man3/Cache::Cache.3pm\n",
            "Installing /usr/local/man/man3/Cache::SizeAwareCache.3pm\n",
            "Installing /usr/local/man/man3/Cache::CacheUtils.3pm\n",
            "Installing /usr/local/man/man3/Cache::NullCache.3pm\n",
            "Installing /usr/local/man/man3/Cache::SizeAwareSharedMemoryCache.3pm\n",
            "Installing /usr/local/man/man3/Cache::SharedMemoryBackend.3pm\n",
            "Installing /usr/local/man/man3/Cache::SizeAwareFileCache.3pm\n",
            "Installing /usr/local/man/man3/Cache::SizeAwareMemoryCache.3pm\n",
            "Installing /usr/local/man/man3/Cache::CacheTester.3pm\n",
            "Installing /usr/local/man/man3/Cache::Object.3pm\n",
            "Installing /usr/local/man/man3/Cache::BaseCache.3pm\n",
            "Installing /usr/local/man/man3/Cache::MemoryBackend.3pm\n",
            "Installing /usr/local/man/man3/Cache::FileCache.3pm\n",
            "Installing /usr/local/man/man3/Cache::BaseCacheTester.3pm\n",
            "Installing /usr/local/man/man3/Cache::SizeAwareCacheTester.3pm\n",
            "Installing /usr/local/man/man3/Cache::SharedMemoryCache.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  RJBS/Cache-Cache-1.08.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "Running install for module 'LWP::Protocol::https'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/O/OA/OALDERS/LWP-Protocol-https-6.10.tar.gz\n",
            "Checksum for /root/.cpan/sources/authors/id/O/OA/OALDERS/LWP-Protocol-https-6.10.tar.gz ok\n",
            "Configuring O/OA/OALDERS/LWP-Protocol-https-6.10.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Warning: prerequisite IO::Socket::SSL 1.54 not found.\n",
            "Warning: prerequisite IO::Socket::SSL::Utils 0 not found.\n",
            "Warning: prerequisite Mozilla::CA 20180117 not found.\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for LWP::Protocol::https\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  OALDERS/LWP-Protocol-https-6.10.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for O/OA/OALDERS/LWP-Protocol-https-6.10.tar.gz\n",
            "---- Unsatisfied dependencies detected during ----\n",
            "----  OALDERS/LWP-Protocol-https-6.10.tar.gz  ----\n",
            "    IO::Socket::SSL [requires]\n",
            "    IO::Socket::SSL::Utils [build_requires]\n",
            "    Mozilla::CA [requires]\n",
            "Running install for module 'IO::Socket::SSL'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/S/SU/SULLR/IO-Socket-SSL-2.081.tar.gz\n",
            "Fetching with LWP:\n",
            "HASH(0x55627e85aba0)authors/id/S/SU/SULLR/CHECKSUMS\n",
            "Fetching with LWP:\n",
            "HASH(0x55627e85aba0)authors/id/S/SU/SULLR/CHECKSUMS.gz\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/S/SU/SULLR/CHECKSUMS\n",
            "Checksum for /root/.cpan/sources/authors/id/S/SU/SULLR/IO-Socket-SSL-2.081.tar.gz ok\n",
            "---- Unsatisfied dependencies detected during ----\n",
            "----     SULLR/IO-Socket-SSL-2.081.tar.gz     ----\n",
            "    Net::SSLeay [build_requires]\n",
            "Running install for module 'Net::SSLeay'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/C/CH/CHRISN/Net-SSLeay-1.92.tar.gz\n",
            "Fetching with LWP:\n",
            "HASH(0x55627eb0af48)authors/id/C/CH/CHRISN/CHECKSUMS\n",
            "Fetching with LWP:\n",
            "HASH(0x55627eb0af48)authors/id/C/CH/CHRISN/CHECKSUMS.gz\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/C/CH/CHRISN/CHECKSUMS\n",
            "Checksum for /root/.cpan/sources/authors/id/C/CH/CHRISN/Net-SSLeay-1.92.tar.gz ok\n",
            "Configuring C/CH/CHRISN/Net-SSLeay-1.92.tar.gz with Makefile.PL\n",
            "Do you want to run external tests?\n",
            "These tests *will* *fail* if you do not have network connectivity. [n] n\n",
            "*** Be sure to use the same compiler and options to compile your OpenSSL, perl,\n",
            "    and Net::SSLeay. Mixing and matching compilers is not supported.\n",
            "*** Found OpenSSL-1.1.1f installed in /usr\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for Net::SSLeay\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  CHRISN/Net-SSLeay-1.92.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for C/CH/CHRISN/Net-SSLeay-1.92.tar.gz\n",
            "cp lib/Net/SSLeay.pm blib/lib/Net/SSLeay.pm\n",
            "AutoSplitting blib/lib/Net/SSLeay.pm (blib/lib/auto/Net/SSLeay)\n",
            "blib/lib/Net/SSLeay.pm: some names are not unique when truncated to 8 characters:\n",
            " directory blib/lib/auto/Net/SSLeay:\n",
            "  do_https3.al, do_https2.al, do_https4.al, do_https.al truncate to do_https\n",
            "  do_httpx3.al, do_httpx2.al, do_httpx4.al truncate to do_httpx\n",
            "  get_https.al, get_https3.al, get_https4.al, get_http.al, get_http3.al, get_http4.al, get_httpx.al, get_httpx3.al, get_httpx4.al truncate to get_http\n",
            "  head_https.al, head_https3.al, head_https4.al, head_http.al, head_http3.al, head_http4.al, head_httpx.al, head_httpx3.al, head_httpx4.al truncate to head_htt\n",
            "  post_https.al, post_https3.al, post_https4.al, post_http.al, post_http3.al, post_http4.al, post_httpx.al, post_httpx3.al, post_httpx4.al truncate to post_htt\n",
            "  put_https.al, put_https3.al, put_https4.al, put_http.al, put_http3.al, put_http4.al, put_httpx.al, put_httpx3.al, put_httpx4.al truncate to put_http\n",
            "  ssl_read_all.al, ssl_read_until.al, ssl_read_CRLF.al truncate to ssl_read\n",
            "  ssl_write_all.al, ssl_write_CRLF.al truncate to ssl_writ\n",
            "  tcp_read_all.al, tcp_read_until.al, tcp_read_CRLF.al truncate to tcp_read\n",
            "  tcp_write_all.al, tcp_write_CRLF.al truncate to tcp_writ\n",
            "cp lib/Net/SSLeay/Handle.pm blib/lib/Net/SSLeay/Handle.pm\n",
            "cp lib/Net/SSLeay.pod blib/lib/Net/SSLeay.pod\n",
            "Running Mkbootstrap for SSLeay ()\n",
            "chmod 644 \"SSLeay.bs\"\n",
            "\"/usr/bin/perl\" -MExtUtils::Command::MM -e 'cp_nonempty' -- SSLeay.bs blib/arch/auto/Net/SSLeay/SSLeay.bs 644\n",
            "\"/usr/bin/perl\" \"/usr/share/perl/5.30/ExtUtils/xsubpp\"  -typemap '/usr/share/perl/5.30/ExtUtils/typemap' -typemap '/root/.cpan/build/Net-SSLeay-1.92-0/typemap'  SSLeay.xs > SSLeay.xsc\n",
            "mv SSLeay.xsc SSLeay.c\n",
            "x86_64-linux-gnu-gcc -c  -I\"/usr/include\" -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\\\"1.92\\\" -DXS_VERSION=\\\"1.92\\\" -fPIC \"-I/usr/lib/x86_64-linux-gnu/perl/5.30/CORE\"   SSLeay.c\n",
            "\u001b[01m\u001b[KSSLeay.xs:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_CTX_tlsv1_new\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.xs:2088:6:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KTLSv1_method\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            " 2088 |      \u001b[01;35m\u001b[KRETVAL\u001b[m\u001b[K = SSL_CTX_new (TLSv1_method());\n",
            "      |      \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/ssl.h:1877:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            " 1877 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(__owur const SSL_METHOD *TLSv1_method(void)) /* TLSv1.0 */\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.xs:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_CTX_tlsv1_1_new\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.xs:2097:6:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KTLSv1_1_method\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            " 2097 |      \u001b[01;35m\u001b[KRETVAL\u001b[m\u001b[K = SSL_CTX_new (TLSv1_1_method());\n",
            "      |      \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/ssl.h:1883:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            " 1883 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(__owur const SSL_METHOD *TLSv1_1_method(void)) /* TLSv1.1 */\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.xs:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_CTX_tlsv1_2_new\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.xs:2108:6:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KTLSv1_2_method\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            " 2108 |      \u001b[01;35m\u001b[KRETVAL\u001b[m\u001b[K = SSL_CTX_new (TLSv1_2_method());\n",
            "      |      \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/ssl.h:1889:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            " 1889 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(__owur const SSL_METHOD *TLSv1_2_method(void)) /* TLSv1.2 */\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.xs:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_RAND_pseudo_bytes\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.xs:3302:9:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KRAND_pseudo_bytes\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            " 3302 |         \u001b[01;35m\u001b[Krc\u001b[m\u001b[K = RAND_pseudo_bytes(random, num);\n",
            "      |         \u001b[01;35m\u001b[K^~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/rand.h:44:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            "   44 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(int RAND_pseudo_bytes(unsigned char *buf, int num))\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_X509_CRL_get_lastUpdate\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.c:7438:2:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KX509_CRL_get_lastUpdate\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            " 7438 |  \u001b[01;35m\u001b[KRETVAL\u001b[m\u001b[K = X509_CRL_get_lastUpdate(x);\n",
            "      |  \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/x509.h:727:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            "  727 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(ASN1_TIME *X509_CRL_get_lastUpdate(X509_CRL *crl))\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_X509_CRL_get_nextUpdate\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.c:7457:2:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KX509_CRL_get_nextUpdate\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            " 7457 |  \u001b[01;35m\u001b[KRETVAL\u001b[m\u001b[K = X509_CRL_get_nextUpdate(x);\n",
            "      |  \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/x509.h:728:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            "  728 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(ASN1_TIME *X509_CRL_get_nextUpdate(X509_CRL *crl))\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.xs:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_X509_get_subjectAltNames\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.xs:4060:26:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KASN1_STRING_data\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            " 4060 |                          \u001b[01;35m\u001b[KPUSHs\u001b[m\u001b[K(sv_2mortal(newSVpv((const char*)ASN1_STRING_data(subjAltNameDN->d.otherName->value->value.utf8string), ASN1_STRING_length(subjAltNameDN->d.otherName->value->value.utf8string))));\n",
            "      |                          \u001b[01;35m\u001b[K^~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/asn1.h:554:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            "  554 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(unsigned char *ASN1_STRING_data(ASN1_STRING *x))\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.xs:4069:26:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KASN1_STRING_data\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            " 4069 |                          \u001b[01;35m\u001b[KPUSHs\u001b[m\u001b[K(sv_2mortal(newSVpv((const char*)ASN1_STRING_data(subjAltNameDN->d.ia5), ASN1_STRING_length(subjAltNameDN->d.ia5))));\n",
            "      |                          \u001b[01;35m\u001b[K^~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/asn1.h:554:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            "  554 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(unsigned char *ASN1_STRING_data(ASN1_STRING *x))\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.xs:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_P_X509_get_crl_distribution_points\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.xs:4133:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KASN1_STRING_data\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            " 4133 |                         \u001b[01;35m\u001b[KXPUSHs\u001b[m\u001b[K(sv_2mortal(newSVpv((char*)ASN1_STRING_data(gn->d.ia5),ASN1_STRING_length(gn->d.ia5))));\n",
            "      |                         \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/asn1.h:554:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            "  554 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(unsigned char *ASN1_STRING_data(ASN1_STRING *x))\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.xs:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_P_X509_get_ocsp_uri\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.xs:4174:3:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KASN1_STRING_data\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            " 4174 |   \u001b[01;35m\u001b[KXPUSHs\u001b[m\u001b[K(sv_2mortal(newSVpv(\n",
            "      |   \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/asn1.h:554:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            "  554 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(unsigned char *ASN1_STRING_data(ASN1_STRING *x))\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.xs:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_P_ASN1_STRING_get\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.xs:4524:9:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KASN1_STRING_data\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            " 4524 |         \u001b[01;35m\u001b[Ku8\u001b[m\u001b[K = newSVpv((const char*)ASN1_STRING_data(s), ASN1_STRING_length(s));\n",
            "      |         \u001b[01;35m\u001b[K^~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/asn1.h:554:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            "  554 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(unsigned char *ASN1_STRING_data(ASN1_STRING *x))\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_TLSv1_method\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.c:10722:2:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KTLSv1_method\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            "10722 |  \u001b[01;35m\u001b[KRETVAL\u001b[m\u001b[K = TLSv1_method();\n",
            "      |  \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/ssl.h:1877:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            " 1877 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(__owur const SSL_METHOD *TLSv1_method(void)) /* TLSv1.0 */\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_TLSv1_server_method\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.c:10739:2:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KTLSv1_server_method\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            "10739 |  \u001b[01;35m\u001b[KRETVAL\u001b[m\u001b[K = TLSv1_server_method();\n",
            "      |  \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/ssl.h:1878:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            " 1878 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(__owur const SSL_METHOD *TLSv1_server_method(void))\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_TLSv1_client_method\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.c:10756:2:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KTLSv1_client_method\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            "10756 |  \u001b[01;35m\u001b[KRETVAL\u001b[m\u001b[K = TLSv1_client_method();\n",
            "      |  \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/ssl.h:1879:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            " 1879 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(__owur const SSL_METHOD *TLSv1_client_method(void))\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_TLSv1_1_method\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.c:10776:2:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KTLSv1_1_method\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            "10776 |  \u001b[01;35m\u001b[KRETVAL\u001b[m\u001b[K = TLSv1_1_method();\n",
            "      |  \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/ssl.h:1883:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            " 1883 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(__owur const SSL_METHOD *TLSv1_1_method(void)) /* TLSv1.1 */\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_TLSv1_1_server_method\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.c:10793:2:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KTLSv1_1_server_method\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            "10793 |  \u001b[01;35m\u001b[KRETVAL\u001b[m\u001b[K = TLSv1_1_server_method();\n",
            "      |  \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/ssl.h:1884:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            " 1884 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(__owur const SSL_METHOD *TLSv1_1_server_method(void))\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_TLSv1_1_client_method\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.c:10810:2:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KTLSv1_1_client_method\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            "10810 |  \u001b[01;35m\u001b[KRETVAL\u001b[m\u001b[K = TLSv1_1_client_method();\n",
            "      |  \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/ssl.h:1885:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            " 1885 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(__owur const SSL_METHOD *TLSv1_1_client_method(void))\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_TLSv1_2_method\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.c:10831:2:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KTLSv1_2_method\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            "10831 |  \u001b[01;35m\u001b[KRETVAL\u001b[m\u001b[K = TLSv1_2_method();\n",
            "      |  \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/ssl.h:1889:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            " 1889 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(__owur const SSL_METHOD *TLSv1_2_method(void)) /* TLSv1.2 */\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_TLSv1_2_server_method\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.c:10848:2:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KTLSv1_2_server_method\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            "10848 |  \u001b[01;35m\u001b[KRETVAL\u001b[m\u001b[K = TLSv1_2_server_method();\n",
            "      |  \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/ssl.h:1890:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            " 1890 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(__owur const SSL_METHOD *TLSv1_2_server_method(void))\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[KSSLeay.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KXS_Net__SSLeay_TLSv1_2_client_method\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSSLeay.c:10865:2:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KTLSv1_2_client_method\u001b[m\u001b[K’ is deprecated [\u001b[01;35m\u001b[K-Wdeprecated-declarations\u001b[m\u001b[K]\n",
            "10865 |  \u001b[01;35m\u001b[KRETVAL\u001b[m\u001b[K = TLSv1_2_client_method();\n",
            "      |  \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/openssl/e_os2.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[K/usr/include/openssl/err.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[KSSLeay.xs:163\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/openssl/ssl.h:1891:1:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdeclared here\n",
            " 1891 | \u001b[01;36m\u001b[KDEPRECATEDIN_1_1_0\u001b[m\u001b[K(__owur const SSL_METHOD *TLSv1_2_client_method(void))\n",
            "      | \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "rm -f blib/arch/auto/Net/SSLeay/SSLeay.so\n",
            "x86_64-linux-gnu-gcc  -shared -L/usr -L/usr/lib64 -L/usr/lib -L/usr/local/lib -fstack-protector-strong  SSLeay.o  -o blib/arch/auto/Net/SSLeay/SSLeay.so  \\\n",
            "   -L/usr -L/usr/lib64 -L/usr/lib -lssl -lcrypto -lz   \\\n",
            "  \n",
            "chmod 755 blib/arch/auto/Net/SSLeay/SSLeay.so\n",
            "Manifying 2 pod documents\n",
            "  CHRISN/Net-SSLeay-1.92.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for CHRISN/Net-SSLeay-1.92.tar.gz\n",
            "\"/usr/bin/perl\" -MExtUtils::Command::MM -e 'cp_nonempty' -- SSLeay.bs blib/arch/auto/Net/SSLeay/SSLeay.bs 644\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/local/*.t t/handle/local/*.t\n",
            "t/handle/local/05_use.t ..................... ok\n",
            "t/local/01_pod.t ............................ skipped: Test::Pod 1.41 required for testing pod\n",
            "t/local/02_pod_coverage.t ................... skipped: These tests are for only for release candidate testing. Enable with RELEASE_TESTING=1\n",
            "t/local/03_use.t ............................ 1/1 # \n",
            "# Testing Net::SSLeay 1.92\n",
            "# \n",
            "# Perl information:\n",
            "#   Version:         '5.030000'\n",
            "#   Executable path: '/usr/bin/perl'\n",
            "# \n",
            "# Library version with OpenSSL_version_num():\n",
            "#   OPENSSL_VERSION_NUMBER: '0x1010106f'\n",
            "# \n",
            "# Library information with SSLeay_version() and OpenSSL_version():\n",
            "#   SSLEAY_VERSION:              'OpenSSL 1.1.1f  31 Mar 2020'\n",
            "#   SSLEAY_CFLAGS:               'compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -Wa,--noexecstack -g -O2 -fdebug-prefix-map=/build/openssl-7mt7n4/openssl-1.1.1f=. -fstack-protector-strong -Wformat -Werror=format-security -DOPENSSL_TLS_SECURITY_LEVEL=2 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2'\n",
            "#   SSLEAY_BUILT_ON:             'built on: Mon Feb  6 17:57:17 2023 UTC'\n",
            "#   SSLEAY_PLATFORM:             'platform: debian-amd64'\n",
            "#   SSLEAY_DIR:                  'OPENSSLDIR: \"/usr/lib/ssl\"'\n",
            "#   OPENSSL_ENGINES_DIR:         'ENGINESDIR: \"/usr/lib/x86_64-linux-gnu/engines-1.1\"'\n",
            "t/local/03_use.t ............................ ok\n",
            "t/local/04_basic.t .......................... ok\n",
            "t/local/05_passwd_cb.t ...................... ok\n",
            "t/local/06_tcpecho.t ........................ ok\n",
            "t/local/07_sslecho.t ........................ ok\n",
            "t/local/08_pipe.t ........................... ok\n",
            "t/local/09_ctx_new.t ........................ ok\n",
            "t/local/10_rand.t ........................... ok\n",
            "t/local/11_read.t ........................... ok\n",
            "t/local/15_bio.t ............................ ok\n",
            "t/local/20_functions.t ...................... ok\n",
            "t/local/21_constants.t ...................... ok\n",
            "t/local/22_provider.t ....................... skipped: no support for providers\n",
            "t/local/22_provider_try_load.t .............. skipped: no support for providers\n",
            "t/local/22_provider_try_load_zero_retain.t .. skipped: no support for providers\n",
            "t/local/30_error.t .......................... ok\n",
            "t/local/31_rsa_generate_key.t ............... ok\n",
            "t/local/32_x509_get_cert_info.t ............. ok\n",
            "t/local/33_x509_create_cert.t ............... ok\n",
            "t/local/34_x509_crl.t ....................... ok\n",
            "t/local/35_ephemeral.t ...................... skipped: LibreSSL and OpenSSL 1.1.0 removed support for ephemeral/temporary RSA private keys\n",
            "t/local/36_verify.t ......................... ok\n",
            "t/local/37_asn1_time.t ...................... ok\n",
            "t/local/38_priv-key.t ....................... ok\n",
            "t/local/39_pkcs12.t ......................... ok\n",
            "t/local/40_npn_support.t .................... ok\n",
            "t/local/41_alpn_support.t ................... ok\n",
            "t/local/42_info_callback.t .................. ok\n",
            "t/local/43_misc_functions.t ................. ok\n",
            "t/local/44_sess.t ........................... ok\n",
            "t/local/45_exporter.t ....................... ok\n",
            "t/local/46_msg_callback.t ................... ok\n",
            "t/local/47_keylog.t ......................... ok\n",
            "t/local/50_digest.t ......................... ok\n",
            "t/local/61_threads-cb-crash.t ............... ok\n",
            "t/local/62_threads-ctx_new-deadlock.t ....... ok\n",
            "t/local/63_ec_key_generate_key.t ............ ok\n",
            "t/local/64_ticket_sharing.t ................. ok\n",
            "t/local/65_security_level.t ................. ok\n",
            "t/local/65_ticket_sharing_2.t ............... ok\n",
            "t/local/66_curves.t ......................... ok\n",
            "t/local/kwalitee.t .......................... skipped: These tests are for only for release candidate testing. Enable with RELEASE_TESTING=1\n",
            "All tests successful.\n",
            "Files=45, Tests=2501,  9 wallclock secs ( 0.43 usr  0.06 sys +  6.68 cusr  0.92 csys =  8.09 CPU)\n",
            "Result: PASS\n",
            "  CHRISN/Net-SSLeay-1.92.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for CHRISN/Net-SSLeay-1.92.tar.gz\n",
            "\"/usr/bin/perl\" -MExtUtils::Command::MM -e 'cp_nonempty' -- SSLeay.bs blib/arch/auto/Net/SSLeay/SSLeay.bs 644\n",
            "Manifying 2 pod documents\n",
            "Files found in blib/arch: installing files in blib/lib into architecture dependent library tree\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/SSLeay.so\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/get_https4.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/get_http4.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/post_https4.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/ssl_write_all.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/sslcat.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/get_httpx4.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/randomize.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/open_proxy_tcp_connection.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/do_https4.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/make_headers.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/put_httpx3.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/put_http.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/post_http.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/get_http3.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/tcp_read_all.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/get_https.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/httpx_cat.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/do_httpx2.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/autosplit.ix\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/ssl_read_until.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/do_https.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/put_https3.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/want_write.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/ssl_read_all.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/want_read.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/post_http3.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/new_x_ctx.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/do_httpx4.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/get_http.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/open_tcp_connection.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/put_https4.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/head_http.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/set_server_cert_and_key.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/post_https.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/do_https3.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/get_httpx.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/tcp_read_until.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/put_http3.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/make_form.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/do_httpx3.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/get_httpx3.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/initialize.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/put_httpx4.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/want_nothing.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/ssl_read_CRLF.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/put_https.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/debug_read.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/post_https3.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/http_cat.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/set_cert_and_key.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/tcp_write_all.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/head_http4.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/put_httpx.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/head_httpx4.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/post_httpx.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/put_http4.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/head_https4.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/get_https3.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/post_httpx3.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/post_httpx4.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/want_X509_lookup.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/set_proxy.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/head_https.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/tcpxcat.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/do_https2.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/head_httpx.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/tcpcat.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/head_httpx3.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/dump_peer_certificate.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/tcp_write_CRLF.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/head_http3.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/tcp_read_CRLF.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/post_http4.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/https_cat.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/ssl_write_CRLF.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/auto/Net/SSLeay/head_https3.al\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/Net/SSLeay.pod\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/Net/SSLeay.pm\n",
            "Installing /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/Net/SSLeay/Handle.pm\n",
            "Installing /usr/local/man/man3/Net::SSLeay.3pm\n",
            "Installing /usr/local/man/man3/Net::SSLeay::Handle.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  CHRISN/Net-SSLeay-1.92.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "  SULLR/IO-Socket-SSL-2.081.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/IO-Socket-SSL-2.081-0\n",
            "Configuring S/SU/SULLR/IO-Socket-SSL-2.081.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Warning: prerequisite Mozilla::CA 0 not found.\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for IO::Socket::SSL\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  SULLR/IO-Socket-SSL-2.081.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for S/SU/SULLR/IO-Socket-SSL-2.081.tar.gz\n",
            "---- Unsatisfied dependencies detected during ----\n",
            "----     SULLR/IO-Socket-SSL-2.081.tar.gz     ----\n",
            "    Mozilla::CA [requires]\n",
            "Running install for module 'Mozilla::CA'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/H/HA/HAARG/Mozilla-CA-20221114.tar.gz\n",
            "Checksum for /root/.cpan/sources/authors/id/H/HA/HAARG/Mozilla-CA-20221114.tar.gz ok\n",
            "Configuring H/HA/HAARG/Mozilla-CA-20221114.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for Mozilla::CA\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  HAARG/Mozilla-CA-20221114.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for H/HA/HAARG/Mozilla-CA-20221114.tar.gz\n",
            "cp lib/Mozilla/CA.pm blib/lib/Mozilla/CA.pm\n",
            "cp lib/Mozilla/CA/cacert.pem blib/lib/Mozilla/CA/cacert.pem\n",
            "Manifying 1 pod document\n",
            "  HAARG/Mozilla-CA-20221114.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for HAARG/Mozilla-CA-20221114.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/locate-file.t .. ok\n",
            "All tests successful.\n",
            "Files=1, Tests=2,  0 wallclock secs ( 0.02 usr  0.01 sys +  0.06 cusr  0.01 csys =  0.10 CPU)\n",
            "Result: PASS\n",
            "  HAARG/Mozilla-CA-20221114.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for HAARG/Mozilla-CA-20221114.tar.gz\n",
            "Manifying 1 pod document\n",
            "Installing /usr/local/share/perl/5.30.0/Mozilla/CA.pm\n",
            "Installing /usr/local/share/perl/5.30.0/Mozilla/CA/cacert.pem\n",
            "Installing /usr/local/man/man3/Mozilla::CA.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  HAARG/Mozilla-CA-20221114.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "  SULLR/IO-Socket-SSL-2.081.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/IO-Socket-SSL-2.081-0\n",
            "  SULLR/IO-Socket-SSL-2.081.tar.gz\n",
            "  Has already been prepared\n",
            "Running make for S/SU/SULLR/IO-Socket-SSL-2.081.tar.gz\n",
            "cp lib/IO/Socket/SSL.pm blib/lib/IO/Socket/SSL.pm\n",
            "cp lib/IO/Socket/SSL/Utils.pm blib/lib/IO/Socket/SSL/Utils.pm\n",
            "cp lib/IO/Socket/SSL.pod blib/lib/IO/Socket/SSL.pod\n",
            "cp lib/IO/Socket/SSL/Intercept.pm blib/lib/IO/Socket/SSL/Intercept.pm\n",
            "cp lib/IO/Socket/SSL/PublicSuffix.pm blib/lib/IO/Socket/SSL/PublicSuffix.pm\n",
            "Manifying 4 pod documents\n",
            "  SULLR/IO-Socket-SSL-2.081.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for SULLR/IO-Socket-SSL-2.081.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t t/external/*.t\n",
            "t/01loadmodule.t .................. 1/3 # openssl version compiled=0x1010106f linked=0x1010106f -- OpenSSL 1.1.1f  31 Mar 2020\n",
            "# Net::SSLeay version=1.92\n",
            "# parent IO::Socket::IP version=0.39\n",
            "t/01loadmodule.t .................. ok\n",
            "t/acceptSSL-timeout.t ............. ok\n",
            "t/alpn.t .......................... ok\n",
            "t/auto_verify_hostname.t .......... ok\n",
            "t/cert_formats.t .................. ok\n",
            "t/cert_no_file.t .................. ok\n",
            "t/compatibility.t ................. ok\n",
            "t/connectSSL-timeout.t ............ ok\n",
            "t/core.t .......................... ok\n",
            "t/dhe.t ........................... ok\n",
            "t/ecdhe.t ......................... ok\n",
            "t/external/ocsp.t ................. # tcp connect to www.chksum.de:443 ok\n",
            "# fingerprint matches\n",
            "# validation with default CA w/o OCSP ok\n",
            "# got stapled response as expected\n",
            "# validation with default CA with OCSP defaults ok\n",
            "# validation with default CA with OCSP full chain ok\n",
            "t/external/ocsp.t ................. 1/3 # tcp connect to www.bild.de:443 ok\n",
            "# tcp connect to revoked.grc.com:443 ok\n",
            "t/external/ocsp.t ................. ok\n",
            "t/external/usable_ca.t ............ # found 124 CA certs\n",
            "# have root CA for www.bild.de in store\n",
            "# 5 connections to www.bild.de ok\n",
            "t/external/usable_ca.t ............ 1/21 # have root CA for www.yahoo.com in store\n",
            "# 5 connections to www.yahoo.com ok\n",
            "# have root CA for www.comdirect.de in store\n",
            "# 5 connections to www.comdirect.de ok\n",
            "t/external/usable_ca.t ............ 7/21 # have root CA for meine.deutsche-bank.de in store\n",
            "# 5 connections to meine.deutsche-bank.de ok\n",
            "# fingerprint meine.deutsche-bank.de matches\n",
            "t/external/usable_ca.t ............ 10/21 # have root CA for www.twitter.com in store\n",
            "# 5 connections to www.twitter.com ok\n",
            "# have root CA for www.facebook.com in store\n",
            "# 5 connections to www.facebook.com ok\n",
            "# have root CA for www.live.com in store\n",
            "# 5 connections to www.live.com ok\n",
            "# fingerprint www.live.com matches\n",
            "t/external/usable_ca.t ............ ok\n",
            "t/io-socket-inet6.t ............... skipped: no IO::Socket::INET6 available\n",
            "t/io-socket-ip.t .................. skipped: (no reason given)\n",
            "t/memleak_bad_handshake.t ......... ok\n",
            "t/mitm.t .......................... ok\n",
            "t/multiple-cert-rsa-ecc.t ......... ok\n",
            "t/nonblock.t ...................... ok\n",
            "t/npn.t ........................... ok\n",
            "t/plain_upgrade_downgrade.t ....... 1/15 # -- test: newINET start_SSL stop_SSL start_SSL\n",
            "# server accepted new client\n",
            "# wait for initial data from client\n",
            "# got 0x666f6f from client\n",
            "# server: got plain data at start of connection\n",
            "# server: TLS upgrade\n",
            "# server: TLS downgrade\n",
            "# server: TLS upgrade#2\n",
            "# -- test: newSSL stop_SSL connect_SSL\n",
            "# server accepted new client\n",
            "# wait for initial data from client\n",
            "# got 0x160301 from client\n",
            "# server: TLS upgrade\n",
            "# server: TLS downgrade\n",
            "# server: TLS upgrade#2\n",
            "# -- test: newSSL:0 connect_SSL stop_SSL connect_SSL\n",
            "# server accepted new client\n",
            "# wait for initial data from client\n",
            "# got 0x666f6f from client\n",
            "# server: got plain data at start of connection\n",
            "# server: TLS upgrade\n",
            "# server: TLS downgrade\n",
            "# server: TLS upgrade#2\n",
            "# -- test: newSSL:0 start_SSL stop_SSL connect_SSL\n",
            "# server accepted new client\n",
            "# wait for initial data from client\n",
            "# got 0x666f6f from client\n",
            "# server: got plain data at start of connection\n",
            "# server: TLS upgrade\n",
            "# server: TLS downgrade\n",
            "# server: TLS upgrade#2\n",
            "# server accepted new client\n",
            "# wait for initial data from client\n",
            "# got 0x656e64 from client\n",
            "# client requested end of tests\n",
            "t/plain_upgrade_downgrade.t ....... ok\n",
            "t/protocol_version.t .............. 1/? # looks like OpenSSL was compiled without SSLv3 support\n",
            "# looks like OpenSSL was compiled without TLSv1 support\n",
            "# failed to accept TLSv1\n",
            "# looks like OpenSSL was compiled without TLSv1_1 support\n",
            "# failed to accept TLSv1_1\n",
            "t/protocol_version.t .............. ok\n",
            "t/public_suffix_lib_encode_idn.t .. ok\n",
            "t/public_suffix_lib_libidn.t ...... ok\n",
            "t/public_suffix_lib_uri.t ......... ok\n",
            "t/public_suffix_ssl.t ............. ok\n",
            "t/readline.t ...................... ok\n",
            "t/session_cache.t ................. ok\n",
            "t/session_ticket.t ................ # listen at 127.0.0.1:59087\n",
            "# listen at 127.0.0.1:52517\n",
            "# connect to 0: success reuse=0 version=TLSv1_3\n",
            "t/session_ticket.t ................ 1/6 # connect to 0: success reuse=1 version=TLSv1_3\n",
            "# connect to 1: success reuse=1 version=TLSv1_3\n",
            "# connect to 1: success reuse=0 version=TLSv1_3\n",
            "# connect to 0: success reuse=0 version=TLSv1_3\n",
            "# connect to 0: success reuse=1 version=TLSv1_3\n",
            "t/session_ticket.t ................ ok\n",
            "t/sessions.t ...................... ok\n",
            "t/set_curves.t .................... ok\n",
            "t/signal-readline.t ............... ok\n",
            "t/sni.t ........................... ok\n",
            "t/sni_verify.t .................... ok\n",
            "t/start-stopssl.t ................. ok\n",
            "t/startssl-failed.t ............... ok\n",
            "t/startssl.t ...................... ok\n",
            "t/sysread_write.t ................. ok\n",
            "t/verify_fingerprint.t ............ ok\n",
            "t/verify_hostname.t ............... ok\n",
            "t/verify_hostname_standalone.t .... ok\n",
            "t/verify_partial_chain.t .......... ok\n",
            "All tests successful.\n",
            "Files=42, Tests=806, 48 wallclock secs ( 0.28 usr  0.06 sys +  5.95 cusr  1.10 csys =  7.39 CPU)\n",
            "Result: PASS\n",
            "  SULLR/IO-Socket-SSL-2.081.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for SULLR/IO-Socket-SSL-2.081.tar.gz\n",
            "Manifying 4 pod documents\n",
            "Installing /usr/local/share/perl/5.30.0/IO/Socket/SSL.pm\n",
            "Installing /usr/local/share/perl/5.30.0/IO/Socket/SSL.pod\n",
            "Installing /usr/local/share/perl/5.30.0/IO/Socket/SSL/Intercept.pm\n",
            "Installing /usr/local/share/perl/5.30.0/IO/Socket/SSL/PublicSuffix.pm\n",
            "Installing /usr/local/share/perl/5.30.0/IO/Socket/SSL/Utils.pm\n",
            "Installing /usr/local/man/man3/IO::Socket::SSL.3pm\n",
            "Installing /usr/local/man/man3/IO::Socket::SSL::PublicSuffix.3pm\n",
            "Installing /usr/local/man/man3/IO::Socket::SSL::Utils.3pm\n",
            "Installing /usr/local/man/man3/IO::Socket::SSL::Intercept.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  SULLR/IO-Socket-SSL-2.081.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "IO::Socket::SSL::Utils is up to date (2.015).\n",
            "  OALDERS/LWP-Protocol-https-6.10.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/LWP-Protocol-https-6.10-0\n",
            "  OALDERS/LWP-Protocol-https-6.10.tar.gz\n",
            "  Has already been prepared\n",
            "Running make for O/OA/OALDERS/LWP-Protocol-https-6.10.tar.gz\n",
            "cp lib/LWP/Protocol/https.pm blib/lib/LWP/Protocol/https.pm\n",
            "Manifying 1 pod document\n",
            "  OALDERS/LWP-Protocol-https-6.10.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for OALDERS/LWP-Protocol-https-6.10.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/00-report-prereqs.t .. # \n",
            "# Versions for all modules listed in MYMETA.json (including optional ones):\n",
            "# \n",
            "# === Configure Requires ===\n",
            "# \n",
            "#     Module              Want Have\n",
            "#     ------------------- ---- ----\n",
            "#     ExtUtils::MakeMaker  any 7.34\n",
            "# \n",
            "# === Configure Suggests ===\n",
            "# \n",
            "#     Module      Want Have\n",
            "#     -------- ------- ----\n",
            "#     JSON::PP 2.27300 4.02\n",
            "# \n",
            "# === Build Requires ===\n",
            "# \n",
            "#     Module              Want Have\n",
            "#     ------------------- ---- ----\n",
            "#     ExtUtils::MakeMaker  any 7.34\n",
            "# \n",
            "# === Test Requires ===\n",
            "# \n",
            "#     Module                 Want     Have\n",
            "#     ---------------------- ---- --------\n",
            "#     ExtUtils::MakeMaker     any     7.34\n",
            "#     File::Spec              any     3.78\n",
            "#     File::Temp              any   0.2309\n",
            "#     IO::Select              any     1.40\n",
            "#     IO::Socket::INET        any     1.40\n",
            "#     IO::Socket::SSL        1.54    2.081\n",
            "#     IO::Socket::SSL::Utils  any    2.015\n",
            "#     LWP::UserAgent         6.06     6.68\n",
            "#     Socket                  any    2.027\n",
            "#     Test::More             0.96 1.302192\n",
            "#     Test::RequiresInternet  any     0.05\n",
            "#     warnings                any     1.44\n",
            "# \n",
            "# === Test Recommends ===\n",
            "# \n",
            "#     Module         Want     Have\n",
            "#     ---------- -------- --------\n",
            "#     CPAN::Meta 2.120900 2.150010\n",
            "# \n",
            "# === Runtime Requires ===\n",
            "# \n",
            "#     Module                  Want     Have\n",
            "#     ------------------- -------- --------\n",
            "#     IO::Socket::SSL         1.54    2.081\n",
            "#     LWP::Protocol::http      any     6.68\n",
            "#     LWP::UserAgent          6.06     6.68\n",
            "#     Mozilla::CA         20180117 20221114\n",
            "#     Net::HTTPS                 6     6.22\n",
            "#     base                     any     2.27\n",
            "#     strict                   any     1.11\n",
            "# \n",
            "# === Other Modules ===\n",
            "# \n",
            "#     Module      Have\n",
            "#     ----------- ----\n",
            "#     Net::SSLeay 1.92\n",
            "# \n",
            "t/00-report-prereqs.t .. ok\n",
            "t/apache.t ............. Use of uninitialized value $proto in socket at /usr/local/share/perl/5.30.0/Test/RequiresInternet.pm line 55.\n",
            "t/apache.t ............. ok\n",
            "t/https_proxy.t ........ ok\n",
            "t/method_in_san.t ...... ok\n",
            "All tests successful.\n",
            "Files=4, Tests=76,  1 wallclock secs ( 0.04 usr  0.01 sys +  0.93 cusr  0.12 csys =  1.10 CPU)\n",
            "Result: PASS\n",
            "  OALDERS/LWP-Protocol-https-6.10.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for OALDERS/LWP-Protocol-https-6.10.tar.gz\n",
            "Manifying 1 pod document\n",
            "Installing /usr/local/share/perl/5.30.0/LWP/Protocol/https.pm\n",
            "Installing /usr/local/man/man3/LWP::Protocol::https.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  OALDERS/LWP-Protocol-https-6.10.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "  CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/Bio-DB-NCBIHelper-1.7.7-0\n",
            "  CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz\n",
            "  Has already been prepared\n",
            "Running make for C/CJ/CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz\n",
            "cp lib/Bio/DB/EntrezGene.pm blib/lib/Bio/DB/EntrezGene.pm\n",
            "cp lib/Bio/DB/NCBIHelper.pm blib/lib/Bio/DB/NCBIHelper.pm\n",
            "cp lib/Bio/DB/Taxonomy/entrez.pm blib/lib/Bio/DB/Taxonomy/entrez.pm\n",
            "cp lib/Bio/DB/Query/GenBank.pm blib/lib/Bio/DB/Query/GenBank.pm\n",
            "cp lib/Bio/DB/GenPept.pm blib/lib/Bio/DB/GenPept.pm\n",
            "cp lib/Bio/DB/GenBank.pm blib/lib/Bio/DB/GenBank.pm\n",
            "cp bin/bp_biofetch_genbank_proxy blib/script/bp_biofetch_genbank_proxy\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_biofetch_genbank_proxy\n",
            "cp bin/bp_download_query_genbank blib/script/bp_download_query_genbank\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_download_query_genbank\n",
            "cp bin/bp_query_entrez_taxa blib/script/bp_query_entrez_taxa\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_query_entrez_taxa\n",
            "Manifying 3 pod documents\n",
            "Manifying 6 pod documents\n",
            "  CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/00-compile.t ......... ok\n",
            "t/author-mojibake.t .... skipped: these tests are for testing by the author\n",
            "t/author-pod-syntax.t .. skipped: these tests are for testing by the author\n",
            "t/EntrezGene.t ......... Use of uninitialized value $proto in socket at /usr/local/share/perl/5.30.0/Test/RequiresInternet.pm line 55.\n",
            "t/EntrezGene.t ......... ok\n",
            "t/GenBank.t ............ Use of uninitialized value $proto in socket at /usr/local/share/perl/5.30.0/Test/RequiresInternet.pm line 55.\n",
            "t/GenBank.t ............ ok\n",
            "t/GenPept.t ............ Use of uninitialized value $proto in socket at /usr/local/share/perl/5.30.0/Test/RequiresInternet.pm line 55.\n",
            "t/GenPept.t ............ ok\n",
            "t/Query-Genbank.t ...... Use of uninitialized value $proto in socket at /usr/local/share/perl/5.30.0/Test/RequiresInternet.pm line 55.\n",
            "t/Query-Genbank.t ...... ok\n",
            "t/RefSeq.t ............. Use of uninitialized value $proto in socket at /usr/local/share/perl/5.30.0/Test/RequiresInternet.pm line 55.\n",
            "t/RefSeq.t ............. ok\n",
            "t/Taxonomy.t ........... Use of uninitialized value $proto in socket at /usr/local/share/perl/5.30.0/Test/RequiresInternet.pm line 55.\n",
            "t/Taxonomy.t ........... 7/? \n",
            "#   Failed test at t/Taxonomy.t line 48.\n",
            "t/Taxonomy.t ........... 43/? \n",
            "#   Failed test at t/Taxonomy.t line 102.\n",
            "#          got: 'Actinomycetes'\n",
            "#     expected: 'Actinomycetia'\n",
            "t/Taxonomy.t ........... 68/? \n",
            "#   Failed test 'Correct: No hit'\n",
            "#   at t/Taxonomy.t line 181.\n",
            "#          got: '2890685'\n",
            "#     expected: 'No hit'\n",
            "t/Taxonomy.t ........... 69/? # Looks like you failed 3 tests of 69.\n",
            "t/Taxonomy.t ........... Dubious, test returned 3 (wstat 768, 0x300)\n",
            "Failed 3/69 subtests \n",
            "\n",
            "Test Summary Report\n",
            "-------------------\n",
            "t/Taxonomy.t         (Wstat: 768 Tests: 69 Failed: 3)\n",
            "  Failed tests:  18, 44, 68\n",
            "  Non-zero exit status: 3\n",
            "Files=9, Tests=175, 154 wallclock secs ( 0.08 usr  0.02 sys +  4.98 cusr  0.64 csys =  5.72 CPU)\n",
            "Result: FAIL\n",
            "Failed 1/9 test programs. 3/175 subtests failed.\n",
            "make: *** [Makefile:945: test_dynamic] Error 3\n",
            "  CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz\n",
            "  /usr/bin/make test -- NOT OK\n",
            "//hint// to see the cpan-testers results for installing this module, try:\n",
            "  reports CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz\n",
            "Running install for module 'Bio::DB::GenPept'\n",
            "  CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/Bio-DB-NCBIHelper-1.7.7-0\n",
            "  CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz\n",
            "  Has already been prepared\n",
            "  CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz\n",
            "  Has already been made\n",
            "  CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz\n",
            "  Won't repeat unsuccessful test during this command\n",
            "Running install for module 'Bio::DB::RefSeq'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/C/CJ/CJFIELDS/Bio-DB-RefSeq-1.7.4.tar.gz\n",
            "Checksum for /root/.cpan/sources/authors/id/C/CJ/CJFIELDS/Bio-DB-RefSeq-1.7.4.tar.gz ok\n",
            "Configuring C/CJ/CJFIELDS/Bio-DB-RefSeq-1.7.4.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for Bio::DB::RefSeq\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  CJFIELDS/Bio-DB-RefSeq-1.7.4.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for C/CJ/CJFIELDS/Bio-DB-RefSeq-1.7.4.tar.gz\n",
            "cp lib/Bio/DB/RefSeq.pm blib/lib/Bio/DB/RefSeq.pm\n",
            "Manifying 1 pod document\n",
            "  CJFIELDS/Bio-DB-RefSeq-1.7.4.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for CJFIELDS/Bio-DB-RefSeq-1.7.4.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/00-compile.t ........... ok\n",
            "t/author-mojibake.t ...... skipped: these tests are for testing by the author\n",
            "t/author-pod-coverage.t .. skipped: these tests are for testing by the author\n",
            "t/author-pod-syntax.t .... skipped: these tests are for testing by the author\n",
            "t/RefSeq.t ............... Use of uninitialized value $proto in socket at /usr/local/share/perl/5.30.0/Test/RequiresInternet.pm line 55.\n",
            "t/RefSeq.t ............... ok\n",
            "All tests successful.\n",
            "Files=5, Tests=7,  3 wallclock secs ( 0.03 usr  0.01 sys +  0.38 cusr  0.09 csys =  0.51 CPU)\n",
            "Result: PASS\n",
            "  CJFIELDS/Bio-DB-RefSeq-1.7.4.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for CJFIELDS/Bio-DB-RefSeq-1.7.4.tar.gz\n",
            "Manifying 1 pod document\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/RefSeq.pm\n",
            "Installing /usr/local/man/man3/Bio::DB::RefSeq.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  CJFIELDS/Bio-DB-RefSeq-1.7.4.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "Running install for module 'Bio::DB::SwissProt'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/C/CJ/CJFIELDS/Bio-DB-SwissProt-1.7.4.tar.gz\n",
            "Checksum for /root/.cpan/sources/authors/id/C/CJ/CJFIELDS/Bio-DB-SwissProt-1.7.4.tar.gz ok\n",
            "Configuring C/CJ/CJFIELDS/Bio-DB-SwissProt-1.7.4.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for Bio::DB::SwissProt\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  CJFIELDS/Bio-DB-SwissProt-1.7.4.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for C/CJ/CJFIELDS/Bio-DB-SwissProt-1.7.4.tar.gz\n",
            "cp lib/Bio/DB/SwissProt.pm blib/lib/Bio/DB/SwissProt.pm\n",
            "Manifying 1 pod document\n",
            "  CJFIELDS/Bio-DB-SwissProt-1.7.4.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for CJFIELDS/Bio-DB-SwissProt-1.7.4.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/00-compile.t ........... ok\n",
            "t/author-eol.t ........... skipped: these tests are for testing by the author\n",
            "t/author-mojibake.t ...... skipped: these tests are for testing by the author\n",
            "t/author-pod-coverage.t .. skipped: these tests are for testing by the author\n",
            "t/author-pod-syntax.t .... skipped: these tests are for testing by the author\n",
            "t/SwissProt.t ............ Use of uninitialized value $proto in socket at /usr/local/share/perl/5.30.0/Test/RequiresInternet.pm line 55.\n",
            "t/SwissProt.t ............ ok\n",
            "All tests successful.\n",
            "Files=6, Tests=23,  6 wallclock secs ( 0.04 usr  0.01 sys +  0.63 cusr  0.12 csys =  0.80 CPU)\n",
            "Result: PASS\n",
            "  CJFIELDS/Bio-DB-SwissProt-1.7.4.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for CJFIELDS/Bio-DB-SwissProt-1.7.4.tar.gz\n",
            "Manifying 1 pod document\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/DB/SwissProt.pm\n",
            "Installing /usr/local/man/man3/Bio::DB::SwissProt.3pm\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  CJFIELDS/Bio-DB-SwissProt-1.7.4.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "Bio::Root::Test is up to date (1.7.8).\n",
            "Bio::Root::Version is up to date (1.7.8).\n",
            "Bio::Seq is up to date (1.7.8).\n",
            "Bio::SeqIO is up to date (1.7.8).\n",
            "Running install for module 'Bio::Tools::Run::RemoteBlast'\n",
            "Fetching with LWP:\n",
            "http://www.cpan.org/authors/id/C/CJ/CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz\n",
            "Checksum for /root/.cpan/sources/authors/id/C/CJ/CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz ok\n",
            "Configuring C/CJ/CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz with Makefile.PL\n",
            "Checking if your kit is complete...\n",
            "Looks good\n",
            "Generating a Unix-style Makefile\n",
            "Writing Makefile for Bio::Tools::Run::RemoteBlast\n",
            "Writing MYMETA.yml and MYMETA.json\n",
            "  CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz\n",
            "  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK\n",
            "Running make for C/CJ/CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz\n",
            "cp lib/Bio/Tools/Run/RemoteBlast.pm blib/lib/Bio/Tools/Run/RemoteBlast.pm\n",
            "cp bin/bp_remote_blast.pl blib/script/bp_remote_blast.pl\n",
            "\"/usr/bin/perl\" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/bp_remote_blast.pl\n",
            "Manifying 1 pod document\n",
            "Manifying 1 pod document\n",
            "  CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/00-compile.t .................... ok\n",
            "t/author-mojibake.t ............... skipped: these tests are for testing by the author\n",
            "t/author-no-tabs.t ................ skipped: these tests are for testing by the author\n",
            "t/author-pod-syntax.t ............. skipped: these tests are for testing by the author\n",
            "t/release-RemoteBlast.t ........... skipped: these tests are for release candidate testing\n",
            "t/release-RemoteBlast_rpsblast.t .. skipped: these tests are for release candidate testing\n",
            "All tests successful.\n",
            "Files=6, Tests=2,  1 wallclock secs ( 0.04 usr  0.01 sys +  0.37 cusr  0.06 csys =  0.48 CPU)\n",
            "Result: PASS\n",
            "  CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz\n",
            "  /usr/bin/make test -- OK\n",
            "Running make install for CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz\n",
            "Manifying 1 pod document\n",
            "Manifying 1 pod document\n",
            "Installing /usr/local/share/perl/5.30.0/Bio/Tools/Run/RemoteBlast.pm\n",
            "Installing /usr/local/man/man1/bp_remote_blast.pl.1p\n",
            "Installing /usr/local/man/man3/Bio::Tools::Run::RemoteBlast.3pm\n",
            "Installing /usr/local/bin/bp_remote_blast.pl\n",
            "Appending installation info to /usr/local/lib/x86_64-linux-gnu/perl/5.30.0/perllocal.pod\n",
            "  CJFIELDS/Bio-Tools-Run-RemoteBlast-1.7.3.tar.gz\n",
            "  /usr/bin/make install  -- OK\n",
            "  CJFIELDS/Bio-Procedural-1.7.4.tar.gz\n",
            "  Has already been unwrapped into directory /root/.cpan/build/Bio-Procedural-1.7.4-0\n",
            "  CJFIELDS/Bio-Procedural-1.7.4.tar.gz\n",
            "  Has already been prepared\n",
            "Running make for C/CJ/CJFIELDS/Bio-Procedural-1.7.4.tar.gz\n",
            "Warning: Prerequisite 'Bio::DB::GenBank => 0' for 'CJFIELDS/Bio-Procedural-1.7.4.tar.gz' failed when processing 'CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz' with 'make_test => NO'. Continuing, but chances to succeed are limited.\n",
            "Warning: Prerequisite 'Bio::DB::GenPept => 0' for 'CJFIELDS/Bio-Procedural-1.7.4.tar.gz' failed when processing 'CJFIELDS/Bio-DB-NCBIHelper-1.7.7.tar.gz' with 'make_test => NO'. Continuing, but chances to succeed are limited.\n",
            "cp lib/Bio/Procedural.pm blib/lib/Bio/Procedural.pm\n",
            "cp lib/Bio/Perl.pm blib/lib/Bio/Perl.pm\n",
            "Manifying 2 pod documents\n",
            "  CJFIELDS/Bio-Procedural-1.7.4.tar.gz\n",
            "  /usr/bin/make -- OK\n",
            "Running make test for CJFIELDS/Bio-Procedural-1.7.4.tar.gz\n",
            "PERL_DL_NONLAZY=1 \"/usr/bin/perl\" \"-MExtUtils::Command::MM\" \"-MTest::Harness\" \"-e\" \"undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')\" t/*.t\n",
            "t/00-compile.t ........... ok\n",
            "t/author-eol.t ........... skipped: these tests are for testing by the author\n",
            "t/author-mojibake.t ...... skipped: these tests are for testing by the author\n",
            "t/author-no-tabs.t ....... skipped: these tests are for testing by the author\n",
            "t/author-pod-coverage.t .. skipped: these tests are for testing by the author\n",
            "t/author-pod-syntax.t .... skipped: these tests are for testing by the author\n",
            "t/Perl.t ................. 1/31 Use of uninitialized value $proto in socket at /usr/local/share/perl/5.30.0/Test/RequiresInternet.pm line 55.\n",
            "t/Perl.t ................. ok\n",
            "All tests successful.\n",
            "Files=7, Tests=33, 17 wallclock secs ( 0.05 usr  0.01 sys +  0.70 cusr  0.11 csys =  0.87 CPU)\n",
            "Result: PASS\n",
            "  CJFIELDS/Bio-Procedural-1.7.4.tar.gz\n",
            "Tests succeeded but 2 dependencies missing (Bio::DB::GenBank,Bio::DB::GenPept)\n",
            "  CJFIELDS/Bio-Procedural-1.7.4.tar.gz\n",
            "  [dependencies] -- NA\n",
            "Cloning into '/root/prokka'...\n",
            "remote: Enumerating objects: 1478, done.\u001b[K\n",
            "remote: Total 1478 (delta 0), reused 0 (delta 0), pack-reused 1478\u001b[K\n",
            "Receiving objects: 100% (1478/1478), 321.93 MiB | 16.07 MiB/s, done.\n",
            "Resolving deltas: 100% (901/901), done.\n",
            "Updating files: 100% (67/67), done.\n",
            "Can't locate XML/Simple.pm in @INC (you may need to install the XML::Simple module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.30.0 /usr/local/share/perl/5.30.0 /usr/lib/x86_64-linux-gnu/perl5/5.30 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.30 /usr/share/perl/5.30 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at /root/prokka/bin/prokka line 29.\n",
            "BEGIN failed--compilation aborted at /root/prokka/bin/prokka line 29.\n"
          ]
        }
      ],
      "source": [
        "!sudo apt-get install libdatetime-perl libxml-simple-perl libdigest-md5-perl git default-jre bioperl\n",
        "!sudo cpan Bio::Perl\n",
        "!git clone https://github.com/tseemann/prokka.git $HOME/prokka\n",
        "!$HOME/prokka/bin/prokka --setupdb"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!apt install libxml-simple-perl"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UQAgjXexIt-r",
        "outputId": "5cf5c187-7d60-4804-8c7d-e41b11deffe6"
      },
      "execution_count": 18,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  libauthen-sasl-perl libdata-dump-perl libencode-locale-perl\n",
            "  libfile-listing-perl libfont-afm-perl libhtml-form-perl libhtml-format-perl\n",
            "  libhtml-parser-perl libhtml-tagset-perl libhtml-tree-perl\n",
            "  libhttp-cookies-perl libhttp-daemon-perl libhttp-date-perl\n",
            "  libhttp-message-perl libhttp-negotiate-perl libio-html-perl\n",
            "  libio-socket-ssl-perl liblwp-mediatypes-perl liblwp-protocol-https-perl\n",
            "  libmailtools-perl libnet-http-perl libnet-smtp-ssl-perl libnet-ssleay-perl\n",
            "  libtry-tiny-perl liburi-perl libwww-perl libwww-robotrules-perl\n",
            "  libxml-libxml-perl libxml-namespacesupport-perl libxml-parser-perl\n",
            "  libxml-sax-base-perl libxml-sax-expat-perl libxml-sax-perl netbase\n",
            "  perl-openssl-defaults\n",
            "Suggested packages:\n",
            "  libdigest-hmac-perl libgssapi-perl libcrypt-ssleay-perl libauthen-ntlm-perl\n",
            "  libxml-sax-expatxs-perl\n",
            "The following NEW packages will be installed:\n",
            "  libauthen-sasl-perl libdata-dump-perl libencode-locale-perl\n",
            "  libfile-listing-perl libfont-afm-perl libhtml-form-perl libhtml-format-perl\n",
            "  libhtml-parser-perl libhtml-tagset-perl libhtml-tree-perl\n",
            "  libhttp-cookies-perl libhttp-daemon-perl libhttp-date-perl\n",
            "  libhttp-message-perl libhttp-negotiate-perl libio-html-perl\n",
            "  libio-socket-ssl-perl liblwp-mediatypes-perl liblwp-protocol-https-perl\n",
            "  libmailtools-perl libnet-http-perl libnet-smtp-ssl-perl libnet-ssleay-perl\n",
            "  libtry-tiny-perl liburi-perl libwww-perl libwww-robotrules-perl\n",
            "  libxml-libxml-perl libxml-namespacesupport-perl libxml-parser-perl\n",
            "  libxml-sax-base-perl libxml-sax-expat-perl libxml-sax-perl\n",
            "  libxml-simple-perl netbase perl-openssl-defaults\n",
            "0 upgraded, 36 newly installed, 0 to remove and 19 not upgraded.\n",
            "Need to get 0 B/2,178 kB of archives.\n",
            "After this operation, 7,174 kB of additional disk space will be used.\n",
            "Extracting templates from packages: 100%\n",
            "Selecting previously unselected package netbase.\n",
            "(Reading database ... 128208 files and directories currently installed.)\n",
            "Preparing to unpack .../00-netbase_6.1_all.deb ...\n",
            "Unpacking netbase (6.1) ...\n",
            "Selecting previously unselected package libdata-dump-perl.\n",
            "Preparing to unpack .../01-libdata-dump-perl_1.23-1_all.deb ...\n",
            "Unpacking libdata-dump-perl (1.23-1) ...\n",
            "Selecting previously unselected package libencode-locale-perl.\n",
            "Preparing to unpack .../02-libencode-locale-perl_1.05-1_all.deb ...\n",
            "Unpacking libencode-locale-perl (1.05-1) ...\n",
            "Selecting previously unselected package libhttp-date-perl.\n",
            "Preparing to unpack .../03-libhttp-date-perl_6.05-1_all.deb ...\n",
            "Unpacking libhttp-date-perl (6.05-1) ...\n",
            "Selecting previously unselected package libfile-listing-perl.\n",
            "Preparing to unpack .../04-libfile-listing-perl_6.04-1_all.deb ...\n",
            "Unpacking libfile-listing-perl (6.04-1) ...\n",
            "Selecting previously unselected package libfont-afm-perl.\n",
            "Preparing to unpack .../05-libfont-afm-perl_1.20-2_all.deb ...\n",
            "Unpacking libfont-afm-perl (1.20-2) ...\n",
            "Selecting previously unselected package libhtml-tagset-perl.\n",
            "Preparing to unpack .../06-libhtml-tagset-perl_3.20-4_all.deb ...\n",
            "Unpacking libhtml-tagset-perl (3.20-4) ...\n",
            "Selecting previously unselected package liburi-perl.\n",
            "Preparing to unpack .../07-liburi-perl_1.76-2_all.deb ...\n",
            "Unpacking liburi-perl (1.76-2) ...\n",
            "Selecting previously unselected package libhtml-parser-perl.\n",
            "Preparing to unpack .../08-libhtml-parser-perl_3.72-5_amd64.deb ...\n",
            "Unpacking libhtml-parser-perl (3.72-5) ...\n",
            "Selecting previously unselected package libio-html-perl.\n",
            "Preparing to unpack .../09-libio-html-perl_1.001-1_all.deb ...\n",
            "Unpacking libio-html-perl (1.001-1) ...\n",
            "Selecting previously unselected package liblwp-mediatypes-perl.\n",
            "Preparing to unpack .../10-liblwp-mediatypes-perl_6.04-1_all.deb ...\n",
            "Unpacking liblwp-mediatypes-perl (6.04-1) ...\n",
            "Selecting previously unselected package libhttp-message-perl.\n",
            "Preparing to unpack .../11-libhttp-message-perl_6.22-1_all.deb ...\n",
            "Unpacking libhttp-message-perl (6.22-1) ...\n",
            "Selecting previously unselected package libhtml-form-perl.\n",
            "Preparing to unpack .../12-libhtml-form-perl_6.07-1_all.deb ...\n",
            "Unpacking libhtml-form-perl (6.07-1) ...\n",
            "Selecting previously unselected package libhtml-tree-perl.\n",
            "Preparing to unpack .../13-libhtml-tree-perl_5.07-2_all.deb ...\n",
            "Unpacking libhtml-tree-perl (5.07-2) ...\n",
            "Selecting previously unselected package libhtml-format-perl.\n",
            "Preparing to unpack .../14-libhtml-format-perl_2.12-1_all.deb ...\n",
            "Unpacking libhtml-format-perl (2.12-1) ...\n",
            "Selecting previously unselected package libhttp-cookies-perl.\n",
            "Preparing to unpack .../15-libhttp-cookies-perl_6.08-1_all.deb ...\n",
            "Unpacking libhttp-cookies-perl (6.08-1) ...\n",
            "Selecting previously unselected package libhttp-daemon-perl.\n",
            "Preparing to unpack .../16-libhttp-daemon-perl_6.06-1ubuntu0.1_all.deb ...\n",
            "Unpacking libhttp-daemon-perl (6.06-1ubuntu0.1) ...\n",
            "Selecting previously unselected package libhttp-negotiate-perl.\n",
            "Preparing to unpack .../17-libhttp-negotiate-perl_6.01-1_all.deb ...\n",
            "Unpacking libhttp-negotiate-perl (6.01-1) ...\n",
            "Selecting previously unselected package perl-openssl-defaults:amd64.\n",
            "Preparing to unpack .../18-perl-openssl-defaults_4_amd64.deb ...\n",
            "Unpacking perl-openssl-defaults:amd64 (4) ...\n",
            "Selecting previously unselected package libnet-ssleay-perl.\n",
            "Preparing to unpack .../19-libnet-ssleay-perl_1.88-2ubuntu1_amd64.deb ...\n",
            "Unpacking libnet-ssleay-perl (1.88-2ubuntu1) ...\n",
            "Selecting previously unselected package libio-socket-ssl-perl.\n",
            "Preparing to unpack .../20-libio-socket-ssl-perl_2.067-1_all.deb ...\n",
            "Unpacking libio-socket-ssl-perl (2.067-1) ...\n",
            "Selecting previously unselected package libnet-http-perl.\n",
            "Preparing to unpack .../21-libnet-http-perl_6.19-1_all.deb ...\n",
            "Unpacking libnet-http-perl (6.19-1) ...\n",
            "Selecting previously unselected package libtry-tiny-perl.\n",
            "Preparing to unpack .../22-libtry-tiny-perl_0.30-1_all.deb ...\n",
            "Unpacking libtry-tiny-perl (0.30-1) ...\n",
            "Selecting previously unselected package libwww-robotrules-perl.\n",
            "Preparing to unpack .../23-libwww-robotrules-perl_6.02-1_all.deb ...\n",
            "Unpacking libwww-robotrules-perl (6.02-1) ...\n",
            "Selecting previously unselected package libwww-perl.\n",
            "Preparing to unpack .../24-libwww-perl_6.43-1_all.deb ...\n",
            "Unpacking libwww-perl (6.43-1) ...\n",
            "Selecting previously unselected package liblwp-protocol-https-perl.\n",
            "Preparing to unpack .../25-liblwp-protocol-https-perl_6.07-2ubuntu2_all.deb ...\n",
            "Unpacking liblwp-protocol-https-perl (6.07-2ubuntu2) ...\n",
            "Selecting previously unselected package libnet-smtp-ssl-perl.\n",
            "Preparing to unpack .../26-libnet-smtp-ssl-perl_1.04-1_all.deb ...\n",
            "Unpacking libnet-smtp-ssl-perl (1.04-1) ...\n",
            "Selecting previously unselected package libmailtools-perl.\n",
            "Preparing to unpack .../27-libmailtools-perl_2.21-1_all.deb ...\n",
            "Unpacking libmailtools-perl (2.21-1) ...\n",
            "Selecting previously unselected package libxml-namespacesupport-perl.\n",
            "Preparing to unpack .../28-libxml-namespacesupport-perl_1.12-1_all.deb ...\n",
            "Unpacking libxml-namespacesupport-perl (1.12-1) ...\n",
            "Selecting previously unselected package libxml-sax-base-perl.\n",
            "Preparing to unpack .../29-libxml-sax-base-perl_1.09-1_all.deb ...\n",
            "Unpacking libxml-sax-base-perl (1.09-1) ...\n",
            "Selecting previously unselected package libxml-sax-perl.\n",
            "Preparing to unpack .../30-libxml-sax-perl_1.02+dfsg-1_all.deb ...\n",
            "Unpacking libxml-sax-perl (1.02+dfsg-1) ...\n",
            "Selecting previously unselected package libxml-libxml-perl.\n",
            "Preparing to unpack .../31-libxml-libxml-perl_2.0134+dfsg-1build1_amd64.deb ...\n",
            "Unpacking libxml-libxml-perl (2.0134+dfsg-1build1) ...\n",
            "Selecting previously unselected package libxml-parser-perl.\n",
            "Preparing to unpack .../32-libxml-parser-perl_2.46-1_amd64.deb ...\n",
            "Unpacking libxml-parser-perl (2.46-1) ...\n",
            "Selecting previously unselected package libxml-sax-expat-perl.\n",
            "Preparing to unpack .../33-libxml-sax-expat-perl_0.51-1_all.deb ...\n",
            "Unpacking libxml-sax-expat-perl (0.51-1) ...\n",
            "Selecting previously unselected package libxml-simple-perl.\n",
            "Preparing to unpack .../34-libxml-simple-perl_2.25-1_all.deb ...\n",
            "Unpacking libxml-simple-perl (2.25-1) ...\n",
            "Selecting previously unselected package libauthen-sasl-perl.\n",
            "Preparing to unpack .../35-libauthen-sasl-perl_2.1600-1_all.deb ...\n",
            "Unpacking libauthen-sasl-perl (2.1600-1) ...\n",
            "Setting up libhttp-date-perl (6.05-1) ...\n",
            "Setting up libfile-listing-perl (6.04-1) ...\n",
            "Setting up libfont-afm-perl (1.20-2) ...\n",
            "Setting up libhtml-tagset-perl (3.20-4) ...\n",
            "Setting up libauthen-sasl-perl (2.1600-1) ...\n",
            "Setting up liblwp-mediatypes-perl (6.04-1) ...\n",
            "Setting up libtry-tiny-perl (0.30-1) ...\n",
            "Setting up perl-openssl-defaults:amd64 (4) ...\n",
            "Setting up libxml-namespacesupport-perl (1.12-1) ...\n",
            "Setting up libencode-locale-perl (1.05-1) ...\n",
            "Setting up libxml-sax-base-perl (1.09-1) ...\n",
            "Setting up libdata-dump-perl (1.23-1) ...\n",
            "Setting up libio-html-perl (1.001-1) ...\n",
            "Setting up netbase (6.1) ...\n",
            "Setting up liburi-perl (1.76-2) ...\n",
            "Setting up libhttp-message-perl (6.22-1) ...\n",
            "Setting up libnet-ssleay-perl (1.88-2ubuntu1) ...\n",
            "Setting up libhttp-negotiate-perl (6.01-1) ...\n",
            "Setting up libhttp-cookies-perl (6.08-1) ...\n",
            "Setting up libnet-http-perl (6.19-1) ...\n",
            "Setting up libxml-sax-perl (1.02+dfsg-1) ...\n",
            "update-perl-sax-parsers: Non-vendor version of XML::SAX is installed.\n",
            "update-perl-sax-parsers: Automatic registration of SAX parsers might not work.\n",
            "update-perl-sax-parsers: Registering Perl SAX parser XML::SAX::PurePerl with priority 10...\n",
            "update-perl-sax-parsers: Non-vendor version of XML::SAX is installed.\n",
            "update-perl-sax-parsers: Automatic registration of SAX parsers might not work.\n",
            "update-perl-sax-parsers: Updating overall Perl SAX parser modules info file...\n",
            "\n",
            "Creating config file /etc/perl/XML/SAX/ParserDetails.ini with new version\n",
            "Setting up libxml-libxml-perl (2.0134+dfsg-1build1) ...\n",
            "update-perl-sax-parsers: Non-vendor version of XML::SAX is installed.\n",
            "update-perl-sax-parsers: Automatic registration of SAX parsers might not work.\n",
            "update-perl-sax-parsers: Registering Perl SAX parser XML::LibXML::SAX::Parser with priority 50...\n",
            "update-perl-sax-parsers: Non-vendor version of XML::SAX is installed.\n",
            "update-perl-sax-parsers: Automatic registration of SAX parsers might not work.\n",
            "update-perl-sax-parsers: Registering Perl SAX parser XML::LibXML::SAX with priority 50...\n",
            "update-perl-sax-parsers: Non-vendor version of XML::SAX is installed.\n",
            "update-perl-sax-parsers: Automatic registration of SAX parsers might not work.\n",
            "update-perl-sax-parsers: Updating overall Perl SAX parser modules info file...\n",
            "Replacing config file /etc/perl/XML/SAX/ParserDetails.ini with new version\n",
            "Setting up libwww-robotrules-perl (6.02-1) ...\n",
            "Setting up libhttp-daemon-perl (6.06-1ubuntu0.1) ...\n",
            "Setting up libhtml-parser-perl (3.72-5) ...\n",
            "Setting up libio-socket-ssl-perl (2.067-1) ...\n",
            "Setting up libhtml-form-perl (6.07-1) ...\n",
            "Setting up libhtml-tree-perl (5.07-2) ...\n",
            "Setting up libhtml-format-perl (2.12-1) ...\n",
            "Setting up libnet-smtp-ssl-perl (1.04-1) ...\n",
            "Setting up libmailtools-perl (2.21-1) ...\n",
            "Setting up libxml-simple-perl (2.25-1) ...\n",
            "Setting up liblwp-protocol-https-perl (6.07-2ubuntu2) ...\n",
            "Setting up libwww-perl (6.43-1) ...\n",
            "Setting up libxml-parser-perl (2.46-1) ...\n",
            "Setting up libxml-sax-expat-perl (0.51-1) ...\n",
            "update-perl-sax-parsers: Non-vendor version of XML::SAX is installed.\n",
            "update-perl-sax-parsers: Automatic registration of SAX parsers might not work.\n",
            "update-perl-sax-parsers: Registering Perl SAX parser XML::SAX::Expat with priority 50...\n",
            "update-perl-sax-parsers: Non-vendor version of XML::SAX is installed.\n",
            "update-perl-sax-parsers: Automatic registration of SAX parsers might not work.\n",
            "update-perl-sax-parsers: Updating overall Perl SAX parser modules info file...\n",
            "Replacing config file /etc/perl/XML/SAX/ParserDetails.ini with new version\n",
            "Processing triggers for man-db (2.9.1-1) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#!wget -N ftp://ftp.ncbi.nih.gov/toolbox/ncbi_tools/converters/by_program/tbl2asn/linux64.tbl2asn.gz\n",
        "#!gunzip linux64.tbl2asn.gz\n",
        "#!chmod +x linux64.tbl2asn\n",
        "#!mv  linux64.tbl2asn /usr/local/bin/tbl2asn\n",
        "!tbl2asn --help\n",
        "\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5K48IkwfMv3t",
        "outputId": "08792174-48a3-441f-c4d2-398b76c8bab7"
      },
      "execution_count": 40,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "tbl2asn: error while loading shared libraries: libidn.so.11: cannot open shared object file: No such file or directory\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!sudo apt-get install parallel\n",
        "!sudo apt install libc6\n",
        "!sudo apt install libidn11\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "34ujW_XDKDp4",
        "outputId": "4d998c21-561b-4d7e-c0c3-faafcb0b6829"
      },
      "execution_count": 43,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "parallel is already the newest version (20161222-1.1).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 19 not upgraded.\n",
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "libc6 is already the newest version (2.31-0ubuntu9.9).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 19 not upgraded.\n",
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following NEW packages will be installed:\n",
            "  libidn11\n",
            "0 upgraded, 1 newly installed, 0 to remove and 19 not upgraded.\n",
            "Need to get 46.2 kB of archives.\n",
            "After this operation, 250 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu focal/main amd64 libidn11 amd64 1.33-2.2ubuntu2 [46.2 kB]\n",
            "Fetched 46.2 kB in 0s (171 kB/s)\n",
            "debconf: unable to initialize frontend: Dialog\n",
            "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 1.)\n",
            "debconf: falling back to frontend: Readline\n",
            "debconf: unable to initialize frontend: Readline\n",
            "debconf: (This frontend requires a controlling tty.)\n",
            "debconf: falling back to frontend: Teletype\n",
            "dpkg-preconfigure: unable to re-open stdin: \n",
            "Selecting previously unselected package libidn11:amd64.\n",
            "(Reading database ... 129210 files and directories currently installed.)\n",
            "Preparing to unpack .../libidn11_1.33-2.2ubuntu2_amd64.deb ...\n",
            "Unpacking libidn11:amd64 (1.33-2.2ubuntu2) ...\n",
            "Setting up libidn11:amd64 (1.33-2.2ubuntu2) ...\n",
            "Processing triggers for libc-bin (2.31-0ubuntu9.9) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ixSfeDzbtXIg",
        "outputId": "9d763fad-c5f3-45b4-b31a-95a7cd653146"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[10:50:59] Appending to PATH: /root/prokka/bin/../binaries/linux\n",
            "[10:50:59] Appending to PATH: /root/prokka/bin/../binaries/linux/../common\n",
            "[10:50:59] Appending to PATH: /root/prokka/bin\n",
            "[10:50:59] Cleaning databases in /root/prokka/db\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Archaea/sprot.phd\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Archaea/sprot.phi\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Archaea/sprot.phr\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Archaea/sprot.pin\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Archaea/sprot.pog\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Archaea/sprot.psd\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Archaea/sprot.psi\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Archaea/sprot.psq\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/AMR.phd\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/AMR.phi\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/AMR.phr\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/AMR.pin\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/AMR.pog\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/AMR.psd\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/AMR.psi\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/AMR.psq\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/IS.phd\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/IS.phi\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/IS.phr\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/IS.pin\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/IS.pog\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/IS.psd\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/IS.psi\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/IS.psq\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/sprot.phd\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/sprot.phi\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/sprot.phr\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/sprot.pin\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/sprot.pog\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/sprot.psd\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/sprot.psi\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Bacteria/sprot.psq\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Mitochondria/sprot.phd\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Mitochondria/sprot.phi\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Mitochondria/sprot.phr\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Mitochondria/sprot.pin\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Mitochondria/sprot.pog\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Mitochondria/sprot.psd\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Mitochondria/sprot.psi\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Mitochondria/sprot.psq\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Viruses/sprot.phd\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Viruses/sprot.phi\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Viruses/sprot.phr\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Viruses/sprot.pin\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Viruses/sprot.pog\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Viruses/sprot.psd\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Viruses/sprot.psi\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/kingdom/Viruses/sprot.psq\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Enterococcus.phd\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Enterococcus.phi\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Enterococcus.phr\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Enterococcus.pin\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Enterococcus.pog\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Enterococcus.psd\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Enterococcus.psi\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Enterococcus.psq\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Escherichia.phd\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Escherichia.phi\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Escherichia.phr\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Escherichia.pin\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Escherichia.pog\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Escherichia.psd\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Escherichia.psi\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Escherichia.psq\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Staphylococcus.phd\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Staphylococcus.phi\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Staphylococcus.phr\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Staphylococcus.pin\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Staphylococcus.pog\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Staphylococcus.psd\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Staphylococcus.psi\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/genus/Staphylococcus.psq\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/hmm/HAMAP.hmm.h3f\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/hmm/HAMAP.hmm.h3i\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/hmm/HAMAP.hmm.h3m\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/hmm/HAMAP.hmm.h3p\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/cm/Archaea.i1f\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/cm/Archaea.i1i\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/cm/Archaea.i1m\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/cm/Archaea.i1p\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/cm/Bacteria.i1f\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/cm/Bacteria.i1i\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/cm/Bacteria.i1m\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/cm/Bacteria.i1p\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/cm/Viruses.i1f\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/cm/Viruses.i1i\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/cm/Viruses.i1m\n",
            "[10:50:59] Deleting unwanted file: /root/prokka/db/cm/Viruses.i1p\n",
            "[10:50:59] Cleaning complete.\n",
            "[10:50:59] Looking for 'makeblastdb' - found /root/prokka/bin/../binaries/linux/makeblastdb\n",
            "[10:50:59] Determined makeblastdb version is 002009 from 'makeblastdb: 2.9.0+'\n",
            "[10:50:59] Making kingdom BLASTP database: /root/prokka/db/kingdom/Archaea/sprot\n",
            "[10:50:59] Running: makeblastdb -hash_index -dbtype prot -in \\/root\\/prokka\\/db\\/kingdom\\/Archaea\\/sprot -logfile /dev/null\n",
            "[10:50:59] Making kingdom BLASTP database: /root/prokka/db/kingdom/Bacteria/sprot\n",
            "[10:50:59] Running: makeblastdb -hash_index -dbtype prot -in \\/root\\/prokka\\/db\\/kingdom\\/Bacteria\\/sprot -logfile /dev/null\n",
            "[10:51:01] Making kingdom BLASTP database: /root/prokka/db/kingdom/Mitochondria/sprot\n",
            "[10:51:01] Running: makeblastdb -hash_index -dbtype prot -in \\/root\\/prokka\\/db\\/kingdom\\/Mitochondria\\/sprot -logfile /dev/null\n",
            "[10:51:02] Making kingdom BLASTP database: /root/prokka/db/kingdom/Viruses/sprot\n",
            "[10:51:02] Running: makeblastdb -hash_index -dbtype prot -in \\/root\\/prokka\\/db\\/kingdom\\/Viruses\\/sprot -logfile /dev/null\n",
            "[10:51:02] Making kingdom BLASTP database: /root/prokka/db/kingdom/Bacteria/IS\n",
            "[10:51:02] Running: makeblastdb -hash_index -dbtype prot -in \\/root\\/prokka\\/db\\/kingdom\\/Bacteria\\/IS -logfile /dev/null\n",
            "[10:51:03] Making kingdom BLASTP database: /root/prokka/db/kingdom/Bacteria/AMR\n",
            "[10:51:03] Running: makeblastdb -hash_index -dbtype prot -in \\/root\\/prokka\\/db\\/kingdom\\/Bacteria\\/AMR -logfile /dev/null\n",
            "[10:51:04] Making genus BLASTP database: /root/prokka/db/genus/Enterococcus\n",
            "[10:51:04] Running: makeblastdb -hash_index -dbtype prot -in \\/root\\/prokka\\/db\\/genus\\/Enterococcus -logfile /dev/null\n",
            "[10:51:04] Making genus BLASTP database: /root/prokka/db/genus/Escherichia\n",
            "[10:51:04] Running: makeblastdb -hash_index -dbtype prot -in \\/root\\/prokka\\/db\\/genus\\/Escherichia -logfile /dev/null\n",
            "[10:51:04] Making genus BLASTP database: /root/prokka/db/genus/Staphylococcus\n",
            "[10:51:04] Running: makeblastdb -hash_index -dbtype prot -in \\/root\\/prokka\\/db\\/genus\\/Staphylococcus -logfile /dev/null\n",
            "[10:51:04] Looking for 'hmmpress' - found /root/prokka/bin/../binaries/linux/hmmpress\n",
            "[10:51:04] Determined hmmpress version is 003001 from '# HMMER 3.1b2 (February 2015); http://hmmer.org/'\n",
            "[10:51:04] Pressing HMM database: /root/prokka/db/hmm/HAMAP.hmm\n",
            "[10:51:04] Running: hmmpress \\/root\\/prokka\\/db\\/hmm\\/HAMAP\\.hmm\n",
            "Working...    done.\n",
            "Pressed and indexed 1536 HMMs (1536 names).\n",
            "Models pressed into binary file:   /root/prokka/db/hmm/HAMAP.hmm.h3m\n",
            "SSI index for binary model file:   /root/prokka/db/hmm/HAMAP.hmm.h3i\n",
            "Profiles (MSV part) pressed into:  /root/prokka/db/hmm/HAMAP.hmm.h3f\n",
            "Profiles (remainder) pressed into: /root/prokka/db/hmm/HAMAP.hmm.h3p\n",
            "[10:51:07] Looking for 'cmpress' - found /root/prokka/bin/../binaries/linux/cmpress\n",
            "[10:51:07] Determined cmpress version is 001001 from '# INFERNAL 1.1.2 (July 2016)'\n",
            "[10:51:07] Pressing CM database: /root/prokka/db/cm/Viruses\n",
            "[10:51:07] Running: cmpress \\/root\\/prokka\\/db\\/cm\\/Viruses\n",
            "Working...    done.\n",
            "Pressed and indexed 239 CMs and p7 HMM filters (239 names and 239 accessions).\n",
            "Covariance models and p7 filters pressed into binary file:  /root/prokka/db/cm/Viruses.i1m\n",
            "SSI index for binary covariance model file:                 /root/prokka/db/cm/Viruses.i1i\n",
            "Optimized p7 filter profiles (MSV part)  pressed into:      /root/prokka/db/cm/Viruses.i1f\n",
            "Optimized p7 filter profiles (remainder) pressed into:      /root/prokka/db/cm/Viruses.i1p\n",
            "[10:51:07] Pressing CM database: /root/prokka/db/cm/Bacteria\n",
            "[10:51:07] Running: cmpress \\/root\\/prokka\\/db\\/cm\\/Bacteria\n",
            "Working...    done.\n",
            "Pressed and indexed 1127 CMs and p7 HMM filters (1127 names and 1127 accessions).\n",
            "Covariance models and p7 filters pressed into binary file:  /root/prokka/db/cm/Bacteria.i1m\n",
            "SSI index for binary covariance model file:                 /root/prokka/db/cm/Bacteria.i1i\n",
            "Optimized p7 filter profiles (MSV part)  pressed into:      /root/prokka/db/cm/Bacteria.i1f\n",
            "Optimized p7 filter profiles (remainder) pressed into:      /root/prokka/db/cm/Bacteria.i1p\n",
            "[10:51:08] Pressing CM database: /root/prokka/db/cm/Archaea\n",
            "[10:51:08] Running: cmpress \\/root\\/prokka\\/db\\/cm\\/Archaea\n",
            "Working...    done.\n",
            "Pressed and indexed 152 CMs and p7 HMM filters (152 names and 152 accessions).\n",
            "Covariance models and p7 filters pressed into binary file:  /root/prokka/db/cm/Archaea.i1m\n",
            "SSI index for binary covariance model file:                 /root/prokka/db/cm/Archaea.i1i\n",
            "Optimized p7 filter profiles (MSV part)  pressed into:      /root/prokka/db/cm/Archaea.i1f\n",
            "Optimized p7 filter profiles (remainder) pressed into:      /root/prokka/db/cm/Archaea.i1p\n",
            "[10:51:08] Looking for databases in: /root/prokka/db\n",
            "[10:51:08] * Kingdoms: Archaea Bacteria Mitochondria Viruses\n",
            "[10:51:08] * Genera: Enterococcus Escherichia Staphylococcus\n",
            "[10:51:08] * HMMs: HAMAP\n",
            "[10:51:08] * CMs: Archaea Bacteria Viruses\n",
            "[10:51:08] This is prokka 1.14.6\n",
            "[10:51:08] Written by Torsten Seemann <torsten.seemann@gmail.com>\n",
            "[10:51:08] Homepage is https://github.com/tseemann/prokka\n",
            "[10:51:08] Local time is Sat Mar  4 10:51:08 2023\n",
            "[10:51:08] You are not telling me who you are!\n",
            "[10:51:08] Operating system is linux\n",
            "[10:51:08] You have BioPerl 1.7.8\n",
            "Argument \"1.7.8\" isn't numeric in numeric lt (<) at /root/prokka/bin/prokka line 259.\n",
            "[10:51:08] System has 2 cores.\n",
            "[10:51:08] Option --cpu asked for 8 cores, but system only has 2\n",
            "[10:51:08] Will use maximum of 2 cores.\n",
            "[10:51:08] Annotating as >>> Bacteria <<<\n",
            "[10:51:08] Generating locus_tag from 'three_libs_spades_out/scaffolds.fasta' contents.\n",
            "[10:51:08] Setting --locustag IAEMKGFD from MD5 2ae640fd6def9ac6a7454ed12b9bdadd\n",
            "[10:51:08] Enabling options to ensure Genbank/ENA/DDJB submission compliance.\n",
            "[10:51:08] Re-using existing --outdir PROKKA_03042023\n",
            "[10:51:08] Using filename prefix: PROKKA_03042023.XXX\n",
            "[10:51:08] Setting HMMER_NCPU=1\n",
            "[10:51:08] Writing log to: PROKKA_03042023/PROKKA_03042023.log\n",
            "[10:51:08] Command: /root/prokka/bin/prokka --force --compliant three_libs_spades_out/scaffolds.fasta\n",
            "[10:51:08] Appending to PATH: /root/prokka/bin/../binaries/linux\n",
            "[10:51:08] Appending to PATH: /root/prokka/bin/../binaries/linux/../common\n",
            "[10:51:08] Appending to PATH: /root/prokka/bin\n",
            "[10:51:08] Looking for 'aragorn' - found /root/prokka/bin/../binaries/linux/aragorn\n",
            "[10:51:08] Determined aragorn version is 001002 from 'ARAGORN v1.2.38 Dean Laslett'\n",
            "[10:51:08] Looking for 'blastp' - found /root/prokka/bin/../binaries/linux/blastp\n",
            "[10:51:08] Determined blastp version is 002009 from 'blastp: 2.9.0+'\n",
            "[10:51:08] Looking for 'cmpress' - found /root/prokka/bin/../binaries/linux/cmpress\n",
            "[10:51:08] Determined cmpress version is 001001 from '# INFERNAL 1.1.2 (July 2016)'\n",
            "[10:51:08] Looking for 'cmscan' - found /root/prokka/bin/../binaries/linux/cmscan\n",
            "[10:51:08] Determined cmscan version is 001001 from '# INFERNAL 1.1.2 (July 2016)'\n",
            "[10:51:08] Looking for 'egrep' - found /usr/bin/egrep\n",
            "[10:51:08] Looking for 'find' - found /usr/bin/find\n",
            "[10:51:08] Looking for 'grep' - found /usr/bin/grep\n",
            "[10:51:08] Looking for 'hmmpress' - found /root/prokka/bin/../binaries/linux/hmmpress\n",
            "[10:51:08] Determined hmmpress version is 003001 from '# HMMER 3.1b2 (February 2015); http://hmmer.org/'\n",
            "[10:51:08] Looking for 'hmmscan' - found /root/prokka/bin/../binaries/linux/hmmscan\n",
            "[10:51:08] Determined hmmscan version is 003001 from '# HMMER 3.1b2 (February 2015); http://hmmer.org/'\n",
            "[10:51:08] Looking for 'java' - found /usr/bin/java\n",
            "[10:51:08] Looking for 'makeblastdb' - found /root/prokka/bin/../binaries/linux/makeblastdb\n",
            "[10:51:08] Determined makeblastdb version is 002009 from 'makeblastdb: 2.9.0+'\n",
            "[10:51:08] Looking for 'minced' - found /root/prokka/bin/../binaries/linux/../common/minced\n",
            "[10:51:08] Determined minced version is 002000 from 'minced 0.2.0'\n",
            "[10:51:08] Looking for 'parallel' - found /usr/bin/parallel\n",
            "[10:51:08] Determined parallel version is 20161222 from 'GNU parallel 20161222'\n",
            "[10:51:08] Looking for 'prodigal' - found /root/prokka/bin/../binaries/linux/prodigal\n",
            "[10:51:08] Determined prodigal version is 002006 from 'Prodigal V2.6.3: February, 2016'\n",
            "[10:51:08] Looking for 'prokka-genbank_to_fasta_db' - found /root/prokka/bin/prokka-genbank_to_fasta_db\n",
            "[10:51:08] Looking for 'sed' - found /usr/bin/sed\n",
            "[10:51:08] Looking for 'tbl2asn' - found /usr/local/bin/tbl2asn\n",
            "[10:51:08] Determined tbl2asn version is 025008 from 'tbl2asn 25.8   arguments:'\n",
            "[10:51:08] Using genetic code table 11.\n",
            "[10:51:08] Loading and checking input file: three_libs_spades_out/scaffolds.fasta\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_206_length_195_cov_53.0286_ID_565434\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_207_length_190_cov_182.089_ID_565474\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_208_length_189_cov_134.187_ID_565652\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_209_length_185_cov_73.9154_ID_565796\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_210_length_185_cov_100.585_ID_565476\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_211_length_184_cov_144.496_ID_565576\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_212_length_184_cov_285.101_ID_565694\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_213_length_181_cov_156.389_ID_565552\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_214_length_180_cov_76.872_ID_565192\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_215_length_180_cov_68.032_ID_565194\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_216_length_179_cov_37.2419_ID_565816\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_217_length_174_cov_39.3361_ID_565530\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_218_length_170_cov_69.2609_ID_565702\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_219_length_165_cov_76.2636_ID_565196\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_220_length_152_cov_87.9278_ID_565538\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_221_length_151_cov_104.438_ID_565512\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_222_length_151_cov_377.802_ID_565418\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_223_length_151_cov_285.802_ID_565544\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_224_length_147_cov_33.7717_ID_565790\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_225_length_147_cov_80.8587_ID_565198\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_226_length_146_cov_63.3956_ID_565818\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_227_length_144_cov_50.9326_ID_565200\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_228_length_143_cov_60.25_ID_565510\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_229_length_143_cov_54.7841_ID_565202\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_230_length_142_cov_222.414_ID_565726\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_231_length_141_cov_56.7791_ID_565662\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_232_length_138_cov_51.4578_ID_565786\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_233_length_134_cov_275.228_ID_565540\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_234_length_132_cov_212.649_ID_565422\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_235_length_130_cov_146.667_ID_565762\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_236_length_129_cov_359.216_ID_565204\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_237_length_128_cov_271.781_ID_565514\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_238_length_125_cov_217.214_ID_565470\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_239_length_123_cov_108.603_ID_565700\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_240_length_120_cov_58.1385_ID_565206\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_241_length_119_cov_123.797_ID_565208\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_242_length_118_cov_61.0794_ID_565438\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_243_length_117_cov_53.4839_ID_565648\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_244_length_116_cov_94.3934_ID_565210\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_245_length_115_cov_61.35_ID_565212\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_246_length_115_cov_128.7_ID_565704\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_247_length_114_cov_105.288_ID_565810\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_248_length_113_cov_93.4655_ID_565668\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_249_length_112_cov_87.6667_ID_565772\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_250_length_112_cov_107.596_ID_565432\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_251_length_111_cov_101.018_ID_565468\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_252_length_111_cov_67.1429_ID_565214\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_253_length_111_cov_167.143_ID_565696\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_254_length_110_cov_115.055_ID_565506\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_255_length_110_cov_76.3455_ID_565436\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_256_length_110_cov_388.309_ID_565684\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_257_length_110_cov_77.9091_ID_565216\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_258_length_110_cov_177.764_ID_565218\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_259_length_109_cov_99.7407_ID_565614\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_260_length_109_cov_136.241_ID_565424\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_261_length_109_cov_154.481_ID_565616\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_262_length_109_cov_88.6481_ID_565532\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_263_length_108_cov_46.8302_ID_565220\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_264_length_107_cov_92.75_ID_565496\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_265_length_106_cov_69.5098_ID_565708\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_266_length_106_cov_109.961_ID_565222\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_267_length_103_cov_64.7708_ID_565820\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_268_length_103_cov_161.333_ID_565373\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_269_length_102_cov_107.447_ID_565656\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_270_length_100_cov_140.267_ID_565375\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_271_length_98_cov_32.3023_ID_565377\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_272_length_98_cov_39.7442_ID_565836\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_273_length_97_cov_52.3571_ID_565502\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_274_length_95_cov_75.05_ID_565586\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_275_length_95_cov_66.5_ID_565379\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_276_length_95_cov_114.175_ID_565426\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_277_length_95_cov_116.625_ID_565381\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_278_length_95_cov_100_ID_565383\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_279_length_90_cov_118.657_ID_565678\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_280_length_88_cov_113.333_ID_565480\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_281_length_87_cov_84.3125_ID_565385\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_282_length_87_cov_32.8438_ID_565572\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_283_length_82_cov_335.111_ID_565536\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_284_length_82_cov_63.7778_ID_565584\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_285_length_80_cov_232.64_ID_565636\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_286_length_79_cov_160.417_ID_565450\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_287_length_78_cov_66.3913_ID_565387\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_288_length_75_cov_210.15_ID_565452\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_289_length_74_cov_252.737_ID_565389\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_290_length_72_cov_383.941_ID_565522\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_291_length_72_cov_160.412_ID_565391\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_292_length_71_cov_153.188_ID_565393\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_293_length_71_cov_198.875_ID_565592\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_294_length_70_cov_367.867_ID_565632\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_295_length_69_cov_298.5_ID_565736\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_296_length_68_cov_84.7692_ID_565395\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_297_length_68_cov_181.538_ID_565660\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_298_length_64_cov_238.556_ID_565658\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_299_length_63_cov_125.75_ID_565397\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_300_length_63_cov_261_ID_565516\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_301_length_61_cov_147.667_ID_565674\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_302_length_60_cov_302.4_ID_565446\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_303_length_60_cov_65.4_ID_565399\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_304_length_60_cov_407_ID_565548\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_305_length_59_cov_42_ID_565720\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_306_length_59_cov_176_ID_565706\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_307_length_57_cov_195.5_ID_565442\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_308_length_57_cov_77.5_ID_565401\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_309_length_57_cov_143_ID_565508\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_310_length_57_cov_93_ID_565403\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_311_length_57_cov_71.5_ID_565594\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_312_length_57_cov_135_ID_565405\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_313_length_57_cov_85.5_ID_565407\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_314_length_57_cov_66_ID_565409\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_315_length_57_cov_147.5_ID_565472\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_316_length_56_cov_137_ID_565566\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_317_length_56_cov_145_ID_565411\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_318_length_56_cov_322_ID_565546\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_319_length_56_cov_37_ID_565676\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_320_length_56_cov_227_ID_565486\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_321_length_56_cov_158_ID_565734\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_322_length_56_cov_259_ID_565742\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_323_length_56_cov_269_ID_565832\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_324_length_56_cov_245_ID_565828\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_325_length_56_cov_183_ID_565794\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_326_length_56_cov_162_ID_565550\n",
            "[10:51:09] Skipping short (<200 bp) contig: NODE_327_length_56_cov_148_ID_565838\n",
            "[10:51:09] Wrote 205 contigs totalling 5424432 bp.\n",
            "[10:51:09] Predicting tRNAs and tmRNAs\n",
            "[10:51:09] Running: aragorn -l -gc11  -w PROKKA_03042023\\/PROKKA_03042023\\.fna\n",
            "[10:51:11] 1 tRNA-SeC [72129,72223] 35 (tca)\n",
            "[10:51:11] 2 tRNA-Ala c[122894,122969] 34 (ggc)\n",
            "[10:51:11] 3 tRNA-Ala c[123009,123084] 34 (ggc)\n",
            "[10:51:11] 4 tRNA-Val [125784,125859] 34 (tac)\n",
            "[10:51:11] 5 tRNA-Glu c[326197,326272] 35 (ttc)\n",
            "[10:51:11] 6 tmRNA [352225,352587] 90,125 ANDENYALAA**\n",
            "[10:51:11] 7 tRNA-Met c[369872,369947] 34 (cat)\n",
            "[10:51:11] 8 tRNA-Arg c[401974,402050] 35 (acg)\n",
            "[10:51:11] 9 tRNA-Arg c[402249,402325] 35 (acg)\n",
            "[10:51:11] 10 tRNA-Arg c[402524,402600] 35 (acg)\n",
            "[10:51:11] 11 tRNA-Ser c[402604,402696] 35 (gct)\n",
            "[10:51:11] 12 tRNA-Met c[494712,494788] 35 (cat)\n",
            "[10:51:11] 13 tRNA-Leu c[498569,498655] 35 (gag)\n",
            "[10:51:11] 14 tRNA-Thr c[592314,592389] 34 (ggt)\n",
            "[10:51:11] 15 tRNA-Pro c[892264,892340] 35 (cgg)\n",
            "[10:51:11] 16 tRNA-Thr [981718,981793] 34 (cgt)\n",
            "[10:51:11] 17 tRNA-Ser [1069278,1069358] 38 (gga)\n",
            "[10:51:11] 18 tRNA-Arg [1286866,1286942] 35 (tct)\n",
            "[10:51:11] 19 tRNA-Arg c[1516746,1516822] 35 (tct)\n",
            "[10:51:11] 20 tRNA-Tyr [1557019,1557103] 35 (gta)\n",
            "[10:51:11] 21 tRNA-Ser [1730723,1730810] 35 (gga)\n",
            "[10:51:11] 22 tRNA-Asn c[1764729,1764804] 34 (gtt)\n",
            "[10:51:11] 23 tRNA-Asn c[1767138,1767213] 34 (gtt)\n",
            "[10:51:11] 24 tRNA-Asn [1768962,1769037] 34 (gtt)\n",
            "[10:51:11] 25 tRNA-Asn c[1808853,1808928] 34 (gtt)\n",
            "[10:51:11] 26 tRNA-Ser [1809920,1810009] 35 (cga)\n",
            "[10:51:11] 27 tRNA-Gly [1864131,1864206] 34 (gcc)\n",
            "[10:51:11] 28 tRNA-Cys [1864261,1864334] 33 (gca)\n",
            "[10:51:11] 29 tRNA-Leu [1864347,1864433] 35 (taa)\n",
            "[10:51:11] 30 tRNA-Val c[2198146,2198222] 35 (gac)\n",
            "[10:51:11] 31 tRNA-Val c[2198227,2198303] 35 (gac)\n",
            "[10:51:11] 32 tRNA-Asp c[2492348,2492424] 35 (gtc)\n",
            "[10:51:11] 33 tRNA-Asp c[2500571,2500647] 35 (gtc)\n",
            "[10:51:11] 34 tRNA-Ala c[2504139,2504214] 34 (tgc)\n",
            "[10:51:11] 35 tRNA-Ile c[2504257,2504333] 35 (gat)\n",
            "[10:51:11] 36 tRNA-Leu [2764127,2764213] 35 (cag)\n",
            "[10:51:11] 37 tRNA-Leu [2764242,2764328] 35 (cag)\n",
            "[10:51:11] 38 tRNA-Leu [2764363,2764449] 35 (cag)\n",
            "[10:51:11] 1 tRNA-Met c[23543,23618] 34 (cat)\n",
            "[10:51:11] 2 tRNA-Phe c[225213,225288] 34 (gaa)\n",
            "[10:51:11] 3 tRNA-Gly [340653,340726] 33 (ccc)\n",
            "[10:51:11] 4 tRNA-Met c[399015,399091] 35 (cat)\n",
            "[10:51:11] 5 tRNA-Met c[399124,399200] 35 (cat)\n",
            "[10:51:11] 6 tRNA-Met c[399233,399309] 35 (cat)\n",
            "[10:51:11] 1 tRNA-Arg c[138625,138701] 35 (tct)\n",
            "[10:51:11] 2 tRNA-Arg c[138715,138791] 35 (tcg)\n",
            "[10:51:11] 3 tRNA-Met c[138801,138876] 34 (cat)\n",
            "[10:51:11] 4 tRNA-Ser [197579,197666] 35 (tga)\n",
            "[10:51:11] 5 tRNA-Ser [304837,304924] 35 (gga)\n",
            "[10:51:11] 6 tRNA-Lys c[496259,496334] 34 (ttt)\n",
            "[10:51:11] 7 tRNA-Val c[496339,496414] 34 (tac)\n",
            "[10:51:11] 8 tRNA-Lys c[496564,496639] 34 (ttt)\n",
            "[10:51:11] 9 tRNA-Val c[496644,496719] 34 (tac)\n",
            "[10:51:11] 10 tRNA-Lys c[496755,496830] 34 (ttt)\n",
            "[10:51:11] 1 tRNA-Pro [204937,205013] 35 (ggg)\n",
            "[10:51:11] 2 tRNA-Arg [369799,369873] 34 (cct)\n",
            "[10:51:11] 1 tRNA-Thr [52721,52796] 34 (tgt)\n",
            "[10:51:11] 2 tRNA-Tyr [52805,52889] 35 (gta)\n",
            "[10:51:11] 3 tRNA-Gly [53006,53080] 34 (tcc)\n",
            "[10:51:11] 4 tRNA-Thr [53087,53162] 34 (ggt)\n",
            "[10:51:11] 1 tRNA-Glu [113579,113654] 35 (ttc)\n",
            "[10:51:11] 2 tRNA-Asp [117017,117093] 35 (gtc)\n",
            "[10:51:11] 3 tRNA-Trp [117102,117177] 34 (cca)\n",
            "[10:51:11] 4 tRNA-Arg [152521,152597] 35 (ccg)\n",
            "[10:51:11] 5 tRNA-His [152656,152731] 34 (gtg)\n",
            "[10:51:11] 6 tRNA-Leu [152752,152838] 35 (cag)\n",
            "[10:51:11] 7 tRNA-Pro [152881,152957] 35 (tgg)\n",
            "[10:51:11] 1 tRNA-Gly c[36411,36486] 34 (gcc)\n",
            "[10:51:11] 2 tRNA-Gly c[36523,36598] 34 (gcc)\n",
            "[10:51:11] 3 tRNA-Gly c[36635,36710] 34 (gcc)\n",
            "[10:51:11] 4 tRNA-Phe [66438,66513] 34 (gaa)\n",
            "[10:51:11] 5 tRNA-Leu c[111903,111987] 35 (caa)\n",
            "[10:51:11] 1 tRNA-Gln c[106291,106365] 33 (ctg)\n",
            "[10:51:11] 2 tRNA-Gln c[106403,106477] 33 (ctg)\n",
            "[10:51:11] 3 tRNA-Met c[106526,106602] 35 (cat)\n",
            "[10:51:11] 4 tRNA-Gln c[106618,106692] 33 (ttg)\n",
            "[10:51:11] 5 tRNA-Gln c[106727,106801] 33 (ttg)\n",
            "[10:51:11] 6 tRNA-Leu c[106825,106909] 35 (tag)\n",
            "[10:51:11] 7 tRNA-Met c[106918,106994] 35 (cat)\n",
            "[10:51:11] 1 tRNA-Arg [62,138] 35 (tcg)\n",
            "[10:51:11] Found 80 tRNAs\n",
            "[10:51:11] Predicting Ribosomal RNAs\n",
            "[10:51:11] You need either Barrnap or RNAmmer installed to predict rRNAs!\n",
            "[10:51:11] Skipping ncRNA search, enable with --rfam if desired.\n",
            "[10:51:11] Total of 79 tRNA + rRNA features\n",
            "[10:51:11] Searching for CRISPR repeats\n",
            "[10:51:13] CRISPR1 gnl|Prokka|IAEMKGFD_2 442127 with 14 spacers\n",
            "[10:51:14] Found 1 CRISPRs\n",
            "[10:51:14] Predicting coding sequences\n",
            "[10:51:14] Contigs total 5424432 bp, so using single mode\n",
            "[10:51:14] Running: prodigal -i PROKKA_03042023\\/PROKKA_03042023\\.fna -c -m -g 11 -p single -f sco -q\n",
            "[10:51:23] Excluding CDS which overlaps existing RNA (tRNA) at gnl|Prokka|IAEMKGFD_1:592237..592560 on + strand\n",
            "[10:51:23] Excluding CDS which overlaps existing RNA (tRNA) at gnl|Prokka|IAEMKGFD_1:1069348..1069593 on + strand\n",
            "[10:51:24] Excluding CDS which overlaps existing RNA (tRNA) at gnl|Prokka|IAEMKGFD_1:1557010..1557282 on - strand\n",
            "[10:51:25] Excluding CDS which overlaps existing RNA (tRNA) at gnl|Prokka|IAEMKGFD_2:225079..225300 on - strand\n",
            "[10:51:26] Found 5064 CDS\n",
            "[10:51:26] Connecting features back to sequences\n",
            "[10:51:26] Not using genus-specific database. Try --usegenus to enable it.\n",
            "[10:51:26] Annotating CDS, please be patient.\n",
            "[10:51:26] Will use 2 CPUs for similarity searching.\n",
            "[10:51:28] There are still 5064 unannotated CDS left (started with 5064)\n",
            "[10:51:28] Will use blast to search against /root/prokka/db/kingdom/Bacteria/IS with 2 CPUs\n",
            "[10:51:28] Running: cat PROKKA_03042023\\/PROKKA_03042023\\.IS\\.tmp\\.94089\\.faa | parallel --gnu --plain -j 2 --block 395525 --recstart '>' --pipe blastp -query - -db /root/prokka/db/kingdom/Bacteria/IS -evalue 1e-30 -qcov_hsp_perc 90 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > PROKKA_03042023\\/PROKKA_03042023\\.IS\\.tmp\\.94089\\.blast 2> /dev/null\n",
            "[10:53:12] Deleting unwanted file: PROKKA_03042023/PROKKA_03042023.IS.tmp.94089.faa\n",
            "[10:53:12] Deleting unwanted file: PROKKA_03042023/PROKKA_03042023.IS.tmp.94089.blast\n",
            "[10:53:13] There are still 4989 unannotated CDS left (started with 5064)\n",
            "[10:53:13] Will use blast to search against /root/prokka/db/kingdom/Bacteria/AMR with 2 CPUs\n",
            "[10:53:13] Running: cat PROKKA_03042023\\/PROKKA_03042023\\.AMR\\.tmp\\.94089\\.faa | parallel --gnu --plain -j 2 --block 390517 --recstart '>' --pipe blastp -query - -db /root/prokka/db/kingdom/Bacteria/AMR -evalue 1e-300 -qcov_hsp_perc 90 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > PROKKA_03042023\\/PROKKA_03042023\\.AMR\\.tmp\\.94089\\.blast 2> /dev/null\n",
            "[10:54:37] Deleting unwanted file: PROKKA_03042023/PROKKA_03042023.AMR.tmp.94089.faa\n",
            "[10:54:37] Deleting unwanted file: PROKKA_03042023/PROKKA_03042023.AMR.tmp.94089.blast\n",
            "[10:54:38] There are still 4989 unannotated CDS left (started with 5064)\n",
            "[10:54:38] Will use blast to search against /root/prokka/db/kingdom/Bacteria/sprot with 2 CPUs\n",
            "[10:54:38] Running: cat PROKKA_03042023\\/PROKKA_03042023\\.sprot\\.tmp\\.94089\\.faa | parallel --gnu --plain -j 2 --block 390517 --recstart '>' --pipe blastp -query - -db /root/prokka/db/kingdom/Bacteria/sprot -evalue 1e-09 -qcov_hsp_perc 80 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > PROKKA_03042023\\/PROKKA_03042023\\.sprot\\.tmp\\.94089\\.blast 2> /dev/null\n"
          ]
        }
      ],
      "source": [
        "!$HOME/prokka/bin/prokka --setupdb\n",
        "!$HOME/prokka/bin/prokka --force --compliant three_libs_spades_out/scaffolds.fasta "
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "mpfWt12sRAd5"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "EpwLZccbtjYA"
      },
      "source": [
        "# 6. Finding the closest relative of E. coli X"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "4UFAq2D8tubO"
      },
      "outputs": [],
      "source": [
        "!git clone https://github.com/tseemann/barrnap.git $HOME/barrnap\n",
        "!$HOME/barrnap/bin/barrnap --help"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "F36dS8kEtvR6"
      },
      "source": [
        "# 7. What is the genetic cause of HUS?"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "3iUcy38ht505"
      },
      "outputs": [],
      "source": []
    }
  ],
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyNArI2kn2PAqJ1axxrbhmg5",
      "include_colab_link": true
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}